<div class="container">

<table style="width: 100%;"><tr>
<td>PACVr.complete</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Execute the complete pipeline of <span class="pkg">PACVr</span>
</h2>

<h3>Description</h3>

<p>This function executes the complete pipeline of <span class="pkg">PACVr</span> 
via a single command.
</p>


<h3>Usage</h3>

<pre><code class="language-R">PACVr.complete(
  gbkFile,
  bamFile,
  windowSize = 250,
  logScale = FALSE,
  threshold = 0.5,
  IRCheck = NA,
  relative = TRUE,
  textSize = 0.5,
  tabularCovStats = FALSE,
  output = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gbkFile</code></td>
<td>
<p>a character string that specifies the name of, and path to, 
the GenBank input file; alternatively, a character string of GenBank data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bamFile</code></td>
<td>
<p>a character string that specifies the name of, and path to, 
the BAM input file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>windowSize</code></td>
<td>
<p>a numeric value that specifies window size in which the 
coverage is calculated</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logScale</code></td>
<td>
<p>a boolean that specifies if the coverage depth is to be 
log-transformed before visualizing it</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>a numeric value that specifies the threshold for plotting 
coverage depth bars in red as opposed to the default black</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>IRCheck</code></td>
<td>
<p>a numeric value that specifies if tests 
for IRs of input genome should be performed, and - if IRs are present - 
which line type to be used for visualizing gene synteny between IRs;
0 = IR presence test but no synteny visualization, 
1 = IR presence test and synteny visualization, with ribbon lines between IRs,
2 = IR presence test and synteny visualization, with solid lines between IRs,
otherwise = neither IR presence test nor synteny visualization</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>relative</code></td>
<td>
<p>a boolean that specifies whether the threshold is a relative 
value of the average coverage instead of an absolute value</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>textSize</code></td>
<td>
<p>a numeric value that specifies the relative font size of the 
text element in the visualization</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tabularCovStats</code></td>
<td>
<p>a boolean, that when TRUE, generates additional files with
detailed genomic region information</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output</code></td>
<td>
<p>a character string that specifies the name of, and path to, 
the output file</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A file in pdf format containing a circular visualization of the 
input plastid genome and its sequence reads. 
As a function, returns 0 in case of visualization success.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
gbkFile &lt;- system.file("extdata", "NC_045072/NC_045072.gb", package="PACVr")
bamFile &lt;- system.file("extdata", "NC_045072/NC_045072_subsampled.bam", package="PACVr")
outFile &lt;- paste(tempdir(), "/NC_045072__all_reads.pdf", sep="")
PACVr.complete(gbkFile=gbkFile, bamFile=bamFile, windowSize=250, logScale=FALSE,
               threshold=0.5, IRCheck=1, relative=TRUE, textSize=0.5,
               tabularCovStats=FALSE, output=outFile)

## End(Not run)
## Not run: 
gbkFile &lt;- system.file("extdata", "MG936619/MG936619.gb", package="PACVr")
bamFile &lt;- system.file("extdata", "MG936619/MG936619_subsampled.bam", package="PACVr")
outFile &lt;- paste(tempdir(), "/MG936619_CoverageViz.pdf", sep="")
PACVr.complete(gbkFile=gbkFile, bamFile=bamFile, windowSize=50, logScale=FALSE,
               threshold=0.5, IRCheck=NA, relative=TRUE, textSize=0.5,
               tabularCovStats=FALSE, output=outFile)

## End(Not run)
</code></pre>


</div>