<div class="container">

<table style="width: 100%;"><tr>
<td>basicP2proc</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Perform basic 'pagoda2' processing, i.e. adjust variance, calculate pca reduction,
make knn graph, identify clusters with multilevel, and generate
largeVis and tSNE embeddings.</h2>

<h3>Description</h3>

<p>Perform basic 'pagoda2' processing, i.e. adjust variance, calculate pca reduction,
make knn graph, identify clusters with multilevel, and generate
largeVis and tSNE embeddings.
</p>


<h3>Usage</h3>

<pre><code class="language-R">basicP2proc(
  cd,
  n.cores = 1,
  n.odgenes = 3000,
  nPcs = 100,
  k = 30,
  perplexity = 50,
  log.scale = TRUE,
  trim = 10,
  keep.genes = NULL,
  min.cells.per.gene = 0,
  min.transcripts.per.cell = 100,
  get.largevis = TRUE,
  get.tsne = TRUE,
  make.geneknn = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cd</code></td>
<td>
<p>count matrix whereby rows are genes, columns are cells.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.cores</code></td>
<td>
<p>numeric Number of cores to use (default=1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.odgenes</code></td>
<td>
<p>numeric Number of top overdispersed genes to use (dfault=3e3)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nPcs</code></td>
<td>
<p>numeric Number of PCs to use (default=100)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>numeric Default number of neighbors to use in kNN graph (default=30)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>perplexity</code></td>
<td>
<p>numeric Perplexity to use in generating tSNE and largeVis embeddings (default=50)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log.scale</code></td>
<td>
<p>boolean Whether to use log scale normalization (default=TRUE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trim</code></td>
<td>
<p>numeric Number of cells to trim in winsorization (default=10)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep.genes</code></td>
<td>
<p>optional set of genes to keep from being filtered out (even at low counts) (default=NULL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.cells.per.gene</code></td>
<td>
<p>numeric Minimal number of cells required for gene to be kept (unless listed in keep.genes) (default=0)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.transcripts.per.cell</code></td>
<td>
<p>numeric Minimumal number of molecules/reads for a cell to be admitted (default=100)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>get.largevis</code></td>
<td>
<p>boolean Whether to caluclate largeVis embedding (default=TRUE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>get.tsne</code></td>
<td>
<p>boolean Whether to calculate tSNE embedding (default=TRUE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>make.geneknn</code></td>
<td>
<p>boolean Whether pre-calculate gene kNN (for gene search) (default=TRUE)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a new 'Pagoda2' object
</p>


</div>