<div class="container">

<table style="width: 100%;"><tr>
<td>pagoo</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create a Pagoo Object</h2>

<h3>Description</h3>

<p>This is the main function to load a pagoo object. It's safer and
more friendly than using pagoo's class constructors (PgR6,
PgR6M, and PgR6MS). This function returns either a
<code>PgR6M</code> class object, or a <code>PgR6MS</code> class object,
depending on the parameters provided. If sequences are provided, it returns
the latter. See below for more details.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pagoo(
  data,
  org_meta,
  cluster_meta,
  sequences,
  core_level = 95,
  sep = "__",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>A <code>data.frame</code> or
<code>DataFrame</code> containing at least the
following columns: <code>gene</code> (gene name), <code>org</code> (organism name to
which the gene belongs to), and <code>cluster</code> (group of orthologous to
which the gene belongs to). More columns can be added as metadata for each
gene.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>org_meta</code></td>
<td>
<p>(optional) A <code>data.frame</code> or
<code>DataFrame</code> containing additional
metadata for organisms. This <code>data.frame</code> must have a column named
"org" with valid organisms names (that is, they should match with those
provided in <code>data</code>, column <code>org</code>), and additional columns will be
used as metadata. Each row should correspond to each organism.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cluster_meta</code></td>
<td>
<p>(optional) A <code>data.frame</code> or
<code>DataFrame</code> containing additional
metadata for clusters. This <code>data.frame</code> must have a column named
"cluster" with valid organisms names (that is, they should match with those
provided in <code>data</code>, column <code>cluster</code>), and additional columns
will be used as metadata. Each row should correspond to each cluster.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sequences</code></td>
<td>
<p>(optional) Can accept: 1) a named <code>list</code> of named
<code>character</code> vector. Name of list are names of organisms, names of
character vector are gene names; or 2) a named <code>list</code> of
<code>DNAStringSetList</code> objects (same
requirements as (1), but with BStringSet names as gene names); or 3) a
<code>DNAStringSetList</code> (same
requirements as (2) but <code>DNAStringSetList</code> names are organisms names).
If this parameter is used, then a <code>PgR6MS</code> class object is
returned.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>core_level</code></td>
<td>
<p>The initial core_level (that's the percentage of organisms a
core cluster must be in to be considered as part of the core genome). Must
be a number between 100 and 85, (default: 95). You can change it later by
using the <code>$core_level</code> field once the object was created.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep</code></td>
<td>
<p>A separator. By default is '__'(two underscores). It will be used
to create a unique <code>gid</code> (gene identifier) for each gene. <code>gid</code>s
are created by pasting <code>org</code> to <code>gene</code>, separated by <code>sep</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p><code>logical</code>. Whether to display progress messages when
loading class.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This package uses
[R6](https://r6.r-lib.org/articles/Introduction.html) classes to provide a
unified, comprehensive, standardized, but at the same time flexible, way to
analyze a pangenome. The idea is to have a single object which contains both
the data and the basic methods to analyze them, as well as manipulate fields,
explore, and to use in harmony with the already existing and extensive list of R
packages available created for comparative genomics and genetics.
</p>
<p>For more information, tutorials, and resources, please visit https://iferres.github.io/pagoo/ .
</p>


<h3>Index</h3>



<h4>Active Bindings</h4>


<ul>
<li> <p><a href="#field-pan_matrix"><code>$pan_matrix</code></a>
</p>
</li>
<li> <p><a href="#field-organisms"><code>$organisms</code></a>
</p>
</li>
<li> <p><a href="#field-clusters"><code>$clusters</code></a>
</p>
</li>
<li> <p><a href="#field-genes"><code>$genes</code></a>
</p>
</li>
<li> <p><a href="#field-sequences"><code>$sequences</code></a>
</p>
</li>
<li> <p><a href="#field-core_level"><code>$core_level</code></a>
</p>
</li>
<li> <p><a href="#field-core_genes"><code>$core_genes</code></a>
</p>
</li>
<li> <p><a href="#field-core_clusters"><code>$core_clusters</code></a>
</p>
</li>
<li> <p><a href="#field-core_sequences"><code>$core_sequences</code></a>
</p>
</li>
<li> <p><a href="#field-shell_genes"><code>$shell_genes</code></a>
</p>
</li>
<li> <p><a href="#field-shell_clusters"><code>$shell_clusters</code></a>
</p>
</li>
<li> <p><a href="#field-shell_sequences"><code>$shell_sequences</code></a>
</p>
</li>
<li> <p><a href="#field-cloud_genes"><code>$cloud_genes</code></a>
</p>
</li>
<li> <p><a href="#field-cloud_clusters"><code>$cloud_clusters</code></a>
</p>
</li>
<li> <p><a href="#field-cloud_sequences"><code>$cloud_sequences</code></a>
</p>
</li>
</ul>
<h3>Active bindings</h3>


<a id="field-pan_matrix"></a>
<ul>
<li>
<p><b><code>$pan_matrix</code></b> The panmatrix. Rows are organisms, and
columns are groups of orthologous. Cells indicates the presence (&gt;=1) or
absence (0) of a given gene, in a given organism. Cells can have values
greater than 1 if contain in-paralogs.
<a id="field-organisms"></a>
</p>
</li>
<li>
<p><b><code>$organisms</code></b> A <code>DataFrame</code> with available
organism names, and organism number identifier as <code>rownames()</code>. (Dropped
organisms will not be displayed in this field, see <code>$dropped</code> below).
Additional metadata will be shown if provided, as additional columns.
<a id="field-clusters"></a>
</p>
</li>
<li>
<p><b><code>$clusters</code></b> A <code>DataFrame</code> with the groups
of orthologous (clusters). Additional metadata will be shown as additional columns,
if provided before. Each row corresponds to each cluster.
<a id="field-genes"></a>
</p>
</li>
<li>
<p><b><code>$genes</code></b> A <code>SplitDataFrameList</code> object with
one entry per cluster. Each element contains a <code>DataFrame</code>
with gene ids (<code>&lt;gid&gt;</code>) and additional metadata, if provided. <code>gid</code> are
created by <code>paste</code>ing organism and gene names, so duplication in gene names
are avoided.
<a id="field-sequences"></a>
</p>
</li>
<li>
<p><b><code>$sequences</code></b> A <code>DNAStringSetList</code> with the
set of sequences grouped by cluster. Each group is accessible as were a list. All
<code>Biostrings</code> methods are available.
<a id="field-core_level"></a>
</p>
</li>
<li>
<p><b><code>$core_level</code></b> The percentage of organisms a gene must be in
to be considered as part of the coregenome. <code>core_level = 95</code> by default.
Can't be set above 100, and below 85 raises a warning.
<a id="field-core_genes"></a>
</p>
</li>
<li>
<p><b><code>$core_genes</code></b> Like <code>genes</code>, but only showing core genes.
<a id="field-core_clusters"></a>
</p>
</li>
<li>
<p><b><code>$core_clusters</code></b> Like <code>$clusters</code>, but only showing core
clusters.
<a id="field-core_sequences"></a>
</p>
</li>
<li>
<p><b><code>$core_sequences</code></b> Like <code>$sequences</code>, but only showing core
sequences.
<a id="field-cloud_genes"></a>
</p>
</li>
<li>
<p><b><code>$cloud_genes</code></b> Like <code>genes</code>, but only showing cloud genes.
These are defined as those clusters which contain a single gene (singletons), plus
those which have more than one but its organisms are probably clonal due to identical
general gene content. Colloquially defined as strain-specific genes.
<a id="field-cloud_clusters"></a>
</p>
</li>
<li>
<p><b><code>$cloud_clusters</code></b> Like <code>$clusters</code>, but only showing cloud
clusters as defined above.
<a id="field-cloud_sequences"></a>
</p>
</li>
<li>
<p><b><code>$cloud_sequences</code></b> Like <code>$sequences</code>, but only showing cloud
sequences as defined above.
<a id="field-shell_genes"></a>
</p>
</li>
<li>
<p><b><code>$shell_genes</code></b> Like <code>genes</code>, but only showing shell genes.
These are defined as those clusters than don't belong neither to the core genome,
nor to cloud genome. Colloquially defined as genes that are present in some but not
all strains, and that aren't strain-specific.
<a id="field-shell_clusters"></a>
</p>
</li>
<li>
<p><b><code>$shell_clusters</code></b> Like <code>$clusters</code>, but only showing shell
clusters, as defined above.
<a id="field-shell_sequences"></a>
</p>
</li>
<li>
<p><b><code>$shell_sequences</code></b> Like <code>$sequences</code>, but only showing shell
sequences, as defined above.
<a id="field-summary_stats"></a>
</p>
</li>
<li>
<p><b><code>$summary_stats</code></b> A <code>DataFrame</code> with
information about the number of core, shell, and cloud clusters, as well as the
total number of clusters.
<a id="field-random_seed"></a>
</p>
</li>
<li>
<p><b><code>$random_seed</code></b> The last <code>.Random.seed</code>. Used for
reproducibility purposes only.
<a id="field-dropped"></a>
</p>
</li>
<li>
<p><b><code>$dropped</code></b> A <code>character</code> vector with dropped organism
names, and organism number identifier as <code>names()</code>
</p>
</li>
</ul>
<h3>Methods</h3>


<p>Below is a comprehensive description of all the methods provided by the object.
</p>
<a id="method-add_metadata"></a>
<dl>
<dt><b>Add metadata</b></dt>
<dd>


<h4>Description:</h4>

<p>Add metadata to the object. You can add metadata to each organism, to each
group of orthologous, or to each gene. Elements with missing data should be filled
by <code>NA</code> (dimensions of the provided data.frame must be coherent with object
data).
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">⁠                  ⁠</code><code>$add_metadata(map = 'org', df)</code>
</p>



<h4>Arguments:</h4>


<ul>
<li>
<p><b><code>map</code></b>: <code>character</code> identifying the metadata to map. Can
be one of <code>"org"</code>, <code>"group"</code>, or <code>"gid"</code>.
</p>
</li>
<li>
<p><b><code>df</code></b>: <code>data.frame</code> or <code>DataFrame</code> with the metadata to
add. For each case, a column named as <code>"map"</code> must exists, which should
contain identifiers for each element. In the case of adding gene (<code>gid</code>)
metadata,each gene should be referenced by the name of the organism and the name
of the gene as provided in the <code>"data"</code> data.frame, separated by the
<code>"sep"</code> argument.
</p>
</li>
</ul>
<h4>Return:</h4>

<p><code>self</code> invisibly, but with additional metadata.
</p>

</dd>
</dl>
<a id="method-drop"></a>
<dl>
<dt><b>Drop an organism</b></dt>
<dd>


<h4>Description:</h4>

<p>Drop an organism from the dataset. This method allows to hide an organism from
the real dataset, ignoring it in downstream analyses. All the fields and
methods will behave as it doesn't exist. For instance, if you decide to drop
organism 1, the <code>$pan_matrix</code> field (see below) would not show it when
called.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">⁠                  ⁠</code><code>$drop(x)</code>
</p>



<h4>Arguments:</h4>


<ul><li>
<p><b><code>x</code></b>:  <code>character</code> or <code>numeric</code>. The name of the
organism wanted to be dropped, or its numeric id as returned in
<code>$organism</code> field (see below).
</p>
</li></ul>
<h4>Return:</h4>

<p><code>self</code> invisibly, but with <code>x</code> dropped. It isn't necessary
to assign the function call to a new object, nor to re-write it as R6 objects
are mutable.
</p>

</dd>
</dl>
<a id="method-recover"></a>
<dl>
<dt><b>Recover a dropped organism</b></dt>
<dd>


<h4>Description:</h4>

<p>Recover a previously <code>$drop()</code>ped organism (see above). All fields
and methods will start to behave considering this organism again.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">⁠                  ⁠</code><code>$recover(x)</code>
</p>



<h4>Arguments:</h4>


<ul><li>
<p><b><code>x</code></b>: <code>character</code> or <code>numeric</code>. The name of the
organism wanted to be recover, or its numeric id as returned in
<code>$dropped</code> field (see below).
</p>
</li></ul>
<h4>Return:</h4>

<p><code>self</code> invisibly, but with <code>x</code> recovered. It isn't necessary
to assign the function call to a new object, nor to re-write it as R6 objects
are mutable.
</p>

</dd>
</dl>
<a id="method-write_pangenome"></a>
<dl>
<dt><b>Write a pangenome as flat (text) files.</b></dt>
<dd>


<h4>Description:</h4>

<p>Write the pangenome data as flat tables (text). Is not the most recommended way
to save a pangenome, since you can loose information as numeric precision,
column classes (factor, numeric, integer), and the state of the object itself
(i.e. dropped organisms, or core_level), loosing reproducibility. Use
<code>save_pangenomeRDS</code> for a more precise way of saving a pagoo object.
Still, it is useful if you want to work with the data outside R, just keep
the above in mind.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">⁠                  ⁠</code><code>$write_pangenome(dir = "pangenome", force = FALSE)</code>
</p>



<h4>Arguments:</h4>


<ul>
<li>
<p><b><code>dir</code></b>: The unexisting directory name where to put the data files. Default
is "pangenome".
</p>
</li>
<li>
<p><b><code>force</code></b>: <code>logical</code>. Whether to overwrite the directory if it already
exists. Default: <code>FALSE</code>.
</p>
</li>
</ul>
<h4>Return:</h4>

<p>A directory with at least 3 files. "data.tsv" contain the basic
pangenome data as it is provided to the <code>data</code> argument in the
initialization method (<code>$new(...)</code>). "clusters.tsv" contain any metadata
associated to the clusters. "organisms.tsv" contain any metadata associated to
the organisms. The latter 2 files will contain a single column if no metadata
was provided.
</p>

</dd>
</dl>
<a id="method-save_pangenomeRDS"></a>
<dl>
<dt><b>Save a pangenome as a RDS (binary) file.</b></dt>
<dd>


<h4>Description:</h4>

<p>Save a pagoo pangenome object. This function provides a method for saving a pagoo
object and its state into a "RDS" file. To load the pangenome, use the
<code>load_pangenomeRDS</code> function in this package. It *should* be compatible between
pagoo versions, so you could update pagoo and still recover the same pangenome. Even
<code>sep</code> and <code>core_level</code> are restored unless the user provides those
arguments in <code>load_pangenomeRDS</code>. <code>dropped</code> organisms also kept hidden, as
you where working with the original object.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">⁠                  ⁠</code><code>$save_pangenomeRDS(file = "pangenome.rds")</code>
</p>



<h4>Arguments:</h4>


<ul><li>
<p><b><code>file</code></b>: The name of the file to save. Default: "pangenome.rds".
</p>
</li></ul>
<h4>Return:</h4>

<p>Writes a list with all the information needed to restore the object by
using the load_pangenomeRDS function, into an RDS (binary) file.
</p>

</dd>
</dl>
<a id="method-clone"></a>
<dl>
<dt><b>Clone a pagoo object.</b></dt>
<dd>


<h4>Description:</h4>

<p>The objects of this class are clonable with this method.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">⁠                  ⁠</code><code>$clone(deep = FALSE)</code>
</p>



<h4>Arguments:</h4>


<ul><li>
<p><b><code>deep</code></b>: <code>character</code> identifying the metadata to map. Can
be one of <code>"org"</code>, <code>"group"</code>, or <code>"gid"</code>.
</p>
</li></ul>
<h4>Return:</h4>

<p>Whether to make a deep clone.
</p>

</dd>
</dl>
<a id="method-dist"></a>
<dl>
<dt><b>Compute distances</b></dt>
<dd>


<h4>Description:</h4>

<p>Compute distance between all pairs of genomes. The default dist method is
<code>"bray"</code> (Bray-Curtis distance). Another used distance method is <code>"jaccard"</code>,
but you should set <code>binary = FALSE</code> (see below) to obtain a meaningful result.
See <code>vegdist</code> for details, this is just a wrapper function.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">⁠                  ⁠</code><code>$dist(
                  method = "bray",
                  binary = FALSE,
                  diag = FALSE,
                  upper = FALSE,
                  na.rm = FALSE,
                  ...
                  )</code>
</p>



<h4>Arguments:</h4>


<ul>
<li>
<p><b><code>method</code></b>: The distance method to use. See vegdist
for available methods, and details for each one.
</p>
</li>
<li>
<p><b><code>binary</code></b>: Transform abundance matrix into a presence/absence
matrix before computing distance.
</p>
</li>
<li>
<p><b><code>diag</code></b>: Compute diagonals.
</p>
</li>
<li>
<p><b><code>upper</code></b>: Return only the upper diagonal.
</p>
</li>
<li>
<p><b><code>na.rm</code></b>: Pairwise deletion of missing observations when
computing dissimilarities.
</p>
</li>
<li>
<p><b><code>...</code></b>: Other parameters. See vegdist for details.
</p>
</li>
</ul>
<h4>Return:</h4>

<p>A <code>dist</code> object containing all pairwise dissimilarities between genomes.
</p>

</dd>
</dl>
<a id="method-pan_pca"></a>
<dl>
<dt><b>Compute a Principal Component Analysis</b></dt>
<dd>


<h4>Description:</h4>

<p>Performs a principal components analysis on the panmatrix.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">⁠                  ⁠</code><code>$pan_pca( center = TRUE, scale. = FALSE, ...)</code>
</p>



<h4>Arguments:</h4>


<ul>
<li>
<p><b><code>center</code></b>: a logical value indicating whether the variables should be shifted
to be zero centered. Alternately, a vector of length equal the number of columns of x can be
supplied. The value is passed to scale.
</p>
</li>
<li>
<p><b><code>scale.</code></b>: a logical value indicating whether the variables should be scaled
to have unit variance before the analysis takes place. The default is TRUE.
</p>
</li>
<li>
<p><b><code>...</code></b>: Other arguments. See prcomp
</p>
</li>
</ul>
<h4>Return:</h4>

<p>Returns a list with class "prcomp". See prcomp for more information.
</p>

</dd>
</dl>
<a id="method-pg_power_law_fit"></a>
<dl>
<dt><b>Fit a Power Law Function for the Pangenome</b></dt>
<dd>


<h4>Description:</h4>

<p>Fits a power law curve for the pangenome rarefaction simulation.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">⁠                  ⁠</code><code>$pg_power_law_fit(raref, ...)</code>
</p>



<h4>Arguments:</h4>


<ul>
<li>
<p><b><code>raref</code></b>: (Optional) A rarefaction matrix, as returned by <code>rarefact()</code>.
</p>
</li>
<li>
<p><b><code>...</code></b>: Further arguments to be passed to <code>rarefact()</code>. If <code>raref</code>
is missing, it will be computed with default arguments, or with the ones provided here.
</p>
</li>
</ul>
<h4>Return:</h4>

<p>A <code>list</code> of two elements: <code>$formula</code> with a fitted function, and <code>$params</code>
with fitted parameters. An attribute <code>"alpha"</code> is also returned (If
<code>alpha&gt;1</code>, then the pangenome is closed, otherwise is open.
</p>

</dd>
</dl>
<a id="method-cg_exp_decay_fit"></a>
<dl>
<dt><b>Fit an Exponential Decay Function for the Coregenome</b></dt>
<dd>


<h4>Description:</h4>

<p>Fits an exponential decay curve for the coregenome rarefaction simulation.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">⁠                  ⁠</code><code>$cg_exp_decay_fit(raref, pcounts = 10, ...)</code>
</p>



<h4>Arguments:</h4>


<ul>
<li>
<p><b><code>raref</code></b>: (Optional) A rarefaction matrix, as returned by <code>rarefact()</code>.
</p>
</li>
<li>
<p><b><code>pcounts</code></b>: An integer of pseudo-counts. This is used to better fit the function
at small numbers, as the linearization method requires to subtract a constant C, which is the
coregenome size, from <code>y</code>. As <code>y</code> becomes closer to the coregenome size, this operation
tends to 0, and its logarithm goes crazy. By default <code>pcounts=10</code>.
</p>
</li>
<li>
<p><b><code>...</code></b>: Further arguments to be passed to <code>rarefact()</code>. If <code>raref</code>
is missing, it will be computed with default arguments, or with the ones provided here.
</p>
</li>
</ul>
<h4>Return:</h4>

<p>A <code>list</code> of two elements: <code>$formula</code> with a fitted function, and <code>$params</code>
with fitted intercept and decay parameters.
</p>

</dd>
</dl>
<a id="method-fluidity"></a>
<dl>
<dt><b>Compute Genomic Fluidity</b></dt>
<dd>


<h4>Description:</h4>

<p>Computes the genomic fluidity, which is a measure of population
diversity. See <code>fluidity</code> for more details.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">⁠                  ⁠</code><code>$fluidity(nsim = 10)</code>
</p>



<h4>Arguments:</h4>


<ul><li>
<p><b><code>nsim</code></b>:An integer specifying the number of random samples
to use in the computations.
</p>
</li></ul>
<h4>Return:</h4>

<p>A list with two elements, the mean fluidity and its sample standard
deviation over the n.sim computed values.
</p>

</dd>
</dl>
<a id="method-gg_barplot"></a>
<dl>
<dt><b>Plot Accessory Frequency Plot</b></dt>
<dd>


<h4>Description:</h4>

<p>Plot a barplot with the frequency of genes within the total number of
genomes.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">⁠                  ⁠</code><code>$gg_barplot()</code>
</p>



<h4>Return:</h4>

<p>A barplot, and a <code>gg</code> object (<code>ggplot2</code> package) invisibly.
</p>

</dd>
</dl>
<a id="method-gg_dist"></a>
<dl>
<dt><b>Plot a Distance Heatmap</b></dt>
<dd>


<h4>Description:</h4>

<p>Plot a heatmap showing the computed distance between all pairs of organisms.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">⁠                  ⁠</code><code>$gg_dist(method = "bray", ...)</code>
</p>



<h4>Arguments:</h4>


<ul>
<li>
<p><b><code>method</code></b>: Distance method. One of "Jaccard" (default), or
"Manhattan",see above.
</p>
</li>
<li>
<p><b><code>...</code></b>: More arguments to be passed to <code>distManhattan</code>
</p>
</li>
</ul>
<h4>Return:</h4>

<p>A heatmap (<code>ggplot2::geom_tile()</code>), and a <code>gg</code> object (<code>ggplot2</code>
package) invisibly.
</p>

</dd>
</dl>
<a id="method-gg_binmap"></a>
<dl>
<dt><b>Plot a Pangenome Binary Map</b></dt>
<dd>


<h4>Description:</h4>

<p>Plot a pangenome binary map representing the presence/absence of each
gene within each organism.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">⁠                  ⁠</code><code>$gg_binmap()</code>
</p>



<h4>Return:</h4>

<p>A binary map (<code>ggplot2::geom_raster()</code>), and a <code>gg</code> object (<code>ggplot2</code>
package) invisibly.
</p>

</dd>
</dl>
<a id="method-gg_pca"></a>
<dl>
<dt><b>Plot a PCA</b></dt>
<dd>


<h4>Description:</h4>

<p>Plot a scatter plot of a Principal Components Analysis.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">⁠                  ⁠</code><code>$gg_pca(colour = NULL, ...))</code>
</p>



<h4>Arguments:</h4>


<ul>
<li>
<p><b><code>colour</code></b>:The name of the column in <code>$organisms</code> field
from which points will take color (if provided). <code>NULL</code> (default) renders
black points.
</p>
</li>
<li>
<p><b><code>...</code></b>: More arguments to be passed to <code>ggplot2::autoplot()</code>.
</p>
</li>
</ul>
<h4>Return:</h4>

<p>A scatter plot (<code>ggplot2::autoplot()</code>), and a <code>gg</code> object
(<code>ggplot2</code> package) invisibly.
</p>

</dd>
</dl>
<a id="method-gg_pie"></a>
<dl>
<dt><b>Plot a Pie with Pangenome Categories</b></dt>
<dd>


<h4>Description:</h4>

<p>Plot a pie chart showing the number of clusters of each pangenome category: core,
shell, or cloud.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">⁠                  ⁠</code><code>$gg_pie()</code>
</p>



<h4>Return:</h4>

<p>A pie chart (<code>ggplot2::geom_bar() + coord_polar()</code>), and a
<code>gg</code> object (<code>ggplot2</code> package) invisibly.
</p>

</dd>
</dl>
<a id="method-gg_curves"></a>
<dl>
<dt><b>Plot Pangenome Curves</b></dt>
<dd>


<h4>Description:</h4>

<p>Plot pangenome and/or coregenome curves with the fitted functions returned by
<code>pg_power_law_fit()</code> and <code>cg_exp_decay_fit()</code>. You can add points by
adding <code>+ geom_points()</code>, of ggplot2 package.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">⁠                  ⁠</code><code>$gg_curves(what = c("pangenome", "coregenome", ...)</code>
</p>



<h4>Arguments:</h4>


<ul>
<li>
<p><b><code>what</code></b>:  <code>"pangenome"</code> and/or <code>"coregenome"</code>.
</p>
</li>
<li>
<p><b><code>...</code></b>: ignored
</p>
</li>
</ul>
<h4>Return:</h4>

<p>A scatter plot, and a <code>gg</code> object (<code>ggplot2</code> package) invisibly.
</p>

</dd>
</dl>
<a id="method-runShinyApp"></a>
<dl>
<dt><b>Run a Shiny App</b></dt>
<dd>


<h4>Description:</h4>

<p>Launch an interactive shiny app. It contains a sidebar
with controls and switches to interact with the pagoo
object. You can drop/recover organisms from the dataset,
modify the core_level, visualize statistics, plots, and
browse cluster and gene information. In the main body, it
contains 2 tabs to switch between summary statistics plots
and core genome information on one side, and accessory
genome plots and information on the other.
</p>
<p>The lower part of each tab contains two tables, side by
side. On the "Summary" tab, the left one contain
information about core clusters, with one cluster per row.
When one of them is selected (click), the one on the right
is updated to show information about its genes (if
provided), one gene per row. On the "Accessory" tab, a
similar configuration is shown, but on this case only
accessory clusters/genes are displayed. There is a slider
on the sidebar where one can select the accessory
frequency range to display.
</p>
<p>Give it a try!
</p>
<p>Take into account that big pangenomes can slow down the
performance of the app. More than 50-70 organisms often
leads to a delay in the update of the plots/tables.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">⁠                  ⁠</code><code>$runShinyApp()</code>
</p>



<h4>Return:</h4>

<p>Opens a shiny app on the browser.
</p>

</dd>
</dl>
<a id="method-core_seqs_4_phylo"></a>
<dl>
<dt><b>Retrieve Core Genes for Phylogeny</b></dt>
<dd>


<h4>Description:</h4>

<p>A field for obtaining core gene sequences is available (see below),
but for creating a phylogeny with this sets is useful to: 1) have the
possibility of extracting just one sequence of each organism on each
cluster, in case paralogues are present, and 2) filling gaps with
empty sequences in case the core_level was set below 100%, allowing
more genes (some not in 100% of organisms) to be incorporated to the
phylogeny. That is the purpose of this special function.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">⁠                  ⁠</code><code>$core_seqs_4_phylo(max_per_org = 1, fill = TRUE)</code>
</p>



<h4>Arguments:</h4>


<ul>
<li>
<p><b><code>max_per_org</code></b>:  Maximum number of sequences of each organism
to be taken from each cluster.
</p>
</li>
<li>
<p><b><code>fill</code></b>: <code>logical</code>. If fill <code>DNAStringSet</code> with
empty <code>DNAString</code> in cases where <code>core_level</code> is set below 100%,
and some clusters with missing organisms are also considered.
</p>
</li>
</ul>
<h4>Return:</h4>

<p>A <code>DNAStringSetList</code> with core genes. Order of organisms on each cluster
is conserved, so it is easier to concatenate them into a super-gene suitable
for phylogenetic inference.
</p>

</dd>
</dl>
</div>