<div class="container">

<table style="width: 100%;"><tr>
<td>panaroo_2_pagoo</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read panaroos's output into a pagoo's R6 class object</h2>

<h3>Description</h3>

<p>This function handle conversion of <a href="https://gtonkinhill.github.io/panaroo/#/">panaroo</a>'s output files into
a pagoo R6 class object. It takes the "gene_presence_absence.csv" file and
(optionally but recommended) gff input file paths, and returns an object of
class <code>PgR6MS</code> (or <code>PgR6M</code> if left
empty the <code>gffs</code> argument). Panaroo identifies some genes with unusual
lengths tagging them with 'stop', 'length', or 'refound' labels. In the
current version, this function discards those genes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">panaroo_2_pagoo(gene_presence_absence_csv, gffs, sep = "__")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gene_presence_absence_csv</code></td>
<td>
<p><code>character</code>, path to the
"gene_presence_absence.csv" file. (Do not confuse with the file with the
same name but with <code>.Rtab</code> extension).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gffs</code></td>
<td>
<p>A <code>character</code> vector with paths to original gff files used
as roary's input. Typically the return value of <code>list.files()</code> function.
This parameter is optional but highly recommended if you want to manipulate
sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep</code></td>
<td>
<p><code>character</code>. Default: <code>"__"</code> (two underscores). See
PgR6MS for a more detail argument description.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A pagoo's R6 class object. Either PgR6M, if <code>gffs</code>
argument is left empty, or PgR6MS if path to gff files is
provided.
</p>


<h3>References</h3>

<p>Tonkin-Hill, G., MacAlasdair, N., Ruis, C. et al. Producing polished
prokaryotic pangenomes with the Panaroo pipeline. Genome Biol 21, 180 (2020).
https://doi.org/10.1186/s13059-020-02090-4
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
gffs &lt;- list.files(path = "path/to/gffs/",
                   pattern = "[.]gff$",
                   full.names = TRUE)
gpa_csv &lt;- "path/to/gene_presence_absence.csv"

library(pagoo)
pg &lt;- panaroo_2_pagoo(gene_presence_absence_csv = gpa_csv,
                    gffs = gffs)

## End(Not run)
</code></pre>


</div>