<div class="container">

<table style="width: 100%;"><tr>
<td>PgR6MS</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>PgR6 class with Methods and Sequences.</h2>

<h3>Description</h3>

<p>PgR6 with Methods and Sequences. Final users should use <code>pagoo</code>
instead of this, since is more easy to understand.
Inherits: <code>PgR6M</code>
</p>


<h3>Super classes</h3>

<p><code>pagoo::PgR6</code> -&gt; <code>pagoo::PgR6M</code> -&gt; <code>PgR6MS</code>
</p>


<h3>Active bindings</h3>

<div class="r6-active-bindings">

<dl>
<dt><code>sequences</code></dt>
<dd>
<p>A <code>DNAStringSetList</code> with the
set of sequences grouped by cluster. Each group is accessible as were a list. All
<code>Biostrings</code> methods are available.</p>
</dd>
<dt><code>core_sequences</code></dt>
<dd>
<p>Like <code>$sequences</code>, but only showing core
sequences.</p>
</dd>
<dt><code>cloud_sequences</code></dt>
<dd>
<p>Like <code>$sequences</code>, but only showing cloud
sequences as defined above.</p>
</dd>
<dt><code>shell_sequences</code></dt>
<dd>
<p>Like <code>$sequences</code>, but only showing shell
sequences, as defined above.</p>
</dd>
</dl>
</div>


<h3>Methods</h3>



<h4>Public methods</h4>


<ul>
<li> <p><a href="#method-PgR6MS-new"><code>PgR6MS$new()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6MS-core_seqs_4_phylo"><code>PgR6MS$core_seqs_4_phylo()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6MS-clone"><code>PgR6MS$clone()</code></a>
</p>
</li>
</ul>
<details><summary>Inherited methods</summary><ul>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6" data-id="add_metadata"><a href="../../pagoo/html/PgR6.html#method-PgR6-add_metadata"><code>pagoo::PgR6$add_metadata()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6" data-id="drop"><a href="../../pagoo/html/PgR6.html#method-PgR6-drop"><code>pagoo::PgR6$drop()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6" data-id="recover"><a href="../../pagoo/html/PgR6.html#method-PgR6-recover"><code>pagoo::PgR6$recover()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6" data-id="save_pangenomeRDS"><a href="../../pagoo/html/PgR6.html#method-PgR6-save_pangenomeRDS"><code>pagoo::PgR6$save_pangenomeRDS()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6" data-id="write_pangenome"><a href="../../pagoo/html/PgR6.html#method-PgR6-write_pangenome"><code>pagoo::PgR6$write_pangenome()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="cg_exp_decay_fit"><a href="../../pagoo/html/PgR6M.html#method-PgR6M-cg_exp_decay_fit"><code>pagoo::PgR6M$cg_exp_decay_fit()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="dist"><a href="../../pagoo/html/PgR6M.html#method-PgR6M-dist"><code>pagoo::PgR6M$dist()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="gg_barplot"><a href="../../pagoo/html/PgR6M.html#method-PgR6M-gg_barplot"><code>pagoo::PgR6M$gg_barplot()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="gg_binmap"><a href="../../pagoo/html/PgR6M.html#method-PgR6M-gg_binmap"><code>pagoo::PgR6M$gg_binmap()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="gg_curves"><a href="../../pagoo/html/PgR6M.html#method-PgR6M-gg_curves"><code>pagoo::PgR6M$gg_curves()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="gg_dist"><a href="../../pagoo/html/PgR6M.html#method-PgR6M-gg_dist"><code>pagoo::PgR6M$gg_dist()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="gg_pca"><a href="../../pagoo/html/PgR6M.html#method-PgR6M-gg_pca"><code>pagoo::PgR6M$gg_pca()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="gg_pie"><a href="../../pagoo/html/PgR6M.html#method-PgR6M-gg_pie"><code>pagoo::PgR6M$gg_pie()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="pan_pca"><a href="../../pagoo/html/PgR6M.html#method-PgR6M-pan_pca"><code>pagoo::PgR6M$pan_pca()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="pg_power_law_fit"><a href="../../pagoo/html/PgR6M.html#method-PgR6M-pg_power_law_fit"><code>pagoo::PgR6M$pg_power_law_fit()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="rarefact"><a href="../../pagoo/html/PgR6M.html#method-PgR6M-rarefact"><code>pagoo::PgR6M$rarefact()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="runShinyApp"><a href="../../pagoo/html/PgR6M.html#method-PgR6M-runShinyApp"><code>pagoo::PgR6M$runShinyApp()</code></a></span></li>
</ul></details><hr>
<a id="method-PgR6MS-new"></a>



<h4>Method <code>new()</code>
</h4>

<p>Create a <code>PgR6MS</code> object.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6MS$new(
  data,
  org_meta,
  cluster_meta,
  core_level = 95,
  sep = "__",
  DF,
  group_meta,
  sequences,
  verbose = TRUE
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>data</code></dt>
<dd>
<p>A <code>data.frame</code> or <code>DataFrame</code> containing at least the
following columns: <code>gene</code> (gene name), <code>org</code> (organism name to which the gene belongs to),
and <code>cluster</code> (group of orthologous to which the gene belongs to). More columns can be added as metadata
for each gene.</p>
</dd>
<dt><code>org_meta</code></dt>
<dd>
<p>(optional) A <code>data.frame</code> or <code>DataFrame</code>
containing additional metadata for organisms. This <code>data.frame</code> must have a column named "org" with
valid organisms names (that is, they should match with those provided in <code>data</code>, column <code>org</code>), and
additional columns will be used as metadata. Each row should correspond to each organism.</p>
</dd>
<dt><code>cluster_meta</code></dt>
<dd>
<p>(optional) A <code>data.frame</code> or <code>DataFrame</code>
containing additional metadata for clusters. This <code>data.frame</code> must have a column named "cluster" with
valid organisms names (that is, they should match with those provided in <code>data</code>, column <code>cluster</code>), and
additional columns will be used as metadata. Each row should correspond to each cluster.</p>
</dd>
<dt><code>core_level</code></dt>
<dd>
<p>The initial core_level (that's the percentage of organisms a core cluster must be in to be
considered as part of the core genome). Must be a number between 100 and 85, (default: 95). You can change it
later by using the <code>$core_level</code> field once the object was created.</p>
</dd>
<dt><code>sep</code></dt>
<dd>
<p>A separator. By default is '__'(two underscores). It will be used to
create a unique <code>gid</code> (gene identifier) for each gene. <code>gid</code>s are created by pasting
<code>org</code> to <code>gene</code>, separated by <code>sep</code>.</p>
</dd>
<dt><code>DF</code></dt>
<dd>
<p>Deprecated. Use <code>data</code> instead.</p>
</dd>
<dt><code>group_meta</code></dt>
<dd>
<p>Deprecated. Use <code>cluster_meta</code> instead.</p>
</dd>
<dt><code>sequences</code></dt>
<dd>
<p>Can accept: 1) a named <code>list</code> of named <code>character</code> vector. Name of list
are names of organisms, names of character vector are gene names; or 2) a named <code>list</code> of <code>DNAStringSetList</code>
objects (same requirements as (1), but with BStringSet names as gene names); or 3) a <code>DNAStringSetList</code>
(same requirements as (2) but <code>DNAStringSetList</code> names are organisms names).</p>
</dd>
<dt><code>verbose</code></dt>
<dd>
<p><code>logical</code>. Whether to display progress messages when loading class.</p>
</dd>
</dl>
</div>



<h5>Returns</h5>

<p>An R6 object of class PgR6MS. It contains basic fields and methods for analyzing a pangenome. It also
contains additional statistical methods for analyze it, methods to make basic exploratory plots, and methods
for sequence manipulation.
</p>


<hr>
<a id="method-PgR6MS-core_seqs_4_phylo"></a>



<h4>Method <code>core_seqs_4_phylo()</code>
</h4>

<p>A field for obtaining core gene sequences is available (see below), but for creating a phylogeny with this
sets is useful to: 1) have the possibility of extracting just one sequence of each organism on each cluster, in
case paralogues are present, and 2) filling gaps with empty sequences in case the core_level was set below 100%,
allowing more genes (some not in 100% of organisms) to be incorporated to the phylogeny. That is the purpose of
this special function.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6MS$core_seqs_4_phylo(max_per_org = 1, fill = TRUE)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>max_per_org</code></dt>
<dd>
<p>Maximum number of sequences of each organism to be taken from each cluster.</p>
</dd>
<dt><code>fill</code></dt>
<dd>
<p><code>logical</code>. If fill <code>DNAStringSet</code> with empty <code>DNAString</code> in cases where
<code>core_level</code> is set below 100%, and some clusters with missing organisms are also considered.</p>
</dd>
</dl>
</div>



<h5>Returns</h5>

<p>A <code>DNAStringSetList</code> with core genes. Order of organisms on each cluster is conserved, so it is easier
to concatenate them into a super-gene suitable for phylogenetic inference.
</p>


<hr>
<a id="method-PgR6MS-clone"></a>



<h4>Method <code>clone()</code>
</h4>

<p>The objects of this class are cloneable with this method.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6MS$clone(deep = FALSE)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>deep</code></dt>
<dd>
<p>Whether to make a deep clone.</p>
</dd>
</dl>
</div>




</div>