<div class="container">

<table style="width: 100%;"><tr>
<td>NBF</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Normal/Bayes factors method for finding associated pathways</h2>

<h3>Description</h3>

<p>A vector of the computed Bayes factors for the tested pathways.</p>


<h3>Usage</h3>

<pre><code class="language-R">NBF(y, G, P, a, b, s2, nu)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>

<p>Response vector of length N</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>G</code></td>
<td>

<p>Genotype matrix, with N rows and L columns (number of tested SNPs) </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>P</code></td>
<td>

<p>Pathway matrix, with L columns and M columns (number of tested pathways) </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>a</code></td>
<td>

<p>Hyper-parameter of the variance assumed for the integrated out SNP effects</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b</code></td>
<td>

<p>Hyper-parameter of the variance assumed for the pathway effects</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>s2</code></td>
<td>

<p>Hyper-parameter of the Inverse-Chi-squared distribution assumed for the variance of the response vector </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nu</code></td>
<td>

<p>Hyper-parameter of the Inverse-Chi-squared distribution assumed for the variance of the response vector </p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A vector of the computed Bayes factors of the same length as the number of tested pathways 
</p>


<h3>References</h3>

<p>Evangelou, M., Dudbridge, F., Wernisch, L. (2014). Two novel pathway analysis methods based on a hierarchical model. Bioinformatics, 30(5), 690 - 697.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
	data(genotypes)
	G=genotypes
	data(pathways)
	data(SNPs)
	data(genes)
	snps.genes=snps.to.genes(SNPs,genes,distance=0)
	snps.paths=snps.to.pathways(pathways,snps.genes)
	P=create.pathway.df(G,snps.paths)
	y=rnorm(nrow(G),mean=0,sd=10)
	NBF(y,G,P,a,b,s2,nu)
## End(Not run)
</code></pre>


</div>