<div class="container">

<table style="width: 100%;"><tr>
<td>exclusionPower</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Power of exclusion</h2>

<h3>Description</h3>

<p>Computes the power (of a single marker) of excluding a claimed relationship,
given the true relationship.
</p>


<h3>Usage</h3>

<pre><code class="language-R">exclusionPower(
  ped_claim,
  ped_true,
  ids,
  markerindex = NULL,
  alleles = NULL,
  afreq = NULL,
  known_genotypes = list(),
  Xchrom = FALSE,
  plot = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ped_claim</code></td>
<td>
<p>a <code>linkdat</code> object, or a list of several linkdat
and/or singleton objects, describing the claimed relationship. If a list,
the sets of ID labels must be disjoint, that is, all ID labels must be
unique.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ped_true</code></td>
<td>
<p>a <code>linkdat</code> object, or a list of several linkdat
and/or singleton objects, describing the true relationship. ID labels must
be consistent with <code>ped_claim</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ids</code></td>
<td>
<p>individuals available for genotyping.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>markerindex</code></td>
<td>
<p>NULL, or a single numeric indicating the index of a marker
of <code>ped_claim</code> from which <code>alleles</code>, <code>afreq</code> and
<code>known_genotypes</code> will be extracted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleles</code></td>
<td>
<p>a numeric or character vector containing marker alleles names.
Ignored if <code>markerindex</code> is non-NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>afreq</code></td>
<td>
<p>a numerical vector with allele frequencies. An error is given if
they don't sum to 1 (rounded to 3 decimals). Ignored if <code>markerindex</code>
is non-NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>known_genotypes</code></td>
<td>
<p>list of triplets <code>(a, b, c)</code>, indicating that
individual <code>a</code> has genotype <code>b/c</code>. Must be NULL if
<code>markerindex</code> is non-NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Xchrom</code></td>
<td>
<p>a logical: Is the marker on the X chromosome? Ignored if
<code>markerindex</code> is non-NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>
<p>either a logical or the character 'plot_only', controlling if a
plot should be produced. If 'plot_only', a plot is drawn, but no further
computations are done.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function computes the 'Power of exclusion', as defined and discussed in
(Egeland et al., 2014).
</p>


<h3>Value</h3>

<p>A single numeric value. If <code>plot='plot_only'</code>, the function
returns NULL after producing the plot.
</p>


<h3>References</h3>

<p>T. Egeland, N. Pinto and M. D. Vigeland, <em>A general approach
to power calculation for relationship testing.</em> Forensic Science
International: Genetics 9 (2014): 186-190. DOI:10.1016/j.fsigen.2013.05.001
</p>


<h3>Examples</h3>

<pre><code class="language-R">
############################################
### A standard case paternity case:
### Compute the power of exclusion when the claimed father is in fact unrelated to the child.
############################################

claim = nuclearPed(noffs=1, sex=2)     # Specifies individual 1 as the father of 3
true = list(singleton(id=1,sex=1), singleton(id=3, sex=2))     # Specifies 1 and 3 as unrelated
available = c(1, 3)     # Individuals 1 and 3 are available for genotyping

# Equifrequent autosomal SNP:
PE1 = exclusionPower(claim, true, available, alleles = 2, afreq=c(0.5,0.5))

# If the child is known to have genotype 1/1:
PE2 = exclusionPower(claim, true, available, alleles = 2, afreq=c(0.5,0.5),
                     known_genotypes=list(c(3,1,1)))

# Equifrequent SNP on the X chromosome:
PE3 = exclusionPower(claim, true, available, alleles = 2, afreq=c(0.5,0.5), Xchrom=TRUE)

stopifnot(PE1==0.125, PE2==0.25, PE3==0.25)

############################################
### Example from Egeland et al. (2012):
### Two females claim to be mother and daughter. Below we compute the power of various
### markers to reject this claim if they in reality are sisters.
############################################

mother_daughter = nuclearPed(1, sex = 2)
sisters = relabel(nuclearPed(2, sex = c(2, 2)), c(101, 102, 2, 3))

# Equifrequent SNP:
PE1 = exclusionPower(ped_claim = mother_daughter, ped_true = sisters, ids = c(2, 3),
                     alleles = 2)

# SNP with MAF = 0.1:
PE2 = exclusionPower(ped_claim = mother_daughter, ped_true = sisters, ids = c(2, 3),
                     alleles = 2, afreq=c(0.9, 0.1))

# Equifrequent tetra-allelic marker:
PE3 = exclusionPower(ped_claim = mother_daughter, ped_true = sisters, ids = c(2, 3),
                     alleles = 4)

# Tetra-allelic marker with one major allele:
PE4 = exclusionPower(ped_claim = mother_daughter, ped_true = sisters, ids = c(2, 3),
                     alleles = 4, afreq=c(0.7, 0.1, 0.1, 0.1))

stopifnot(round(c(PE1,PE2,PE3,PE4), 5) == c(0.03125, 0.00405, 0.08203, 0.03090))

####### How does the power change if the true pedigree is inbred?
sisters_LOOP = addParents(sisters, 101, father = 201, mother = 202)
sisters_LOOP = addParents(sisters_LOOP, 102, father = 201, mother = 203)

# Equifrequent SNP:
PE5 = exclusionPower(ped_claim = mother_daughter, ped_true = sisters_LOOP,
                     ids = c(2, 3), alleles = 2)

# SNP with MAF = 0.1:
PE6 = exclusionPower(ped_claim = mother_daughter, ped_true = sisters_LOOP,
                     ids = c(2, 3), alleles = 2, afreq=c(0.9, 0.1))

stopifnot(round(c(PE5,PE6), 5) == c(0.03125, 0.00765))

## Not run: 
# Equifrequent tetra-allelic marker:
PE7 = exclusionPower(ped_claim = mother_daughter, ped_true = sisters_LOOP,
                     ids = c(2, 3), alleles = 4)

# Tetra-allelic marker with one major allele:
PE8 = exclusionPower(ped_claim = mother_daughter, ped_true = sisters_LOOP,
                     ids = c(2, 3), alleles = 4, afreq=c(0.7, 0.1, 0.1, 0.1))

stopifnot(round(c(PE7,PE8), 5) == c(0.07617, 0.03457))

## End(Not run)

</code></pre>


</div>