<div class="container">

<table style="width: 100%;"><tr>
<td>dominant</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Example pedigree for linkage analysis</h2>

<h3>Description</h3>

<p>Medical pedigree with 23 individuals of which 15 are genotyped with 650 SNP
markers. Eleven family members are affected by a disease, showing an
autosomal dominant inheritance pattern.
</p>


<h3>Usage</h3>

<pre><code class="language-R">dominant
</code></pre>


<h3>Format</h3>

<p>A data frame with 23 rows and 1306 columns, describing the pedigree and
marker data in pre-makeped format. The first 6 columns contain the pedigree
structure and affection status, while the final 1300 columns hold the marker alleles.
</p>

<ul>
<li>
<p> FAMID. Family ID
</p>
</li>
<li>
<p> ID. Individual ID
</p>
</li>
<li>
<p> FID. Father ID
</p>
</li>
<li>
<p> MID. Mother ID
</p>
</li>
<li>
<p> SEX. Gender (male=1, female=2)
</p>
</li>
<li>
<p> AFF. Affection status (unaffected=1, affected=2, unknown=0)
</p>
</li>
<li>
<p> M1_1. First allele of marker 1
</p>
</li>
<li>
<p> M1_2. Second allele of marker 1
</p>
</li>
<li>
<p> ...
</p>
</li>
<li>
<p> M650_1. First allele of marker 650
</p>
</li>
<li>
<p> M650_2. Second allele of marker 650
</p>
</li>
</ul>
<p>All markers are SNPs, whose alleles are written as 1 and 2.
Missing alleles are denoted by 0.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
x = linkdat(dominant)
summary(x)

</code></pre>


</div>