<div class="container">

<table style="width: 100%;"><tr>
<td>lod</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Two-point LOD score</h2>

<h3>Description</h3>

<p>Calculates the two-point LOD scores of a pedigree for the specified markers.
The recombination ratio between the disease and marker loci can be either
fixed at specific values, or optimized.
</p>


<h3>Usage</h3>

<pre><code class="language-R">lod(
  x,
  markers = seq_len(x$nMark),
  theta = 0,
  loop_breakers = NULL,
  max.only = FALSE,
  verbose = FALSE,
  tol = 0.01
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a <code>linkdat</code> object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>markers</code></td>
<td>
<p>an integer vector denoting which markers to use.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>theta</code></td>
<td>
<p>either a numeric containing specific recombination ratio(s), or
the word 'max', indicating that the recombination ratio should be optimized
by the program.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loop_breakers</code></td>
<td>
<p>a numeric containing IDs of individuals to be used as
loop breakers. Relevant only if the pedigree has loops. See
<code>breakLoops</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.only</code></td>
<td>
<p>a logical indicating whether only the maximum LOD score
should be returned.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>a logical: verbose output or not.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tol</code></td>
<td>
<p>a numeric passed on to <code>optimize</code> as its tolerance
parameter.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The LOD score of a marker is defined as </p>
<p style="text-align: center;"><code class="reqn">LOD(\theta) =
\log[10]\frac{L(\theta)}{L(0.5)}</code>
</p>
<p> where <code class="reqn">L(\theta)</code> denotes the
likelihood of the observed marker genotypes given a recombination ratio
<code class="reqn">\theta</code> between the marker and the disease locus.
</p>


<h3>Value</h3>

<p>If <code>max.only=TRUE</code>, the highest computed LOD score is returned,
as a single number.
</p>
<p>Otherwise a <code>linkres</code> object, which is essentially a matrix containing
the LOD scores. The details depend on the other parameters:
</p>
<p>If <code>theta</code> is numeric, the matrix has dimensions <code>length(theta) *
  length(markers)</code>, and the entry in row <code>t</code>, column <code>m</code> is the lod
score of the pedigree for marker <code>m</code> assuming a recombination rate of
<code>t</code>.
</p>
<p>If <code>theta='max'</code>, the <code>linkres</code> matrix has one column per marker
and two rows: The first containing the LOD score and the second the optimal
recombination ratio for each marker.
</p>
<p>If a marker has incompatible values (i.e. if a child of homozygous 1/1
parents has a 2 allele), the corresponding output entries are <code>NaN</code>.
</p>


<h3>See Also</h3>

<p><code>likelihood</code>, <code>optimize</code>,
<code>breakLoops</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
x = linkdat(toyped, model=1)
res = lod(x)
res_theta = lod(x, theta=c(0, 0.1, 0.2, 0.5))
res_max = lod(x, theta='max')
stopifnot(all(0.3 == round(c(res, res_theta['0',], res_max['LOD',]), 1)))

# bigger pedigree with several markers
y = linkdat(dominant)
y = setModel(y, model=1, penetrances=c(.001, .9, .99))
lod(y, markers=305:310)
lod(y, markers=305:310, theta='max')

# Example with pedigree with loops:
z = linkdat(twoloops, model=2) # fully penetrant autosomal recessive model.

# add SNP for which individuals 15 and 16 are homozygous for the rare allele.
m = marker(z, 15:16, c(1,1), alleles=1:2, afreq=c(0.001, 0.999))
z = addMarker(z, m)
res1 = lod(z)
# manual specification of loop breakers gives same result
res2 = lod(z, loop_breakers=c(8,12))

# making the marker triallelic and adding some genotypes.
z = modifyMarker(z, marker=1, ids=c(7,9,11,13), genotype=3, alleles=1:3, afreq=c(0.001, 0.499, 0.5))
plot(z, 1)
res3 = lod(z)

z = modifyMarker(z, marker=1, alleles=1:4, afreq=c(0.001, 0.499, 0.25, 0.25))
res4 = lod(z)

stopifnot(all(3 == round(c(res1, res2, res3, res4), 1)))

</code></pre>


</div>