<div class="container">

<table style="width: 100%;"><tr>
<td>markerSim</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Marker simulation</h2>

<h3>Description</h3>

<p>Simulates marker genotypes conditional on the pedigree structure, affection
statuses and disease model.
</p>


<h3>Usage</h3>

<pre><code class="language-R">markerSim(
  x,
  N = 1,
  available = NULL,
  alleles = NULL,
  afreq = NULL,
  partialmarker = NULL,
  loop_breakers = NULL,
  eliminate = 0,
  seed = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a <code>linkdat</code> object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>N</code></td>
<td>
<p>a positive integer: the number of markers to be simulated</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>available</code></td>
<td>
<p>a vector containing IDs of the available individuals, i.e.
those whose genotypes should be simulated. By default, all individuals are
included.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleles</code></td>
<td>
<p>a vector containing the alleles for the marker to be
simulation. If a single integer is given, this is interpreted as the number
of alleles, and the actual alleles as <code>1:alleles</code>. Must be NULL if
<code>partialmarker</code> is non-NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>afreq</code></td>
<td>
<p>a vector of length 2 containing the population frequencies for
the marker alleles. Must be NULL if <code>partialmarker</code> is non-NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>partialmarker</code></td>
<td>
<p>Either NULL (resulting in unconditional simulation), a
marker object (on which the simulation should be conditioned) or the index
of an existing marker of <code>x</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loop_breakers</code></td>
<td>
<p>a numeric containing IDs of individuals to be used as
loop breakers. Relevant only if the pedigree has loops, and only if
<code>partialmarker</code> is non-NULL. See <code>breakLoops</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eliminate</code></td>
<td>
<p>A non-negative integer, indicating the number of iterations
in the internal genotype-compatibility algorithm. Positive values can save
time if <code>partialmarker</code> is non-NULL and the number of alleles is
large.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>NULL, or a numeric seed for the random number generator.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>a logical.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This implements (with various time savers) the algorithm used in SLINK of the
LINKAGE/FASTLINK suite. If <code>partialmarker</code> is NULL, genotypes are
simulated by simple gene dropping, using <code>simpleSim</code>.
</p>


<h3>Value</h3>

<p>a <code>linkdat</code> object equal to <code>x</code> except its
<code>markerdata</code> entry, which consists of the <code>N</code> simulated markers.
</p>


<h3>References</h3>

<p>G. M. Lathrop, J.-M. Lalouel, C. Julier, and J. Ott,
<em>Strategies for Multilocus Analysis in Humans</em>, PNAS 81(1984), pp.
3443-3446.
</p>


<h3>See Also</h3>

<p><code>simpleSim</code>, <code>linkage.power</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
x = nuclearPed(2)
partial = marker(x, 3, 1, alleles=1:3)
markerSim(x, N=1, alleles=1:3)
markerSim(x, N=1, partialmarker=partial)
markerSim(x, N=1, partialmarker=partial)
markerSim(x, N=1, available=4, partialmarker=partial)

</code></pre>


</div>