<div class="container">

<table style="width: 100%;"><tr>
<td>simpleSim</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Unconditional marker simulation</h2>

<h3>Description</h3>

<p>Unconditional simulation of unlinked markers
</p>


<h3>Usage</h3>

<pre><code class="language-R">simpleSim(
  x,
  N,
  alleles,
  afreq,
  available,
  Xchrom = FALSE,
  mutmat = NULL,
  seed = NULL,
  verbose = T
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a <code>linkdat</code> object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>N</code></td>
<td>
<p>a positive integer: the number of markers to be simulated</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleles</code></td>
<td>
<p>a vector containing the allele names. If missing, the alleles
are taken to be <code>seq_along(afreq)</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>afreq</code></td>
<td>
<p>a vector of length 2 containing the population frequencies for
the alleles. If missing, the alleles are assumed equifrequent.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>available</code></td>
<td>
<p>a vector containing IDs of the available individuals, i.e.
those whose genotypes should be simulated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Xchrom</code></td>
<td>
<p>a logical: X linked markers or not?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mutmat</code></td>
<td>
<p>a mutation matrix, or a list of two such matrices named
'female' and 'male'. The matrix/matrices must be square, with the allele
labels as dimnames, and each row must sum to 1 (after rounding to 3
decimals).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>NULL, or a numeric seed for the random number generator.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>a logical.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This simulation is done by distributing alleles randomly to all founders,
followed by unconditional gene dropping down throughout the pedigree (i.e.
for each non-founder a random allele is selected from each of the parents).
Finally the genotypes of any individuals not included in <code>available</code> are
removed.
</p>


<h3>Value</h3>

<p>a <code>linkdat</code> object equal to <code>x</code> in all respects except its
<code>markerdata</code> entry, which consists of the <code>N</code> simulated markers.
</p>


<h3>See Also</h3>

<p><code>markerSim</code>, <code>linkageSim</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
x = nuclearPed(1)
simpleSim(x, N=3, afreq=c(0.5, 0.5))

y = addOffspring(cousinPed(1), father=7, mother=8, noffs=1)
simpleSim(y, N=3, alleles=LETTERS[1:10])

</code></pre>


</div>