<div class="container">

<table style="width: 100%;"><tr>
<td>dominant1</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Dominant linkage analysis example</h2>

<h3>Description</h3>

<p>A dataset with SNP genotypes for 14 members of a pedigree affected with a
dominant disorder.
</p>


<h3>Usage</h3>

<pre><code class="language-R">dominant1
</code></pre>


<h3>Format</h3>

<p>A list with 3 elements:
</p>

<ul>
<li> <p><code>ped</code>: A pedtools::ped object describing a pedigree with 19 individuals,
including genotypes for 14 members at 248 markers on Chromosome 1.
</p>
</li>
<li> <p><code>aff</code>: A vector indicating the affected pedigree members.
</p>
</li>
<li> <p><code>map</code>: A data frame with 3 columns (<code>chrom</code>, <code>marker</code>, <code>cm</code>) describing the
centiMorgan positions of the markers.
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">
data(dominant1)
ped = dominant1$ped
aff = dominant1$aff
model = diseaseModel("AD")

# Compute singlepoint LODs
lods = lod(ped, aff = aff, model = model)

# LOD score graph
plot(lods)


</code></pre>


</div>