<div class="container">

<table style="width: 100%;"><tr>
<td>relationLR</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Relationship Likelihood Ratio</h2>

<h3>Description</h3>

<p>Computes likelihood for two pedigrees and their ratio, the likelihood ratio
(LR).
</p>


<h3>Usage</h3>

<pre><code class="language-R">relationLR(
  ped_numerator,
  ped_denominator,
  ids,
  alleles,
  afreq = NULL,
  known_genotypes = list(),
  loop_breakers = NULL,
  Xchrom = FALSE,
  plot = TRUE,
  title1 = "",
  title2 = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ped_numerator</code></td>
<td>
<p>a <code>linkdat</code> object, or a list of several
linkdat and/or singleton objects, describing the relationship corresponding
to the hypothesis H1 (numerator).  If a list, the sets of ID labels must be
disjoint, that is, all ID labels must be unique.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ped_denominator</code></td>
<td>
<p>a <code>linkdat</code> object, or a list of several
linkdat and/or singleton objects, describing the relationship corresponding
to the hypothesis H2 (denominator). ID labels must be consistent with
<code>ped_claim</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ids</code></td>
<td>
<p>genotyped individuals.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleles</code></td>
<td>
<p>a numeric or character vector containing marker alleles names</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>afreq</code></td>
<td>
<p>a numerical vector with allele frequencies. An error is given if
they don't sum to 1 (rounded to 3 decimals).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>known_genotypes</code></td>
<td>
<p>list of triplets <code>(a, b, c)</code>, indicating that
individual <code>a</code> has genotype <code>b/c</code>. Missing value is 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loop_breakers</code></td>
<td>
<p>Not yet implemented, only default value NULL currently
handled</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Xchrom</code></td>
<td>
<p>a logical: Is the marker on the X chromosome?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>
<p>either a logical or the character 'plot_only', controlling if a
plot should be produced. If 'plot_only', a plot is drawn, but no further
computations are done.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title1</code></td>
<td>
<p>a character, title of leftmost plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title2</code></td>
<td>
<p>a character, title of rightmost plot.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function computes the likelihood of two pedigrees (each corresponding to
a hypothesis describing a family relationship).  The likelihood ratio is also
reported. Unlike other implementations we are aware of, partial DNA profiles
are allowed here. For instance, if the genotype of a person is reported as
1/0 (0 is 'missing') for a triallelic marker with uniform allele frequencies,
the possible ordered genotypes (1,1), (2,1), (1,2), (1,3) and (3,1) are
treated as equally likely. (For general allele frequencies, genotype
probabilities are obtained by assuming Hardy-Weinberg equilibrium.) A
reasonable future extension would be to allow the user to weigh these
genotypes; typically (1,1) may be more likely than the others.  If
<code>plot='plot_only'</code>, the function returns NULL after producing the plot.
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>lik.numerator</code></td>
<td>
<p>likelihood of data given ped_numerator</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lik.denominator</code></td>
<td>
<p>likelihood of data given ped_denominator</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LR</code></td>
<td>
<p>likelihood ratio lik.numerator/lik.denominator</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Thore Egeland, Magnus Dehli Vigeland
</p>


<h3>See Also</h3>

<p><code>exclusionPower</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
############################################
# A partial DNA profile is obtained from the person
# denoted 4 in the figure produced below
# There are two possibilities:
# H1: 4 is the missing relative of 3 and 6 (as shown to the left) or
# H2: 4 is unrelated to 3 and 6.
############################################
p = c(0.2, 0.8)
alleles = 1:length(p)
g3 = c(1,1); g4 = c(1,0); g6 = c(2,2)
x1 = nuclearPed(2)
x1 = addOffspring(x1, father = 4, sex = 1, noff = 1)
m = marker(x1, 3, g3, 4, g4, 6, g6, alleles = alleles, afreq = p)
x1 = addMarker(x1, m)
x2 = nuclearPed(2)
x2 = addOffspring(x2, father = 4, sex = 1, noff = 1)
m = marker(x2, 3, g3, 6, g6, alleles = alleles, afreq = p)
x2 = addMarker(x2, m)
missing = singleton(4, sex = 1)
m.miss = marker(missing, g4, alleles = alleles, afreq = p)
missing = addMarker(missing, m.miss)
x2 = relabel(x2, c(1:3, 99, 5:6), 1:6)
known = list(c(3, g3), c(4,g4), c(6, g6))
LR = relationLR(x1, list(x2, missing), ids = c(3,4,6),
                alleles = alleles, afreq = p, known = known,
                title1 = 'H1: Missing person 4 related',
                title2 = 'H2:Missing person 4 unrelated')$LR
# Formula:
p = p[1]
LR.a = (1+p)/(2*p*(2-p))
stopifnot(abs(LR - LR.a) &lt; 1e-10)

</code></pre>


</div>