<div class="container">

<table style="width: 100%;"><tr>
<td>score_terms</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate Agglomerated Scores of Enriched Terms for Each Subject</h2>

<h3>Description</h3>

<p>Calculate Agglomerated Scores of Enriched Terms for Each Subject
</p>


<h3>Usage</h3>

<pre><code class="language-R">score_terms(
  enrichment_table,
  exp_mat,
  cases = NULL,
  use_description = FALSE,
  plot_hmap = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>enrichment_table</code></td>
<td>
<p>a data frame that must contain the 3 columns below: </p>

<dl>
<dt>Term_Description</dt>
<dd>
<p>Description of the enriched term (necessary if <code>use_description = TRUE</code>)</p>
</dd>
<dt>ID</dt>
<dd>
<p>ID of the enriched term (necessary if <code>use_description = FALSE</code>)</p>
</dd>
<dt>Up_regulated</dt>
<dd>
<p>the up-regulated genes in the input involved in the given term's gene set, comma-separated</p>
</dd>
<dt>Down_regulated</dt>
<dd>
<p>the down-regulated genes in the input involved in the given term's gene set, comma-separated</p>
</dd>
</dl>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exp_mat</code></td>
<td>
<p>the experiment (e.g., gene expression/methylation) matrix.
Columns are samples and rows are genes. Column names must contain sample
names and row names must contain the gene symbols.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cases</code></td>
<td>
<p>(Optional) A vector of sample names that are cases in the
case/control experiment. (default = NULL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use_description</code></td>
<td>
<p>Boolean argument to indicate whether term descriptions
(in the 'Term_Description' column) should be used. (default = <code>FALSE</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_hmap</code></td>
<td>
<p>Boolean value to indicate whether or not to draw the
heatmap plot of the scores. (default = TRUE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional arguments for <code>plot_scores</code> for aesthetics
of the heatmap plot</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Matrix of agglomerated scores of each enriched term per sample.
Columns are samples, rows are enriched terms. Optionally, displays a heatmap
of this matrix.
</p>


<h3>Conceptual Background</h3>

<p>For an experiment matrix (containing expression, methylation, etc. values),
the rows of which are genes and the columns of which are samples,
we denote: </p>

<ul>
<li>
<p> E as a matrix of size m x n
</p>
</li>
<li>
<p> G as the set of all genes in the experiment G = E<sub>i.</sub>,  i ∈ [1, m]
</p>
</li>
<li>
<p> S as the set of all samples in the experiment S = E<sub>.j</sub>,  i ∈ [1, n]
</p>
</li>
</ul>
<p>We next define the gene score matrix GS (the standardized experiment matrix,
also of size m x n) as:
</p>
GS<sub>gs</sub> = (E<sub>gs</sub> - ē<sub>g</sub>) / s<sub>g</sub><p>where g ∈ G, s ∈ S,
ē<sub>g</sub> is the mean of
all values for gene g and s<sub>g</sub>
is the standard deviation of all values for gene g.
</p>
<p>We next denote T to be a set of terms (where each t ∈ T
is a set of term-related genes, i.e.,
t = {g<sub>x</sub>, ..., g<sub>y</sub>} ⊂ G)
and finally define the agglomerated term scores matrix TS (where rows
correspond to genes and columns corresponds to samples s.t. the matrix has size
|T| x n) as:
</p>
TS<sub>ts</sub> = 1/|t| ∑ <sub>g ∈ t</sub> GS<sub>gs</sub><p>,
where t ∈ T and s ∈ S.
</p>


<h3>Examples</h3>

<pre><code class="language-R">score_matrix &lt;- score_terms(
  example_pathfindR_output,
  example_experiment_matrix,
  plot_hmap = FALSE
)
</code></pre>


</div>