<div class="container">

<table style="width: 100%;"><tr>
<td>active_snw_search</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Perform Active Subnetwork Search</h2>

<h3>Description</h3>

<p>Perform Active Subnetwork Search
</p>


<h3>Usage</h3>

<pre><code class="language-R">active_snw_search(
  input_for_search,
  pin_name_path = "Biogrid",
  snws_file = "active_snws",
  dir_for_parallel_run = NULL,
  score_quan_thr = 0.8,
  sig_gene_thr = 0.02,
  search_method = "GR",
  seedForRandom = 1234,
  silent_option = TRUE,
  use_all_positives = FALSE,
  geneInitProbs = 0.1,
  saTemp0 = 1,
  saTemp1 = 0.01,
  saIter = 10000,
  gaPop = 400,
  gaIter = 10000,
  gaThread = 5,
  gaCrossover = 1,
  gaMut = 0,
  grMaxDepth = 1,
  grSearchDepth = 1,
  grOverlap = 0.5,
  grSubNum = 1000
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>input_for_search</code></td>
<td>
<p>input the input data that active subnetwork search uses. The input
must be a data frame containing at least these 2 columns: </p>

<dl>
<dt>GENE</dt>
<dd>
<p>Gene Symbol</p>
</dd>
<dt>P_VALUE</dt>
<dd>
<p>p value obtained through a test, e.g. differential expression/methylation</p>
</dd>
</dl>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pin_name_path</code></td>
<td>
<p>Name of the chosen PIN or absolute/path/to/PIN.sif. If PIN name,
must be one of c('Biogrid', 'STRING', 'GeneMania', 'IntAct', 'KEGG', 'mmu_STRING'). If
path/to/PIN.sif, the file must comply with the PIN specifications. (Default = 'Biogrid')</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snws_file</code></td>
<td>
<p>name for active subnetwork search output data
<strong>without file extension</strong> (default = 'active_snws')</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dir_for_parallel_run</code></td>
<td>
<p>(previously created) directory for a parallel run iteration.
Used in the wrapper function (see ?run_pathfindR) (Default = NULL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>score_quan_thr</code></td>
<td>
<p>active subnetwork score quantile threshold. Must be
between 0 and 1 or set to -1 for not filtering. (Default = 0.8)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sig_gene_thr</code></td>
<td>
<p>threshold for the minimum proportion of significant genes in
the subnetwork (Default = 0.02) If the number of genes to use as threshold is
calculated to be &lt; 2 (e.g. 50 signif. genes x 0.01 = 0.5), the threshold number
is set to 2</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>search_method</code></td>
<td>
<p>algorithm to use when performing active subnetwork
search. Options are greedy search (GR), simulated annealing (SA) or genetic
algorithm (GA) for the search (default = 'GR').</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seedForRandom</code></td>
<td>
<p>seed for reproducibility while running the java modules (applies for GR and SA)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>silent_option</code></td>
<td>
<p>boolean value indicating whether to print the messages
to the console (FALSE) or not (TRUE, this will print to a temp. file) during
active subnetwork search (default = TRUE). This option was added because
during parallel runs, the console messages get disorderly printed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use_all_positives</code></td>
<td>
<p>if TRUE: in GA, adds an individual with all positive
nodes. In SA, initializes candidate solution with all positive nodes. (default = FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneInitProbs</code></td>
<td>
<p>For SA and GA, probability of adding a gene in initial solution (default = 0.1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>saTemp0</code></td>
<td>
<p>Initial temperature for SA (default = 1.0)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>saTemp1</code></td>
<td>
<p>Final temperature for SA (default = 0.01)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>saIter</code></td>
<td>
<p>Iteration number for SA (default = 10000)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gaPop</code></td>
<td>
<p>Population size for GA (default = 400)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gaIter</code></td>
<td>
<p>Iteration number for GA (default = 200)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gaThread</code></td>
<td>
<p>Number of threads to be used in GA (default = 5)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gaCrossover</code></td>
<td>
<p>Applies crossover with the given probability in GA (default = 1, i.e. always perform crossover)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gaMut</code></td>
<td>
<p>For GA, applies mutation with given mutation rate (default = 0, i.e. mutation off)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grMaxDepth</code></td>
<td>
<p>Sets max depth in greedy search, 0 for no limit (default = 1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grSearchDepth</code></td>
<td>
<p>Search depth in greedy search (default = 1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grOverlap</code></td>
<td>
<p>Overlap threshold for results of greedy search (default = 0.5)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grSubNum</code></td>
<td>
<p>Number of subnetworks to be presented in the results (default = 1000)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list of genes in every identified active subnetwork that has a score greater than
the 'score_quan_thr'th quantile and that has at least 'sig_gene_thr' affected genes.
</p>


<h3>Examples</h3>

<pre><code class="language-R">processed_df &lt;- example_pathfindR_input[1:15, -2]
colnames(processed_df) &lt;- c('GENE', 'P_VALUE')
GR_snws &lt;- active_snw_search(
  input_for_search = processed_df,
  pin_name_path = 'KEGG',
  search_method = 'GR',
  score_quan_thr = 0.8
)
# clean-up
unlink('active_snw_search', recursive = TRUE)
</code></pre>


</div>