<div class="container">

<table style="width: 100%;"><tr>
<td>Drosophila</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Drosophila datasets and QTL mapping study
</h2>

<h3>Description</h3>

<p>Two datasets with genotypes and phenotypes for backcrossed Drosophilas.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(Drosophila)</code></pre>


<h3>Format</h3>

<p>Two datasets with genotypes and phenotypes for backcrossed Drosophilas.
</p>
<p>The set of 41 markers describes genotypes while 5 variables describe phenotypes. See references for more details.
</p>

<dl>
<dt><code>bm</code></dt>
<dd>
<p>A data.frame with 370 obs. and 46 variables, first 41 are genotypes of gene markers, last five describes genotypes</p>
</dd>
<dt><code>bs</code></dt>
<dd>
<p>A data.frame with 402 obs. and 46 variables, first 41 are genotypes of gene markers, last five describes genotypes</p>
</dd>
<dt><code>chr</code></dt>
<dd>
<p>Factor w/ 4 levels CBFbeta, FLT3, None, Other</p>
</dd>
<dt><code>pos</code></dt>
<dd>
<p>Markers position on chromosom in centimorgnas</p>
</dd>
</dl>
<h3>Details</h3>

<p>The phonotype pc1 is nicely described by genotype in both backcrossed datasets.
</p>


<h3>Source</h3>

<p>Genetic Architecture of a Morphological Shape Difference Between Two Drosophila Species
Zhao-Bang Zenga, Jianjun Liu, Lynn F. Stamb, Chen-Hung Kao, John M. Mercer, Cathy C. Laurie
Genetics, Vol. 154, 299-310, January 2000
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(Drosophila)
library(lattice)
# calculate log likelihoods
pval1 = numeric(41)
for (i in 1:41) {
  y = Drosophila$bm$pc1
  x = factor(Drosophila$bm[,i])
  pval1[i] = logLik(lm(y~x))
}
# loglikelihood plot
xyplot(pval1~pos|chr, data=Drosophila, type=c("p","l"), 
     pch=19, ylab="log likelihood")
</code></pre>


</div>