<div class="container">

<table style="width: 100%;"><tr>
<td>vcf2pcadapt</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>vcfR-based converter</h2>

<h3>Description</h3>

<p><code>vcf2pcadapt</code> uses the package <code>vcfR</code> to extract the genotype information 
from a vcf file and exports it under the format required by <code>pcadapt</code>.
Note that this VCF converter is not maintained anymore. Please use PLINK.
</p>


<h3>Usage</h3>

<pre><code class="language-R">vcf2pcadapt(input, output = "tmp.pcadapt", allele.sep = c("/", "|"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>input</code></td>
<td>
<p>a character string specifying the name of the file to be converted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output</code></td>
<td>
<p>a character string indicating the name of the output file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele.sep</code></td>
<td>
<p>a vector of characters indicating what delimiters are used
to separate alleles.</p>
</td>
</tr>
</table>
</div>