<div class="container">

<table style="width: 100%;"><tr>
<td>plot.pcadapt</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>pcadapt visualization tool</h2>

<h3>Description</h3>

<p><code>plot.pcadapt</code> is a method designed for objects of class <code>pcadapt</code>.
It provides plotting options for quick visualization of <code>pcadapt</code> 
objects. Different options are currently available : <code>"screeplot"</code>, 
<code>"scores"</code>, <code>"stat.distribution"</code>, <code>"manhattan"</code> and 
<code>"qqplot"</code>. <code>"screeplot"</code> shows the decay of the genotype matrix 
singular values and provides a figure to help with the choice of <code>K</code>.
<code>"scores"</code> plots the projection of the individuals onto the first two 
principal components. <code>"stat.distribution"</code> displays the histogram of 
the selected test statistics, as well as the estimated distribution for the 
neutral SNPs. <code>"manhattan"</code> draws the Manhattan plot of the p-values 
associated with the statistic of interest. <code>"qqplot"</code> draws a Q-Q plot 
of the p-values associated with the statistic of interest.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'pcadapt'
plot(
  x,
  ...,
  option = "manhattan",
  i = 1,
  j = 2,
  pop,
  col,
  chr.info = NULL,
  snp.info = NULL,
  plt.pkg = "ggplot",
  K = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of class "pcadapt" generated with <code>pcadapt</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>...</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>option</code></td>
<td>
<p>a character string specifying the figures to be displayed. If 
<code>NULL</code> (the default), all three plots are printed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>i</code></td>
<td>
<p>an integer indicating onto which principal component the individuals
are projected when the "scores" option is chosen.
Default value is set to <code>1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>j</code></td>
<td>
<p>an integer indicating onto which principal component the individuals 
are projected when the "scores" option is chosen.
Default value is set to <code>2</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pop</code></td>
<td>
<p>a list of integers or strings specifying which subpopulation the 
individuals belong to.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>a list of colors to be used in the score plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr.info</code></td>
<td>
<p>a list containing the chromosome information for each marker.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp.info</code></td>
<td>
<p>a list containing the names of all genetic markers present in
the input.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plt.pkg</code></td>
<td>
<p>a character string specifying the package to be used to 
display the graphical outputs. Use <code>"plotly"</code> for interactive plots, or 
<code>"ggplot"</code> for static plots.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K</code></td>
<td>
<p>an integer specifying the principal component of interest. <code>K</code> 
has to be specified only when using the <code>"componentwise"</code> method.</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">## see ?pcadapt for examples

</code></pre>


</div>