<div class="container">

<table style="width: 100%;"><tr>
<td>pcadapt</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Principal Component Analysis for outlier detection</h2>

<h3>Description</h3>

<p><code>pcadapt</code> performs principal component analysis and computes p-values to
test for outliers. The test for outliers is based on the correlations between
genetic variation and the first <code>K</code> principal components. <code>pcadapt</code>
also handles Pool-seq data for which the statistical analysis is performed on
the genetic markers frequencies. Returns an object of class <code>pcadapt</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pcadapt(
  input,
  K = 2,
  method = "mahalanobis",
  min.maf = 0.05,
  ploidy = 2,
  LD.clumping = NULL,
  pca.only = FALSE,
  tol = 1e-04
)

## S3 method for class 'pcadapt_matrix'
pcadapt(
  input,
  K = 2,
  method = c("mahalanobis", "componentwise"),
  min.maf = 0.05,
  ploidy = 2,
  LD.clumping = NULL,
  pca.only = FALSE,
  tol = 1e-04
)

## S3 method for class 'pcadapt_bed'
pcadapt(
  input,
  K = 2,
  method = c("mahalanobis", "componentwise"),
  min.maf = 0.05,
  ploidy = 2,
  LD.clumping = NULL,
  pca.only = FALSE,
  tol = 1e-04
)

## S3 method for class 'pcadapt_pool'
pcadapt(
  input,
  K = (nrow(input) - 1),
  method = "mahalanobis",
  min.maf = 0.05,
  ploidy = NULL,
  LD.clumping = NULL,
  pca.only = FALSE,
  tol
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>input</code></td>
<td>
<p>The output of function <code>read.pcadapt</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K</code></td>
<td>
<p>an integer specifying the number of principal components to retain.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>a character string specifying the method to be used to compute
the p-values. Two statistics are currently available, <code>"mahalanobis"</code>,
and <code>"componentwise"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.maf</code></td>
<td>
<p>Threshold of minor allele frequencies above which p-values are 
computed. Default is <code>0.05</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ploidy</code></td>
<td>
<p>Number of trials, parameter of the binomial distribution. 
Default is 2, which corresponds to diploidy, such as for the human genome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LD.clumping</code></td>
<td>
<p>Default is <code>NULL</code> and doesn't use any SNP thinning.
If you want to use SNP thinning, provide a named list with parameters 
<code>$size</code> and <code>$thr</code> which corresponds respectively to the window 
radius and the squared correlation threshold. A good default value would 
be <code>list(size = 500, thr = 0.1)</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pca.only</code></td>
<td>
<p>a logical value indicating whether PCA results should be 
returned (before computing any statistic).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tol</code></td>
<td>
<p>Convergence criterion of <code>RSpectra::svds()</code>. 
Default is <code>1e-4</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>First, a principal component analysis is performed on the scaled and 
centered genotype data. Depending on the specified <code>method</code>, different 
test  statistics can be used.
</p>
<p><code>mahalanobis</code> (default): the robust Mahalanobis distance is computed for 
each genetic marker using a robust estimate of both mean and covariance 
matrix between the <code>K</code> vectors of z-scores.
</p>
<p><code>communality</code>: the communality statistic measures the proportion of 
variance explained by the first <code>K</code> PCs. Deprecated in version 4.0.0.
</p>
<p><code>componentwise</code>: returns a matrix of z-scores.
</p>
<p>To compute p-values, test statistics (<code>stat</code>) are divided by a genomic 
inflation factor (<code>gif</code>) when <code>method="mahalanobis"</code>. When using 
<code>method="mahalanobis"</code>, the scaled statistics 
(<code>chi2_stat</code>) should follow a chi-squared distribution with <code>K</code> 
degrees of freedom. When using <code>method="componentwise"</code>, the z-scores 
should follow a chi-squared distribution with <code>1</code> degree of freedom. For 
Pool-seq data, <code>pcadapt</code> provides p-values based on the Mahalanobis 
distance for each SNP.
</p>


<h3>Value</h3>

<p>The returned value is an object of class <code>pcadapt</code>.
</p>


</div>