<div class="container">

<table style="width: 100%;"><tr>
<td>methylation</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Methylation values around BLK gene</h2>

<h3>Description</h3>

<p>A dataset containing methylation values for cell-separated samples. The methylation was measured
using bisulfite sequencing. The data also contains the genomic position of these CpG sites, as
well as a binary phenotype (i.e. whether the sample comes from a B cell).
</p>


<h3>Usage</h3>

<pre><code class="language-R">methylation

pheno

position

index

pheno2

position2

methylation2
</code></pre>


<h3>Format</h3>

<p>The data comes in four objects: 
</p>
 
<dl>
<dt>methylation</dt>
<dd>
<p>Matrix of methylation
values at 5,986 sites measured on 40 samples</p>
</dd> 
<dt>pheno</dt>
<dd>
<p>Vector of phenotype, indicating
whether the sample comes from a B cell</p>
</dd> 
<dt>position</dt>
<dd>
<p>Data frame recording the position of
each CpG site along the chromosome</p>
</dd> 
<dt>index</dt>
<dd>
<p>Index vector used in the computation of
PCEV-block</p>
</dd> 
<dt>methylation2</dt>
<dd>
<p>Matrix of methylation values at 1000 sites measured on 40
samples</p>
</dd> 
<dt>pheno2</dt>
<dd>
<p>Vector of phenotype, indicating the cell type of the sample (B cell, T cell, or Monocyte)</p>
</dd> 
<dt>position2</dt>
<dd>
<p>Data frame recording the position of each CpG site along the chromosome</p>
</dd> </dl>
<h3>Details</h3>

<p>Methylation was first measured at 24,068 sites, on 40 samples. Filtering was performed to keep
the 25% most variable sites. See the vignette for more detail.
</p>
<p>A second sample of the methylation dataset was extracted. This second dataset contains
methylation values at 1000 CpG dinucleotides.
</p>


<h3>Source</h3>

<p>Tomi Pastinen, McGill University, Genome Quebec.
</p>


</div>