<div class="container">

<table style="width: 100%;"><tr>
<td>pcRes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>The pc algorithm applied to residuals</h2>

<h3>Description</h3>

<p>The standard pc algorithm applied to GBLUP residuals, or to the GBLUP itself.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pcRes(suffStat, alpha= 0.01, K = NULL, m.max = Inf, verbose = FALSE,
  covariates = NULL, QTLs = integer(), cov.method = "uni",
  use.GBLUP = FALSE, return.pvalues = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>suffStat</code></td>
<td>
<p>A data.frame, of which the first column is the factor G (genotype),  and subsequent columns contain the traits, and optionally some QTLs. The name of the first column should be G. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>The significance level used in the test. Default is 0.01.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K</code></td>
<td>
<p>A genetic relatedness matrix. If <code>NULL</code> (the default), independent genetic effects are assumed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m.max</code></td>
<td>
<p>Maximum size of the conditioning set, in the pc-algorithm on the residuals. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>If <code>TRUE</code>, p-values for the conditional independence tests are printed. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>covariates</code></td>
<td>
<p>A data.frame containing covariates, that should always be used in each conditional independence test. Should be either <code>NULL</code> (default) or a data.frame with the same number of rows as <code>suffStat</code>. An intercept is already included for each trait in <code>suffStat</code>; covariates should not contain a column of ones.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>QTLs</code></td>
<td>
<p>Column numbers in suffStat that correspond to QTLs. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cov.method</code></td>
<td>
<p> A string, specifying which method should be used to compute the GBLUP. Options are <code>'us'</code> (unstructured multi-trait model fitted using sommer) and <code>'uni'</code> (based on univariate GBLUPs).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use.GBLUP</code></td>
<td>
<p>Use the GBLUP itself, instead of the residuals</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return.pvalues</code></td>
<td>
<p>If <code>TRUE</code>, the maximal p-value for each edge is returned.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If <code>use.GBLUP = FALSE</code>, GBLUP residuals are used as input for the pc-stable algorithm of Colombo and Maathuis (2014). This closely resembles the residual networks of Valente et al., (2010) and Topner et al., (2017) (who used different ways to predict the genetic effects, and applied other causal inference algorithms to the residuals). When <code>use.GBLUP = TRUE</code>, pc-stable is applied to the GBLUP itself, which resembles the genomic networks of Topner et al., (2017). If <code>cov.method = "uni"</code>, the GBLUP and the residuals are computed separately for each trait in suffStat. The covariance of each trait is assumed to be
</p>
<p style="text-align: center;"><code class="reqn">\sigma_G^2 Z K Z^t + \sigma_E^2 I_n</code>
</p>

<p>where <code class="reqn">Z</code> is a binary incidence matrix, assigning plants or plots to genotypes. <code class="reqn">Z</code> is based on the first column in suffStat. If there is a single observation per genotype (typically a genotypic mean), <code class="reqn">Z</code> is the identity matrix, and the relatedness matrix <code class="reqn">K</code> should be specified. If there are replicates
for at least some of the genotypes, and no <code class="reqn">K</code> is provided, independent genetic effects are assumed (<code class="reqn">K</code> will be the identity matrix). It is also possible to have replicates and specify a non-diagonal <code class="reqn">K</code>. Whenever  <code class="reqn">K</code> is specified, sommer (mmer2) will be used; otherwise lmer (lme4).
mmer2 is also used when <code>cov.method = "us"</code>, in which case the multivariate GBLUP is computed, for all traits in suffStat simultaneously. This is only possible for a limited number of traits.
</p>


<h3>Value</h3>

<p>If <code>return.pvalues = FALSE</code>, the output is a graph (an object with S3 class <code>"pcgen"</code>). If <code>return.pvalues = TRUE</code>, the output is a list with elements <code>gr</code> (the graph) and <code>pMax</code> (a matrix with the p-values).
</p>


<h3>Author(s)</h3>

<p>Willem Kruijer and Pariya Behrouzi.
Maintainers: Willem Kruijer <a href="mailto:willem.kruijer@wur.nl">willem.kruijer@wur.nl</a> and
Pariya Behrouzi <a href="mailto:pariya.behrouzi@gmail.com">pariya.behrouzi@gmail.com</a>
</p>


<h3>References</h3>

<p>1. Colombo, D. and Maathuis, M.H., 2014. Order-independent constraint-based causal structure learning. <em>The Journal of Machine Learning Research</em>, 15(1), pp.3741-3782. <br>
2. Kruijer, W., Behrouzi, P., Rodriguez-Alvarez, M. X., Wit, E. C., Mahmoudi, S. M., Yandell, B., Van Eeuwijk, F., (2018, in preparation), Reconstruction of networks with direct and indirect genetic effects. <br>
3. Topner, K., Rosa, G.J., Gianola, D. and Schon, C.C., 2017. Bayesian Networks Illustrate Genomic and Residual Trait Connections in Maize (Zea mays L.). <em>G3: Genes, Genomes, Genetics</em>, pp.g3-117. <br>
4. Valente, B.D., Rosa, G.J., Gustavo, A., Gianola, D. and Silva, M.A., 2010. Searching for recursive causal structures in multivariate quantitative genetics mixed models. <em>Genetics</em>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(simdata)
out &lt;- pcRes(suffStat = simdata, alpha = 0.01, verbose= FALSE)

</code></pre>


</div>