<div class="container">

<table style="width: 100%;"><tr>
<td>ChIPreads</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Reads aligned to hg19 from two ChIP-seq experiments
</h2>

<h3>Description</h3>

<p>These data are raw aligned reads which have been mapped to the human
genome, hg19. One is sample ID McGill0004, experiment H3K36me3, chr9,
chunk H3K36me3_AM_immune/8. The other is sample ID McGill0002,
experiment H3K4me3, chr2, chunk H3K4me3_PGP_immune/7.
The chunk ID numbers refer to parts of the McGill benchmark data set,
<a href="https://rcdata.nau.edu/genomic-ml/chip-seq-chunk-db/">https://rcdata.nau.edu/genomic-ml/chip-seq-chunk-db/</a>
</p>


<h3>Usage</h3>

<pre><code class="language-R">data("ChIPreads")</code></pre>


<h3>Format</h3>

<p>A data frame with 40396 observations on the following 4 variables.
</p>

<dl>
<dt><code>experiment</code></dt>
<dd>
<p>either H3K36me3 or H3K4me3</p>
</dd>
<dt><code>chrom</code></dt>
<dd>
<p>either chr9 or chr2</p>
</dd>
<dt><code>chromStart</code></dt>
<dd>
<p>0-based start position of read</p>
</dd>
<dt><code>chromEnd</code></dt>
<dd>
<p>1-based end position of read</p>
</dd>
<dt><code>count</code></dt>
<dd>
<p>number of times a read occured with the given
chromStart/end in this sample/experiment</p>
</dd>
</dl>
<h3>Details</h3>

<p>Peak detection algorithm are typically run on a sequence of
non-negative integer
count data, one data point for each genomic position.
These data are useful for proving that peak detection methods are
robust to different sequences: (1) spatially correlated,
non-independent aligned read coverage; (2) un-correlated, independent
representations such as first or last read.
</p>


</div>