<div class="container">

<table style="width: 100%;"><tr>
<td>chr11ChIPseq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
ChIP-seq aligned read coverage for 4 samples on a subset of chr11
</h2>

<h3>Description</h3>

<p>A ChIP-seq experiment was performed to locate the genomic positions of
a histone (H3K4me3) in 2 B cell samples (McGill0091, McGill0322) and
2 T cell samples (McGill0002, McGill0004). The short sequence reads
(about 100 base pairs each) were aligned to the hg19 reference genome,
and the "coverage" in this data set contains
the total count of aligned reads at each base pair. It also contains
annotated regions determined by an expert who examined scatterplots of
the coverage profiles.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data("chr11ChIPseq")</code></pre>


<h3>Format</h3>

<p>A named list of 2 data.frames: regions contains annotations
about which regions contain or do not contain peaks, and coverage
contains the noisy signal.</p>


<h3>Source</h3>

<p>H3K4me3_TDH_immune chunk 5 in
http://cbio.ensmp.fr/~thocking/chip-seq-chunk-db/
which in turn comes from
http://epigenomesportal.ca/
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(chr11ChIPseq)
library(ggplot2)
ann.colors &lt;-
  c(noPeaks="#f6f4bf",
    peakStart="#ffafaf",
    peakEnd="#ff4c4c",
    peaks="#a445ee")

if(interactive() &amp;&amp; require(ggplot2)){

ggplot()+
  scale_fill_manual("annotation", values=ann.colors,
                    breaks=names(ann.colors))+
  penaltyLearning::geom_tallrect(aes(xmin=chromStart/1e3, xmax=chromEnd/1e3,
                    fill=annotation),
                data=chr11ChIPseq$regions, alpha=1/2)+
  theme_bw()+
  theme(panel.margin=grid::unit(0, "cm"))+
  facet_grid(sample.id ~ ., scales="free")+
  geom_step(aes(chromStart/1e3, count), data=chr11ChIPseq$coverage)+
  xlab("position on chr11 (kilo base pairs)")

}

</code></pre>


</div>