<div class="container">

<table style="width: 100%;"><tr>
<td>writeBedGraph</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Write bedGraph file</h2>

<h3>Description</h3>

<p>Write a data.frame in R to a bedGraph file on disk. This must be a
plain text file with the following four columns: chrom (character
chromosome name), chromStart (integer start position), chromEnd
(integer end position), count (integer aligned read count on chrom
from chromStart+1 to chromEnd); see also
https://genome.ucsc.edu/goldenPath/help/bedgraph.html</p>


<h3>Usage</h3>

<pre><code class="language-R">writeBedGraph(count.df, 
    coverage.bedGraph)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>count.df</code></td>
<td>
<p>data.frame with four columns: chrom, chromStart, chromEnd, count.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coverage.bedGraph</code></td>
<td>
<p>file path where data will be saved in plain text / bedGraph format.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>NULL (same as write.table).</p>


<h3>Author(s)</h3>

<p>Toby Dylan Hocking &lt;toby.hocking@r-project.org&gt; [aut, cre]</p>


<h3>Examples</h3>

<pre><code class="language-R">
library(PeakSegDisk)
data(Mono27ac, envir=environment())
coverage.bedGraph &lt;- file.path(
  tempfile(),
  "H3K27ac-H3K4me3_TDHAM_BP",
  "samples",
  "Mono1_H3K27ac",
  "S001YW_NCMLS",
  "problems",
  "chr11-60000-580000",
  "coverage.bedGraph")
dir.create(
  dirname(coverage.bedGraph),
  recursive=TRUE, showWarnings=FALSE)
writeBedGraph(Mono27ac$coverage, coverage.bedGraph)
fread.first(coverage.bedGraph, col.name.list$coverage)
fread.last(coverage.bedGraph, col.name.list$coverage)

</code></pre>


</div>