<div class="container">

<table style="width: 100%;"><tr>
<td>pedgene</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Compute Kernel and Burden Statistics for Pedigree Data (possibly
with unrelated subjects) 
</h2>

<h3>Description</h3>

<p>Compute linear kernel and burden statistics for gene-level analysis of
data that includes pedigree-related subjects, and possibly unrelated
subjects.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pedgene(ped, geno, map=NULL, male.dose=2,
                    checkpeds=TRUE, verbose.return=FALSE,
                    weights=NULL, weights.beta=c(1,25),
                    weights.mb=FALSE, relation=NULL,
                    method="kounen", acc.davies=1e-5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ped</code></td>
<td>
<p>A data.frame with variables that define the pedigree structure
(typical format used by LINKAGE and PLINK), trait (phenotype), and
optionally a covariate-adjusted trait (for covariate-adjusted gene
level statistics). The columns in the data.frame must be named as
follows:
</p>

<ul>
<li>
<p>famid: pedigree ID, character or numeric allowed
</p>
</li>
<li>
<p>person: person ID, a unique ID within each pedigree, numeric
or character allowed
</p>
</li>
<li>
<p>father: father ID, 0 if no father
</p>
</li>
<li>
<p>mother: mother ID, 0 if no mother
</p>
</li>
<li>
<p>sex: coded as 1 for male, 2 for female
</p>
</li>
<li>
<p>trait: phenotype, either case-control status coded as 1
for affected and 0 for unaffected, or a continuous
value. Subjects with missing (NA) will be removed from the analysis
</p>
</li>
<li>
<p>trait.adjusted: an optional variable for covariate-adjusted
trait. If trait.adjusted is present in the data.frame, then
gene-level tests are adjusted for covariates using
residuals = (trait - trait.adjusted). Otherwise, gene-level tests
are not adjusted for covariates, in which case
residuals = trait - mean(trait) where the mean is taken on all
subjects passed into pedgene before removing subjects who do not
have genotype data.
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>

<p>Data.frame or matrix with genotypes for subjects (rows) at each variant position
(columns). The first two columns are required to be named <code>famid</code> and
<code>person</code>, which are used to match subjects to their data in the
<code>ped</code> data.frame. The genotypes are coded as 0, 1, 2 for autosomal
markers (typically a count of the number of the less-frequent
allele). For X-chromosome markers, females are coded 0, 1, 2, and
males coded 0, 1. Missing genotypes (NA) are allowed.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>

<p>Optional data.frame with columns "chrom" and  "gene", one row per
variant column in geno. The gene name can be any identifier for the
gene. The chromosome can be either numeric or character, where the
calculations will differ between autosomes vs X chromosome (allow
"X"/"x"/23, converted to "X" in results)
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>male.dose</code></td>
<td>
<p>When analyzing the X-chromosome, male.dose defines
how male genotypes should be analyzed. male.dose can be between 0
and 2, but is typically either 1 or 2. Ozbek and Clayton show that
male.dose = 2 is powerful in the presence of X-chromosome dosage
compensation in females.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>checkpeds</code></td>
<td>
<p>logical, if FALSE, the method will skip the pedigree
checking step, which can be intensive for large studies</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose.return</code></td>
<td>
<p>logical, if TRUE, return the pedigree, geno,
and map objects used in the tests after initial cleanup, e.g., the
removal of monomorphic variants. They are returned in the pedgene
object in a list called "save"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights.beta</code></td>
<td>
<p> Weights based on a function of the minor allele
frequency (maf) and the Beta distribution</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights</code></td>
<td>
<p>optional user-specified weights, a vector of weights
for each variant column of geno. If none given, the Beta weights
are applied</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights.mb</code></td>
<td>
<p>logical, if TRUE and no user-given weights, apply the
Madsen-Browning weights per variant: 1/sqrt(maf*(1-maf)). The
hierarchy of weights used is 1) user-specified weights, 2)
Madsen-Browning if weights.mb=TRUE and weights=NULL, 3) Beta density
weights (default if no other weights are set)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>relation</code></td>
<td>

<p>Optional data.frame/matrix with 4 columns (id, person1, person2, code) specifying 
special relationships between pairs of individuals and used by the kinship 
function. Codes are : 
1=Monozygotic twin, 
2=Dizygotic twin, 
3=Twin of unknown zygosity, and
4=Spouse. 
The last is necessary in order to place a marriage with no children into the 
plot. See kinship2::pedigree for details.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>method for calculating the kernel test p-value. Kounen's
saddlepoint approximation (default) is based on the survey package,
and has been found to have less faults (e.g., returned missing
value) than  Davies' method (see Chen et al., 2012). The Davies method, 
which computes an exact p-value for a mixture of chi-square
distributions, is also provided. The accuracy of the Davies method
depends on the numerical accuracy parameter (acc.davies), which can
be difficult to specify ahead of time.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>acc.davies</code></td>
<td>
<p>Numerical accuracy parameter used in the Davies'
method for calculating the kernel test p-value. In some instances, a
p-value from the kernel test is out of range, in which case the
p-value is set to 0 or 1, depending on which direction the p-value
was out of range.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The pedgene function is a wrapper function to call pedgene.stats on
one gene at a time. The pedgene.stats function calculates gene-level
tests for associations with a trait among subjects, accounting for relationships
among subjects based on known pedigree relationships (see Schaid et al). 
This is achieved by the kinship function in the kinship2 package. The kernel
association statistic uses a weighted linear kernel, with default
weights based on the beta distribution and the sample minor allele
frequency. The burden statistic is based on a weighted sum of variants. If
a gene only has one variant, the kernel test reduces to the burden statistic.
Variant positions that have zero variance are removed from the
analysis because they do not contribute information.
</p>
<p>Note that if ped contains extra people that are not necessary to
define relationships of people with genotype data, their trait value will
still be used in mean(trait) in calculating trait.adjusted if trait.adjusted is
not given as a column in ped. 
</p>


<h3>Value</h3>

<p>An object of the pedgene S3 class, with the following elements:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>call: </code></td>
<td>
<p>function call</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pgdf: </code></td>
<td>
<p>data.frame with gene name, chromosome, n-variants per
gene(after removing uncessary variants), n-variants removed per
gene, kernel and burden test statistics and p-values. Kernel
p-values are based on either Kuonen (1999) or Davies (1980) method.
The burden statistic has a standard normal distribution, so the sign
of the burden statistic gives information on the direction of
association (positive value implies large burden score are positively 
associated with larger trait values).  When a gene has only 1 variant, 
the kernel test reduces to the burden test. In this instance, the
kernel statistic (chi-square) is the square of the burden statistic
(standard normal), with both having the same p-value. When a gene
has no markers after removing zero-variance markers, the gene test
stastistics and p-values are all NA. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save: </code></td>
<td>
<p>If verbose.return was set to TRUE, a list containing the
cleaned form of the data that was input to pedgene and is used
in the tests: ped, geno, and map</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Daniel J. Schaid, Jason P. Sinnwell, Mayo Clinic (contact: schaid@mayo.edu).
Alessia Visconti, King's College London (alessia.visconti@kcl.ac.uk).
</p>


<h3>References</h3>

<p>Schaid DJ, McDonnell SK, Sinnwell JP, Thibodeau SN (2013)
Multiple Genetic Variant Association Testing by Collapsing and Kernel
Methods With Pedigree or Population Structured Data. Genetic
Epidemiology, 37(5):409-18.
</p>
<p>Ozbek U (2012) Statistics for X-chromosome association. 62nd Annual
Meeting of The American Society of Human Genetics; Program #22. San
Francisco, California.
</p>
<p>Clayton D (2008) Testing for association on the X chromosome.
Biostatistics 9:593-600
</p>
<p>Chen H, Meigs J, Dupuis J (2012) Sequence kernel association test for
quantitative traits in family samples. Genetic Epidem 37:196-204
</p>
<p>Kounen D (1999) Saddle point approximatinos for distributions of
quadratic forms in normal variables. Biometrika 86:929 -935
</p>
<p>Davies RB (1980) Algorithm AS 155: The Distribution of a Linear
Combination of chi-2 Random Variables, Journal of the Royal
Statistical Society. Series C (Applied Statistics), 29(3):323-33
</p>


<h3>See Also</h3>

<p>pedigreeChecks, example.ped
</p>


<h3>Examples</h3>

<pre><code class="language-R"># example data with the same 10 variants for an autosome and X chromosome
# pedigree data on 39 subjects including 3 families and unrelateds
data(example.ped)
data(example.geno)
data(example.map)

# gene tests (chroms 1 and X) with male.dose=2
pg.m2 &lt;- pedgene(example.ped, example.geno, example.map, male.dose=2)
# same genes, with male.dose=1
pg.m1 &lt;- pedgene(example.ped, example.geno, example.map, male.dose=1)

## print and summary methods
print(pg.m2, digits=3)
summary(pg.m1, digits=3)
</code></pre>


</div>