<div class="container">

<table style="width: 100%;"><tr>
<td>example.geno</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Example datasets for pedgene </h2>

<h3>Description</h3>

<p>example.geno: a data frame with minor allele count for subjects (rows)
at variant positions (columns); example.ped: pedigree and trait data for subjects in
example.geno; example.map: gene and chromosome for variant positions in example.geno;  
example.relation: special (twin) relationships for individuals in example.ped
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(example.geno)
data(example.ped)
data(example.map)
data(example.relation)
</code></pre>


<h3>Format</h3>

<p><code>example.geno</code> -data frame with minor allele count for 20 variant
positions:
</p>

<dl>
<dt><code>ped </code></dt>
<dd>
<p>pedigree ID, character or numeric</p>
</dd>
<dt><code>person </code></dt>
<dd>
<p>person ID, used with ped to match subjects to
their row in example.ped</p>
</dd>
<dt><code>AA.1-AA.10,AX.1-AX.10 </code></dt>
<dd>
<p>genotype columns at 10 positions for
each of 2 simulated genes</p>
</dd>
</dl>
<p><code>example.ped</code> -data frame with pedigree structure and trait
values in the following columns:
</p>

<dl>
<dt><code>ped </code></dt>
<dd>
<p> pedigree ID, character or numeric</p>
</dd>
<dt><code>person </code></dt>
<dd>
<p> person ID, a unique ID within each pedigree</p>
</dd>
<dt><code>father </code></dt>
<dd>
<p>father ID, 0 if no father</p>
</dd>
<dt><code>mother </code></dt>
<dd>
<p>mother ID, 0 if no mother</p>
</dd>
<dt><code>sex </code></dt>
<dd>
<p>coded as 1 for male, 2 for female</p>
</dd>
<dt><code>trait </code></dt>
<dd>
<p>phenotype, either case-control status coded as 1
for affected and 0 for unaffected, or a continuous
value. Subjects with missing (NA) will be removed from the analysis</p>
</dd>
<dt><code>trait.adjusted </code></dt>
<dd>
<p>an optional variable for covariate-adjusted
trait. If trait.adjusted is present in the data.frame, then
gene-level tests are adjusted for covariates using
residuals = (trait - trait.adjusted). Otherwise, gene-level tests
are not adjusted for covariates, in which case residuals = trait - mean(trait)</p>
</dd>
</dl>
<p><code>example.map </code> - data frame with columns for gene name and chromosome:
</p>

<dl>
<dt><code>chrom </code></dt>
<dd>
<p>chromosome code (1-23,X allowed) where the gene is located</p>
</dd>
<dt><code>gene </code></dt>
<dd>
<p>gene identifier</p>
</dd>
</dl>
<p><code>example.relation </code> - matrix specifying special relationships between pairs 
of individuals in the following columns:
</p>

<dl>
<dt><code>ped </code></dt>
<dd>
<p>pedigree ID, character or numeric</p>
</dd>
<dt><code>id1 </code></dt>
<dd>
<p>person1 ID, used with ped to match subjects to
their row in example.ped</p>
</dd>
<dt><code>id2 </code></dt>
<dd>
<p>person2 ID, used with ped to match subjects to
their row in example.ped</p>
</dd>
<dt><code>code </code></dt>
<dd>
<p>any of the following: 1=Monozygotic twin, 2=Dizygotic twin, 
3=Twin of unknown zygosity, and 4=Spouse</p>
</dd>
</dl>
<h3>Source</h3>

<p>Simulated data for testing purposes
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(example.geno)
data(example.ped)
data(example.map)
data(example.relation)
</code></pre>


</div>