<div class="container">

<table style="width: 100%;"><tr>
<td>calcG</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Function to calculate a relationship matrix from marker data
(usually allele count data), G matrix.</h2>

<h3>Description</h3>

<p>Function to calculate a relationship matrix from marker data. Option
to return the inverse of matrix. Inverse calculated using
<code>Matrix</code> package.
</p>


<h3>Usage</h3>

<pre><code class="language-R">calcG(M, data = NULL,solve = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>M</code></td>
<td>
<p>Matrix of marker genotypes, usually the count of one of the
two SNP alleles at each markers (0, 1, or 2).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>Optional logical vector which can tell of which
individuals we have phenotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>solve</code></td>
<td>
<p>Logic, if TRUE then function returns the inverse of the
relationship matrix.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Matrix of class <code>dgeMatrix</code>.</p>


<h3>See Also</h3>

<p><code>SamplePedigree</code>, <code>gblup</code>,
<code>makeAinv</code>,<code>blup</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">example(gblup)
G &lt;- calcG(M)
Ginv &lt;- calcG(M,solve = TRUE)
</code></pre>


</div>