<div class="container">

<table style="width: 100%;"><tr>
<td>oneMarkerDistribution</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Genotype distribution for a single marker</h2>

<h3>Description</h3>

<p>Computes the genotype probability distribution of one or several pedigree
members, possibly conditional on known genotypes for the marker.
</p>


<h3>Usage</h3>

<pre><code class="language-R">oneMarkerDistribution(
  x,
  ids,
  partialmarker,
  loopBreakers = NULL,
  eliminate = 0,
  grid.subset = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A <code>ped</code> object or a list of such.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ids</code></td>
<td>
<p>A vector of ID labels of one or more members of <code>x</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>partialmarker</code></td>
<td>
<p>Either a <code>marker</code> object or the name (or index) of a
marker attached to <code>x</code>. If <code>x</code> has multiple components, only the latter is
allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loopBreakers</code></td>
<td>
<p>(Only relevant if the pedigree has loops). A vector with
ID labels of individuals to be used as loop breakers. If NULL (default)
loop breakers are selected automatically. See <code>breakLoops()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eliminate</code></td>
<td>
<p>Deprecated, not used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grid.subset</code></td>
<td>
<p>(Optional; not relevant for most users.) A numeric matrix
describing a subset of all marker genotype combinations for the <code>ids</code>
individuals. The matrix should have one column for each of the <code>ids</code>
individuals, and one row for each combination: The genotypes are described
in terms of the matrix <code>M = allGenotypes(n)</code>, where <code>n</code> is the number of
alleles for the marker. If the entry in column <code>j</code> is the integer <code>k</code>, this
means that the genotype of individual <code>ids[j]</code> is row <code>k</code> of <code>M</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>A logical.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A named <code>k</code>-dimensional array, where <code>k = length(ids)</code>, with the
joint genotype distribution for the <code>ids</code> individuals. The probabilities
are conditional on the known genotypes and the allele frequencies of
<code>partialmarker</code>.
</p>


<h3>Author(s)</h3>

<p>Magnus Dehli Vigeland
</p>


<h3>See Also</h3>

<p><code>twoMarkerDistribution()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# Trivial example giving Hardy-Weinberg probabilities
s = singleton(id = 1)
m = marker(s, alleles = 1:2) # equifrequent SNP
oneMarkerDistribution(s, ids = 1, partialmarker = m)

# Conditioning on a partial genotype
genotype(m, id = 1) = "1/-"
oneMarkerDistribution(s, ids = 1, partialmarker = m)

# Genotype distribution for a child of heterozygous parents
trio = nuclearPed(father = "fa", mother = "mo", child = "ch")
m1 = marker(trio, fa = "1/2", mo = "1/2")
oneMarkerDistribution(trio, ids = "ch", partialmarker = m1)

# Joint distribution of the parents, given that the child is heterozygous
m2 = marker(trio, ch = "1/2", afreq = c("1" = 0.5, "2" = 0.5))
oneMarkerDistribution(trio, ids = c("fa", "mo"), partialmarker = m2)

# A different example: The genotype distribution of an individual (id = 8)
# whose half cousin (id = 9) is homozygous for a rare allele.
y = halfCousinPed(degree = 1) |&gt;
  addMarker("9" = "a/a", afreq = c(a = 0.01, b = 0.99))

oneMarkerDistribution(y, ids = 8, partialmarker = 1)

</code></pre>


</div>