<div class="container">

<table style="width: 100%;"><tr>
<td>likelihood</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Pedigree likelihood</h2>

<h3>Description</h3>

<p>The <code>likelihood()</code> and <code>likelihood2()</code> functions constitute the heart of
<strong>pedprobr</strong>. The former computes the pedigree likelihood for each indicated
marker. The latter computes the likelihood for a pair of linked markers
separated by a given recombination rate.
</p>


<h3>Usage</h3>

<pre><code class="language-R">likelihood(x, ...)

## S3 method for class 'ped'
likelihood(
  x,
  markers = NULL,
  peelOrder = NULL,
  lump = TRUE,
  eliminate = 0,
  logbase = NULL,
  loopBreakers = NULL,
  allX = NULL,
  verbose = FALSE,
  theta = 0,
  ...
)

## S3 method for class 'list'
likelihood(x, markers = NULL, logbase = NULL, ...)

likelihood2(x, ...)

## S3 method for class 'ped'
likelihood2(
  x,
  marker1,
  marker2,
  rho = NULL,
  peelOrder = NULL,
  eliminate = 0,
  logbase = NULL,
  loopBreakers = NULL,
  verbose = FALSE,
  ...
)

## S3 method for class 'list'
likelihood2(x, marker1, marker2, logbase = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A <code>ped</code> object, a <code>singleton</code> object, or a list of such objects.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Further arguments.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>markers</code></td>
<td>
<p>One or several markers compatible with <code>x</code>. Several input
forms are possible:
</p>

<ul>
<li>
<p> A <code>marker()</code> object compatible with <code>x</code>.
</p>
</li>
<li>
<p> A list of marker objects.
</p>
</li>
<li>
<p> A vector of names or indices of markers attached to <code>x</code>. If <code>x</code> is a
list, this is the only valid input.
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>peelOrder</code></td>
<td>
<p>For internal use.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lump</code></td>
<td>
<p>Activate allele lumping, i.e., merging unobserved alleles. This
is an important time saver, and should be applied in nearly all cases. (The
parameter exists mainly for debugging purposes.) The lumping algorithm will
detect (and complain) if any markers use a non-lumpable mutation model.
Default: TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eliminate</code></td>
<td>
<p>Deprecated, not used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logbase</code></td>
<td>
<p>Either NULL (default) or a positive number indicating the
basis for logarithmic output. Typical values are <code>exp(1)</code> and 10.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loopBreakers</code></td>
<td>
<p>A vector of ID labels indicating loop breakers. If NULL
(default), automatic selection of loop breakers will be performed. See
<code>breakLoops()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allX</code></td>
<td>
<p>For internal use; set to TRUE if all markers are X-chromosomal.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>A logical.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>theta</code></td>
<td>
<p>Theta correction.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>marker1, marker2</code></td>
<td>
<p>Single markers compatible with <code>x</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rho</code></td>
<td>
<p>The recombination rate between <code>marker1</code> and <code>marker2</code>. To make
biological sense <code>rho</code> should be between 0 and 0.5.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The implementation is based on the peeling algorithm of Elston and Stewart
(1971). A variety of situations are covered; see the Examples section for
some demonstrations.
</p>

<ul>
<li>
<p> autosomal and X-linked markers
</p>
</li>
<li>
<p> 1 marker or 2 linked markers
</p>
</li>
<li>
<p> complex inbred pedigrees
</p>
</li>
<li>
<p> markers with mutation models
</p>
</li>
<li>
<p> pedigrees with inbred founders
</p>
</li>
</ul>
<p>For more than two linked markers, see <code>likelihoodMerlin()</code>.
</p>


<h3>Value</h3>

<p>A numeric with the same length as the number of markers indicated by
<code>markers</code>. If <code>logbase</code> is a positive number, the output is
<code>log(likelihood, logbase)</code>.
</p>


<h3>Author(s)</h3>

<p>Magnus Dehli Vigeland
</p>


<h3>References</h3>

<p>Elston and Stewart (1971). <em>A General Model for the Genetic
Analysis of Pedigree Data</em>. <a href="https://doi.org/10.1159/000152448">doi:10.1159/000152448</a>
</p>


<h3>See Also</h3>

<p><code>likelihoodMerlin()</code>, for likelihoods involving more than 2 linked markers.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
### Simple likelihood ###
p = 0.1
q = 1 - p
afr = c("1" = p, "2" = q)

# Singleton
s = singleton() |&gt; addMarker(geno = "1/2", afreq = afr)

stopifnot(all.equal(likelihood(s), 2*p*q))

# Trio
trio = nuclearPed() |&gt;
  addMarker(geno = c("1/1", "1/2", "1/1"), afreq = afr)

stopifnot(all.equal(likelihood(trio), p^2 * 2*p*q * 0.5))


### Example of calculation with inbred founders ###

### Case 1: Trio with inbred father
x = cousinPed(0, child = TRUE)
x = addSon(x, 5)
x = relabel(x, old = 5:7, new = c("father", "mother", "child"))

# Add equifrequent SNP; father homozygous, child heterozygous
x = addMarker(x, father = "1/1", child = "1/2")

# Plot with genotypes
plot(x, marker = 1)

# Compute the likelihood
lik1 = likelihood(x, markers = 1)


### Case 2: Using founder inbreeding
# Remove ancestry of father
y = subset(x, c("father", "mother", "child"))

# Indicate that the father has inbreeding coefficient 1/4
founderInbreeding(y, "father") = 1/4

# Plot (notice the inbreeding coefficient)
plot(y, marker = 1)

# Likelihood should be the same as above
lik2 = likelihood(y, markers = 1)

stopifnot(all.equal(lik1, lik2))



</code></pre>


</div>