<div class="container">

<table style="width: 100%;"><tr>
<td>twoMarkerDistribution</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Genotype distribution for two linked markers</h2>

<h3>Description</h3>

<p>Computes the joint genotype distribution of two markers for a specified
pedigree member, conditional on known genotypes and the recombination rate
between the markers.
</p>


<h3>Usage</h3>

<pre><code class="language-R">twoMarkerDistribution(
  x,
  id,
  partialmarker1,
  partialmarker2,
  rho = NULL,
  loopBreakers = NULL,
  eliminate = 0,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A <code>ped</code> object or a list of such.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p>A single ID label.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>partialmarker1, partialmarker2</code></td>
<td>
<p>Either a <code>marker</code> object, or the name (or
index) of a marker attached to <code>x</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rho</code></td>
<td>
<p>A single numeric in the interval <code style="white-space: pre;">⁠[0, 0.5]⁠</code>: the recombination
fraction between the two markers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loopBreakers</code></td>
<td>
<p>(Only relevant if the pedigree has loops). A vector with
ID labels of individuals to be used as loop breakers. If NULL (default)
loop breakers are selected automatically. See <code>breakLoops()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eliminate</code></td>
<td>
<p>Deprecated, not used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>A logical.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A named matrix giving the joint genotype distribution.
</p>


<h3>Author(s)</h3>

<p>Magnus Dehli Vigeland
</p>


<h3>See Also</h3>

<p><code>oneMarkerDistribution()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# A sib-pair pedigree
x = nuclearPed(children = c("bro1", "bro2"))

# Two SNP markers; first brother homozygous for the `1` allele
SNP1 = SNP2 = marker(x, bro1 = "1/1", afreq = c("1" = 0.5, "2" = 0.5))

plot(x, marker = list(SNP1, SNP2))

# Genotype distribution for the brother depends on linkage
twoMarkerDistribution(x, id = "bro2", SNP1, SNP2, rho = 0)
twoMarkerDistribution(x, id = "bro2", SNP1, SNP2, rho = 0.5)

# X-linked
chrom(SNP1) = chrom(SNP2) = "X"

plot(x, marker = list(SNP1, SNP2))

twoMarkerDistribution(x, id = "bro2", SNP1, SNP2, rho = 0)
twoMarkerDistribution(x, id = "bro2", SNP1, SNP2, rho = 0.5)

</code></pre>


</div>