<div class="container">

<table style="width: 100%;"><tr>
<td>marker_attach</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Attach markers to pedigrees</h2>

<h3>Description</h3>

<p>In many applications it is useful to <em>attach</em> markers to their associated
<code>ped</code> object. In particular for bigger projects with many markers, this makes
it easier to manipulate the dataset as a unit. The function <code>setMarkers()</code>
replaces all existing markers with the supplied ones, while <code>addMarkers()</code>
appends the supplied markers to any existing ones. Note that there is also
the function <code>addMarker()</code>, which creates and attaches a single marker in one
go.
</p>


<h3>Usage</h3>

<pre><code class="language-R">setMarkers(
  x,
  m = NULL,
  alleleMatrix = NULL,
  locusAttributes = NULL,
  missing = 0,
  sep = NULL,
  checkCons = TRUE
)

addMarkers(
  x,
  m = NULL,
  alleleMatrix = NULL,
  locusAttributes = NULL,
  missing = 0,
  sep = NULL,
  checkCons = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A <code>ped</code> object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m</code></td>
<td>
<p>Either a single <code>marker</code> object or a list of <code>marker</code> objects</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleleMatrix</code></td>
<td>
<p>A matrix with <code>pedsize(x)</code> rows, containing the observed
alleles for one or several markers. The matrix must have either 1 or 2
columns per marker. If the former, then a <code>sep</code> string must be a given, and
will be used to split all entries.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>locusAttributes</code></td>
<td>
<p>A list of lists, with attributes for each marker. See
Details for possible attributes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>missing</code></td>
<td>
<p>A single character (or coercible to one) indicating the symbol
for missing alleles.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep</code></td>
<td>
<p>If this is a single string, each entry of <code>alleleMatrix</code> is
interpreted as a genotype, and will be split by calling <code>strsplit(..., split = sep, fixed = TRUE)</code>. If <code>alleleMatrix</code> contains entries with "/",
this will be taken as separator by default. (To override this behaviour,
put <code>sep = FALSE</code>.)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>checkCons</code></td>
<td>
<p>A logical. If TRUE (default), each marker is checked for
consistency with <code>x</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The most general format of <code>locusAttributes</code> a list of lists, one for each
marker, where possible entries in the inner lists are as follows (default
values in parenthesis):
</p>

<ul>
<li> <p><code>alleles</code> : a character vector with allele labels
</p>
</li>
<li> <p><code>afreq</code> :  a numeric vector with allele frequencies (<code>rep.int(1/L, L)</code>,
where <code>L = length(alleles)</code>)
</p>
</li>
<li> <p><code>chrom</code> : chromosome number (NA)
</p>
</li>
<li> <p><code>posMb</code> : physical location in megabases (NA)
</p>
</li>
<li> <p><code>name</code> : marker name (NA)
</p>
</li>
<li> <p><code>mutmod</code> : mutation model, or model name (NULL)
</p>
</li>
<li> <p><code>rate</code> : mutation model parameter (NULL)
</p>
</li>
</ul>
<p>If <code>locusAttributes</code> is a single list of attributes (not a list of lists),
then it is repeated to match the number of markers.
</p>


<h4>Alternative formats of <code>locusAttributes</code>:</h4>


<ul>
<li>
<p> data frame or matrix. In this case an attempt is made to interpret it as a
frequency database in <code style="white-space: pre;">⁠allelic ladder⁠</code> format.
</p>
</li>
<li>
<p> A list of frequency vectors. All vectors should sum to 1, and be named
(with allele labels)
</p>
</li>
<li>
<p> Shortcut for simple SNP data: The argument <code>locusAttributes = "snp-AB"</code>
sets all markers to be equifrequent SNPs with alleles A and B. The letters A
and B may be replaced by other single-character letters or numbers.
</p>
</li>
</ul>
<h3>Value</h3>

<p>A <code>ped</code> object.
</p>


<h3>See Also</h3>

<p><code>addMarker()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">x = singleton(1)
m1 = marker(x, `1` = "1/2")
m2 = marker(x, `1` = "a/b")

# Attach to x
x1 = setMarkers(x, list(m1, m2))

# Reversing the order of the markers
setMarkers(x, list(m2, m1))

# Alternative syntax, adding one marker at a time
x2 = x |&gt;
  addMarker(`1` = "1/2") |&gt;
  addMarker(`1` = "a/b")

stopifnot(identical(x1, x2))

</code></pre>


</div>