<div class="container">

<table style="width: 100%;"><tr>
<td>setSNPs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Attach SNP loci to a pedigree</h2>

<h3>Description</h3>

<p>Create and attach a list of empty SNP markers with specified position and
allele frequencies.
</p>


<h3>Usage</h3>

<pre><code class="language-R">setSNPs(x, snpData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A <code>ped</code> object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snpData</code></td>
<td>
<p>A data frame with 6 columns. See Details.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The data frame <code>snpData</code> should contain the following columns, in order:
</p>

<ul>
<li> <p><code>CHROM</code>: Chromosome (character)
</p>
</li>
<li> <p><code>MARKER</code>: Marker name (character)
</p>
</li>
<li> <p><code>MB</code>: Physical position in megabases (numeric)
</p>
</li>
<li> <p><code>A1</code>: First allele (single-letter character)
</p>
</li>
<li> <p><code>A2</code>: Second allele (single-letter character)
</p>
</li>
<li> <p><code>FREQ1</code>: Allele frequency of <code>A1</code> (number in <code style="white-space: pre;">⁠[0,1]⁠</code>)
</p>
</li>
</ul>
<p>The actual column names do not matter.
</p>
<p>Each column must be of the stated type, or coercible to it. (For example,
<code>CHROM</code>, <code>A1</code> and <code>A2</code> may be given as numbers, but will be internally
converted to characters.)
</p>


<h3>Value</h3>

<p>A copy of <code>x</code> with the indicated SNP markers attached.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
snps = data.frame(
  CHROM  = 1:2,
  MARKER = c("M1", "M2"),
  MB     = c(1.23, 2.34),
  A1     = c("A", "G"),
  A2     = c("C", "C"),
  FREQ1  = c(0.7, 0.12))

x = setSNPs(nuclearPed(), snpData = snps)

# Inspect the results:
getMap(x)
getFreqDatabase(x)

</code></pre>


</div>