<div class="container">

<table style="width: 100%;"><tr>
<td>marker</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Marker objects</h2>

<h3>Description</h3>

<p>Creating a marker object associated with a pedigree. The function <code>marker()</code>
returns a marker object, while <code>addMarker()</code> first creates the marker and
then attaches it to <code>x</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">marker(
  x,
  ...,
  geno = NULL,
  allelematrix = NULL,
  alleles = NULL,
  afreq = NULL,
  chrom = NA,
  posMb = NA,
  name = NA,
  mutmod = NULL,
  rate = NULL,
  NAstrings = c(0, "", NA, "-"),
  validate = TRUE,
  validateMut = validate
)

addMarker(
  x,
  ...,
  geno = NULL,
  allelematrix = NULL,
  alleles = NULL,
  afreq = NULL,
  chrom = NA,
  posMb = NA,
  name = NA,
  mutmod = NULL,
  rate = NULL,
  locusAttr = NULL,
  NAstrings = c(0, "", NA, "-"),
  validate = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A <code>ped</code> object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>One or more expressions of the form <code>id = genotype</code>, where <code>id</code> is
the ID label of a member of <code>x</code>, and <code>genotype</code> is a numeric or character
vector of length 1 or 2 (see Examples).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>A character vector of length <code>pedsize(x)</code>, with genotypes written
in the format "a/b".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allelematrix</code></td>
<td>
<p>A matrix with 2 columns and <code>pedsize(x)</code> rows. If this is
non-NULL, then <code>...</code> must be empty.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleles</code></td>
<td>
<p>A character containing allele names. If not given, and <code>afreq</code>
is named, <code>names(afreq)</code> is used. The default action is to take the sorted
vector of distinct alleles occurring in <code>allelematrix</code>, <code>geno</code> or <code>...</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>afreq</code></td>
<td>
<p>A numeric of the same length as <code>alleles</code>, indicating the
population frequency of each allele. A warning is issued if the frequencies
don't sum to 1 after rounding to 3 decimals. If the vector is named, and
<code>alleles</code> is not NULL, an error is raised if <code>setequal(names(afreq), alleles)</code> is not TRUE. If <code>afreq</code> is not specified, all alleles are given
equal frequencies.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrom</code></td>
<td>
<p>A single integer: the chromosome number. Default: NA.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>posMb</code></td>
<td>
<p>A nonnegative real number: the physical position of the marker,
in megabases. Default: NA.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>name</code></td>
<td>
<p>A character string: the name of the marker. Default: NA.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mutmod, rate</code></td>
<td>
<p>Mutation model parameters to be passed on to
<code>pedmut::mutationModel()</code>; see there for details. Note: <code>mutmod</code>
corresponds to the <code>model</code> parameter. Default: NULL (no mutation model).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>NAstrings</code></td>
<td>
<p>A character vector containing strings to be treated as
missing alleles. Default: <code>c("", "0", NA, "-")</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>validate</code></td>
<td>
<p>A logical indicating if the validity of the marker object
should be checked. Default: TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>validateMut</code></td>
<td>
<p>A logical indicating if the mutation model (if present)
should be checked.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>locusAttr</code></td>
<td>
<p>A list with names <code>alleles</code>, <code>afreq</code>, <code>chrom</code>, <code>name</code>,
<code>posMb</code>, <code>mutmod</code>, <code>rate</code> (or a subset of these). This can be used as an
alternative to entering the arguments as function parameters.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An object of class <code>marker</code>. This is an integer matrix with 2 columns
and one row per individual, and the following attributes:
</p>

<ul>
<li> <p><code>alleles</code> (a character vector with allele labels)
</p>
</li>
<li> <p><code>afreq</code> (allele frequencies; default <code>rep.int(1/length(alleles),   length(alleles))</code>)
</p>
</li>
<li> <p><code>chrom</code> (chromosome number; default = NA)
</p>
</li>
<li> <p><code>posMb</code> (physical location in megabases; default = NA)
</p>
</li>
<li> <p><code>name</code> (marker identifier; default = NA)
</p>
</li>
<li> <p><code>mutmod</code> (a list of two (male and female) mutation matrices; default =
NULL)
</p>
</li>
</ul>
<h3>See Also</h3>

<p>Get/set marker attributes: marker_getattr, marker_setattr.
</p>
<p>Retrieve various marker properties: marker_prop, <code>nMarkers()</code>,
</p>
<p>Add alleles to an existing marker: <code>addAllele()</code>
</p>
<p>Attach multiple markers: marker_attach
</p>


<h3>Examples</h3>

<pre><code class="language-R">x = nuclearPed(father = "fa", mother = "mo", children = "child")

# An empty SNP with alleles "A" and "B"
marker(x, alleles = c("A", "B"))

# Creating and attaching to `x`
addMarker(x, alleles = c("A", "B"))

# Alleles/frequencies can be given jointly or separately
stopifnot(identical(
  marker(x, afreq = c(A = 0.01, B = 0.99)),
  marker(x, alleles = c("A", "B"), afreq = c(0.01, 0.99)),
  ))

# Genotypes can be assigned individually ...
marker(x, fa = "1/1", mo = "1/2")

# ... or using the `geno` vector (all members in order)
marker(x, geno = c("1/1", "1/2", NA))


# Attaching a marker to the pedigree
m = marker(x) # By default a SNP with alleles 1,2
x = setMarkers(x, m)

# A marker with a "proportional" mutation model,
# with different rates for males and females
mutrates = list(female = 0.1, male = 0.2)
marker(x, alleles = 1:2, mutmod = "prop", rate = mutrates)

</code></pre>


</div>