<div class="container">

<table style="width: 100%;"><tr>
<td>marker_setattr</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Set marker attributes</h2>

<h3>Description</h3>

<p>These functions set or modify various attributes of markers attached to a
pedigree. They are sometimes more convenient (and pipe-friendly) than the
in-place modifiers described in marker_inplace.
</p>


<h3>Usage</h3>

<pre><code class="language-R">setGenotype(x, marker = NULL, ids = NULL, geno = NULL, id = NULL)

setAfreq(x, marker, afreq, strict = TRUE)

setAlleleLabels(x, marker, alleles)

setMarkername(x, marker = NULL, name)

setChrom(x, marker = NULL, chrom)

setPosition(x, marker = NULL, posMb)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A <code>ped</code> object or a list of <code>ped</code> objects.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>marker</code></td>
<td>
<p>A vector of indices or names of one or several markers attached
to <code>x</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>A character vector of length <code>pedsize(x)</code>, with genotypes written
in the format "a/b".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id, ids</code></td>
<td>
<p>A vector naming one or several pedigree members, or a function
(e.g., <code>founders()</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>afreq</code></td>
<td>
<p>A numeric of the same length as <code>alleles</code>, indicating the
population frequency of each allele. A warning is issued if the frequencies
don't sum to 1 after rounding to 3 decimals. If the vector is named, and
<code>alleles</code> is not NULL, an error is raised if <code>setequal(names(afreq), alleles)</code> is not TRUE. If <code>afreq</code> is not specified, all alleles are given
equal frequencies.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>strict</code></td>
<td>
<p>A logical. If TRUE (default) the new frequencies cannot remove
or add any alleles.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleles</code></td>
<td>
<p>A character containing allele names. If not given, and <code>afreq</code>
is named, <code>names(afreq)</code> is used. The default action is to take the sorted
vector of distinct alleles occurring in <code>allelematrix</code>, <code>geno</code> or <code>...</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>name</code></td>
<td>
<p>A character of the same length as <code>marker</code>, containing marker
names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrom</code></td>
<td>
<p>A character of the same length as <code>marker</code>, containing
chromosome labels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>posMb</code></td>
<td>
<p>A numeric of the same length as <code>marker</code>, containing the
physical marker positions in megabases (or NA).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A copy of <code>x</code> with modified attributes.
</p>


<h3>Examples</h3>

<pre><code class="language-R">x = nuclearPed() |&gt;
  addMarker(alleles = 1:2) |&gt;
  setMarkername(marker = 1, name = "M") |&gt;
  setGenotype(marker = "M", ids = 1, geno = "1/2") |&gt;
  setAfreq(marker = "M", afreq = c(`1` = 0.1, `2` = 0.9)) |&gt;
  setChrom(marker = "M", chrom = 1) |&gt;
  setPosition(marker = "M", posMb = 123.45)

# Alternatively, all of this could have been done on creation:
y = addMarker(nuclearPed(),
              `1` = "1/2",
              afreq = c(`1` = 0.1, `2` = 0.9),
              name = "M",
              chrom = 1,
              posMb = 123.45)
stopifnot(identical(x, y))

</code></pre>


</div>