<div class="container">

<table style="width: 100%;"><tr>
<td>genome_sim</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>A function to simulate QTL and/or SNP data.</h2>

<h3>Description</h3>

<p>Simulates a chromosome of arbitrary length with arbitrary numbers, types, and spacings of genetic loci over arbitrary pedigrees.
</p>


<h3>Usage</h3>

<pre><code class="language-R">genome_sim(
  pedigree,
  founders = NULL,
  positions = NULL,
  initHe = NULL,
  mutationType = NULL,
  mutationRate = NULL,
  phenotyped = NULL,
  founderHaplotypes = NULL,
  genotyped = NULL,
  returnG = "n",
  initFreqs = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pedigree</code></td>
<td>
<p>A pedigree</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>founders</code></td>
<td>
<p>A vector of indicator variables denoting founder status (1=founder, 0=non-founder)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>positions</code></td>
<td>
<p>Genome locations in cM for markers</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>initHe</code></td>
<td>
<p>Initial levels of expected heterozygosity</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mutationType</code></td>
<td>
<p>A vector of locus types - see details</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mutationRate</code></td>
<td>
<p>A vector of mutation rates</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phenotyped</code></td>
<td>
<p>A vector of IDs of those individuals for which to return phenotypic data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>founderHaplotypes</code></td>
<td>
<p>A matrix or dataframe containing founder haplotypes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotyped</code></td>
<td>
<p>A vector of IDs of those individuals for which to return genotypic data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>returnG</code></td>
<td>
<p>If 'y' then genotypic data for all loci (including <code>cIAM</code> loci) will be returned.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>initFreqs</code></td>
<td>
<p>A list of allele frequencies for all loci.  If <code>initFreqs</code> is specified, it will override information from <code>initHe</code>.  <code>extractA</code> from package <code>MasterBayes</code> can be used to obtain obtain <code>initFreqs</code> form a sample of genotypes.  For <code>cIAM</code> loci, allele names in <code>initFreqs</code> should be allelic substitution effects.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Valid mutation types are <code style="white-space: pre;">⁠Micro', ⁠</code>Dom', <code style="white-space: pre;">⁠dIAM' and ⁠</code>cIAM', for microsatellite, dominant (AFLP), discrete infinite alleles mutation model loci (SNPs), and continuous infinite alleles mutation model loci (polymorphisms effecting phenotypic variation).  cIAM loci have mutational allelic substitution effects taken drawn from a normal distribution with mean 0 and variance 1.
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Phenotypes</code></td>
<td>
<p>A vector of phenotypes.  Calculated as the sum of all allelic effects.  Scaling is currently left to be done post-hoc.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MarkerData</code></td>
<td>
<p>A vector of marker genotypes, i.e. alleles at all loci except those designated ‘cIAM’</p>
</td>
</tr>
</table>
<h3>See Also</h3>

<p><code>phen_sim</code>, <code>micro_sim</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">testData &lt;- as.data.frame(matrix(
  c(
    1,      NA,     NA,     1,      1,      1,      2,      2,
    2,      NA,     NA,     1,      1,      1,      2,      2,
    3,      NA,     NA,     1,      1,      1,      2,      2,
    4,      NA,     NA,     1,      0,      1,      2,      2,
    5,      NA,     NA,     1,      0,      1,      2,      2,
    6,      1,      4,      0,      -1,     2,      3,      3,
    7,      1,      4,      0,      -1,     2,      3,      3,
    8,      1,      4,      0,      -1,     2,      3,      3,
    9,      1,      4,      0,      -1,     2,      3,      3,
    10,     2,      5,      0,      -1,     2,      3,      3,
    11,     2,      5,      0,      -1,     2,      3,      3,
    12,     2,      5,      0,      -1,     2,      3,      3,
    13,     2,      5,      0,      -1,     2,      3,      3,
    14,     3,      5,      0,      -1,     2,      3,      3,
    15,     3,      5,      0,      -1,     2,      3,      3,
    16,     3,      5,      0,      -1,     2,      3,      3,
    17,     3,      5,      0,      -1,     2,      3,      3
  ),
  17, 8,
  byrow = TRUE
))

names(testData) &lt;- c(
  "id", "dam", "sire", "founder", "sex",
  "cohort", "first", "last"
)
pedigree &lt;- as.data.frame(cbind(
  testData$id, testData$dam,
  testData$sire
))
for (x in 1:3) pedigree[, x] &lt;- as.factor(pedigree[, x])
names(pedigree) &lt;- c("id", "dam", "sire")
pedigree

## make up some microsatellite and gene allele frquencies:
sampleGenotypes &lt;- as.data.frame(matrix(c(
  1, 2, -1.32, 0.21, 2, 1, 0.21, 0.21
), 2, 4, byrow = TRUE))
testFreqs &lt;- extractA(sampleGenotypes)

## note that alleles at the gene locus are given as their
## allelic substitution effects:
testFreqs

## simulate data for these indivdiuals based on a single QTL
## with two equally alleles with balanced frequencies in the
## founders, linked (2 cM) to a highly  polymorphic microsatellite:
genome_sim(
  pedigree = pedigree, founders = testData$founder, positions = c(0, 2),
  mutationType = c("Micro", "cIAM"), mutationRate = c(0, 0),
  initFreqs = testFreqs, returnG = "y"
)
## since we specified returnG='y', we can check that
## the phenotypes add up to the
## allelic substitution effects for the second locus.

</code></pre>


</div>