<div class="container">

<table style="width: 100%;"><tr>
<td>makemakePedigreeFactor</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Deprecated functions from package <span class="pkg">pedantics</span>.</h2>

<h3>Description</h3>

<p>The functions listed below are deprecated and are available for
backward compatibility only. They might be defunct in a future version
When possible, alternative functions with similar
functionality are also mentioned.
</p>


<h3>Usage</h3>

<pre><code class="language-R">makemakePedigreeFactor(id, sire, dam, key = NULL)

makePedigreeNumeric(id, sire, dam, missingVal = NA)

drawPedigree(
  Ped,
  cohorts = NULL,
  sex = NULL,
  dat = NULL,
  dots = "n",
  plotfull = "y",
  writeCohortLabels = "n",
  links = "all",
  sexInd = c(0, 1),
  dotSize = 0.001,
  dataDots = "n",
  dataDots.cex = 2,
  cohortLabs.cex = 1,
  retain = "informative",
  focal = NULL,
  sexColours = c("red", "blue"),
  ...
)

fixPedigree(ped, dat = NULL)

genomesim(
  pedigree,
  founders = NULL,
  positions = NULL,
  initHe = NULL,
  mutationType = NULL,
  mutationRate = NULL,
  phenotyped = NULL,
  founderHaplotypes = NULL,
  genotyped = NULL,
  returnG = "n",
  initFreqs = NULL
)

microsim(
  pedigree,
  genFreqs = NULL,
  genotypesSample = NULL,
  knownGenotypes = NULL,
  records = NULL,
  eRate1 = 0,
  eRate2 = 0,
  eRate3 = 0
)

pedigreeStat()

pedStatSummary()

phensim(
  pedigree,
  traits = 1,
  randomA = NULL,
  randomE = NULL,
  parentalA = NULL,
  parentalE = NULL,
  sampled = NULL,
  records = NULL,
  returnAllEffects = FALSE
)
</code></pre>


<h3>Value</h3>

<p>dependant on the deprecated function see pedantics help for details
</p>


<h3><code>makePedigreeFactor</code></h3>

<p>the function is now a special case of the <code>convert_ped</code> function when type is factor
</p>


<h3><code>makePedigreeNumeric</code></h3>

<p>the function is now a special case of the <code>convert_ped</code> function when type is numeric
</p>


<h3><code>drawPedigree</code></h3>

<p>the function has just been renamed with no other changes for the moment, but will soon be replace by "ggpedigree" function
</p>


<h3><code>fixPedigree</code></h3>

<p>the function has just been renamed with no
other changes for the moment
</p>


<h3><code>genomesim</code></h3>

<p>the function has just been renamed with no other changes for the moment
</p>


<h3><code>microsim</code></h3>

<p>the function has just been renamed with no other changes for the moment
</p>


<h3><code>pedigreeStats</code></h3>

<p>the function has been simplified but only the functionality are still available via <code>ped_stats</code> and its summary and plot methods
</p>


<h3><code>pedStatSummary</code></h3>

<p>the function has been simplified but only the functionality are still available via <code>ped_stats</code> and its summary and plot methods
</p>


<h3><code>phensim</code></h3>

<p>the function has just been renamed with no other changes for the moment
</p>


</div>