<div class="container">

<table style="width: 100%;"><tr>
<td>readPed</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read a pedigree from file</h2>

<h3>Description</h3>

<p>Reads a text file in pedigree format, or something fairly close to it.
</p>


<h3>Usage</h3>

<pre><code class="language-R">readPed(
  pedfile,
  colSep = "",
  header = NA,
  famid_col = NA,
  id_col = NA,
  fid_col = NA,
  mid_col = NA,
  sex_col = NA,
  marker_col = NA,
  locusAttributes = NULL,
  missing = 0,
  sep = NULL,
  colSkip = NULL,
  sexCodes = NULL,
  addMissingFounders = FALSE,
  validate = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pedfile</code></td>
<td>
<p>A file name</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colSep</code></td>
<td>
<p>A column separator character, passed on as the <code>sep</code> argument
of <code>read.table()</code>. The default is to separate on white space, that is, one
or more spaces, tabs, newlines or carriage returns. (Note: the parameter
<code>sep</code> is used to indicate allele separation in genotypes.)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>header</code></td>
<td>
<p>A logical. If NA, the program will interpret the first line as
a header line it contains both "id" and "sex" as part of some entries
(ignoring case).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>famid_col</code></td>
<td>
<p>Index of family ID column. If NA, the program looks for a
column named "famid" (ignoring case).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id_col</code></td>
<td>
<p>Index of individual ID column. If NA, the program looks for a
column named "id" (ignoring case).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fid_col</code></td>
<td>
<p>Index of father ID column. If NA, the program looks for a
column named "fid" (ignoring case).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mid_col</code></td>
<td>
<p>Index of mother ID column. If NA, the program looks for a
column named "mid" (ignoring case).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sex_col</code></td>
<td>
<p>Index of column with gender codes (0 = unknown; 1 = male; 2 =
female). If NA, the program looks for a column named "sex" (ignoring case).
If this is not found, genders of parents are deduced from the data, leaving
the remaining as unknown.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>marker_col</code></td>
<td>
<p>Index vector indicating columns with marker alleles. If NA,
all columns to the right of all pedigree columns are used. If <code>sep</code> (see
below) is non-NULL, each column is interpreted as a genotype column and
split into separate alleles with <code>strsplit(..., split = sep, fixed = TRUE)</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>locusAttributes</code></td>
<td>
<p>Passed on to <code>setMarkers()</code> (see explanation there).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>missing</code></td>
<td>
<p>Passed on to <code>setMarkers()</code> (see explanation there).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep</code></td>
<td>
<p>Passed on to <code>setMarkers()</code> (see explanation there).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colSkip</code></td>
<td>
<p>Columns to skip, given as a vector of indices or columns
names. If given, these columns are removed directly after <code>read.table()</code>,
before any other processing.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sexCodes</code></td>
<td>
<p>A list with optional entries "male", "female" and "unknown",
indicating how non-default entries in the <code>sex</code> column should be
interpreted. Default values: male = 1, female = 2, unknown = 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>addMissingFounders</code></td>
<td>
<p>A logical. If TRUE, any parent not included in the
<code>id</code> column is added as a founder of corresponding sex. By default, missing
founders result in an error.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>validate</code></td>
<td>
<p>A logical indicating if the pedigree structure should be
validated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Further parameters passed on to <code>read.table()</code>, e.g.
<code>comment.char</code> and <code>quote</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If there are no headers, and no column information is provided by the user,
the program assumes the following column order:
</p>

<ul>
<li>
<p> family ID (optional; guessed from the data)
</p>
</li>
<li>
<p> individual ID
</p>
</li>
<li>
<p> father's ID
</p>
</li>
<li>
<p> mother's ID
</p>
</li>
<li>
<p> sex
</p>
</li>
<li>
<p> marker data (remaining columns)
</p>
</li>
</ul>
<h4>Reading SNP data</h4>

<p>Adding the argument <code>locusAttributes = "snp-AB"</code>, sets all markers to be
equifrequent SNPs with alleles A and B. Moreover, the letters A and B may be
replaced by other single-character letters or numbers, e.g., "snp-12" gives
alleles 1 and 2.
</p>



<h3>Value</h3>

<p>A ped object or a list of such.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
tf = tempfile()

### Write and read a trio
trio = data.frame(id = 1:3, fid = c(0,0,1), mid = c(0,0,2), sex = c(1,2,1))
write.table(trio, file = tf, row.names = FALSE)
readPed(tf)

# With marker data in one column
trio.marker = cbind(trio, M = c("1/1", "2/2", "1/2"))
write.table(trio.marker, file = tf, row.names = FALSE)
readPed(tf)

# With marker data in two allele columns
trio.marker2 = cbind(trio, M.1 = c(1,2,1), M.2 = c(1,2,2))
write.table(trio.marker2, file = tf, row.names = FALSE)
readPed(tf)

### Two singletons in the same file
singles = data.frame(id = c("S1", "S2"),
                     fid = c(0,0), mid = c(0,0), sex = c(2,1),
                     M = c("9/14.2", "9/9"))
write.table(singles, file = tf, row.names = FALSE)
readPed(tf)

### Two trios in the same file
trio2 = cbind(famid = rep(c("trio1", "trio2"), each = 3), rbind(trio, trio))

# With column names
write.table(trio2, file = tf, col.names = TRUE, row.names = FALSE)
readPed(tf)

# Without column names
write.table(trio2, file = tf, col.names = FALSE, row.names = FALSE)
readPed(tf)

### With non-standard `sex` codes
trio3 = data.frame(id = 1:3, fid = c(0,0,1), mid = c(0,0,2),
                   sex = c("male","female","?"))
write.table(trio3, file = tf, row.names = FALSE)
readPed(tf, sexCodes = list(male = "male", female = "female", unknown = "?"))

# Cleanup
unlink(tf)

</code></pre>


</div>