<div class="container">

<table style="width: 100%;"><tr>
<td>theta.h</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Population Parameter THETA using Homozygosity</h2>

<h3>Description</h3>

<p>This function computes the population parameter THETA using the
homozygosity (or mean heterozygosity) from gene frequencies.
</p>


<h3>Usage</h3>

<pre><code class="language-R">theta.h(x, standard.error = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a vector or a factor.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>standard.error</code></td>
<td>
<p>a logical indicating whether the standard error
of the estimated theta should be returned (<code>TRUE</code>), the default
being <code>FALSE</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The argument <code>x</code> can be either a factor or a vector. If it is a
factor, then it is taken to give the individual alleles in the
population. If it is a numeric vector, then its values are taken to be
the numbers of each allele in the population. If it is a non-numeric
vector, it is a coerced as a factor.
</p>
<p>The standard error is computed with an approximation due to
Chakraborty and Weiss (1991).
</p>


<h3>Value</h3>

<p>A numeric vector of length one with the estimated theta (the default),
or of length two if the standard error is returned
(<code>standard.error = TRUE</code>).
</p>


<h3>Author(s)</h3>

<p>Emmanuel Paradis</p>


<h3>References</h3>

<p>Zouros, E. (1979) Mutation rates, population sizes and amounts of
electrophoretic variation at enzyme loci in natural
populations. <em>Genetics</em>, <b>92</b>, 623–646.
</p>
<p>Chakraborty, R. and Weiss, K. M. (1991) Genetic variation of the
mitochondrial DNA genome in American Indians is at mutation-drift
equilibrium. <em>American Journal of Physical Anthropology</em>,
<b>86</b>, 497–506.
</p>


<h3>See Also</h3>

<p><code>heterozygosity</code>, <code>theta.s</code>,
<code>theta.k</code>, <code>theta.tree</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(jaguar)
## compute frequencies:
S &lt;- summary(jaguar)
## compute THETA for all loci:
sapply(S, function(x) theta.h(x$allele))
</code></pre>


</div>