<div class="container">

<table style="width: 100%;"><tr>
<td>dist.asd</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Allelic Sharing Distance</h2>

<h3>Description</h3>

<p>This function computes the allelic sharing distance (ASD) for diploid
genotypes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">dist.asd(x, scaled = TRUE, pairwise.deletion = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of class <code>"loci"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scaled</code></td>
<td>
<p>a logical value specifying whether the distances should
be scaled by the number of loci.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pairwise.deletion</code></td>
<td>
<p>a logical value: whether to check for missing
values for each pairwise comparison (see details).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The ASD between two diploid genotypes is (Gao and Martin, 2009):
</p>
<p style="text-align: center;"><code class="reqn">\frac{1}{L}\sum_{j=1}^L d_j</code>
</p>

<p>where <code class="reqn">L</code> is the number loci, <code class="reqn">d_j</code> is the value for the
<code class="reqn">j</code>th locus: 0 if both genotypes are identical, 1 if they have one
allele in common, or 2 if they have no allele in common.
</p>
<p><code>dist.asd</code> works for all diploid genotypes (phased or unphased,
with two alleles or more). Note that the required conditions are not
checked by the present function: see the functions below.
</p>
<p>The pairwise deletion is done with respect to missing values coded as
<code>NA</code>, not on the ‘null alleles’ (‘0’ or ‘.’). You
may need to use the function <code>nullAlleles2NA</code> first if
your data has genotypes with null alleles that you want to treat as
missing values.
</p>


<h3>Value</h3>

<p>an object of class <code>"dist"</code>.</p>


<h3>Author(s)</h3>

<p>Emmanuel Paradis</p>


<h3>References</h3>

<p>Gao, X. and Martin, E. R. (2009) Using allele sharing distance for
detecting human population stratification. <em>Human Hederity</em>,
<b>68</b>, 182–191.
</p>


<h3>See Also</h3>

<p><code>is.snp</code>, <code>is.phased</code>, <code>getPloidy</code>,
<code>nullAlleles2NA</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(jaguar)
## ASD for micro-satellites:
d &lt;- dist.asd(jaguar)
co &lt;- rainbow(nlevels(jaguar$pop))
plot(nj(d), "u", tip.color = co[jaguar$pop], font = 2, lab4 = "a")
legend("topleft", legend = levels(jaguar$pop), text.col = co, text.font = 2)
</code></pre>


</div>