<div class="container">

<table style="width: 100%;"><tr>
<td>utilities</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Utily Functions for pegas</h2>

<h3>Description</h3>

<p>The first three functions extract information on loci,
<code>expand.genotype</code> creates a table of all possible genotypes given
a set of alleles, <code>proba.genotype</code> calculates expected
probabilities of genotypes under Hardyâ€“Weinberg equilibrium,
<code>is.snp</code> tests whether a locus is a SNP, <code>is.phased</code> tests
whether a gentotype is phased, and <code>unphase</code> unphase phased
genotypes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getPloidy(x)
getAlleles(x)
getGenotypes(x)
expand.genotype(n, alleles = NULL, ploidy = 2, matrix = FALSE)
proba.genotype(alleles = c("1", "2"), p, ploidy = 2)
is.snp(x)
## S3 method for class 'loci'
is.snp(x)
is.phased(x)
unphase(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of class <code>"loci"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>an integer giving how many alleles to consider (ignored if
<code>alleles</code> is used).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleles</code></td>
<td>
<p>the allele names as a vector of mode character.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ploidy</code></td>
<td>
<p>an integer giving the ploidy level (either 2 or 4 for
the moment).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>matrix</code></td>
<td>
<p>a logical specifying whether to return the genotypes in
a matrix or as a character vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>
<p>a vector of allele probabilities; if missing, equal
probabilities are assumed.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>expand.genotype</code> and <code>proba.genotype</code> accept any level of
ploidy and any number of alleles.
</p>
<p>For <code>is.snp</code>, a locus is defined as a SNP if it has two alleles
and their labels are made of a single character (e.g., A and T, or 1
and 2, but not A and AT).
</p>


<h3>Value</h3>

<p><code>getPloidy</code> returns the ploidy level of all genotypes as a matrix
of integers with rownames and colnames taken from <code>x</code>.
</p>
<p><code>getAlleles</code> and <code>getGenotypes</code> return the alleles and
genotypes, respectively, observed in all loci in an object of class
<code>"loci"</code> as a list.
</p>
<p><code>expand.genotype</code> returns a character vector (the default) or a
matrix where the rows are the genotypes and the columns are the
alleles. The matrix is numeric by default, or character if the
argument <code>alleles</code> is given.
</p>
<p><code>proba.genotype</code> returns a numeric vector with names set as the
genotypes.
</p>
<p><code>is.snp</code> returns a logical vector specifying whether each locus
is a SNP.
</p>
<p><code>is.phased</code> returns a matrix of the same size than the original
data specifying whether each genotype is phased or not.
</p>
<p><code>unphase</code> unphases the genotypes and eventually pools those that
become identical once unphased (e.g., A|T and T|A).
</p>


<h3>Author(s)</h3>

<p>Emmanuel Paradis</p>


<h3>Examples</h3>

<pre><code class="language-R">data(jaguar)
X &lt;- jaguar[, 1:2]
getAlleles(X)
getGenotypes(X)
expand.genotype(2)
expand.genotype(2, LETTERS[1:3])
expand.genotype(3, ploidy = 4)
proba.genotype() # classical HWE with 2 alleles
## an octoploid with a six-allele locus (1287 possible genotypes):
length(p &lt;- proba.genotype(alleles = LETTERS[1:6], ploidy = 8))
max(p) # ~ 0.006
## back to the jaguar data:
s &lt;- summary(X)
## allele counts from the first locus:
p &lt;- s[[1]]$allele
## expected probabilities for the 136 possible genotypes...
proba.genotype(names(p), p/sum(p))
## ... to be compared with s[[1]]$genotype
</code></pre>


</div>