<div class="container">

<table style="width: 100%;"><tr>
<td>LDscan</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Multi-Locus Linkage Disequilibrium</h2>

<h3>Description</h3>

<p><code>LDscan</code> computes a matrix of pairwise linkage disequilibrium (LD)
coefficients (<code class="reqn">|r|</code>) from a set of loci (which must be bi-allelic;
if not, the results are not guaranteed to be meaningful). The
genotypes must be phased.
</p>
<p><code>LDmap</code> plots a matrix of LD coefficients, optionally with the
positions of the loci.
</p>


<h3>Usage</h3>

<pre><code class="language-R">LDscan(x, ...)

## S3 method for class 'DNAbin'
LDscan(x, quiet = FALSE, what = c("r", "Dprime"), ...)
## S3 method for class 'loci'
LDscan(x, depth = NULL, quiet = FALSE,
        what = c("r", "Dprime"), ...)

LDmap(d, POS = NULL, breaks = NULL, col = NULL, border = NA,
      angle = 0, asp = 1, cex = 1, scale.legend = 0.8, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of class <code>"loci"</code> with phased genotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>depth</code></td>
<td>
<p>a vector of integers giving the the depth(s) (or lags) at
which the <code class="reqn">r</code>'s are calculated. By default, all possible
depths are considered.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>a logical: should the progress of the operation be
printed?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>what</code></td>
<td>
<p>the quantity to be computed. Two choices are possible:
<code>"r"</code> (the default) for the absolute value of the correlation
between alleles and <code>"Dprime"</code> for the (scaled) coefficients.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d</code></td>
<td>
<p>a correlation matrix (can be an object of class <code>"dist"</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>POS</code></td>
<td>
<p>an optional vector of locus positions (e.g., from a VCF
file; see examples).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>breaks</code></td>
<td>
<p>a vector of break intervals to count the values in
<code>d</code>; by default, ten equally-sized intervals are used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>an optional vector of colours; a scale from lightyellow to
red is used by default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>border</code></td>
<td>
<p>the border of the rectangles: the default is to have no
border (this is not the same than default in
<code>rect</code>; see examples).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>angle</code></td>
<td>
<p>value (in degrees) to rotate the graphic.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>asp</code></td>
<td>
<p>the aspect ratio of the graphic; one by default so the
elements are squares (not rectangles).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex</code></td>
<td>
<p>the scaling of the labels and text.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale.legend</code></td>
<td>
<p>the scaling of the legend rectangles.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments passed to methods (<code>LDscan</code>) or to
<code>plot.default</code> (<code>LDmap</code>).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The LD coefficient <code class="reqn">r</code> is well defined when the two loci have
only two alleles. In other cases, LD is well defined (see
<code>LD</code>) but the definition of <code class="reqn">r</code> is not clear.
</p>
<p>All levels of ploidy are accepted, but all loci should have the same
ploidy level.
</p>
<p>If <code>depth</code> is used, the <code class="reqn">r</code>'s are calculated only for the
pairs of loci that are distant by these values in <code>x</code>, but
necessarily on the chromosome. The returned list has names set with
the values of <code>depth</code>.
</p>
<p>The value returned is actually <code class="reqn">|r|</code> (not <code class="reqn">r^2</code>).
</p>


<h3>Value</h3>

<p><code>LDscan</code> returns an object of class <code>"dist"</code> by default, or
a list if <code>depth</code> is used.
</p>


<h3>Author(s)</h3>

<p>Emmanuel Paradis</p>


<h3>See Also</h3>

<p><code>LD</code>, <code>read.vcf</code></p>


<h3>Examples</h3>

<pre><code class="language-R">data(woodmouse)
d &lt;- LDscan(woodmouse)
LDmap(d, seg.sites(woodmouse), seq(0, 1, .1))

## Not run: 
## Download the VCF file from Dryad:
## https://doi.org/10.5061/dryad.446sv.2

## the VCF file should have this name:
fl &lt;- "global.pop.GATK.SNP.hard.filters.V3.phased_all.pop.maf.05.recode.vcf.gz"

info.fly &lt;- VCFloci(fl)

## LD map from the first 100 loci:
x &lt;- read.vcf(fl, to = 100) # read only 100 loci
res &lt;- LDscan(x)
bks &lt;- seq(0, 1, 0.2)
LDmap(res, info.fly$POS[1:100], bks, scale.legend = 3)

## check the chromosomes:
table(info.fly$CHROM)

## LD map from 100 loci randomly distributed on the chromosome:
s &lt;- ceiling(seq(1, 224253, length.out = 100))
xs &lt;- read.vcf(fl, which.loci = s)
res2 &lt;- LDscan(xs)
LDmap(res2, info.fly$POS[s], bks, scale.legend = 3)

## something simpler with 10 loci:
x10 &lt;- x[, 1:10]
## the VCF file has no locus IDs, so we give some here:
names(x10) &lt;- paste0("Loc", 1:10)
res10 &lt;- LDscan(x10, quiet = TRUE)
LDmap(res10, angle = 45, border = NULL)

## End(Not run)
</code></pre>


</div>