<div class="container">

<table style="width: 100%;"><tr>
<td>LD</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Linkage Disequilibrium</h2>

<h3>Description</h3>

<p>These two functions analyse linkage disequilibrium in the case of
phased (<code>LD</code>) or unphased (<code>LD2</code>) genotypes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">LD(x, locus = c(1, 2), details = TRUE)
LD2(x, locus = c(1, 2), details = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of class <code>"loci"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>locus</code></td>
<td>
<p>a vector of two integers giving the loci to analyse.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>details</code></td>
<td>
<p>a logical value indicating whether to print the
correlation matrix among alleles.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>These functions consider a pair of loci and compute the correlations
among pairs of alleles.
</p>
<p><code>LD</code> first scans the data for unphased genotypes: all individuals
with at least one unphased genotype are dropped with a warning. It is
based on the observed frequencies of haplotypes (Zaykin et
al. 2008). <code>LD2</code> is based on the observed frequencies of
different genotypes (Schaid 2004).
</p>
<p>Both functions accept any number of alleles. <code>LD</code> can work with
any level of ploidy; <code>LD2</code> works with diploid data.
</p>
<p>The present version does not test the significance of the <code class="reqn">T_2</code>
test (Zaykin et al. 2008) with permutations. These authors present
simulation results suggesting that the chi-squared approximation has
similar type I error rates and power than the test based on
permutations even for small sample sizes. Furthermore, this test has
better statistical properties than alternatives such as those reported
here (LRT and Pearson's test).
</p>


<h3>Value</h3>

<p>For both functions, if <code>details = FALSE</code>, only the T2 test is
returned.
</p>
<p>For <code>LD</code>: if <code>details = TRUE</code>, a named list with the
following elements:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>Observed frequencies</code></td>
<td>
<p>the counts of haplotypes in the data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Expected frequencies</code></td>
<td>
<p>the expected frequencies of haplotypes
computed from the observed proportions of alleles under the
assumption of no linkage disequilibrium.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Correlations among alleles</code></td>
<td>
<p>the observed correlations among
alleles from both loci.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LRT (G-squared)</code></td>
<td>
<p>the likelihood-ratio test of the null
hypothesis of no linkage disequilibrium.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Pearson's test (chi-squared)</code></td>
<td>
<p>the chi-squared test based on
haplotypes counts.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>T2</code></td>
<td>
<p>the <code class="reqn">T_2</code> test with its number of degrees of freedom
(df).</p>
</td>
</tr>
</table>
<p>For <code>LD2</code>: if <code>details = TRUE</code>, a named list with two
elements:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>Delta</code></td>
<td>
<p>the correlations among alleles (denoted <code class="reqn">Delta</code> in
Schaid 2004).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>T2</code></td>
<td>
<p>the <code class="reqn">T_2</code> test with its number of degrees of freedom
(df).</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Emmanuel Paradis</p>


<h3>References</h3>

<p>Schaid, D. J. (2004) Linkage disequilibrium testing when linkage phase
is unknown. <em>Genetics</em>, <b>166</b>, 505–512.
</p>
<p>Zaykin, D. V., Pudovkin, A. and  Weir, B. S. (2008) Correlation-based
inference for linkage disequilibrium with multiple alleles.
<em>Genetics</em>, <b>180</b>, 533–545.
</p>


<h3>See Also</h3>

<p><code>haplotype.loci</code>, <code>is.phased</code>,
<code>LDscan</code></p>


<h3>Examples</h3>

<pre><code class="language-R">data(jaguar)
LD2(jaguar, details = FALSE)
LD2(jaguar, locus = 8:9, details = FALSE)
</code></pre>


</div>