<div class="container">

<table style="width: 100%;"><tr>
<td>VCFloci</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Information From VCF Files</h2>

<h3>Description</h3>

<p>These functions help to extract information from VCF files and to
select which loci to read with <code>read.vcf</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">VCFloci(file, what = "all", chunk.size = 1e9, quiet = FALSE)
## S3 method for class 'VCFinfo'
print(x, ...)
VCFheader(file)
VCFlabels(file)
## S3 method for class 'VCFinfo'
is.snp(x)
rangePOS(x, from, to)
selectQUAL(x, threshold = 20)
getINFO(x, what = "DP", as.is = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>file name of the VCF file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>what</code></td>
<td>
<p>a character specifying the information to be extracted
(see details).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chunk.size</code></td>
<td>
<p>the size of data in bytes read at once.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>a logical: should the progress of the operation be printed?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of class <code>"VCFinfo"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>from, to</code></td>
<td>
<p>integer values giving the range of position values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>a numerical value indicating the minimum value of
quality for selecting loci.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>as.is</code></td>
<td>
<p>a logical. By default, <code>getINFO</code> tries to convert
its output as numeric: if too many NA's are produced, the output is
returned as character. Use <code>as.is = TRUE</code> to force the output
to be in character mode.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments passed to and from other methods.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The variant call format (VCF) is described in details in the
References. Roughly, a VCF file is made of two parts: the header and
the genotypes. The last line of the header gives the labels of the
genotypes: the first nine columns give information for each locus and
are (always) "CHROM", "POS", "ID", "REF", "ALT", "QUAL", "FILTER",
"INFO", and "FORMAT". The subsequent columns give the labels
(identifiers) of the individuals; these may be missing if the file
records only the variants. Note that the data are arranged as the
transpose of the usual way: the individuals are as columns and the
loci are as rows.
</p>
<p><code>VCFloci</code> is the main function documented here: it reads the
information relative to each locus. The option <code>what</code> specifies
which column(s) to read. By default, all of them are read. If the user
is interested in only the locus positions, the option <code>what =
  "POS"</code> would be used.
</p>
<p>Since VCF files can be very big, the data are read in portions of
<code>chunk.size</code> bytes. The default (1 Gb) should be appropriate in
most situations. This value should not exceed 2e9.
</p>
<p><code>VCFheader</code> returns the header of the VCF file (excluding the
line of labels). <code>VCFlabels</code> returns the individual labels.
</p>
<p>The output of <code>VCFloci</code> is a data frame with as many rows as
there are loci in the VCF file and storing the requested
information. The other functions help to extract specific information
from this data frame: their outputs may then be used to select which
loci to read with <code>read.vcf</code>.
</p>
<p><code>is.snp</code> tests whether each locus is a SNP (i.e., the reference
allele, REF, is a single charater and the alternative allele, ALT,
also). It returns a logical vector with as many values as there are
loci. Note that some VCF files have the information VT (variant type)
in the INFO column.
</p>
<p><code>rangePOS</code> and <code>selectQUAL</code> select some loci with respect to
values of position or quality. They return the indices (i.e., row
numbers) of the loci satisfying the conditions.
</p>
<p><code>getINFO</code> extracts a specific information from the INFO
column. By default, these are the total depths (DP) which can be
changed with the option <code>what</code>. The meaning of these information
should be described in the header of the VCF file.
</p>


<h3>Value</h3>

<p><code>VCFloci</code> returns an object of class <code>"VCFinfo"</code> which is a
data frame with a specific print method.
</p>
<p><code>VCFheader</code> returns a single character string which can be
printed nicely with <code>cat</code>.
</p>
<p><code>VCFlabels</code> returns a vector of mode character.
</p>
<p><code>is.snp</code> returns a vector of mode logical.
</p>
<p><code>rangePOS</code> and <code>selectQUAL</code> return a vector of mode
numeric.
</p>
<p><code>getINFO</code> returns a vector of mode character or numeric (see above).
</p>


<h3>Note</h3>

<p><code>VCFloci</code> is able to read either compressed (*.gz) or
uncompressed files.
</p>


<h3>Author(s)</h3>

<p>Emmanuel Paradis</p>


<h3>References</h3>

<p><a href="https://www.internationalgenome.org/wiki/Analysis/vcf4.0">https://www.internationalgenome.org/wiki/Analysis/vcf4.0</a>
</p>
<p><a href="https://github.com/samtools/hts-specs">https://github.com/samtools/hts-specs</a>
</p>


<h3>See Also</h3>

<p><code>read.vcf</code></p>


<h3>Examples</h3>

<pre><code class="language-R">## see ?read.vcf
</code></pre>


</div>