<div class="container">

<table style="width: 100%;"><tr>
<td>haplotype.loci</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Haplotype Extraction and Frequencies From Allelic Data</h2>

<h3>Description</h3>

<p>This function extracts haplotypes from phased genotypes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'loci'
haplotype(x, locus = 1:2, quiet = FALSE, compress = TRUE,
           check.phase = TRUE, ...)
## S3 method for class 'haplotype.loci'
plot(x, ...)
dist.haplotype.loci(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of class <code>"loci"</code> or of class
<code>"haplotype.loci"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>locus</code></td>
<td>
<p>a vector of integers giving the loci to analyse.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>a logical value specifying whether to not print the
progress of the analysis (<code>FALSE</code> by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>compress</code></td>
<td>
<p>by default only the unique haplotypes are returned
with their frequencies. If <code>compress = FALSE</code>, a matrix with
all observed haplotypes is returned (with the number of columns
equals to the number of individuals times the ploidy level).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check.phase</code></td>
<td>
<p>a logical value specifying whether to check if the
individual genotypes are phased.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>arguments passed to and from methods.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The individuals with at least one unphased genotype are ignored with a
warning.
</p>
<p><code>dist.haplotype.loci</code> computes pairwise distances among
haplotypes by counting the number of different alleles.
</p>
<p>Checking whether the genotypes are phased can be time consuming with
very big data sets. It may be useful to set <code>check.phase = FALSE</code>
if several analyses are done on the same data and no warning was
issued after the first scan, or you are sure that the genotypes are phased.
</p>


<h3>Value</h3>

<p><code>haplotype</code> returns a matrix of mode character with the loci as
rows and the haplotypes as columns. The attribute <code>"freq"</code> gives
the counts of each haplotype and the class is <code>"haplotype.loci"</code>.
</p>
<p><code>dist.haplotype.loci</code> returns an object of class <code>"dist"</code>.
</p>


<h3>Note</h3>

<p><code>haplotype</code> is a generic function with methods for objects of
class <code>"DNAbin"</code> and of class <code>"loci"</code>. Note that the class
returned by these methods is different: <code>c("haplotype", "DNAbin")</code>
and <code>"haplotype.loci"</code>, respectively. This and other details are
likely to change in the future.
</p>


<h3>Author(s)</h3>

<p>Emmanuel Paradis</p>


<h3>See Also</h3>

 <p><code>haplotype</code>, <code>LD</code></p>


</div>