<div class="container">

<table style="width: 100%;"><tr>
<td>Fst</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>F-Statistics</h2>

<h3>Description</h3>

<p><code>Fst</code> computes the <code class="reqn">F_{IT}</code>, <code class="reqn">F_{ST}</code> and
<code class="reqn">F_{IS}</code> for each locus in the data. <code>Rst</code> computes the
<code class="reqn">R_{ST}</code> for microsatellites.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Fst(x, pop = NULL, quiet = TRUE, na.alleles = "")
Rst(x, pop = NULL, quiet = TRUE, na.alleles = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of class <code>"loci"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pop</code></td>
<td>
<p>a vector or factor giving the population assignment of each
row of <code>x</code>, or a single numeric value specifying which column
of <code>x</code> to use as population indicator. By default, the column
labelled <code>"population"</code> is used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>a logical value: should calculations be quiet?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na.alleles</code></td>
<td>
<p>by default, only genotypes coded as
NA are considered as missing data. This option
is to specify if some alleles code for missing data.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>Fst</code> uses the formulae in Weir and Cockerham (1984) for each
allele, and then averaged within each locus over the different alleles
as suggested by these authors.
</p>
<p><code>Rst</code> uses the formulae in Slatkin (1995).
</p>


<h3>Value</h3>

<p>A matrix with genes (loci) as rows and the three <em>F</em>-statistics
as columns.
</p>


<h3>Author(s)</h3>

<p>Emmanuel Paradis</p>


<h3>References</h3>

<p>Slatkin, M. (1995) A measure of population subdivision based on
microsatellite allele frequencies.  <em>Genetics</em>, <b>139</b>,
457–462.
</p>
<p>Weir, B. S. and Cockerham, C. C. (1984) Estimating <em>F</em>-statistics
for the analysis of population structure. <em>Evolution</em>, <b>38</b>,
1358–1370.
</p>
<p>Weir, B. S. and Hill, W. G. (2002) Estimating <em>F</em>-statistics.
<em>Annual Review of Genetics</em>, <b>36</b>, 721–750.
</p>


<h3>See Also</h3>

<p><code>fstat</code> in package <span class="pkg">hierfstat</span>; package <span class="pkg">dirmult</span>
on CRAN that implements various estimators of the
Dirichlet-multinomial distribution, including maximum likekihood and
the moments estimator of Weir and Hill (2002); <code>Fst</code> in
<span class="pkg">Biodem</span> that caculates <code class="reqn">F_{ST}</code> from a “kinship
matrix”.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(jaguar)
Fst(jaguar)
Rst(jaguar)

## no Fst but Fit and Fis in case of single population:
jaguar_corridor &lt;- jaguar[jaguar$population == "Green Corridor", ]
Fst(jaguar_corridor) 
</code></pre>


</div>