<div class="container">

<table style="width: 100%;"><tr>
<td>as.loci</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Conversion Among Allelic Data Classes</h2>

<h3>Description</h3>

<p>These functions do conversion among different allelic data classes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">as.loci(x, ...)
## S3 method for class 'genind'
as.loci(x, ...)
genind2loci(x)
## S3 method for class 'data.frame'
as.loci(x, allele.sep = "/|", col.pop = NULL, col.loci = NULL, ...)
loci2genind(x, ploidy = 2, na.alleles = c("0", "."), unphase = TRUE)
## S3 method for class 'factor'
as.loci(x, allele.sep = "/|", ...)
## S3 method for class 'character'
as.loci(x, allele.sep = "/|", ...)
loci2SnpMatrix(x, checkSNP = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of class <code>"loci"</code> or <code>"genind"</code>, a data
frame, a factor, or a vector of mode character.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele.sep</code></td>
<td>
<p>the character(s) separating the alleles for each locus
in the data file (a forward slash by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col.pop</code></td>
<td>
<p>specifies whether one of the column of the data file
identifies the population; default <code>NULL</code>, otherwise an integer
or a character giving the number or the name of the column.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col.loci</code></td>
<td>
<p>a vector of integers or of characters specifying the
indices or the names of the columns that are loci. By default, all
columns are taken as loci except the one labelled "population", if
present or specified.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ploidy</code></td>
<td>
<p>the ploidy level (see details).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na.alleles</code></td>
<td>
<p>a vector of charater strings giving the alleles
to be treated as missing data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>unphase</code></td>
<td>
<p>a logical value; by default, the genotypes are unphased
before conversion (this should not be changed).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments to be passed to or from other methods.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>checkSNP</code></td>
<td>
<p>a logical value. If you are sure that all data in the
<code>"loci"</code> object are SNPs, using <code>checkSNP = FALSE</code> makes
it faster.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The main objectives of these functions is to provide easy conversion
between the data structures of <span class="pkg">adegenet</span> and <span class="pkg">pegas</span>, so both
packages can be used together smoothly. In addition, it is possible to
create a <code>"loci"</code> object directly from a data frame, a vector, or
a factor.
</p>
<p><code>genind2loci(x)</code> and <code>as.loci(x)</code> are the same if <code>x</code>
is of class <code>"genind"</code>.
</p>
<p>The ploidy level specified in <code>loci2genind</code> can be a vector in
which case it should be of length equal to the number of individuals
and will be interpreted as giving the ploidy of each of them. Note
that this is different from <code>getPloidy</code> which returns the
ploidy level of each locus.
</p>


<h3>Value</h3>

<p>An object of class <code>c("loci", "data.frame")</code> for <code>as.loci</code>
and <code>genind2loci</code>; an object of class <code>"genind"</code> for
<code>loci2genind</code>; an object of class <code>"SnpMatrix"</code> for
<code>loci2SnpMatrix</code>.
</p>


<h3>Author(s)</h3>

<p>Emmanuel Paradis</p>


<h3>See Also</h3>

<p><code>read.loci</code>, <code>genind</code>,
<code>df2genind</code> for converting data frames to
<code>"genind"</code>, <code>alleles2loci</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">x &lt;- c("A-A", "A-a", "a-a")
as.loci(x, allele.sep = "-")
## Not run: 
require(adegenet)
data(nancycats)
x &lt;- as.loci(nancycats)
y &lt;- loci2genind(x) # back to "genind"
identical(nancycats@tab, y@tab)
identical(nancycats@pop, y@pop)

## End(Not run)</code></pre>


</div>