<div class="container">

<table style="width: 100%;"><tr>
<td>subset.haplotype</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Subsetting and Filtering Haplotypes</h2>

<h3>Description</h3>

<p>This function selects haplotypes based on their (absolute) frequencies
and/or proportions of missing nucleotides.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'haplotype'
subset(x, minfreq = 1, maxfreq = Inf, maxna = Inf, na = c("N", "?"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of class <code>c("haplotype", "DNAbin")</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minfreq, maxfreq</code></td>
<td>
<p>the lower and upper limits of (absolute)
haplotype frequencies. By default, all haplotypes are selected
whatever their frequency.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxna</code></td>
<td>
<p>the maximum frequency (absolute or relative; see details)
of missing nucleotides within a given haplotype.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na</code></td>
<td>
<p>a vector of mode character specifying which nucleotide
symbols should be treated as missing data; by default, unknown
nucleotide (N) and completely unknown site (?) (can be lower- or
uppercase). There are two shortcuts: see details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>unused.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The value of <code>maxna</code> can be either less than one, or greater or
equal to one. In the former case, it is taken as specifying the
maximum proportion (relative frequency) of missing data within a given
haplotype. In the latter case, it  is taken as the maximum number
(absolute frequency).
</p>
<p><code>na = "all"</code> is a shortcut for all ambiguous nucleotides
(including N) plus alignment gaps and completely unknown site (?).
</p>
<p><code>na = "ambiguous"</code> is a shortcut for only ambiguous nucleotides
(including N).
</p>


<h3>Value</h3>

<p>an object of class <code>c("haplotype", "DNAbin")</code>.
</p>


<h3>Author(s)</h3>

<p>Emmanuel Paradis</p>


<h3>See Also</h3>

<p><code>haplotype</code></p>


<h3>Examples</h3>

<pre><code class="language-R">data(woodmouse)
h &lt;- haplotype(woodmouse)
subset(h, maxna = 20)
subset(h, maxna = 20/ncol(h)) # same thing than above
</code></pre>


</div>