<div class="container">

<table style="width: 100%;"><tr>
<td>site.spectrum</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Site Frequency Spectrum</h2>

<h3>Description</h3>

<p><code>site.spectrum</code> computes the (un)folded site frequency spectrum
of a set of aligned DNA sequences or SNPs.
</p>


<h3>Usage</h3>

<pre><code class="language-R">site.spectrum(x, ...)
## S3 method for class 'DNAbin'
site.spectrum(x, folded = TRUE, outgroup = 1, ...)
## S3 method for class 'loci'
site.spectrum(x, folded = TRUE, ancestral = NULL, ...)
## S3 method for class 'spectrum'
plot(x, col = "red", main = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a set of DNA sequences (as an object of class
<code>"DNAbin"</code>), or an object of class <code>"spectrum"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>folded</code></td>
<td>
<p>a logical specifying whether to compute the folded site
frequency spectrum (the default), or the unfolded spectrum if
<code>folded = FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outgroup</code></td>
<td>
<p>a single integer value giving which sequence is
ancestral; ignored if <code>folded = TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ancestral</code></td>
<td>
<p>a vector of ancestral alleles (required if
<code>folded = FALSE</code>), typically from an output of
<code>VCFloci</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>the colour of the barplot (red by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>main</code></td>
<td>
<p>a character string for the title of the plot; a generic
title is given by default (use <code>main = ""</code> to have no title).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments passed to
<code>barplot</code>, or to other mehods.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Under the infinite sites model of mutation, mutations occur on
distinct sites, so every segregating (polymorphic) site defines a
partition of the <code class="reqn">n</code> sequences (see Wakeley, 2009). The <em>site
frequency spectrum</em> is a series of values where the <code class="reqn">i</code>th element
is the number of segregating sites defining a partition of <code class="reqn">i</code> and
<code class="reqn">n - i</code> sequences. The <em>unfolded</em> version requires to define
an ancestral state with an external (outgroup) sequence, so <code class="reqn">i</code>
varies between 1 and <code class="reqn">n - 1</code>. If no ancestral state can be
defined, the <em>folded</em> version is computed, so <code class="reqn">i</code> varies
between 1 and <code class="reqn">n/2</code> or <code class="reqn">(n - 1)/2</code>, for <code class="reqn">n</code> even or odd,
respectively.
</p>
<p>If <code>folded = TRUE</code>, sites with more than two states are ignored
and a warning is returned giving how many were found.
</p>
<p>If <code>folded = FALSE</code>, sites with an ambiguous state at the
external sequence are ignored and a warning is returned giving how
many were found. Note that it is not checked if some sites have more
than two states.
</p>
<p>If <code>x</code> is an object of class <code>"loci"</code>, the loci which are
not biallelic (e.g., SNPs) are dropped with a warning.
</p>


<h3>Value</h3>

<p><code>site.spectrum</code> returns an object of class <code>"spectrum"</code>
which is a vector of integers (some values may be equal to zero) with
the attributes <code>"sample.size"</code> and <code>"folded"</code> (a logical
value) indicating which version of the spectrum has been computed.
</p>


<h3>Author(s)</h3>

<p>Emmanuel Paradis</p>


<h3>References</h3>

<p>Wakeley, J. (2009) <em>Coalescent Theory: An Introduction.</em>
Greenwood Village, CO: Roberts and Company Publishers.
</p>


<h3>See Also</h3>

<p><code>DNAbin</code> for manipulation of DNA sequences in R,
<code>haplotype</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">require(ape)
data(woodmouse)
(sp &lt;- site.spectrum(woodmouse))
plot(sp)
</code></pre>


</div>