<div class="container">

<table style="width: 100%;"><tr>
<td>read.loci</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read Allelic Data Files</h2>

<h3>Description</h3>

<p>This function reads allelic data from a text file: rows are
individuals, and columns are loci and optional variables. By default,
the first line of the file gives the locus names. If one column is
labelled ‘population’, it is taken as a population variable.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read.loci(file, header = TRUE, loci.sep = "", allele.sep = "/|",
          col.pop = NULL, col.loci = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>a file name specified by either a variable of mode character,
or a quoted string.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>header</code></td>
<td>
<p>a logical specifying whether the first line of the data
file gives the names of the loci (<code>TRUE</code> by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loci.sep</code></td>
<td>
<p>the character(s) separating the loci (columns) in the
data file (a white space by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele.sep</code></td>
<td>
<p>the character(s) separating the alleles for each locus
in the data file (a forward slash by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col.pop</code></td>
<td>
<p>specifies whether one of the column of the data file
identifies the population. By default, if one column is labelled
‘population’ (case-insensitive), it is taken as the population
variable; otherwise an integer giving the number of the column or a
character string giving its name. It is eventually renamed
‘population’ and transformed as a factor.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col.loci</code></td>
<td>
<p>a vector of integers or characters specifying the
indices or the names of the columns that are loci. By default, all
columns are taken as loci except the population one, if present or
specified.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments passed to <code>read.table</code> (e.g.,
<code>row.names</code>).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The rownames of the returned object identify the individual genotypes;
they are either taken from the data file if present, or given the
values <code>"1"</code>, <code>"2"</code>, ... Similarly for the colnames: if
absent in the file (in which case <code>header = FALSE</code> must be set),
they are given the values <code>"V1"</code>, <code>"V2"</code>, ...
</p>
<p>In the returned genotypes, alleles are separated by <code>"/"</code>, even
if it is not the case in the data file.
</p>
<p>The vignette “Reading Genetic Data Files Into R with <span class="pkg">adegenet</span>
and <span class="pkg">pegas</span>” explains how to read various file formats including
Excel files (type <code>vignette("ReadingFiles")</code> in R).
</p>


<h3>Value</h3>

<p>A data frame with class <code>c("loci", "data.frame")</code>. It is a data
frame with an attribute <code>"locicol"</code> specifying the columns that
must be treated as loci. The latter are factors. The other columns can
be of any type.
</p>
<p>Details on the structure can be found in
<a href="https://emmanuelparadis.github.io/pegas/DefinitionDataClassesPegas.pdf">https://emmanuelparadis.github.io/pegas/DefinitionDataClassesPegas.pdf</a>
</p>


<h3>Author(s)</h3>

<p>Emmanuel Paradis</p>


<h3>See Also</h3>

<p><code>read.gtx</code>, <code>read.vcf</code>,
<code>write.loci</code>, <code>summary.loci</code>
</p>


</div>