<div class="container">

<table style="width: 100%;"><tr>
<td>summary.loci</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Print and Summaries of Loci Objects</h2>

<h3>Description</h3>

<p>These functions print and summarize table of alleles and loci (objects
of class <code>"loci"</code>).
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'loci'
print(x, details = FALSE, ...)
## S3 method for class 'loci'
summary(object, ...)
## S3 method for class 'summary.loci'
print(x, ...)
## S3 method for class 'loci'
x[i, j, drop = FALSE]
## S3 method for class 'summary.loci'
plot(x, loci, what = "both", layout = 1, col = c("blue", "red"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x, object</code></td>
<td>
<p>an object of class <code>"loci"</code> or <code>"summary.loci"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>details</code></td>
<td>
<p>a logical value: if <code>TRUE</code> the data are printed as
a data frame; the default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>i, j</code></td>
<td>
<p>indices of the rows and/or columns to select or to drop.
They may be numeric, logical, or character (in the same way than for
standard R objects).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>drop</code></td>
<td>
<p>a logical specifying whether to returned an object of
the smallest dimension possible, i.e., may return a vector or a
factor if <code>drop = TRUE</code> (this is not the default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loci</code></td>
<td>
<p>the loci (genes) to be plotted. By default, all loci are
plotted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>what</code></td>
<td>
<p>the frequencies to be plotted. Three choices are possible:
<code>"alleles"</code>, <code>"genotypes"</code>, and <code>"both"</code> (the default),
or any unambiguous abbreviations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>layout</code></td>
<td>
<p>the number of graphs to be plotted simultaneously.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>the colours used for the barplots.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments to be passed to or from other methods.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Genotypes not observed in the data frame are not counted.
</p>
<p>When using the <code>[</code> method, if only one column is extracted and 
the option <code>drop = TRUE</code>, or if the returned data frame has no ‘locus’ 
column, then the class <code>"loci"</code> is dropped. The option <code>drop = FALSE</code> 
(default) keeps the class (see examples).
</p>
<p>An object of class <code>"loci"</code> can be edited in the R data editor
with, e.g., <code>fix(x)</code> or <code>x &lt;- edit(x)</code>.
</p>
<p><code>summary.loci</code> computes the absolute frequencies (counts); see
the examples on how to compute the relative frequencies (proportions).
</p>


<h3>Value</h3>

<p><code>summary.loci</code> returns a list with the genes as names and each
element made a list with two vectors <code>"genotype"</code> and
<code>"allele"</code> with the frequencies (numbers) of genotypes and
alleles, respectively. The names of these two vectors are the observed
genotypes and alleles.
</p>
<p><code>print</code> and <code>plot</code> methods return NULL.
</p>


<h3>Author(s)</h3>

<p>Emmanuel Paradis</p>


<h3>See Also</h3>

<p><code>read.loci</code>, <code>getAlleles</code>, <code>edit.loci</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(jaguar)
s &lt;- summary(jaguar)
## Not run: 
## works if the device is large enough:
plot(s, layout = 30, las = 2)
layout(1)

## End(Not run)
## compute the relative frequencies:
rapply(s, function(x) x/sum(x), how = "replace")
## extract a single locus:
jaguar[, 1]
jaguar[, 1, drop = TRUE] # returns a vector
jaguar[[1]]              # also returns a vector
</code></pre>


</div>