<div class="container">

<table style="width: 100%;"><tr>
<td>opt.splitval</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Parallelized calculation of split training/test set predictions from L1/L2/Elastic
Net penalized regression.
</h2>

<h3>Description</h3>

<p>uses a single training/test split to train a penalized
regression model in the training samples, then use the model to
calculate values of the linear risk score in the test samples.  This
function is used by opt.nested.crossval, but can also be used on its own.
</p>
<p>This function support z-score scaling of training data, and application
of these scaling and shifting coefficients to the test data.  It also
supports repeated tuning of the penalty parameters and selection of the
model with greatest cross-validated likelihood.
</p>


<h3>Usage</h3>

<pre><code class="language-R">opt.splitval(optFUN="opt1D",testset="equal",scaling=TRUE,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>optFUN</code></td>
<td>

<p>"opt1D" for Lasso or Ridge regression, "opt2D" for Elastic Net.  See
the help pages for these functions for additional arguments.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>testset</code></td>
<td>

<p>For the opt.splitval function ONLY.
"equal" for randomly assigned equal training and test sets, or an
integer vector defining the positions of the test samples in the
response, penalized, and unpenalized arguments which are passed to
the optL1, optL2, or cvl functions of the penalized R package.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scaling</code></td>
<td>

<p>If TRUE, each feature (column) of the training samples (in
matrix/dataframe specified by the penalized argument) are scaled to
z-scores, then these scaling and shifting factors are applied to the
test data.
If FALSE, no scaling is done.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>Additional arguments are required, to be passed to the
optL1 or optL2 function of the penalized R package.  See those help
pages, and it may be desirable to test these arguments directly on
optL1 or optL2 before using this more CPU-consuming and complex
function.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function does split sample model training and testing for a
single split of the data, using the optL1 or optL2 functions
of the penalized R package, for each
iteration of the cross-validation.  Scaling of the test samples is
done independently, using scale factors determined from the training
samples.  Repeated starts of model training can be parallelized as
documented in the opt1D and opt2D functions.  This function is used
for nested cross-validation by the opt.nested.crossval function.
</p>


<h3>Value</h3>

<p>Returns a vector of cross-validated continuous risk score predictions.
</p>


<h3>Note</h3>

<p>Depends on the R packages: penalized, parallel, rlecuyer
</p>


<h3>Author(s)</h3>

<p>Levi Waldron et al.
</p>


<h3>References</h3>

<p>Waldron L, Pintilie M, Tsao M-S, Shepherd FA, Huttenhower C*, Jurisica
I*: Optimized application of penalized regression methods to diverse
genomic data. Bioinformatics 2011, 27:3399-3406.  (*equal contribution)
</p>


<h3>See Also</h3>

<p>opt1D, opt2D, opt.nested.crossval
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(beer.exprs)
data(beer.survival)

## select just 250 genes to speed computation:
set.seed(1)
beer.exprs.sample &lt;- beer.exprs[sample(1:nrow(beer.exprs), 250), ]

gene.quant &lt;- apply(beer.exprs.sample, 1, quantile, probs = 0.75)
dat.filt &lt;- beer.exprs.sample[gene.quant &gt; log2(100),]
gene.iqr &lt;- apply(dat.filt, 1, IQR)
dat.filt &lt;- as.matrix(dat.filt[gene.iqr &gt; 0.5,])
dat.filt &lt;- t(dat.filt)

library(survival)
surv.obj &lt;- Surv(beer.survival$os, beer.survival$status)

## Single split training/test evaluation.  Ideally nsim would be 50 and
## fold=10, but this requires 100x more resources.
set.seed(1)
preds50 &lt;- opt.splitval(
  optFUN = "opt1D",
  scaling = TRUE,
  testset = "equal",
  setpen = "L1",
  nsim = 1,
  nprocessors = 1,
  response = surv.obj,
  penalized = dat.filt,
  fold = 5,
  positive = FALSE,
  standardize = FALSE,
  trace = FALSE
)

preds50.dichot &lt;- preds50 &gt; median(preds50)

surv.obj.50 &lt;-
  surv.obj[match(names(preds50), rownames(beer.survival))]
coxfit50.continuous &lt;- coxph(surv.obj.50 ~ preds50)
coxfit50.dichot &lt;- coxph(surv.obj.50 ~ preds50.dichot)
summary(coxfit50.continuous)
summary(coxfit50.dichot)
</code></pre>


</div>