<div class="container">

<table style="width: 100%;"><tr>
<td>aaComp</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute the amino acid composition of a protein sequence</h2>

<h3>Description</h3>

<p>This function calculates the amount of amino acids of a particular class and classified as: Tiny, Small, Aliphatic, Aromatic, Non-polar, Polar, Charged, Basic and Acidic based on their size and R-groups using same function implemented in EMBOSS 'pepstat'. 
The output is a matrix with the number and percentage of amino acids of a particular class
</p>


<h3>Usage</h3>

<pre><code class="language-R">aaComp(seq)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>seq</code></td>
<td>
<p>An amino-acid sequence</p>
</td>
</tr></table>
<h3>Details</h3>

<p>Amino acids are zwitterionic molecules with an amine and a carboxyl group present in their structure. 
Some amino acids possess side chains with specific properties that allow grouping them in different ways. 
The <code>aaComp</code> function classifies amino acids based on their size, side chains, hydrophobicity, charge and their response to pH 7.
</p>


<h3>Value</h3>

<p>The output is a matrix with the number and percentage of amino acids of a particular class:
</p>

<ul>
<li>
<p> Tiny (A + C + G + S + T)
</p>
</li>
<li>
<p> Small (A + B + C + D + G + N + P + S + T + V)
</p>
</li>
<li>
<p> Aliphatic (A + I + L + V)
</p>
</li>
<li>
<p> Aromatic (F + H + W + Y)
</p>
</li>
<li>
<p> Non-polar (A + C + F + G + I + L + M + P + V + W + Y)
</p>
</li>
<li>
<p> Polar (D + E + H + K + N + Q + R + S + T + Z)
</p>
</li>
<li>
<p> Charged (B + D + E + H + K + R + Z)
</p>
</li>
<li>
<p> Basic (H + K + R)
</p>
</li>
<li>
<p> Acidic (B + D + E + Z)
</p>
</li>
</ul>
<h3>Note</h3>

<p>This function was originally written by Alan Bleasby (ajb@ebi.ac.uk) for the EMBOSS package. 
Further information: http://emboss.sourceforge.net/apps/cvs/emboss/apps/pepstats.html
</p>


<h3>References</h3>

<p>Rice, Peter, Ian Longden, and Alan Bleasby. "EMBOSS: the European molecular biology open software suite." Trends in genetics 16.6 (2000): 276-277.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># COMPARED TO PEPSTATS
# http://emboss.bioinformatics.nl/cgi-bin/emboss/pepstats
# Property      Residues              Number  Mole%
# Tiny          (A+C+G+S+T)             4   19.048
# Small         (A+B+C+D+G+N+P+S+T+V)   4   19.048
# Aliphatic     (A+I+L+V)               5   23.810
# Aromatic      (F+H+W+Y)               5   23.810
# Non-polar     (A+C+F+G+I+L+M+P+V+W+Y) 11  52.381
# Polar         (D+E+H+K+N+Q+R+S+T+Z)   9   42.857
# Charged       (B+D+E+H+K+R+Z)         8   38.095
# Basic         (H+K+R)                 8   38.095
# Acidic        (B+D+E+Z)               0   00.000

## AA composition of PDB: 1D9J Cecropin Peptide
aaComp(seq= "KWKLFKKIGIGKFLHSAKKFX")

## Output
#           Number  Mole %
# Tiny           4 19.048
# Small          4 19.048
# Aliphatic      5 23.810
# Aromatic       5 23.810
# NonPolar      11 52.381
# Polar          9 42.857
# Charged        8 38.095
# Basic          8 38.095
# Acidic         0  0.000
</code></pre>


</div>