<div class="container">

<table style="width: 100%;"><tr>
<td>AAdata</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Properties, scales and indices for the 20 naturally occurring amino acids from various sources</h2>

<h3>Description</h3>

<p>A list with a collection of properties, scales and indices for the 20 naturally occurring amino acids from various sources.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(AAdata)
</code></pre>


<h3>Format</h3>

<p>A list as follows: </p>

<ul>
<li>
<p>Hydrophobicity The hydrophobicity is an important stabilization force in protein folding; this force changes depending on the solvent in which the protein is found. </p>

<ul>
<li>
<p>Aboderin: Aboderin, A. A. (1971). An empirical hydrophobicity scale for alpha-amino-acids and some of its applications. International Journal of Biochemistry, 2(11), 537-544.
</p>
</li>
<li>
<p>AbrahamLeo: Abraham D.J., Leo A.J. Hydrophobicity (delta G1/2 cal). Proteins: Structure, Function and Genetics 2:130-152(1987).
</p>
</li>
<li>
<p>Argos: Argos, P., Rao, J. K., &amp; Hargrave, P. A. (1982). Structural Prediction of Membrane-Bound Proteins. European Journal of Biochemistry, 128(2-3), 565-575.
</p>
</li>
<li>
<p>BlackMould: Black S.D., Mould D.R. Hydrophobicity of physiological L-alpha amino acids. Anal. Biochem. 193:72-82(1991).
</p>
</li>
<li>
<p>BullBreese: Bull H.B., Breese K. Hydrophobicity (free energy of transfer to surface in kcal/mole). Arch. Biochem. Biophys. 161:665-670(1974).
</p>
</li>
<li>
<p>Casari: Casari, G., &amp; Sippl, M. J. (1992). Structure-derived hydrophobic potential: hydrophobic potential derived from X-ray structures of globular proteins is able to identify native folds. Journal of molecular biology, 224(3), 725-732.
</p>
</li>
<li>
<p>Chothia: Chothia, C. (1976). The nature of the accessible and buried surfaces in proteins. Journal of molecular biology, 105(1), 1-12.
</p>
</li>
<li>
<p>Cid: Cid, H., Bunster, M., Canales, M., &amp; Gazitua, F. (1992). Hydrophobicity and structural classes in proteins. Protein engineering, 5(5), 373-375.
</p>
</li>
<li>
<p>Cowan3.4: Cowan R., Whittaker R.G. Hydrophobicity indices at pH 3.4 determined by HPLC. Peptide Research 3:75-80(1990).
</p>
</li>
<li>
<p>Cowan7.5: Cowan R., Whittaker R.G. Hydrophobicity indices at pH 7.5 determined by HPLC. Peptide Research 3:75-80(1990).
</p>
</li>
<li>
<p>Eisenberg: Eisenberg D., Schwarz E., Komarony M., Wall R. Normalized consensus hydrophobicity scale. J. Mol. Biol. 179:125-142(1984).
</p>
</li>
<li>
<p>Engelman: Engelman, D. M., Steitz, T. A., &amp; Goldman, A. (1986). Identifying nonpolar transbilayer helices in amino acid sequences of membrane proteins. Annual review of biophysics and biophysical chemistry, 15(1), 321-353.
</p>
</li>
<li>
<p>Fasman: Fasman, G. D. (Ed.). (1989). Prediction of protein structure and the principles of protein conformation. Springer.
</p>
</li>
<li>
<p>Fauchere: Fauchere J.-L., Pliska V.E. Hydrophobicity scale (pi-r). Eur. J. Med. Chem. 18:369-375(1983).
</p>
</li>
<li>
<p>Goldsack: Goldsack, D. E., &amp; Chalifoux, R. C. (1973). Contribution of the free energy of mixing of hydrophobic side chains to the stability of the tertiary structure of proteins. Journal of theoretical biology, 39(3), 645-651.
</p>
</li>
<li>
<p>Guy: Guy H.R. Hydrophobicity scale based on free energy of transfer (kcal/mole). Biophys J. 47:61-70(1985).
</p>
</li>
<li>
<p>HoppWoods: Hopp T.P., Woods K.R. Hydrophilicity. Proc. Natl. Acad. Sci. U.S.A. 78:3824-3828(1981).
</p>
</li>
<li>
<p>Janin: Janin J. Free energy of transfer from inside to outside of a globular protein. Nature 277:491-492(1979).
</p>
</li>
<li>
<p>Jones: Jones, D. D. (1975). Amino acid properties and side-chain orientation in proteins: a cross correlation approach. Journal of theoretical biology, 50(1), 167-183.
</p>
</li>
<li>
<p>Juretic: Juretic, D., Lucic, B., Zucic, D., &amp; Trinajstic, N. (1998). Protein transmembrane structure: recognition and prediction by using hydrophobicity scales through preference functions. Theoretical and computational chemistry, 5, 405-445.
</p>
</li>
<li>
<p>Kidera: Kidera, A., Konishi, Y., Oka, M., Ooi, T., &amp; Scheraga, H. A. (1985). Statistical analysis of the physical properties of the 20 naturally occurring amino acids. Journal of Protein Chemistry, 4(1), 23-55.
</p>
</li>
<li>
<p>Kuhn: Kuhn, L. A., Swanson, C. A., Pique, M. E., Tainer, J. A., &amp; Getzoff, E. D. (1995). Atomic and residue hydrophilicity in the context of folded protein structures. Proteins: Structure, Function, and Bioinformatics, 23(4), 536-547.
</p>
</li>
<li>
<p>KyteDoolittle: Kyte J., Doolittle R.F. Hydropathicity. J. Mol. Biol. 157:105-132(1982).
</p>
</li>
<li>
<p>Levitt: Levitt, M. (1976). A simplified representation of protein conformations for rapid simulation of protein folding. Journal of molecular biology, 104(1), 59-107.
</p>
</li>
<li>
<p>Manavalan: Manavalan P., Ponnuswamy Average surrounding hydrophobicity. P.K. Nature 275:673-674(1978).
</p>
</li>
<li>
<p>Miyazawa: Miyazawa S., Jernigen R.L. Hydrophobicity scale (contact energy derived from 3D data). Macromolecules 18:534-552(1985).
</p>
</li>
<li>
<p>Parker: Parker J.M.R., Guo D., Hodges R.S. Hydrophilicity scale derived from HPLC peptide retention times. Biochemistry 25:5425-5431(1986).
</p>
</li>
<li>
<p>Ponnuswamy: Ponnuswamy, P. K. (1993). Hydrophobic charactesristics of folded proteins. Progress in biophysics and molecular biology, 59(1), 57-103.
</p>
</li>
<li>
<p>Prabhakaran: Prabhakaran, M. (1990). The distribution of physical, chemical and conformational properties in signal and nascent peptides. Biochem. J, 269, 691-696.
</p>
</li>
<li>
<p>Rao: Rao M.J.K., Argos P. Membrane buried helix parameter. Biochim. Biophys. Acta 869:197-214(1986).
</p>
</li>
<li>
<p>Rose: Rose G.D., Geselowitz A.R., Lesser G.J., Lee R.H., Zehfus M.H. Mean fractional area loss (f) [average area buried/standard state area]. Science 229:834-838(1985)
</p>
</li>
<li>
<p>Roseman: Roseman M.A. Hydrophobicity scale (pi-r). J. Mol. Biol. 200:513-522(1988).
</p>
</li>
<li>
<p>Sweet: Sweet R.M., Eisenberg D. Optimized matching hydrophobicity (OMH). J. Mol. Biol. 171:479-488(1983).
</p>
</li>
<li>
<p>Tanford: Tanford C. Hydrophobicity scale (Contribution of hydrophobic interactions to the stability of the globular conformation of proteins). J. Am. Chem. Soc. 84:4240-4274(1962).
</p>
</li>
<li>
<p>Welling: Welling G.W., Weijer W.J., Van der Zee R., Welling-Wester S. Antigenicity value X 10. FEBS Lett. 188:215-218(1985).
</p>
</li>
<li>
<p>Wilson: Wilson K.J., Honegger A., Stotzel R.P., Hughes G.J. Hydrophobic constants derived from HPLC peptide retention times. Biochem. J. 199:31-41(1981).
</p>
</li>
<li>
<p>Wolfenden: Wolfenden R.V., Andersson L., Cullis P.M., Southgate C.C.F. Hydration potential (kcal/mole) at 25C. Biochemistry 20:849-855(1981).
</p>
</li>
<li>
<p>Zimmerman: Zimmerman, J. M., Eliezer, N., &amp; Simha, R. (1968). The characterization of amino acid sequences in proteins by statistical methods. Journal of theoretical biology, 21(2), 170-201.
</p>
</li>
<li>
<p>interfaceScale_pH8 White, Stephen (2006-06-29). "Experimentally Determined Hydrophobicity Scales". University of California, Irvine. Retrieved 2017-05-25
</p>
</li>
<li>
<p>interfaceScale_pH2 White, Stephen (2006-06-29). "Experimentally Determined Hydrophobicity Scales". University of California, Irvine. Retrieved 2017-05-25
</p>
</li>
<li>
<p>octanolScale_pH8 White, Stephen (2006-06-29). "Experimentally Determined Hydrophobicity Scales". University of California, Irvine. Retrieved 2017-05-25
</p>
</li>
<li>
<p>octanolScale_pH2 White, Stephen (2006-06-29). "Experimentally Determined Hydrophobicity Scales". University of California, Irvine. Retrieved 2017-05-25
</p>
</li>
<li>
<p>oiScale_pH8 White, Stephen (2006-06-29). "Experimentally Determined Hydrophobicity Scales". University of California, Irvine. Retrieved 2017-05-25
</p>
</li>
<li>
<p>oiScale_pH2 White, Stephen (2006-06-29). "Experimentally Determined Hydrophobicity Scales". University of California, Irvine. Retrieved 2017-05-25
</p>
</li>
</ul>
</li>
<li>
<p>crucianiProperties: The three Cruciani et. al (2004) properties, are the scaled principal component scores that summarize a broad set of descriptors calculated based on the interaction of each amino acid residue with several chemical groups (or "probes"), such as charged ions, methyl, hydroxyl groups, and so forth. </p>

<ul>
<li>
<p>PP1: Polarity, 
</p>
</li>
<li>
<p>PP2: Hydrophobicity, 
</p>
</li>
<li>
<p>PP3: H-bonding
</p>
</li>
</ul>
</li>
<li>
<p>kideraFactors: The Kidera Factors were originally derived by applying multivariate analysis to 188 physical properties of the 20 amino acids and using dimension reduction techniques. A 10-dimensional vector of orthogonal factors was then obtained for each amino acid.The first four factors are essentially pure physical properties; the remaining six factors are superpositions of several physical properties, and are labelled for convenience by the name of the most heavily weighted component </p>

<ul>
<li>
<p>helix.bend.pref: Helix/bend preference
</p>
</li>
<li>
<p>side.chain.size: Side-chain size
</p>
</li>
<li>
<p>extended.str.pref: Extended structure preference
</p>
</li>
<li>
<p>hydrophobicity: Hydrophobicity
</p>
</li>
<li>
<p>double.bend.pref: Double-bend preference
</p>
</li>
<li>
<p>partial.spec.vol: Partial specific volume
</p>
</li>
<li>
<p>flat.ext.pref: Flat extended preference
</p>
</li>
<li>
<p>occurrence.alpha.reg: Occurrence in alpha region
</p>
</li>
<li>
<p>pK.C: pK-C
</p>
</li>
<li>
<p>surrounding.hydrop: Surrounding hydrophobicity
</p>
</li>
</ul>
</li>
<li>
<p>pK</p>

<ul>
<li>
<p>Bjellqvist: Bjellqvist, B., Hughes, G.J., Pasquali, Ch., Paquet, N., Ravier, F., Sanchez, J.Ch., Frutige,r S., Hochstrasser D. (1993) The focusing positions of polypeptides in immobilized pH gradients can be predicted from their amino acid sequences. Electrophoresis, 14:1023-1031.
</p>
</li>
<li>
<p>Dawson: Dawson, R. M. C.; Elliot, D. C.; Elliot, W. H.; Jones, K. M. Data for biochemical research. Oxford University Press, 1989; p. 592.
</p>
</li>
<li>
<p>EMBOSS: EMBOSS data are from http://emboss.sourceforge.net/apps/release/5.0/emboss/apps/iep.html.
</p>
</li>
<li>
<p>Lehninger: Nelson, D. L.; Cox, M. M. Lehninger Principles of Biochemistry, Fourth Edition; W. H. Freeman, 2004; p. 1100.
</p>
</li>
<li>
<p>Murray: Murray, R.K., Granner, D.K., Rodwell, V.W. (2006) Harper's illustrated Biochemistry. 27th edition. Published by The McGraw-Hill Companies.
</p>
</li>
<li>
<p>Rodwell: Rodwell, J. Heterogeneity of component bands in isoelectric focusing patterns. Analytical Biochemistry, 1982, 119 (2), 440-449.
</p>
</li>
<li>
<p>Sillero: Sillero, A., Maldonado, A. (2006) Isoelectric point determination of proteins and other macromolecules: oscillating method. Comput Biol Med., 36:157-166.
</p>
</li>
<li>
<p>Solomon: Solomon, T.W.G. (1998) Fundamentals of Organic Chemistry, 5th edition. Published by Wiley.
</p>
</li>
<li>
<p>Stryer: Stryer L. (1999) Biochemia. czwarta edycja. Wydawnictwo Naukowe PWN.
</p>
</li>
</ul>
</li>
<li>
<p>zScales The five Sandberg et al. (1998) Z-scales describe each amino acid with numerical values, descriptors, which represent the physicochemical properties of the amino acids including NMR data and thin-layer chromatography (TLC) data.</p>

<ul>
<li>
<p>Z1: Lipophilicity
</p>
</li>
<li>
<p>Z2: Steric properties (Steric bulk/Polarizability)
</p>
</li>
<li>
<p>Z3: Electronic properties (Polarity / Charge)
</p>
</li>
<li>
<p>Z4: Related to  electronegativity, heat of formation, electrophilicity and hardness.
</p>
</li>
<li>
<p>Z5: Related to electronegativity, heat of formation, electrophilicity and hardness.
</p>
</li>
</ul>
</li>
<li>
<p>FASGAI Factor Analysis Scale of Generalized Amino Acid Information (FASGAI) proposed by Liang and Li (2007), is a set of amino acid descriptors, that reflects hydrophobicity, alpha and turn propensities, bulky properties, compositional characteristics, local flexibility, and electronic properties, was derived from multi-dimensional properties of 20 naturally occurring amino acids. </p>

<ul>
<li>
<p>F1: Hydrophobicity index
</p>
</li>
<li>
<p>F2: Alpha and turn propensities
</p>
</li>
<li>
<p>F3: Bulky properties
</p>
</li>
<li>
<p>F4: Compositional characteristic index
</p>
</li>
<li>
<p>F5: Local flexibility
</p>
</li>
<li>
<p>F6: Electronic properties
</p>
</li>
</ul>
</li>
<li>
<p>VHSE The principal components score Vectors of Hydrophobic, Steric, and Electronic properties, is derived from principal components analysis (PCA) on independent families of 18 hydrophobic properties, 17 steric properties, and 15 electronic properties, respectively, which are included in total 50 physicochemical variables of 20 coded amino acids. </p>

<ul>
<li>
<p>VHSE1 and VHSE2: Hydrophobic properties
</p>
</li>
<li>
<p>VHSE3 and VHSE4: Steric properties
</p>
</li>
<li>
<p>VHSE5 to VHSE8: Electronic properties
</p>
</li>
</ul>
</li>
</ul>
<h3>Source</h3>


<ul>
<li>
<p>Hydrophobicity</p>

<ul>
<li>
<p> ExPASy-Protscale (http://web.expasy.org/protscale/)
</p>
</li>
<li>
<p> AAIndex Database (http://www.genome.jp/aaindex/)
</p>
</li>
</ul>
</li>
<li>
<p>pK</p>

<ul><li>
<p> Kiraga, J. (2008) Analysis and computer simulations of variability of isoelectric point of proteins in the proteomes. PhD thesis, University of Wroclaw, Poland.
</p>
</li></ul>
</li>
</ul>
<h3>References</h3>


<ul>
<li>
<p>Hydrophobicity</p>

<ul>
<li>
<p> Nakai, K., Kidera, A., and Kanehisa, M.; Cluster analysis of amino acid indices for prediction of protein structure and function. Protein Eng. 2, 93-100 (1988).
</p>
</li>
<li>
<p> Tomii, K. and Kanehisa, M.; Analysis of amino acid indices and mutation matrices for sequence comparison and structure prediction of proteins. Protein Eng. 9, 27-36 (1996). 
</p>
</li>
<li>
<p> Kawashima, S., Ogata, H., and Kanehisa, M.; AAindex: amino acid index database. Nucleic Acids Res. 27, 368-369 (1999).
</p>
</li>
<li>
<p> Kawashima, S. and Kanehisa, M.; AAindex: amino acid index database. Nucleic Acids Res. 28, 374 (2000).
</p>
</li>
<li>
<p> Kawashima, S., Pokarowski, P., Pokarowska, M., Kolinski, A., Katayama, T., and Kanehisa, M.; AAindex: amino acid index database, progress report 2008. Nucleic Acids Res. 36, D202-D205 (2008). 
</p>
</li>
</ul>
</li>
<li>
<p>crucianiProperties: </p>

<ul><li>
<p> Cruciani, G., Baroni, M., Carosati, E., Clementi, M., Valigi, R., and Clementi, S. (2004) Peptide studies by means of principal properties of amino acids derived from MIF descriptors. J. Chemom. 18, 146-155.
</p>
</li></ul>
</li>
<li>
<p>kideraFactors: </p>

<ul><li>
<p> Kidera, A., Konishi, Y., Oka, M., Ooi, T., &amp; Scheraga, H. A. (1985). Statistical analysis of the physical properties of the 20 naturally occurring amino acids. Journal of Protein Chemistry, 4(1), 23-55.
</p>
</li></ul>
</li>
<li>
<p>pK: </p>

<ul>
<li>
<p> Aronson, J. N. The Henderson-Hasselbalch equation revisited. Biochemical Education, 1983, 11 (2), 68.
</p>
</li>
<li>
<p> Moore, D. S.. Amino acid and peptide net charges: A simple calculational procedure. Biochemical Education, 1986, 13 (1), 10-12.
</p>
</li>
<li>
<p> Goloborodko, A.A.; Levitsky, L.I.; Ivanov, M.V.; and Gorshkov, M.V. (2013) "Pyteomics - a Python Framework for Exploratory Data Analysis and Rapid Software Prototyping in Proteomics", Journal of The American Society for Mass Spectrometry, 24(2), 301-304.
</p>
</li>
<li>
<p> Kiraga, J. (2008) Analysis and computer simulations of variability of isoelectric point of proteins in the proteomes. PhD thesis, University of Wroclaw, Poland.
</p>
</li>
</ul>
</li>
<li>
<p>zScales </p>

<ul><li>
<p> Sandberg M, Eriksson L, Jonsson J, Sjostrom M, Wold S: New chemical descriptors relevant for the design of biologically active peptides. A multivariate characterization of 87 amino acids. J Med Chem 1998, 41:2481-2491.
</p>
</li></ul>
</li>
<li>
<p>FASGAI </p>

<ul><li>
<p> Liang, G., &amp; Li, Z. (2007). Factor analysis scale of generalized amino acid information as the source of a new set of descriptors for elucidating the structure and activity relationships of cationic antimicrobial peptides. Molecular Informatics, 26(6), 754-763.
</p>
</li></ul>
</li>
<li>
<p>VHSE </p>

<ul><li>
<p> Mei, H. U., Liao, Z. H., Zhou, Y., &amp; Li, S. Z. (2005). A new set of amino acid descriptors and its application in peptide QSARs. Peptide Science, 80(6), 775-786.
</p>
</li></ul>
</li>
</ul>
</div>