<div class="container">

<table style="width: 100%;"><tr>
<td>perm.path</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Perform Permutation Based Pathway Analysis
</h2>

<h3>Description</h3>

<p>This is the package main function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">perm.path(expr, y, local.test, global.test="wilcoxon", B, gset, min.num=2, max.num, 
imputeval=NULL, transfun=function(x){x}, sort="pval", anno=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>expr</code></td>
<td>

<p>An <code class="reqn">K \times n</code> matrix of gene expression data, where <code class="reqn">K</code> is the number of genes 
and <code class="reqn">n</code> is the number of samples.  
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>

<p>An outcome vector of length <code class="reqn">n</code>. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>local.test</code></td>
<td>

<p>Local test statistic of each gene. Current possible choices are <code class="reqn">t-test</code>, <code class="reqn">Wilcoxon</code>
test, <code class="reqn">Pearson</code>, <code class="reqn">Spearman</code> and <code class="reqn">JT</code> test.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>global.test</code></td>
<td>

<p>Global test statictic, used to compute the score. Current possible choices are <code class="reqn">mean</code>,
<code class="reqn">meanabs</code> (mean of absolute values) and <code class="reqn">maxmean</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>B</code></td>
<td>

<p>specifies the number of random permutations to be performed.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gset</code></td>
<td>

<p>A list of pathways. Each element is a vector of gene names. 
The list element names are the pathway names.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.num</code></td>
<td>

<p>Specifies the minimum number of genes that a pathway should have. 
Pathways with smaller number of genes will be excluded.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.num</code></td>
<td>

<p>Specifies the maximum number of genes that a pathway should have. 
Pathways with larger number of genes will be excluded.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>imputeval</code></td>
<td>

<p>The gene expression value to be imputed in case of missing values. 
The default choice is <code class="reqn">NULL</code> in which case no imputation is done.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>transfun</code></td>
<td>

<p>Specifies transformation of the gene expression data. 
The default option is untransformed gene expression data.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sort</code></td>
<td>

<p>Specifies sorting of the results. 
If <code class="reqn">sort="pval"</code> sorting is done in order of increasing <code class="reqn">p-values</code>. 
If <code class="reqn">sort="score"</code> sorting is done in order of decreasing <code class="reqn">scores</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anno</code></td>
<td>

<p>If <code class="reqn">TRUE</code> the output contains annotation of each pathway.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>This function returns a list consisting of the following elements:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>res</code></td>
<td>
<p>Data frame consisting of the pathway names (Pathway), the genes 
involved in each pathway (Genes), the number of genes in each pathway (Size), 
the score for each pathway (Score), the permutation raw p-value (pval), 
the FWER-adjusted permutation p-value (pfwer), the FDR-adjusted permutation p-value, 
the Bonferroni-adjusted permutation p-value (bonferroni)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>stats</code></td>
<td>
<p>The individual test statistic for each gene</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scores</code></td>
<td>
<p>A matrix of scores. The matrix is of dimension <code class="reqn">(B+1)\times K</code>, 
where K is the number of pathways. The first column contains the unpermuted scores, 
the remaining <code class="reqn">B</code> columns contain the scores computed after each permutation.</p>
</td>
</tr>
</table>
<h3>References</h3>

<p>B. Efron, R. Tibshirani (2007) 
On Testing the Significance of Sets of Genes. <em>The Annals of Applied Statistics</em>. 
<b>Vol. 1</b>, No 1, 107–129.
</p>
<p>A. Subramanian, P. Tamayo, V. K. Mootha, S. Mukherjee, B. L. Ebert, M. A. Gillette, 
A. Paulovich, S. L. Pomeroy, T. R. Golub, E. S. Lander and J. P. Mesirov (2005), 
Gene Set Enrichment Analysis: A knowledge-based Approach for Interpreting 
Genome-Wide Expression Profiles. <em>Proc. Natl. Acad. Sci. USA</em>. <b>Vol. 102</b>, No 43, 15545–15550.
</p>


<h3>Examples</h3>

<pre><code class="language-R">set.seed(1234)

## Generate toy phenotype and gene expression data sets
## This example consists of 40 genes grouped into 5 pathways and 100 patients
## grp is a binary trait (e.g., case vs control)
## bp is a continuous trait (e.g., blood pressure)
## g is a group indicator

n = 100
K = 40
grp = rep(1:0,each=n/2)
bp = rnorm(n)
g = rep(1:(n/20), rep(20,n/20))

pdat = data.frame(grp, bp, g)
rm(grp, bp)
expdat = matrix(rnorm(K*n),K,n)

## Assign marker names g1,...,gK to the expression data set and
## patient ids id1,...,idn to the expression and phenotype data
gnames = paste("g",1:K,sep="")
rownames(expdat) = gnames
patid = paste("id",1:n,sep="")
rownames(pdat) = patid
colnames(expdat) = patid

#Group the K genes into M pathways of sizes n1,...,nM
M = 5
p = runif(M)
p = p/sum(p)
nM = rmultinom(1, size=K, prob=p)
gset = lapply(nM, function(x){gnames[sample(x)]})
names(gset) = paste("pathway",1:M,sep="")

## Carry out permutation analysis with grp as the outcome
## using the two-sample Wilcoxon with B=100 random permutations
perm.path(expdat, y=pdat[["grp"]], local.test="wilcoxon", global.test="maxmean", B=100, 
gset=gset, min.num=2, max.num=50, sort="score")

## Carry out permutation analysis with g as the outcome
## using the JT test with B=100 random permutations
perm.path(expdat, y=pdat[["g"]], local.test="jt", global.test="maxmean", B=100, 
gset=gset, min.num=2, max.num=50, sort="score")
</code></pre>


</div>