<div class="container">

<table style="width: 100%;"><tr>
<td>pez.metrics</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Phylogenetic and functional trait metrics within pez</h2>

<h3>Description</h3>

<p>Using these functions, you can calculate any of the phylogenetic
metrics within pez, using <code>comparative.comm</code>
objects. While you can call each individually, using the
<code>pez.shape</code>, <code>pez.evenness</code>,
<code>pez.dispersion</code>, and <code>pez.dissimilarity</code>
wrapper functions (and the more flexible
<code>generic.metrics</code> and null model functions) are probably
your best bet. Note that *all of these functions* take a common
first parameter: a <code>comparative.comm</code> object. There are
additional parameters that can be passed, which are described
below.
</p>


<h3>Usage</h3>

<pre><code class="language-R">.hed(x, ...)

.eed(x, na.rm = TRUE, ...)

.psv(x, ...)

.psr(x, ...)

.mpd(x, dist = NULL, abundance.weighted = FALSE, ...)

.vpd(x, dist = NULL, abundance.weighted = FALSE, ...)

.vntd(x, dist = NULL, abundance.weighted = FALSE, ...)

.pd(x, include.root = TRUE, abundance.weighted = FALSE, ...)

.mntd(x, dist = NULL, abundance.weighted = FALSE, ...)

.gamma(x, ...)

.taxon(x, dist = NULL, abundance.weighted = FALSE, ...)

.eigen.sum(x, dist = NULL, which.eigen = 1, ...)

.dist.fd(x, method = "phy", abundance.weighted = FALSE, ...)

.sqrt.phy(x)

.phylo.entropy(x, ...)

.aed(x, ...)

.haed(x, ...)

.simpson.phylogenetic(x)

.iac(x, na.rm = TRUE, ...)

.pae(x, na.rm = TRUE, ...)

.scheiner(x, q = 0, abundance.weighted = FALSE, ...)

.pse(x, ...)

.rao(x, ...)

.lambda(x, ...)

.delta(x, ...)

.kappa(x, ...)

.eaed(x, ...)

.unifrac(x, ...)

.pcd(x, permute = 1000, ...)

.comdist(x, dist = NULL, abundance.weighted = FALSE, ...)

.phylosor(x, dist = NULL, abundance.weighted = FALSE, ...)

.d(x, permute = 1000, ...)

.ses.mpd(
  x,
  dist = NULL,
  null.model = "taxa.labels",
  abundance.weighted = FALSE,
  permute = 1000,
  ...
)

.ses.mntd(
  x,
  dist = NULL,
  null.model = "taxa.labels",
  abundance.weighted = FALSE,
  permute = 1000,
  ...
)

.ses.vpd(
  x,
  dist = NULL,
  null.model = "taxa.labels",
  abundance.weighted = FALSE,
  permute = 1000,
  ...
)

.ses.vntd(
  x,
  dist = NULL,
  null.model = "taxa.labels",
  abundance.weighted = FALSE,
  permute = 1000,
  ...
)

.ses.mipd(
  x,
  dist = NULL,
  null.model = "taxa.labels",
  abundance.weighted = FALSE,
  permute = 1000,
  ...
)

.ses.innd(
  x,
  dist = NULL,
  null.model = "taxa.labels",
  abundance.weighted = FALSE,
  permute = 1000,
  ...
)

.mipd(x, dist = NULL, abundance.weighted = FALSE, ...)

.innd(x, dist = NULL, abundance.weighted = FALSE, ...)

.innd(x, dist = NULL, abundance.weighted = FALSE, ...)

.pe(x, ...)

.bed(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p><code>comparative.comm</code> object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>ignored</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na.rm</code></td>
<td>
<p>remove NAs in calculations (altering this can obscure
errors that are meaningful; I would advise leaving alone)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dist</code></td>
<td>
<p>distance matrix for use with calculations; could be
generated from traits, a square-root-transformed distance matrix
(see <code>.sqrt.phy</code> for creating a
<code>comparative.comm</code> object with a square-root
transformed phylogeny). Default: NULL (–&gt; calculate distance
matrix from phylogeny)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>abundance.weighted</code></td>
<td>
<p>whether to include species' abundances in
metric calculation, often dictating whether you're calculating a
<code>pez.shape</code> or <code>pez.evenness</code>
metric. Default: FALSE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>include.root</code></td>
<td>
<p>include root in PD calculations (default is
TRUE, as in picante, but within <code>pez.shape</code> I specify
FALSE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>which.eigen</code></td>
<td>
<p>which phylo-eigenvector to be used for PVR
metric</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>whether to calculate using phylogeny ("phy";
default) or trait data ("traits")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>q</code></td>
<td>
<p>the q parameter for <code>.scheiner</code>; default 0.0001</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>permute</code></td>
<td>
<p>number of permutations of null randomisations
(mostly only applies to <code>dispersion
metrics</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>null.model</code></td>
<td>
<p>one of "taxa.labels", "richness", "frequency",
"sample.pool", "phylogeny.pool", "independentswap", or
"independentswap". These correspond to the null models available in
<code>picante</code>; only <code>d</code> does not use these null models</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>.pd</code> returns two metrics: Faith's PD (which does not take
into account abundance) and Faith's PD corrected for species
richness or total abundance (depending on
<code>abundance.weighted</code>). I am almost certain that I got the idea
for this from somewhere, but I can't find the reference: if you
published on this before 2012, please get in touch with me.
</p>
<p><code>.scheiner</code> has a different formula for the case where
<code>q</code> is equal to 1 (check the code if interested). The nature
of its definition means that values very close to, but not exactly
equal to, 1 may be extremely large or extremely small. This is a
feature, not a bug, and an inherent aspect of its definition. Check
the formula in the code for more information!
</p>


<h3>Note</h3>

<p>Many (but not all) of these functions are fairly trivial
wrappers around functions in other packages. In the citations for
each metric, * indicates a function that's essentially written in
<code>picante</code>. The Pagel family of measures are also fairly
trivial wrapper around <code>caper</code> code, functional
dissimilarity <code>FD</code> code, <code>gamma</code>, and <code>ape</code>
code. I can't demand it, but I would be grateful if you would cite these
authors when using these wrappers.
</p>
<p>The <code>pez.shape</code>, <code>pez.evenness</code>,
<code>pez.dispersion</code>, and <code>pez.dissimilarity</code>
wrapper functions go to some trouble to stop you calculating
metrics using inappropriate data (see their notes). These functions
give you access to the underlying code within <code>pez</code>; there is
nothing I can do to stop you calculating a metric that, in my
opinion, doesn't make any sense. You have been warned :D
</p>
<p>If you're a developer hoping to make your metric(s) work in this
framework, please use the argument naming convention for arguments
described in this help file, and use the <code>...</code> operator in
your definition. That way functions that don't need particular
arguments can co-exist peacefully with those that do. The first
argument to one of these functions should <em>always</em> be a
<code>comparative.comm</code> object; there is no method dispatch
on any of these functions and I foresee future pain without this
rule.
</p>


<h3>References</h3>

<p><code>eed,hed</code> (i.e., <em>Eed, Hed</em>) Cadotte M.W.,
Davies T.J., Regetz J., Kembel S.W., Cleland E. &amp; Oakley
T.H. (2010). Phylogenetic diversity metrics for ecological
communities: integrating species richness, abundance and
evolutionary history. Ecology Letters, 13, 96-105.
</p>
<p><code>PSV,PSR,PSE</code> Helmus M.R., Bland T.J., Williams
C.K. &amp; Ives A.R. (2007). Phylogenetic measures of
biodiversity. American Naturalist, 169, E68-E83.
</p>
<p><code>PD</code> Faith D.P. (1992). Conservation evaluation
and phylogenetic diversity. Biological Conservation, 61, 1-10.
</p>
<p><code>gamma</code> Pybus O.G. &amp; Harvey P.H. (2000) Testing
macro-evolutionary models using incomplete molecular
phylogenies. _Proceedings of the Royal Society of London. Series
B. Biological Sciences 267: 2267–2272.
</p>
<p><code>taxon</code> Clarke K.R. &amp; Warwick R.M. (1998). A
taxonomic distinctness index and its statistical
properties. J. Appl. Ecol., 35, 523-531.
</p>
<p><code>eigen.sum</code> Diniz-Filho J.A.F., Cianciaruso M.V.,
Rangel T.F. &amp; Bini L.M. (2011). Eigenvector estimation of
phylogenetic and functional diversity. Functional Ecology, 25,
735-744.
</p>
<p><code>entropy</code> Allen B., Kon M. &amp; Bar-Yam Y. (2009). A
New Phylogenetic Diversity Measure Generalizing the Shannon Index
and Its Application to Phyllostomid Bats. The American Naturalist,
174, 236-243.
</p>
<p><code>pae,aed,iac,haed,eaed</code> Cadotte M.W., Davies T.J.,
Regetz J., Kembel S.W., Cleland E. &amp; Oakley
T.H. (2010). Phylogenetic diversity metrics for ecological
communities: integrating species richness, abundance and
evolutionary history. Ecology Letters, 13, 96-105.
</p>
<p><code>scheiner</code> Scheiner, S.M. (20120). A metric of
biodiversity that integrates abundance, phylogeny, and function.
Oikos, 121, 1191-1202.
</p>
<p><code>rao</code> Webb C.O. (2000). Exploring the phylogenetic
structure of ecological communities: An example for rain forest
trees. American Naturalist, 156, 145-155.
</p>
<p><code>lambda,delta,kappa</code> Mark Pagel (1999) Inferring
the historical patterns of biological evolution. Nature 6756(401):
877–884.
</p>
<p><code>unifrac</code> Lozupone C.A. &amp; Knight
R. (2005). UniFrac: a new phylogenetic method for comparing
microbial communities. Applied and Environmental Microbiology, 71,
8228-8235.
</p>
<p><code>pcd</code> Ives A.R. &amp; Helmus M.R. (2010). Phylogenetic
metrics of community similarity. The American Naturalist, 176,
E128-E142.
</p>
<p><code>comdist</code> C.O. Webb, D.D. Ackerly, and
S.W. Kembel. 2008. Phylocom: software for the analysis of
phylogenetic community structure and trait
evolution. Bioinformatics 18:2098-2100.
</p>
<p><code>phylosor</code> Bryant J.A., Lamanna C., Morlon H.,
Kerkhoff A.J., Enquist B.J. &amp; Green J.L. (2008). Microbes on
mountainsides: Contrasting elevational patterns of bacterial and
plant diversity. Proceedings of the National Academy of Sciences of
the United States of America, 105, 11505-11511.
</p>
<p><code>d</code> Fritz S.A. &amp; Purvis A. (2010). Selectivity in
Mammalian Extinction Risk and Threat Types: a New Measure of
Phylogenetic Signal Strength in Binary Traits. Conservation
Biology, 24, 1042-1051.
</p>
<p><code>sesmpd,sesmntd</code> Webb C.O. (2000). Exploring the
phylogenetic structure of ecological communities: An example for
rain forest trees. American Naturalist, 156, 145-155.
</p>
<p><code>innd,mipd</code> Ness J.H., Rollinson E.J. &amp; Whitney
K.D. (2011). Phylogenetic distance can predict susceptibility to
attack by natural enemies. Oikos, 120, 1327-1334.
</p>
<p><code>PE</code> Rosauer, D. A. N., Laffan, S. W., Crisp,
M. D., Donnellan, S. C., &amp; Cook, L. G. (2009). Phylogenetic
endemism: a new approach for identifying geographical
concentrations of evolutionary history. Molecular Ecology,
18(19), 4061-4072.
</p>
<p><code>BED</code> Cadotte, M. W., &amp; Jonathan Davies,
T. (2010). Rarest of the rare: advances in combining
evolutionary distinctiveness and scarcity to inform
conservation at biogeographical scales. Diversity and
Distributions, 16(3), 376-385.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(laja)
data &lt;- comparative.comm(invert.tree, river.sites)
.psv(data)
</code></pre>


</div>