<div class="container">

<table style="width: 100%;"><tr>
<td>read.phyDat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Import and export sequence alignments</h2>

<h3>Description</h3>

<p>These functions read and write sequence alignments.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read.phyDat(file, format = "phylip", type = "DNA", ...)

write.phyDat(x, file, format = "phylip", colsep = "", nbcol = -1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>a file name specified by either a variable of mode character, or
a double-quoted string.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>format</code></td>
<td>
<p>File format of the sequence alignment (see details).  Several
popular formats are supported: "phylip", "interleaved", "sequential",
"clustal", "fasta" or "nexus", or any unambiguous abbreviation of these.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>Type of sequences ("DNA", "AA", "CODON" or "USER").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An object of class <code>phyDat</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colsep</code></td>
<td>
<p>a character used to separate the columns (a single space by
default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nbcol</code></td>
<td>
<p>a numeric specifying the number of columns per row (-1 by
default); may be negative implying that the nucleotides are printed on a
single line.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>write.phyDat</code> calls the function <code>write.dna</code> or
<code>write.nexus.data</code> and <code>read.phyDat</code> calls the function
<code>read.dna</code> or <code>read.nexus.data</code>, so see
for more details over there.
</p>
<p>You may import data directly with <code>read.dna</code> or
<code>read.nexus.data</code> and convert the data to class phyDat.
</p>


<h3>Value</h3>

<p><code>read.phyDat</code> returns an object of class phyDat,
<code>write.phyDat</code> write an alignment to a file.
</p>


<h3>Author(s)</h3>

<p>Klaus Schliep <a href="mailto:klaus.schliep@gmail.com">klaus.schliep@gmail.com</a>
</p>


<h3>References</h3>


<p><a href="https://www.ncbi.nlm.nih.gov/blast/fasta.shtml">https://www.ncbi.nlm.nih.gov/blast/fasta.shtml</a> Felsenstein, J. (1993)
Phylip (Phylogeny Inference Package) version 3.5c. Department of Genetics,
University of Washington.
<a href="https://phylipweb.github.io/phylip/">https://phylipweb.github.io/phylip/</a>
</p>


<h3>See Also</h3>

<p><code>read.dna</code>, <code>read.GenBank</code>,
<code>phyDat</code>, <code>read.alignment</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">fdir &lt;- system.file("extdata/trees", package = "phangorn")
primates &lt;- read.phyDat(file.path(fdir, "primates.dna"),
                        format = "interleaved")
</code></pre>


</div>