<div class="container">

<table style="width: 100%;"><tr>
<td>bab</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Branch and bound for finding all most parsimonious trees</h2>

<h3>Description</h3>

<p><code>bab</code> finds all most parsimonious trees.
</p>


<h3>Usage</h3>

<pre><code class="language-R">bab(data, tree = NULL, trace = 0, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>an object of class phyDat.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tree</code></td>
<td>
<p>a phylogenetic tree an object of class phylo, otherwise a
pratchet search is performed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trace</code></td>
<td>
<p>defines how much information is printed during optimization.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Further arguments passed to or from other methods</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This implementation is very slow and depending on the data may take very
long time. In the worst case all <code class="reqn">(2n-5)!! = 1 \times 3 \times 5
\times \ldots \times (2n-5)</code> possible
trees have to be examined, where n is the number of species / tips. For ten
species there are already 2027025 tip-labelled unrooted trees. It only uses
some basic strategies to find a lower and upper bounds similar to penny from
phylip. <code>bab</code> uses a very basic heuristic approach of MinMax Squeeze
(Holland et al. 2005) to improve the lower bound.  On the positive side
<code>bab</code> is not like many other implementations restricted to binary or
nucleotide data.
</p>


<h3>Value</h3>

<p><code>bab</code> returns all most parsimonious trees in an object of class
<code>multiPhylo</code>.
</p>


<h3>Author(s)</h3>

<p>Klaus Schliep <a href="mailto:klaus.schliep@gmail.com">klaus.schliep@gmail.com</a> based on work on Liam
Revell
</p>


<h3>References</h3>

<p>Hendy, M.D. and Penny D. (1982) Branch and bound algorithms to
determine minimal evolutionary trees.  <em>Math. Biosc.</em> <b>59</b>,
277-290
</p>
<p>Holland, B.R., Huber, K.T. Penny, D. and Moulton, V. (2005) The MinMax
Squeeze: Guaranteeing a Minimal Tree for Population Data, <em>Molecular
Biology and Evolution</em>, <b>22</b>, 235–242
</p>
<p>White, W.T. and Holland, B.R. (2011) Faster exact maximum parsimony search
with XMP. <em>Bioinformatics</em>, <b>27(10)</b>,1359–1367
</p>


<h3>See Also</h3>

<p><code>pratchet</code>, <code>dfactorial</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(yeast)
dfactorial(11)
# choose only the first two genes
gene12 &lt;- yeast[, 1:3158]
trees &lt;- bab(gene12)

</code></pre>


</div>