<div class="container">

<table style="width: 100%;"><tr>
<td>phyDat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Conversion among Sequence Formats</h2>

<h3>Description</h3>

<p>These functions transform several DNA formats into the <code>phyDat</code> format.
<code>allSitePattern</code> generates an alignment of all possible site patterns.
</p>


<h3>Usage</h3>

<pre><code class="language-R">phyDat(data, type = "DNA", levels = NULL, return.index = TRUE, ...)

as.phyDat(x, ...)

## S3 method for class 'factor'
as.phyDat(x, ...)

## S3 method for class 'DNAbin'
as.phyDat(x, ...)

## S3 method for class 'AAbin'
as.phyDat(x, ...)

## S3 method for class 'alignment'
as.phyDat(x, type = "DNA", ...)

phyDat2alignment(x)

## S3 method for class 'MultipleAlignment'
as.phyDat(x, ...)

## S3 method for class 'AAStringSet'
as.phyDat(x, ...)

## S3 method for class 'DNAStringSet'
as.phyDat(x, ...)

as.StringSet(x, ...)

## S3 method for class 'phyDat'
as.StringSet(x, ...)

## S3 method for class 'phyDat'
as.MultipleAlignment(x, ...)

## S3 method for class 'phyDat'
as.character(x, allLevels = TRUE, ...)

## S3 method for class 'phyDat'
as.data.frame(x, ...)

## S3 method for class 'phyDat'
as.DNAbin(x, ...)

## S3 method for class 'phyDat'
as.AAbin(x, ...)

genlight2phyDat(x, ambiguity = NA)

acgt2ry(obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>An object containing sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>Type of sequences ("DNA", "AA", "CODON" or "USER").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>levels</code></td>
<td>
<p>Level attributes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return.index</code></td>
<td>
<p>If TRUE returns a index of the site patterns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An object containing sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allLevels</code></td>
<td>
<p>return original data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ambiguity</code></td>
<td>
<p>character for ambiguous character and no contrast is
provided.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>obj</code></td>
<td>
<p>as object of class phyDat</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If <code>type</code> "USER" a vector has to be give to <code>levels</code>. For example
c("a", "c", "g", "t", "-") would create a data object that can be used in
phylogenetic analysis with gaps as fifth state.  There is a more detailed
example for specifying "USER" defined data formats in the vignette
"phangorn-specials".
</p>
<p><code>acgt2ry</code> converts a <code>phyDat</code> object of nucleotides into an binary
ry-coded dataset.
</p>


<h3>Value</h3>

<p>The functions return an object of class <code>phyDat</code>.
</p>


<h3>Author(s)</h3>

<p>Klaus Schliep <a href="mailto:klaus.schliep@gmail.com">klaus.schliep@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code>DNAbin</code>, <code>as.DNAbin</code>,
<code>baseFreq</code>, <code>glance.phyDat</code>,
<code>read.dna</code>, <code>read.nexus.data</code>
and the chapter 1 in the <code>vignette("phangorn-specials",
package="phangorn")</code> and the example of <code>pmlMix</code> for the use of
<code>allSitePattern</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(Laurasiatherian)
class(Laurasiatherian)
Laurasiatherian
# transform as characters
LauraChar &lt;- as.character(Laurasiatherian)
# and back
Laura &lt;- phyDat(LauraChar)
all.equal(Laurasiatherian, Laura)
LauraDNAbin &lt;- as.DNAbin(Laurasiatherian)
all.equal(Laurasiatherian, as.phyDat(LauraDNAbin))

</code></pre>


</div>