<div class="container">

<table style="width: 100%;"><tr>
<td>plotAnc</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot ancestral character on a tree</h2>

<h3>Description</h3>

<p><code>plotAnc</code> plots a phylogeny and adds character to the nodes. Either
takes output from  <code>ancestral.pars</code> or <code>ancestral.pml</code> or from an
alignment where there are node labels in the tree match the constructed
sequences in the alignment.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plotAnc(x, i = 1, type = "phylogram", ..., col = NULL, cex.pie = 0.5,
  pos = "bottomright", scheme = NULL)

plotSeqLogo(x, node = getRoot(x$tree), start = 1, end = 10,
  scheme = "Ape_NT", ...)

add_mutations(x, frame = "none", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of class <code>ancestral</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>i</code></td>
<td>
<p>plots the i-th site.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>a character string specifying the type of phylogeny to be drawn;
it must be one of "phylogram" (the default), "cladogram", "fan", "unrooted",
"radial", "tidy", or any unambiguous abbreviation of these.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>a vector containing the colors for all possible states.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.pie</code></td>
<td>
<p>a numeric defining the size of the pie graphs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos</code></td>
<td>
<p>a character string defining the position of the legend.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scheme</code></td>
<td>
<p>a predefined color scheme. For amino acid options are "Ape_AA",
"Zappo_AA", "Clustal", "Polarity" and "Transmembrane_tendency", for
nucleotides "Ape_NT" and"RY_NT". Names can be abbreviated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>node</code></td>
<td>
<p>to plot for which the probabilities should be plotted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start</code></td>
<td>
<p>start position to plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>end</code></td>
<td>
<p>end position to plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>frame</code></td>
<td>
<p>a character string specifying the kind of frame to be printed
around the text. See <code>edgelabels</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For further details see vignette("Ancestral").
</p>


<h3>Value</h3>

<p><code>plotAnc</code> returns silently x.
</p>
<p><code>plotSeqLogo</code> returns a ggplot object.
</p>
<p><code>add_mutations</code> adds the position and and changes of possible
mutations to a phylogeny.
</p>


<h3>Author(s)</h3>

<p>Klaus Schliep <a href="mailto:klaus.schliep@gmail.com">klaus.schliep@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code>ancestral.pml</code>, <code>plot.phylo</code>,
<code>image.DNAbin</code>, <code>image.AAbin</code>
<code>ggseqlogo</code>, <code>edgelabels</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
example(NJ)
# generate node labels to ensure plotting will work
tree &lt;- makeNodeLabel(tree)
anc.p &lt;- anc_pars(tree, Laurasiatherian)
# plot the third character
plotAnc(anc.p, 3, pos="topright")
plotSeqLogo(anc.p, node="Node10", 1, 25)

data(chloroplast)
tree &lt;- pratchet(chloroplast,  maxit=10, trace=0)
tree &lt;- makeNodeLabel(tree)
anc.ch &lt;- anc_pars(tree, chloroplast)
image(as.phyDat(anc.ch)[, 1:25])
plotAnc(anc.ch, 21, scheme="Ape_AA", pos="topleft")
plotAnc(anc.ch, 21, scheme="Clustal", pos="topleft")
plotSeqLogo(anc.ch, node="Node1", 1, 25, scheme="Clustal")


data(woodmouse)
tree &lt;- pml_bb(woodmouse, "JC", rearrangement="NNI")$tree |&gt; midpoint()
woodmouse_aa &lt;- trans(woodmouse, 2) |&gt; as.phyDat()
anc_aa &lt;- anc_pars(tree, woodmouse_aa)
plot(tree, direction="downwards")
add_mutations(anc_aa)

</code></pre>


</div>