<div class="container">

<table style="width: 100%;"><tr>
<td>dna2codon</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Translate nucleic acid sequences into codons</h2>

<h3>Description</h3>

<p>The function transforms <code>dna2codon</code> DNA sequences to codon sequences,
<code>codon2dna</code> transform the other way. <code>dna2codon</code> translates
nucleotide to amino acids using the function <code>trans</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">dna2codon(x, codonstart = 1, code = 1, ambiguity = "---", ...)

codon2dna(x)

dna2aa(x, codonstart = 1, code = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An object containing sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>codonstart</code></td>
<td>
<p>an integer giving where to start the translation. This
should be 1, 2, or 3, but larger values are accepted and have for effect to
start the translation further within the sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>code</code></td>
<td>
<p>The ncbi genetic code number for translation (see details).
By default the standard genetic code is used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ambiguity</code></td>
<td>
<p>character for ambiguous character and no contrast is
provided.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The following genetic codes are described here. The number preceding each
corresponds to the code argument.
</p>

<table>
<tr>
<td style="text-align: right;">
  1 </td>
<td style="text-align: left;"> standard   </td>
</tr>
<tr>
<td style="text-align: right;">
  2 </td>
<td style="text-align: left;"> vertebrate.mitochondrial </td>
</tr>
<tr>
<td style="text-align: right;">
  3 </td>
<td style="text-align: left;"> yeast.mitochondrial </td>
</tr>
<tr>
<td style="text-align: right;">
  4 </td>
<td style="text-align: left;"> protozoan.mitochondrial+mycoplasma </td>
</tr>
<tr>
<td style="text-align: right;">
  5 </td>
<td style="text-align: left;"> invertebrate.mitochondrial </td>
</tr>
<tr>
<td style="text-align: right;">
  6 </td>
<td style="text-align: left;"> ciliate+dasycladaceal </td>
</tr>
<tr>
<td style="text-align: right;">
  9 </td>
<td style="text-align: left;"> echinoderm+flatworm.mitochondrial </td>
</tr>
<tr>
<td style="text-align: right;">
  10 </td>
<td style="text-align: left;"> euplotid </td>
</tr>
<tr>
<td style="text-align: right;">
  11 </td>
<td style="text-align: left;"> bacterial+plantplastid </td>
</tr>
<tr>
<td style="text-align: right;">
  12 </td>
<td style="text-align: left;"> alternativeyeast </td>
</tr>
<tr>
<td style="text-align: right;">
  13 </td>
<td style="text-align: left;"> ascidian.mitochondrial </td>
</tr>
<tr>
<td style="text-align: right;">
  14 </td>
<td style="text-align: left;"> alternativeflatworm.mitochondrial </td>
</tr>
<tr>
<td style="text-align: right;">
  15 </td>
<td style="text-align: left;"> blepharism </td>
</tr>
<tr>
<td style="text-align: right;">
  16 </td>
<td style="text-align: left;"> chlorophycean.mitochondrial </td>
</tr>
<tr>
<td style="text-align: right;">
  21 </td>
<td style="text-align: left;"> trematode.mitochondrial </td>
</tr>
<tr>
<td style="text-align: right;">
  22 </td>
<td style="text-align: left;"> scenedesmus.mitochondrial </td>
</tr>
<tr>
<td style="text-align: right;">
  23 </td>
<td style="text-align: left;"> thraustochytrium.mitochondria </td>
</tr>
<tr>
<td style="text-align: right;">
  24 </td>
<td style="text-align: left;"> Pterobranchia.mitochondrial </td>
</tr>
<tr>
<td style="text-align: right;">
  25 </td>
<td style="text-align: left;"> CandidateDivision.SR1+Gracilibacteria </td>
</tr>
<tr>
<td style="text-align: right;">
  26 </td>
<td style="text-align: left;"> Pachysolen.tannophilus
</td>
</tr>
</table>
<p>Alignment gaps and ambiguities are currently ignored and sites containing
these are deleted.
</p>


<h3>Value</h3>

<p>The functions return an object of class <code>phyDat</code>.
</p>


<h3>Author(s)</h3>

<p>Klaus Schliep <a href="mailto:klaus.schliep@gmail.com">klaus.schliep@gmail.com</a>
</p>


<h3>References</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/Taxonomy/taxonomyhome.html/index.cgi?chapter=cgencodes">https://www.ncbi.nlm.nih.gov/Taxonomy/taxonomyhome.html/index.cgi?chapter=cgencodes</a>
</p>


<h3>See Also</h3>

<p><code>trans</code>, <code>phyDat</code> and the chapter 4 in
the <code>vignette("phangorn-specials", package="phangorn")</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(Laurasiatherian)
class(Laurasiatherian)
Laurasiatherian
dna2codon(Laurasiatherian)

</code></pre>


</div>