<div class="container">

<table style="width: 100%;"><tr>
<td>readStandardGenotypes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read genotypes from file.</h2>

<h3>Description</h3>

<p>readStandardGenotypes can read genotypes from a number of input
formats for standard GWAS (binary plink, snptest, bimbam, gemma) or
simulation software (binary plink, hapgen2, genome). Alternatively, simple
text files (with specified delimiter) can be read. For more information on
the different file formats see <em>External genotype software and formats</em>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">readStandardGenotypes(
  N,
  filename,
  format = NULL,
  verbose = TRUE,
  sampleID = "ID_",
  snpID = "SNP_",
  delimiter = ","
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>N</code></td>
<td>
<p>Number [integer] of samples to simulate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filename</code></td>
<td>
<p>path/to/genotypefile [string] in plink, oxgen
(impute2/snptest/hapgen2), genome, bimbam or [delimiter]-delimited format (
for format information see <em>External genotype software and formats</em>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>format</code></td>
<td>
<p>Name [string] of genotype file format. Options are: "plink",
"oxgen", "genome", "bimbam" or "delim".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>[boolean] If TRUE, progress info is printed to standard out.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sampleID</code></td>
<td>
<p>Prefix [string] for naming samples (will be followed by
sample number from 1 to N when constructing id_samples).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snpID</code></td>
<td>
<p>Prefix [string] for naming SNPs (will be followed by SNP number
from 1 to NrSNP when constructing id_snps).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>delimiter</code></td>
<td>
<p>Field separator [string] of genotype file when
format == "delim".</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The file formats and software formates supported are described
below. For large external genotypes, consider the option to only read
randomly selected, causal SNPs into memory via getCausalSNPs.
</p>


<h3>Value</h3>

<p>Named list of [NrSamples X NrSNPs] genotypes (genotypes), their
[NrSNPs] SNP IDs (id_snps), their [NrSamples] sample IDs (id_samples) and
format-specific additional files (such as format-specific genotypes encoding
or sample information; might be used for output writing).
</p>


<h3>External genotype software and formats</h3>



<h4>Formats:</h4>


<ul>
<li>
<p> PLINK: consists of three files: .bed, .bim and .fam. When specifying
the filepath, only the core of the name without the ending should be
specified (i.e. for geno.bed, geno.bim and geno.fam, geno should be
specified). When reading data from plink files, the absolute path to the
plink-format file has to be provided, tilde expansion not provided.
From <a href="https://www.cog-genomics.org/plink/1.9/formats">https://www.cog-genomics.org/plink/1.9/formats</a>:  The .bed
files contain the  primary representation of genotype calls at biallelic
variants in a binary format. The .bim is a text file with no header
line, and one line  per variant with the following six fields: i) Chromosome
code (either an integer, or 'X'/'Y'/'XY'/'MT'; '0' indicates unknown) or
name, ii) Variant identifier, iii) Position in morgans or centimorgans (safe
to use dummy value of '0'), iv) Base-pair coordinate (normally 1-based, but
0 ok; limited to 231-2), v) Allele 1 (corresponding to clear bits in .bed;
usually minor), vi) Allele 2 (corresponding to set bits in .bed; usually
major). The .fam file is a text file with no header line, and one line per
sample with the following six fields: i) Family ID ('FID'), ii), Within-
family ID ('IID'; cannot be '0'), iii) Within-family ID of father ('0' if
father isn't in dataset, iv) within-family ID of mother ('0' if mother isn't
in dataset), v) sex code ('1' = male, '2' = female, '0' = unknown), vi)
Phenotype value ('1' = control, '2' = case, '-9'/'0'/non-numeric = missing
data if case/control)
</p>
</li>
<li>
<p> oxgen: consists of two files: the space-separated genotype file ending
in .gen and the space-separated sample file ending in .sample. When
specifying the filepath, only the core of the name without the ending should
be specified (i.e. for geno.gen and geno.sample, geno should be specified).
From
<a href="https://www.well.ox.ac.uk/~gav/snptest/#input_file_formats">https://www.well.ox.ac.uk/~gav/snptest/#input_file_formats</a>:
The genotype file stores data on a one-line-per-SNP format. The first five
entries of each line should be the SNP ID, RS ID of the SNP, base-pair
position of the SNP, the allele coded A and the allele coded B. The SNP ID
can be used to denote the chromosome number of each SNP. The next three
numbers on the line should be the probabilities of the three genotypes AA,
AB and BB at the SNP for the first individual in the cohort. The next three
numbers should be the genotype probabilities for the second individual in the
cohort. The next three numbers are for the third individual and so on. The
order of individuals in the genotype file should match the order of the
individuals in the sample file. The sample file has three parts (a) a header
line detailing the names of the columns in the file, (b) a line detailing the
types of variables stored in each column, and (c) a line for each individual
detailing the information for that individual. For more information on the
sample file visit the above url or see writeStandardOutput.
</p>
</li>
<li>
<p> genome: The entire output of genome can be saved via 'genome -options &gt;
outputfile'. The /path/to/outputfile should be provided and this function
extracts the relevant genotype information from this output file.
<a href="http://csg.sph.umich.edu/liang/genome/">http://csg.sph.umich.edu/liang/genome/</a>
</p>
</li>
<li>
<p> bimbam: Mean genotype file format of bimbam which is a single, comma-
separated file, without information on individuals. From the documentation
for bimbam at <a href="http://www.haplotype.org/software.html">http://www.haplotype.org/software.html</a>: the first column
of the mean genotype files is the SNP ID, the second and third columns are
allele types with minor allele first. The remaining columns are the mean
genotypes of different individuals â€“ numbers between 0 and 2 that represents
the (posterior) mean genotype, or dosage of the minor allele. </p>
</li>
<li>
<p> delim: a
[delimter]-delimited file of [(NrSNPs+1) x (NrSamples+1)] genotypes with the
snpIDs in the first column and the sampleIDs in the first row and genotypes
encoded as numbers between 0 and 2 representing the (posterior) mean
genotype, or dosage of the minor allele. Can be user-genotypes or genotypes
simulated with foward-time algorithms such as simupop
(<a href="http://simupop.sourceforge.net/Main/HomePage">http://simupop.sourceforge.net/Main/HomePage</a>)
that allow for user-specified output formats.
</p>
</li>
</ul>
<h4>Genotype simulation characteristics:</h4>


<ul>
<li>
<p> PLINK: simple, bi-allelelic genotypes without LD structure, details can
be found at <a href="https://www.cog-genomics.org/plink/1.9/input#simulate">https://www.cog-genomics.org/plink/1.9/input#simulate</a>.
</p>
</li>
<li>
<p> Hapgen2: resampling-based genotype simulation, details can be found at
<a href="http://mathgen.stats.ox.ac.uk/genetics_software/hapgen/hapgen2.html">http://mathgen.stats.ox.ac.uk/genetics_software/hapgen/hapgen2.html</a>.
</p>
</li>
<li>
<p> Genome: coalescent-based genotype simulation, details can be found at
<a href="http://csg.sph.umich.edu/liang/genome/GENOME-manual.pdf">http://csg.sph.umich.edu/liang/genome/GENOME-manual.pdf</a>.
</p>
</li>
</ul>
<p>Examples on how to call these genotype simulation tools can be found in the
vignette <em>sample-scripts-external-genotype-simulation</em>.</p>



<h3>Examples</h3>

<pre><code class="language-R"># Genome format
filename_genome  &lt;- system.file("extdata/genotypes/genome/",
"genotypes_genome.txt",
package = "PhenotypeSimulator")
data_genome &lt;- readStandardGenotypes(N=100, filename_genome, format ="genome")

filename_hapgen  &lt;- system.file("extdata/genotypes/hapgen/",
"genotypes_hapgen.controls.gen",
package = "PhenotypeSimulator")
filename_hapgen &lt;- gsub("\\.gen", "", filename_hapgen)
data_hapgen &lt;- readStandardGenotypes(N=100, filename_hapgen, format='oxgen')

filename_plink  &lt;- system.file("extdata/genotypes/plink/",
"genotypes_plink.bed",
package = "PhenotypeSimulator")
filename_plink &lt;- gsub("\\.bed", "", filename_plink)
data_plink &lt;- readStandardGenotypes(N=100, filename=filename_plink,
format="plink")

filename_delim  &lt;- system.file("extdata/genotypes/",
"genotypes_chr22.csv",
package = "PhenotypeSimulator")
data_delim &lt;- readStandardGenotypes(N=50, filename=filename_delim,
format="delim")
</code></pre>


</div>