<div class="container">

<table style="width: 100%;"><tr>
<td>simulateGenotypes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate bi-allelic genotypes.</h2>

<h3>Description</h3>

<p>Simulate bi-allelic genotypes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">simulateGenotypes(
  N,
  NrSNP = 5000,
  frequencies = c(0.1, 0.2, 0.4),
  sampleID = "ID_",
  snpID = "SNP_",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>N</code></td>
<td>
<p>Number of samples for which to simulate bi-allelic genotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>NrSNP</code></td>
<td>
<p>Number of SNPs to simulate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>frequencies</code></td>
<td>
<p>Vector of allele frequencies [double] from which to
sample.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sampleID</code></td>
<td>
<p>Prefix [string] for naming samples (will be followed by
sample number from 1 to N when constructing id_samples).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snpID</code></td>
<td>
<p>Prefix [string] for naming SNPs (will be followed by SNP number
from 1 to NrSNP when constructing id_snps).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>[boolean] If TRUE, progress info is printed to standard out.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Named list with [N x NrSNP] matrix of simulated genotypes
(genotypes), their SNP frequencies (freq), a vector of sample IDs
(id_samples) and a vector of SNP IDs (id_snps).
</p>


<h3>See Also</h3>

<p><code>standardiseGenotypes</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">N10NrSNP10 &lt;- simulateGenotypes(N=10, NrSNP=10)
N10NrSNP10 &lt;- simulateGenotypes(N=10, NrSNP=10,
frequencies=c(0.2,0.3,0.4))
</code></pre>


</div>