<div class="container">

<table style="width: 100%;"><tr>
<td>standardiseGenotypes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Standardise genotypes.</h2>

<h3>Description</h3>

<p>Genotypes are standardised as described in Yang et al:
snp_standardised = (snp - 2 * ref_allele_freq)/
sqrt(2 * ref_allele_freq * alt_allele_freq).
</p>


<h3>Usage</h3>

<pre><code class="language-R">standardiseGenotypes(geno, impute = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>[N x NrSNP] Matrix/dataframe of genotypes [integer]/[double].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>impute</code></td>
<td>
<p>[logical] Indicating if missing genotypes should be imputed; if
set FALSE and data contains missing values,  <code>standardiseGenotypes</code> will
return an error.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Missing genotypes can be mean-imputed and rounded to nearest integer
before standardisation. If genotypes contain missing values and impute is set
to FALSE, <code>standardiseGenotypes</code> will return an error.
</p>


<h3>Value</h3>

<p>[N x NrSNP] Matrix of standardised genotypes [double].
</p>


<h3>References</h3>

<p>Yang, J., Lee, S.H., Goddard, M.E., Visscher, P.M. (2011) GCTA:
a tool for genome-wide complex trait analysis, AJHG: 88
</p>


<h3>See Also</h3>

<p><code>getAlleleFrequencies</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">geno &lt;- cbind(rbinom(2000, 2, 0.3), rbinom(2000, 2, 0.4),rbinom(2000, 2, 0.5))
geno_sd &lt;- standardiseGenotypes(geno)
</code></pre>


</div>