<div class="container">

<table style="width: 100%;"><tr>
<td>getGeneticAssociations</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Perform association tests between phenotype risk scores and genotypes</h2>

<h3>Description</h3>

<p>The association test for each disease-variant pair is based on a linear
model, with the phenotype risk score as the dependent variable.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getGeneticAssociations(
  scores,
  genotypes,
  demos,
  diseaseVariantMap,
  lmFormula,
  modelType = c("genotypic", "additive", "dominant", "recessive"),
  level = 0.95,
  dopar = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>scores</code></td>
<td>
<p>A data.table of phenotype risk scores. Must have columns
<code>person_id</code>, <code>disease_id</code>, <code>score</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotypes</code></td>
<td>
<p>A matrix or 'BEDMatrix' object containing genetic data, with
rownames corresponding to <code>person_id</code>s in <code>demos</code> and <code>scores</code>, and
colnames corresponding to <code>variant_id</code>s in <code>diseaseVariantMap</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>demos</code></td>
<td>
<p>A data.table of characteristics for each person in the cohort.
Must have column <code>person_id</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>diseaseVariantMap</code></td>
<td>
<p>A data.table indicating which genetic variants to
test for association with phenotype risk scores for which diseases. Must
have columns <code>disease_id</code> and <code>variant_id</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lmFormula</code></td>
<td>
<p>A formula representing the linear model (excluding the term
for genotype) to use for the association tests. All terms in the formula
must correspond to columns in <code>demos</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>modelType</code></td>
<td>
<p>A string indicating how to encode genotype in the model.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>level</code></td>
<td>
<p>A number indicating the level of the confidence interval.
Default is 0.95.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dopar</code></td>
<td>
<p>Logical indicating whether to run calculations in parallel if
a parallel backend is already set up, e.g., using
<code>doParallel::registerDoParallel()</code>. Recommended to minimize runtime.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data.table of statistics for the association tests (if a model
fails to converge, NAs will be reported):
</p>

<ul>
<li> <p><code>disease_id</code>: Disease identifier
</p>
</li>
<li> <p><code>variant_id</code>: Variant identifier
</p>
</li>
<li> <p><code>n_total</code>: Number of persons with non-missing genotype data for the
given variant.
</p>
</li>
<li> <p><code>n_wt</code>: Number of persons homozygous for the wild-type allele.
</p>
</li>
<li> <p><code>n_het</code>: Number of persons having one copy of the alternate allele.
</p>
</li>
<li> <p><code>n_hom</code>: Number of persons homozygous for the alternate allele.
</p>
</li>
<li> <p><code>beta</code>: Coefficient for the association of genotype with score
</p>
</li>
<li> <p><code>se</code>: Standard error for <code>beta</code>
</p>
</li>
<li> <p><code>pval</code>: P-value for <code>beta</code> being non-zero
</p>
</li>
<li> <p><code>ci_lower</code>: Lower bound of the confidence interval for <code>beta</code>
</p>
</li>
<li> <p><code>ci_upper</code>: Upper bound of the confidence interval for <code>beta</code>
</p>
</li>
</ul>
<p>If <code>modelType</code> is "genotypic", the data.table will include separate
statistics for heterozygous and homozygous genotypes.
</p>


<h3>See Also</h3>

<p><code>stats::lm()</code>, <code>stats::confint()</code>, <code>getScores()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">library('data.table')
library('BEDMatrix')

# map ICD codes to phecodes
phecodeOccurrences = getPhecodeOccurrences(icdSample)

# calculate weights
weights = getWeights(demoSample, phecodeOccurrences)

# OMIM disease IDs for which to calculate phenotype risk scores
diseaseId = 154700

# map diseases to phecodes
diseasePhecodeMap = mapDiseaseToPhecode()

# calculate scores
scores = getScores(weights, diseasePhecodeMap[disease_id == diseaseId])

# map diseases to genetic variants
nvar = 10
diseaseVariantMap = data.table(disease_id = diseaseId, variant_id = paste0('snp', 1:nvar))

# load sample genetic data
npop = 50
genoSample = BEDMatrix(system.file('extdata', 'geno_sample.bed', package = 'phers'))
colnames(genoSample) = paste0('snp', 1:nvar)
rownames(genoSample) = 1:npop

# run genetic association tests
genoStats = getGeneticAssociations(
  scores, genoSample, demoSample, diseaseVariantMap, lmFormula = ~ sex,
  modelType = 'additive')
</code></pre>


</div>