<div class="container">

<table style="width: 100%;"><tr>
<td>phyclust-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Phyloclustering – Phylogenetic Clustering</h2>

<h3>Description</h3>

<p>This package <span class="pkg">phyclust</span> (Chen 2011) implements an novel approach
combining model-based clusterings
and phylogenetics to classify DNA sequences and SNP sequences.
Based on evolution models, sequences are assumed to follow a mutation
process/distribution clouding around an unknown center ancestor.
Based on Continuous Time Markov Chain Theory, mixture distributions are
established to model/classify subpopulations or population structures.
</p>
<p>The kernel part of the package are implemented in C.
EM algorithms are performed to find the maximum likelihood estimators.
Initialization methods for EM algorithms are also established.
Several evolution models are also developed.
</p>
<p><code>ms</code> (Hudson 2002) and <code>seq-gen</code> (Rambaut and Grassly 1997)
are two useful programs to generate coalescent trees and sequences, and
both are merged into <span class="pkg">phyclust</span>. <code>baseml</code> of PAML (Yang 1997, 2007)
is also ported into <span class="pkg">phyclust</span> and it is a program to find a phylogenetic
tree by maximizing likelihood. Hap-Clustering method (Tzeng 2005) for
haplotype grouping is also incorporated into <span class="pkg">phyclust</span>.
</p>
<p>Type <code>help(package = phyclust)</code> to see a list of major
functions for which further documentations are available. The on-line
detail instructions are also available and the link is given below in the
‘References’ section.
</p>
<p>Some C and R functions and R classes of the <span class="pkg">ape</span> package are also
required and modified in <span class="pkg">phyclust</span>.
</p>


<h3>Details</h3>

<p>The main function is <code>phyclust</code> controlled by an object
<code>.EMC</code> generated by a function <code>.EMControl</code>,
and <code>find.best</code> can find the best solution by repeating
<code>phyclust</code> with different initializations.
</p>
<p><code>ms</code> and <code>seqgen</code> can generate trees and sequences
based on varied conditions, and they can jointly perform simulations.
</p>
<p><code>paml.baseml</code> can estimate trees based on sequences.
</p>
<p><code>haplo.post.prob</code> is a modified version of Tzeng's method
for haplotype grouping which uses a evolution approach to group SNP
sequences.
</p>
<p>Some tool functions of the <span class="pkg">ape</span> package are utilized in this package
to perform trees in plots, check object types, and read sequence data.
</p>


<h3>Author(s)</h3>

<p>Wei-Chen Chen <a href="mailto:wccsnow@gmail.com">wccsnow@gmail.com</a>
</p>


<h3>References</h3>

<p>Phylogenetic Clustering Website:
<a href="https://snoweye.github.io/phyclust/">https://snoweye.github.io/phyclust/</a>
</p>
<p>Chen, W.-C. (2011)
“Overlapping codon model, phylogenetic clustering, and alternative
partial expectation conditional maximization algorithm”,
<em>Ph.D. Diss., Iowa Stat University</em>.
</p>
<p>Hudson, R.R. (2002)
“Generating Samples under a Wright-Fisher Neutral Model of Genetic
Variation”,
<em>Bioinformatics</em>, <b>18</b>, 337-338.
<a href="http://home.uchicago.edu/~rhudson1/source.html">http://home.uchicago.edu/~rhudson1/source.html</a>
</p>
<p>Rambaut, A. and Grassly, N.C. (1997)
“Seq-Gen: An Application for the Monte Carlo Simulation of DNA
Sequence Evolution along Phylogenetic Trees”,
<em>Computer Applications In The Biosciences</em>, <b>13:3</b>, 235-238.
<a href="http://tree.bio.ed.ac.uk/software/seqgen/">http://tree.bio.ed.ac.uk/software/seqgen/</a>
</p>
<p>Yang, Z. (1997)
“PAML: a program package for phylogenetic analysis by maximum
likelihood”,
<em>Computer Applications in BioSciences</em>, <b>13</b>, 555-556.
<a href="http://abacus.gene.ucl.ac.uk/software/paml.html">http://abacus.gene.ucl.ac.uk/software/paml.html</a>
</p>
<p>Yang, Z. (2007)
“PAML 4: a program package for phylogenetic analysis by maximum
likelihood”,
<em>Molecular Biology and Evolution</em>, <b>24</b>, 1586-1591.
</p>
<p>Tzeng, J.Y. (2005)
“Evolutionary-Based Grouping of Haplotypes in Association Analysis”,
<em>Genetics Epidemiology</em>, <b>28</b>, 220-231.
<a href="https://www4.stat.ncsu.edu/~jytzeng/software.php">https://www4.stat.ncsu.edu/~jytzeng/software.php</a>
</p>
<p>Paradis E., Claude J., and Strimmer K. (2004)
“APE: analyses of phylogenetics and evolution in R language”,
<em>Bioinformatics</em>, <b>20</b>, 289-290.
<a href="http://ape-package.ird.fr/">http://ape-package.ird.fr/</a>
</p>


<h3>See Also</h3>

<p><code>phyclust</code>,
<code>.EMC</code>,
<code>.EMControl</code>,
<code>find.best</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
library(phyclust, quiet = TRUE)

demo(package = "phyclust")
demo("ex_trees", package = "phyclust")

## End(Not run)
</code></pre>


</div>