<div class="container">

<table style="width: 100%;"><tr>
<td>file.read</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read Data from Files by Formats and Return a seq.data Object</h2>

<h3>Description</h3>

<p>Two major file formats are supported in <span class="pkg">phyclust</span>,
<code>Format phylip</code> and <code>Format fasta</code>.
<strong>These functions only read files in basic syntax</strong>, and
return an object in <code>Class seq.data</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read.fasta(filename, byrow = TRUE, code.type = .code.type[1], aligned = TRUE,
           sep = "")
read.fasta.format(filename, byrow = TRUE, aligned = TRUE, sep = "")

read.phylip(filename, byrow = TRUE, code.type = .code.type[1], sep = "")
read.phylip.format(filename, byrow = TRUE, sep = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>filename</code></td>
<td>
<p>a file name where data is read from.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>byrow</code></td>
<td>
<p>advanced option, default = TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>code.type</code></td>
<td>
<p>either "NUCLEOTIDE" (default) or "SNP".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aligned</code></td>
<td>
<p>indicate aligned data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep</code></td>
<td>
<p>use to split sites, "" (default) and "," for "CODON".</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For unaligned sequences, <code>read.fasta</code> returns a list storing data.
<code>read.phylip</code> is only for aligned data and returns a matrix.
</p>
<p><code>read.fasta.format</code> and <code>read.phylip.format</code> will read in
original coding without any transformation as <code>code.type = NULL</code>
in <code>write.fasta</code> and <code>write.phylip</code>. Suppose these functions
return an object <code>ret</code>, one can write other functions <code>ret2aa()</code>
to post transform the coding and replace <code>ret$org</code> by the results of
<code>ret2aa(ret$org.code)</code>.
</p>
<p><code>byrow</code> indicates the data will be store by row or not. Usually,
the default is TRUE. The FALSE is only for advance users
with careful manipulations and for speeding up the bootstraps.
</p>
<p><code>sep</code> can specify a character which is used to split sites in file.
By default, "" denote no character between sites. Only "CODON" id requires
a character to avoid ambiguity
</p>


<h3>Value</h3>

<p>Return an object in <code>Class seq.data</code>.
</p>


<h3>Author(s)</h3>

<p>Wei-Chen Chen <a href="mailto:wccsnow@gmail.com">wccsnow@gmail.com</a>
</p>


<h3>References</h3>

<p>Phylogenetic Clustering Website:
<a href="https://snoweye.github.io/phyclust/">https://snoweye.github.io/phyclust/</a>
</p>


<h3>See Also</h3>

<p><code>write.fasta</code>,
<code>write.phylip</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
library(phyclust, quiet = TRUE)

# PHYLIP
data.path &lt;- paste(.libPaths()[1], "/phyclust/data/crohn.phy", sep = "")
(my.snp &lt;- read.phylip(data.path, code.type = "SNP"))

# FASTA
data.path &lt;- paste(.libPaths()[1], "/phyclust/data/pony625.fas", sep = "")
(my.pony &lt;- read.fasta(data.path))

## End(Not run)
</code></pre>


</div>