<div class="container">

<table style="width: 100%;"><tr>
<td>nid.aid.cid</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Transfer nids (0, 1, ..., 4) , aids (0, 1, ..., 21) and cids (0, 1, ..., 64)</h2>

<h3>Description</h3>

<p>Transfer nids (0, 1, ..., 4) , aids (0, 1, ..., 21) and cids (0, 1, ..., 64).
</p>


<h3>Usage</h3>

<pre><code class="language-R">### S3 methods for a list, vector or matrix (default).
nid2aid(nidseq, start = 1, end = NULL, drop.gap = FALSE, byrow = TRUE)
nid2cid(nidseq, start = 1, end = NULL, drop.gap = FALSE, byrow = TRUE)
cid2aid(cidseq)
aid2acode(aidseq, lower.case = FALSE)
acode2aid(acodeseq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nidseq</code></td>
<td>
<p>a numerical vector contains nucleotide ids, 0, 1, 2, 3, or 4.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cidseq</code></td>
<td>
<p>a numerical vector contains codon ids, 0, 1, ..., or 64.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aidseq</code></td>
<td>
<p>a numerical vector contains amino acid ids, 0, 1, ..., or 21.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>acodeseq</code></td>
<td>
<p>a character vector contains amino acid codes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start</code></td>
<td>
<p>the start site to translate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>end</code></td>
<td>
<p>the end site to translate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>drop.gap</code></td>
<td>
<p>ignore gaps if TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>byrow</code></td>
<td>
<p>advanced option, default = TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lower.case</code></td>
<td>
<p>transfer in lower cases.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>These functions are based on the internal object <code>.nucleotide</code>,
<code>.codon</code>, <code>.amino.acid</code>, and <code>.genetic.code</code> to
transfer sequences.
</p>


<h3>Value</h3>

<p><code>nid2aid</code> and <code>cid2aid</code> returns a numerical vector containing
amino acid ids, and <code>nid2cid</code> returns a numerical vector containing
codon ids, <code>aid2acode</code> returns a character vector containing
amino acid codes, and <code>acode2aid</code> returns a numerical vector containing
amino acid ids.
</p>


<h3>Author(s)</h3>

<p>Wei-Chen Chen <a href="mailto:wccsnow@gmail.com">wccsnow@gmail.com</a>
</p>


<h3>References</h3>

<p>Phylogenetic Clustering Website:
<a href="https://snoweye.github.io/phyclust/">https://snoweye.github.io/phyclust/</a>
</p>


<h3>See Also</h3>

<p><code>.nucleotide</code>,
<code>.amino.acid</code>,
<code>.codon</code>,
<code>.genetic.code</code>,
<code>code2nid</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
library(phyclust, quiet = TRUE)

### Test S3 methods by a vector
a.vector &lt;- c("A", "C", "G", "-", "T", "A")
code2nid(a.vector)
nid2cid(code2nid(a.vector))
cid2aid(nid2cid(code2nid(a.vector)))
nid2aid(code2nid(a.vector))
aid2acode(nid2aid(code2nid(a.vector)))
acode2aid(aid2acode(nid2aid(code2nid(a.vector))))

### Test S3 methods by a matrix
a.matrix &lt;- rbind(a.vector, a.vector, a.vector)
code2nid(a.matrix)
nid2cid(code2nid(a.matrix))
cid2aid(nid2cid(code2nid(a.matrix)))
nid2aid(code2nid(a.matrix))
aid2acode(nid2aid(code2nid(a.matrix)))
acode2aid(aid2acode(nid2aid(code2nid(a.matrix))))

### Test S3 methods by a list
a.list &lt;- list(a.vector, a.vector)
code2nid(a.list)
nid2cid(code2nid(a.list))
cid2aid(nid2cid(code2nid(a.list)))
nid2aid(code2nid(a.list))
aid2acode(nid2aid(code2nid(a.list)))
acode2aid(aid2acode(nid2aid(code2nid(a.list))))

## End(Not run)
</code></pre>


</div>