<div class="container">

<table style="width: 100%;"><tr>
<td>plotgaps</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Gaps Plots of Sequences for Visual Comparisons</h2>

<h3>Description</h3>

<p>This function provides gaps plots of data set to identify regions where
gaps enriched. The plot show the proportions of context by sites and the
diverse may be caused by mutations, sequencing errors, or alignment
errors.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plotgaps(X, code.type = .code.type[1], main = "Gaps Plot",
    xlab = "Sites", ylab = "Proportion", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>numerical data matrix with <code class="reqn">N</code> rows/sequences and <code class="reqn">L</code> columns/sites.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>code.type</code></td>
<td>
<p>either "NUCLEOTIDE" (default) or "SNP".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>main</code></td>
<td>
<p>main label, default = "Gaps Plot".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab</code></td>
<td>
<p>x-axis label, default = "Sites". </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylab</code></td>
<td>
<p>y-axis label, default = "Proportion". </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>other options passed to <code>plot</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Proportions of gaps will be drawn.
</p>


<h3>Value</h3>

<p>A gaps plot will be drawn.
</p>


<h3>Author(s)</h3>

<p>Wei-Chen Chen <a href="mailto:wccsnow@gmail.com">wccsnow@gmail.com</a>
</p>


<h3>References</h3>

<p>Phylogenetic Clustering Website:
<a href="https://snoweye.github.io/phyclust/">https://snoweye.github.io/phyclust/</a>
</p>


<h3>See Also</h3>

<p><code>plotdots</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
library(phyclust, quiet = TRUE)

# For nucleotide
set.seed(1234)
X &lt;- seq.data.toy$org
X[sample(c(T, F), length(X), replace = TRUE, prob = c(0.05, 0.95))] &lt;-
  .nucleotide$nid[.nucleotide$code == "-"]
plotgaps(X)

## End(Not run)
</code></pre>


</div>