<div class="container">

<table style="width: 100%;"><tr>
<td>data.phylip.crohn</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Crohn's Disease SNP Dataset in the phylip Format</h2>

<h3>Description</h3>

<p>Crohn's disease dataset is published by Hugot, et al. (2001).
</p>


<h3>Format</h3>

<p>A text file in phylip format is stored in the data subdirectory.
</p>


<h3>Details</h3>

<p>Crohn's disease dataset is used to perform haplotype grouping used
in Tzeng's paper (2005).
</p>
<p>Totally, 1102 haplotypes/SNP sequences and 8 sites.
</p>


<h3>Author(s)</h3>

<p>Hugot, J.P., et al. (2001).
</p>


<h3>References</h3>

<p>Phylogenetic Clustering Website:
<a href="https://snoweye.github.io/phyclust/">https://snoweye.github.io/phyclust/</a>
</p>
<p>Hugot, J.P., et al. (2001)
“Association of NOD2 Leucine-Rich Repeat Variants with
Susceptibility to Crohn's Disease”,
<em>Nature</em>, <b>411</b>, 599-603.
</p>
<p>Tzeng, J.Y. (2005)
“Evolutionary-Based Grouping of Haplotypes in Association Analysis”,
<em>Genetics Epidemiology</em>, <b>28</b>, 220-231.
<a href="https://www4.stat.ncsu.edu/~jytzeng/software.php">https://www4.stat.ncsu.edu/~jytzeng/software.php</a>
</p>


<h3>See Also</h3>

<p><code>read.phylip</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
library(phyclust, quiet = TRUE)

data.path &lt;- paste(.libPaths()[1], "/phyclust/data/crohn.phy", sep = "")
# edit(file = data.path)
my.snp &lt;- read.phylip(data.path, code.type = "SNP")
str(my.snp)

## End(Not run)
</code></pre>


</div>