<div class="container">

<table style="width: 100%;"><tr>
<td>haplo.post.prob</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Tzeng's Method: Haplotype Grouping for SNP Sequences</h2>

<h3>Description</h3>

<p>For SNP sequences only, Tzeng's method (2005) uses an evolution approach
to group haplotypes based on a deterministic transformation of haplotype
frequency. This is a modified version of original function,
<code>haplo.score.RD.unphased.fun</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  haplo.post.prob(X, ploidy = 2, skip.haplo = 1e-07, K = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>sid matrix with <code class="reqn">N</code> rows/sequences and <code class="reqn">L</code> columns/sites.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ploidy</code></td>
<td>
<p>ploidy, no effect for phase known, keep consistence only.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>skip.haplo</code></td>
<td>
<p>lower bound of haplotypes frequencies.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K</code></td>
<td>
<p>number of clusters.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>X</code> should be a phase known haplotype data. For phase unknown and
Tzeng's method (2006) are not tested yet.
</p>
<p>If <code>K</code> is NULL, the result of <code>getcut.fun</code> will be used.
</p>


<h3>Value</h3>

<p>See the original paper and source codes' documents for details.
The function returns a list contains:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>'haplo'</code></td>
<td>
<p>summarized data set in a list contains:
</p>

<table>
<tr>
<td style="text-align: left;">
      'haplotype' </td>
<td style="text-align: left;"> unique haplotypes, dim = <code class="reqn">N_{unique} \times L</code>. </td>
</tr>
<tr>
<td style="text-align: left;">
      'hap.prob'  </td>
<td style="text-align: left;"> frequency of haplotypes. </td>
</tr>
<tr>
<td style="text-align: left;">
      'post'      </td>
<td style="text-align: left;"> posterior probabilities of phase unknown haplotypes. </td>
</tr>
<tr>
<td style="text-align: left;">
      'hap1code'  </td>
<td style="text-align: left;"> unique ids of 'haplotype'. </td>
</tr>
<tr>
<td style="text-align: left;">
      'hap2code'  </td>
<td style="text-align: left;"> unique ids of 'haplotype', no effect if ploidy = 2. </td>
</tr>
<tr>
<td style="text-align: left;">
      'indx.subj' </td>
<td style="text-align: left;"> id of subjects. </td>
</tr>
<tr>
<td style="text-align: left;">
    </td>
</tr>
</table>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>'FD.id'</code></td>
<td>
<p>unique ids of 'haplotype' for full dimension analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>'RD.id'</code></td>
<td>
<p>unique ids of 'haplotype' for reduced dimension analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>'FD.post'</code></td>
<td>
<p>posterior probabilities for full dimension analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>'RD.post'</code></td>
<td>
<p>posterior probabilities for reduced dimension analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>'g.truncate'</code></td>
<td>
<p>number of clusters</p>
</td>
</tr>
</table>
<h3>ToDo(s)</h3>


<ul><li>
<p> test codes for phased unknown cases.
</p>
</li></ul>
<h3>Author(s)</h3>

<p>Jung-Ying Tzeng.
</p>
<p>Maintain: Wei-Chen Chen <a href="mailto:wccsnow@gmail.com">wccsnow@gmail.com</a>
</p>


<h3>References</h3>

<p>Phylogenetic Clustering Website:
<a href="https://snoweye.github.io/phyclust/">https://snoweye.github.io/phyclust/</a>
</p>
<p>Tzeng, J.Y. (2005)
“Evolutionary-Based Grouping of Haplotypes in Association Analysis”,
<em>Genetics Epidemiology</em>, <b>28</b>, 220-231.
<a href="https://www4.stat.ncsu.edu/~jytzeng/software.php">https://www4.stat.ncsu.edu/~jytzeng/software.php</a>
</p>


<h3>See Also</h3>

<p><code>getcut.fun</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
library(phyclust, quiet = TRUE)

data.path &lt;- paste(.libPaths()[1], "/phyclust/data/crohn.phy", sep = "")
my.snp &lt;- read.phylip(data.path, code.type = "SNP")
ret &lt;- haplo.post.prob(my.snp$org, ploidy = 1)
str(ret)

## End(Not run)
</code></pre>


</div>