<div class="container">

<table style="width: 100%;"><tr>
<td>.substitution.model</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Substitution Models for Mutation Processes</h2>

<h3>Description</h3>

<p>An object stores substitution models for mutation processes
for Continuous Time Markov Chain theory.
<strong>This is a read-only object and the elemental order is followed in C.</strong>
</p>


<h3>Usage</h3>

<pre><code class="language-R">.substitution.model</code></pre>


<h3>Format</h3>

<p>A data frame contains two character vectors, <code>mode</code> and <code>code.type</code>.
</p>


<h3>Details</h3>

<p>This data frame indicates substitution models implemented in C.
</p>

<ul>
<li>
<p> 'model': names of substitution models for mutations.
</p>
</li>
<li>
<p> 'code.type': code types of substitution models, either NUCLEOTIDE or SNP.
</p>
</li>
</ul>
<p>The major models are:
</p>

<table>
<tr>
<td style="text-align: left;">
    Model </td>
<td style="text-align: left;"> Author and Publication            </td>
<td style="text-align: left;"> Parameter </td>
</tr>
<tr>
<td style="text-align: left;">
    JC69  </td>
<td style="text-align: left;"> Jukes and Cantor 1969.            </td>
<td style="text-align: left;"> <code class="reqn">t</code> </td>
</tr>
<tr>
<td style="text-align: left;">
    K80   </td>
<td style="text-align: left;"> Kimura 1980.                      </td>
<td style="text-align: left;"> <code class="reqn">\kappa, t</code> </td>
</tr>
<tr>
<td style="text-align: left;">
    F81   </td>
<td style="text-align: left;"> Felsenstein 1981.                 </td>
<td style="text-align: left;"> <code class="reqn">\pi, t</code> </td>
</tr>
<tr>
<td style="text-align: left;"> 
    HKY85 </td>
<td style="text-align: left;"> Hasegawa, Kishino, and Yano 1985. </td>
<td style="text-align: left;"> <code class="reqn">\pi, \kappa, t</code> </td>
</tr>
<tr>
<td style="text-align: left;">
  </td>
</tr>
</table>
<p>Other models starting with <code>E_</code> use empirical frequencies for
equilibrium probabilities.
</p>


<h3>Author(s)</h3>

<p>Wei-Chen Chen <a href="mailto:wccsnow@gmail.com">wccsnow@gmail.com</a>
</p>


<h3>References</h3>

<p>Phylogenetic Clustering Website:
<a href="https://snoweye.github.io/phyclust/">https://snoweye.github.io/phyclust/</a>
</p>
<p>Jukes, T. H. and Cantor, C. R. (1969)
“Evolution of Protein Molecules”,
<em>Mammalian Protein Metabolism</em>, <b>3</b>, 21-132
</p>
<p>Kimura, M. (1980)
“A Simple Method for Estimating Evolutionary Rates of Base
Substitutions through Comparative Studies of Nucleotide Sequences”,
<em>Journal of Molecular Evolution</em>, <b>16</b>, 111-120
</p>
<p>Felsenstein, J. (1981)
“Evolutionary Trees from DNA Sequences: A Maximum Likelihood
Approach”,
<em>Journal of Molecular Evolution</em>, <b>17</b>, 368-376
</p>
<p>Hasegawa, M. and Kishino, H. and Yano, T. (1985)
“Dating of the Human-Ape Splitting by a Molecular Clock of
Mitochondrial DNA”,
<em>Journal of Molecular Evolution</em>, <b>22:2</b>, 160-174
</p>


<h3>See Also</h3>

<p><code>.show.option</code>,
<code>.code.type</code>,
<code>.identifier</code>,
<code>.EMControl</code>,
<code>phyclust</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
library(phyclust, quiet = TRUE)

.substitution.model

## End(Not run)
</code></pre>


</div>