<div class="container">

<table style="width: 100%;"><tr>
<td>detect.outliers</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Detection of outliers in 1D and 2D data</h2>

<h3>Description</h3>

<p>Functions to detect outliers, in matrices or in arrays.
</p>


<h3>Usage</h3>

<pre><code class="language-R">detect.outliers(X, k = 3, test.island = TRUE, normalizeby = "row")

detect.outliers.array(arr, nbsp, k = 3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>2D matrix (gene x species) obtained with the <code>Dist2WR()</code> function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>strength of outlier detection. High values (typically 3) 
correspond to less outliers detected than with lower ones (e.g. 1.5).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>test.island</code></td>
<td>
<p>should islands of outliers be treated as such. 
If TRUE (the default), only the highest value in an island of outliers is
removed. This prevents erroneous outliers due to hicthiking effect to be removed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>normalizeby</code></td>
<td>
<p>Should the 2D matrix be normalized prior to outlier detection, and how.
Can be "row" (the default),"col" or "none". Normalization is done by dividing columns or rows 
by their median.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>arr</code></td>
<td>
<p>Array of values, typically the weight of each gene matrix (alpha values).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nbsp</code></td>
<td>
<p>Number of species in the analysis</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>These functions detect outliers either in matrices or in arrays. For the method 
to be adapted to skewed data, as is the case here, the outlier detection 
method used is the adjusted Tukey proposed by Hubert and Vandervieren (2008).
</p>


<h3>Value</h3>

<p><code>detect.outliers</code>: A matrix with outliers detected in the 2D matric. Each row <code>x</code> contains the 
gene (<code>x[1]</code>) where the species (<code>x[2]</code>) is an outlier.
</p>
<p><code>detect.outliers.array</code>: An array listing the outliers detected (if any)
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>detect.outliers()</code>: detect outliers in 2D matrix
</p>
</li>
<li> <p><code>detect.outliers.array()</code>: detects outliers in 1D array
</p>
</li>
</ul>
<h3>References</h3>

<p>de Vienne D.M., Ollier S. et Aguileta G. (2012) Phylo-MCOA: 
A Fast and Efficient Method to Detect Outlier Genes and Species
in Phylogenomics Using Multiple Co-inertia Analysis. Molecular 
Biology and Evolution 29 : 1587-1598.
</p>
<p>Hubert, M. and Vandervieren, E. (2008). An adjusted boxplot for skewed
distributions. Computational Statistics and Data Analysis, 52, 5186-5201.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Get the initial gene x species matrix
# from the carnivora dataset
data(carnivora) 
mat &lt;- phylter(carnivora, InitialOnly = TRUE, parallel = FALSE)$WR

# detect outliers in this matrix
outliers&lt;-detect.outliers(mat)
outliers$cells # matrix where each row represents one cell in the input matrix

</code></pre>


</div>