<div class="container">

<table style="width: 100%;"><tr>
<td>supermat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Build PHYLIP supermatrix and RAxML partition file using aligned FASTA or PHYLIP files.
</h2>

<h3>Description</h3>

<p>Build PHYLIP supermatrix and create RAxML partition file using aligned fasta or phylip files.
</p>


<h3>Usage</h3>

<pre><code class="language-R">supermat(infiles, outfile = "supermat.out.phy",
         partition.file = "gene_partition.txt")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>infiles</code></td>
<td>

<p>a character string vector for phylip or aligned fasta file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outfile</code></td>
<td>

<p>the name of the PHYLIP supermatrix
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>partition.file</code></td>
<td>

<p>partition data summary describing the genes.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Supermatrix here means a phylip file with combined aligned sequences. The missing sequences should be replaced with either "?" or "-".
</p>


<h3>Value</h3>

<p>A list containing:
(1)supermat.dat:a list containing all the data frames read by read.phylip or read.fasta
(2)res.super.dat: a data frame containing the sequences and the names
(3)partition.dat: summary for all the fasta or phylip files
(4)partition.dat.vector: character string vector for the partition file for RAxML
</p>


<h3>Note</h3>

<p>Punctuation characters and white space in the names of the sequences will be replaced by "_". More information can be found at <code>regex</code>.
Type of the sequence in the RAxML partition file should be changed manually according to the manual of RAxML.
</p>


<h3>Author(s)</h3>

<p>Jinlong Zhang &lt;jinlongzhang01@gmail.com&gt;
</p>


<h3>References</h3>

<p>Kress, W. J., Erickson, D. L., Jones, F. A., Swenson, N. G., Perez, R., Sanjur, O., &amp; Bermingham, E. (2009). Plant DNA barcodes and a community phylogeny of a tropical forest dynamics plot in Panama. Proceedings of the National Academy of Sciences, 106(44), 18621-18626.
</p>
<p>de Queiroz, A.and Gatesy, J. (2007). The supermatrix approach to systematics. Trends in Ecology &amp; Evolution, 22(1), 34-41.
</p>
<p><a href="https://github.com/stamatak/standard-RAxML">https://github.com/stamatak/standard-RAxML</a>
</p>


<h3>See Also</h3>

<p><code>read.fasta</code>,<code>read.phylip</code>,<code>dat2phylip</code>,
</p>


<h3>Examples</h3>

<pre><code class="language-R">
  cat("6 22",
  "seq_1    --TTACAAATTGACTTATTATA",
  "seq_2    GATTACAAATTGACTTATTATA",
  "seq_3    GATTACAAATTGACTTATTATA",
  "seq_5    GATTACAAATTGACTTATTATA",
  "seq_8    GATTACAAATTGACTTATTATA",
  "seq_10   ---TACAAATTGAATTATTATA",
  file = "matk.phy", sep = "\n")

  cat("5 15",
  "seq_1     GATTACAAATTGACT",
  "seq_3     GATTACAAATTGACT",
  "seq_4     GATTACAAATTGACT",
  "seq_5     GATTACAAATTGACT",
  "seq_8     GATTACAAATTGACT",
  file = "rbcla.phy", sep = "\n")

  cat("5 50",
  "seq_2          GTCTTATAAGAAAGAATAAGAAAG--AAATACAAA-------AAAAAAGA",
  "seq_3          GTCTTATAAGAAAGAAATAGAAAAGTAAAAAAAAA-------AAAAAAAG",
  "seq_5          GACATAAGACATAAAATAGAATACTCAATCAGAAACCAACCCATAAAAAC",
  "seq_8          ATTCCAAAATAAAATACAAAAAGAAAAAACTAGAAAGTTTTTTTTCTTTG",
  "seq_9          ATTCTTTGTTCTTTTTTTTCTTTAATCTTTAAATAAACCTTTTTTTTTTA",
  file = "trn1.phy", sep = "\n")

supermat(infiles = c("matk.phy", "rbcla.phy", "trn1.phy"))
unlink(c("matk.phy", "rbcla.phy", "trn1.phy"))
unlink(c("supermat.out.phy","gene_partition.txt"))

cat(
  "&gt;seq_1",  "--TTACAAATTGACTTATTATA",
  "&gt;seq_2",  "GATTACAAATTGACTTATTATA",
  "&gt;seq_3",  "GATTACAAATTGACTTATTATA",
  "&gt;seq_5",  "GATTACAAATTGACTTATTATA",
  "&gt;seq_8",  "GATTACAAATTGACTTATTATA",
  "&gt;seq_10", "---TACAAATTGAATTATTATA",
  file = "matk.fasta", sep = "\n")

cat(
  "&gt;seq_1", "GATTACAAATTGACT",
  "&gt;seq_3", "GATTACAAATTGACT",
  "&gt;seq_4", "GATTACAAATTGACT",
  "&gt;seq_5", "GATTACAAATTGACT",
  "&gt;seq_8", "GATTACAAATTGACT",
  file = "rbcla.fasta", sep = "\n")

cat(
  "&gt;seq_2", "GTCTTATAAGAAAGAATAAGAAAG--AAATACAAA-------AAAAAAGA",
  "&gt;seq_3", "GTCTTATAAGAAAGAAATAGAAAAGTAAAAAAAAA-------AAAAAAAG",
  "&gt;seq_5", "GACATAAGACATAAAATAGAATACTCAATCAGAAACCAACCCATAAAAAC",
  "&gt;seq_8", "ATTCCAAAATAAAATACAAAAAGAAAAAACTAGAAAGTTTTTTTTCTTTG",
  "&gt;seq_9", "ATTCTTTGTTCTTTTTTTTCTTTAATCTTTAAATAAACCTTTTTTTTTTA",
  file = "trn1.fasta", sep = "\n")

supermat(infiles = c("matk.fasta", "rbcla.fasta", "trn1.fasta"))
unlink(c("matk.fasta", "rbcla.fasta", "trn1.fasta"))

unlink(c("supermat.out.phy","gene_partition.txt"))

</code></pre>


</div>