<div class="container">

<table style="width: 100%;"><tr>
<td>mrp.supertree</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Matrix representation parsimony supertree estimation</h2>

<h3>Description</h3>

<p>This function estimates the MRP (matrix representation parsimony) supertree from a set of input trees (Baum 1992; Ragan 1992).
</p>


<h3>Usage</h3>

<pre><code class="language-R">mrp.supertree(trees, method=c("pratchet","optim.parsimony"), ...)
compute.mr(trees, type=c("phyDat","matrix"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>trees</code></td>
<td>
<p>an object of class <code>"multiPhylo"</code>  that consists of a list of phylogenetic trees.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>an argument specifying whether to optimize the tree using the <span class="pkg">phangorn</span> parsimony optimizer <code>pratchet</code> or <code>optim.parsimony</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>for <code>compute.mr</code>, the type of object to return (e.g., <code>"phyDat"</code> or <code>"matrix"</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>optional arguments - mostly to be passed to <code>pratchet</code> or <code>optim.parsimony</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>mrp.supertree</code> uses <code>pratchet</code> or <code>optim.parsimony</code> from the <span class="pkg">phangorn</span> package (Schliep, 2011) for optimization, and <code>prop.part</code> from <span class="pkg">ape</span> package (Paradis et al. 2004). 
</p>
<p>See <code>pratchet</code> or <code>optim.parsimony</code> for optional arguments, which vary slightly depending on the method. All optional arguments of these methods are available to the user with one exception. The argument <code>tree</code> in <code>optim.parsimony</code> is supplied instead as <code>start</code>. In addition to being an object of class <code>"phylo"</code>, <code>start</code> can also be assigned the string values of <code>"NJ"</code> or <code>"random"</code>, in which case either a neighbor-joining or random tree will be used as the starting tree for optimization.
</p>
<p>The function <code>compute.mr</code> computes the matrix-representation matrix of the input trees. It is used internally by <code>mrp.supertree</code>, but can also be used to export an object that can be written to file if desired.
</p>


<h3>Value</h3>

<p>An object of class <code>"phylo"</code> or <code>"multiPhylo"</code> that is the MP or set of MP MRP trees.
</p>
<p>In the case of <code>compute.mr</code>, an object of class <code>"phyDat"</code> or a matrix.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Baum, B. R., (1992) Combining trees as a way of combining data sets for phylogenetic inference, and the desirability of combining gene trees. <em>Taxon</em>, <b>41</b>, 3-10.
</p>
<p>Felsenstein, J. (2004) <em>Inferring Phylogenies</em>. Sinauer.
</p>
<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics and evolution in R language. <em>Bioinformatics</em>, <b>20</b>, 289-290.
</p>
<p>Ragan, M. A. (1992) Phylogenetic inference based on matrix representation of trees. <em>Molecular Phylogenetics and Evolution</em>, <em>1</em>, 53-58.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Schliep, K. P. (2011) phangorn: phylogenetic analysis in R. <em>Bioinformatics</em>, <b>27</b>, 592-593.
</p>


<h3>See Also</h3>

<p><code>exhaustiveMP</code>, <code>optim.parsimony</code>, <code>pratchet</code>
</p>


</div>