<div class="container">

<table style="width: 100%;"><tr>
<td>read.simmap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read SIMMAP style trees from file</h2>

<h3>Description</h3>

<p>This reads one or multiple SIMMAP style trees from file.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read.simmap(file="", text, format="nexus", rev.order=TRUE, version=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>name of text file with one or multiple SIMMAP v1.0 or v1.5 style trees containing the mapped history of a discrete character.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>text</code></td>
<td>
<p>character string containing the tree. If <code>version=1.5</code> this argument is ignored. (This format tree can only be read from file in the present version.)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>format</code></td>
<td>
<p>format of the trees: either <code>"phylip"</code> or <code>"nexus"</code> - the latter is the default output from SIMMAP. If <code>version=1.5</code> this argument is ignored.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rev.order</code></td>
<td>
<p>a logical value indicating whether the states and times along each branch is given (from root to tip) in right-to-left order (if TRUE) or in left-to-right order. If <code>version=1.5</code> this argument is ignored.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>version</code></td>
<td>
<p>version of SIMMAP for input tree. If the tree(s) was/were simulated in SIMMAP v1.0 or written to file by <code>link{make.simmap}</code> then <code>version=1.0</code>; if the tree(s) was/were simulated using SIMMAP v1.5 then <code>version=1.5</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function now accepts trees in both SIMMAP v1.0 and SIMMAP v1.5 format. In addition, it can read a more flexible format than is produced by SIMMAP (for instance, multi-character mapped states and more than 7 mapped states).
</p>
<p>The function uses some modified code from <code>read.nexus</code> from the <span class="pkg">ape</span> package to read the NEXUS block created by SIMMAP. Also creates the attribute <code>"map.order"</code> which indicates whether the stochastic map was read in from left to right or right to left.  This attribute is used by default by <code>write.simmap</code> to write the tree in the same order. 
</p>


<h3>Value</h3>

<p>An object of class <code>"simmap"</code> (or list of such objects with class <code>"multiSimmap"</code>), consisting of a modified object of class <code>"phylo"</code> with at least the following additional elements:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>maps</code></td>
<td>
<p>a list of named vectors containing the times spent in each state on each branch, in the order in which they occur.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mapped.edge</code></td>
<td>
<p>a matrix containing the total time spent in each state along each edge of the tree.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Bollback, J. P. (2006) Stochastic character mapping of discrete traits on phylogenies. <em>BMC Bioinformatics</em>, <b>7</b>, 88.
</p>
<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics and evolution in R language. <em>Bioinformatics</em>, <b>20</b>, 289-290.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>See Also</h3>

<p><code>brownie.lite</code>, <code>evol.vcv</code>, <code>read.tree</code>, <code>read.nexus</code>
</p>


</div>