<div class="container">

<table style="width: 100%;"><tr>
<td>phylomorphospace</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Creates phylomorphospace plot</h2>

<h3>Description</h3>

<p>Project a phylogeny into morphospace.
</p>


<h3>Usage</h3>

<pre><code class="language-R">phylomorphospace(tree, X, A=NULL, label=c("radial","horizontal","off"),
    control=list(), ...)
project.phylomorphospace(tree, X, nsteps=200, sleep=0,
    direction=c("to","from","both"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>tree</code></td>
<td>
<p>a phylogenetic tree in <code>"phylo"</code> format, or a modified <code>"phylo"</code> object with a mapped discrete character.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>an <code>n</code> x 2 matrix of tip values for two characters in <code>n</code> species.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>A</code></td>
<td>
<p>an optional <code>m</code> x 2 matrix (for <code>m</code> nodes) of values for two traits at internal nodes in the tree - if not supplied, these values will be estimated using <code>fastAnc</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>string indicating whether to plot the tip labels in the same direction as the terminal edge (<code>label="radial"</code>), horizontally <code>label="horizontal"</code>, or not at all <code>"off"</code>. <code>label=TRUE</code> and <code>label=FALSE</code> are also acceptable, for compatibility with phytools &lt;= 0.3-03.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>control</code></td>
<td>
<p>a list containing the following optional control parameters: <code>col.edge</code>: a vector of edge colors; and <code>col.node</code>: a vector of node colors.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nsteps</code></td>
<td>
<p>for <code>project.phylomorphospace</code> the number of frames in the animation between the phylogeny &amp; the phylomorphospace or <em>vice versa</em>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sleep</code></td>
<td>
<p>for <code>project.phylomorphospace</code> the time between frames.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>direction</code></td>
<td>
<p>for <code>project.phylomorphospace</code> whether to morph <code>"to"</code> a phylomorphospace, <code>"from"</code> a phylomorphospace, or there &amp; back again (<code>"both"</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>optional arguments for plotting, including <code>xlim</code>, <code>ylim</code>, <code>xlab</code>, <code>ylab</code>, <code>lwd</code>, <code>colors</code>, <code>fsize</code>, and <code>node.by.map</code>. <code>colors</code> is only used when there is a mapped discrete character on the tree, in which case <code>control$col.edge</code> is ignored. <code>fsize</code> is relative to the default, which is <code>textxy(...,cx=0.75)</code>. <code>node.by.map</code> is a logical value (defaults to <code>FALSE</code> which tells the function whether or not to plot the node colors using the colors of the mapped discrete character. Setting this option to <code>TRUE</code> will cause <code>control$col.node</code> to be ignored. <code>node.size</code> is a vector containing the point size relative to the default (see <code>par</code> for plotted internal nodes and tips, respectively. Defaults to <code>node.size=c(1,1.3)</code>. If only one number is provided it will be recycled. <code>axes</code> is a logical value indicating whether or not axes should be plotted (see <code>plot.default</code>. Finally, <code>add</code> indicates whether to add the phylomorphospace to the current plot.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function creates a phylomorphospace plot (a projection of the tree into morphospace) for two characters following Sidlauskas (2008; <em>Evolution</em>). It will also plot a discrete character mapped on tree.
</p>
<p><code>project.phylomorphospace</code> animates the phylomorphospace projection.
</p>


<h3>Value</h3>

<p>This function creates a phylomorphospace plot
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Paradis, E., J. Claude, and K. Strimmer (2004) APE: Analyses of phylogenetics and evolution in R language. <em>Bioinformatics</em>, <b>20</b>, 289-90.
</p>
<p>Revell, L. J. (2014) Graphical methods for visualizing comparative data on phylogenies. Chapter 4 in <em>Modern phylogenetic comparative methods and their application in evolutionary biology: Concepts and practice</em> (L. Z. Garamszegi ed.), pp. 77-103.
</p>
<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>
<p>Sidlauskas, B. (2008) Continuous and arrested morphological diversification in sister clades of characiform fishes: A phylomorphospace approach. <em>Evolution</em>, <b>62</b>, 3135-3156.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## load tree &amp; data from Revell &amp; Collar (2009)
data(sunfish.tree)
data(sunfish.data)
## set colors for mapped discrete character
cols&lt;-setNames(c("blue","red"),
    levels(sunfish.data$feeding.mode))
phylomorphospace(sunfish.tree,sunfish.data[,3:2],
    colors=cols,bty="l",ftype="off",node.by.map=TRUE,
    node.size=c(0,1.2),xlab="relative buccal length",
    ylab="relative gape width")
title(main="Phylomorphospace of buccal morphology in Centrarchidae",
    font.main=3)	
</code></pre>


</div>