<div class="container">

<table style="width: 100%;"><tr>
<td>genSeq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate a DNA alignment on the tree under a model</h2>

<h3>Description</h3>

<p>Simulates DNA sequence on <code>tree</code> under the specified model.
</p>


<h3>Usage</h3>

<pre><code class="language-R">genSeq(tree, l=1000, Q=NULL, rate=1, format="DNAbin", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>tree</code></td>
<td>
<p>object of class <code>"phylo"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>l</code></td>
<td>
<p>length of desired sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Q</code></td>
<td>
<p>transition matrix for the simulation. Row and column names (<code>c("a","c","g","t")</code>, although not necessarily in that order) should be provided. If <code>NULL</code>, a single rate is assumed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rate</code></td>
<td>
<p>multiplier for <code>Q</code>, or a vector for <code class="reqn">\Gamma</code> rate heterogeneity.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>format</code></td>
<td>
<p>format of the output object. Can be <code>"DNAbin"</code>, <code>"phyDat"</code>, or <code>"matrix"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>optional arguments.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Uses <code>sim.Mk</code> internally.
</p>


<h3>Value</h3>

<p>An object of class <code>"DNAbin"</code> or <code>"phyDat"</code>, or a matrix of nucleotides.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>References</h3>

<p>Revell, L. J. (2024) phytools 2.0: an updated R ecosystem for phylogenetic comparative methods (and other things). <em>PeerJ</em>, <b>12</b>, e16505.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(mammal.tree)
mammal.tree$edge.length&lt;-mammal.tree$edge.length/
    max(nodeHeights(mammal.tree))*0.2 ## rescale tree
## simulate gamma rate heterogeneity
gg&lt;-rgamma(n=100,shape=0.25,rate=0.25)
dna&lt;-genSeq(mammal.tree,l=100,rate=gg)
</code></pre>


</div>