<div class="container">

<table style="width: 100%;"><tr>
<td>anoletree</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Phylogenetic datasets</h2>

<h3>Description</h3>

<p>Various phylogenetic datasets for comparative analysis.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(anole.data)
data(anoletree)
data(ant.geog)
data(ant.tree)
data(bat.tree)
data(bat_virus.data)
data(betaCoV.tree)
data(bonyfish.data)
data(bonyfish.tree)
data(butterfly.data)
data(butterfly.tree)
data(cordylid.data)
data(cordylid.tree)
data(darter.tree)
data(eel.data)
data(eel.tree)
data(elapidae.tree)
data(flatworm.data)
data(flatworm.tree)
data(liolaemid.tree)
data(mammal.data)
data(mammal.geog)
data(mammal.tree)
data(primate.data)
data(primate.tree)
data(salamanders)
data(sunfish.data)
data(sunfish.tree)
data(tortoise.geog)
data(tortoise.tree)
data(tropidurid.data)
data(tropidurid.tree)
data(vertebrate.data)
data(vertebrate.tree)
data(wasp.data)
data(wasp.trees)
data(whale.tree)
</code></pre>


<h3>Format</h3>

<p><code>anoletree</code> is an object of class <code>"simmap"</code>. <code>anole.data</code> is a data frame.
</p>
<p><code>ant.tree</code> is an object of class <code>"phylo"</code>. <code>ant.geog</code> is a vector.
</p>
<p><code>bat.tree</code> and <code>betaCoV.tree</code> are objects of class <code>"phylo"</code>. <code>bat_virus.data</code> is a data frame.
</p>
<p><code>bonyfish.tree</code> is an object of class <code>"phylo"</code>. <code>bonyfish.data</code> is a data frame.
</p>
<p><code>butterfly.tree</code> is an object of class <code>"phylo"</code>. <code>butterfly.data</code> is a data frame.
</p>
<p><code>cordylid.tree</code> is an object of class <code>"phylo"</code>. <code>cordylid.data</code> is a data frame.
</p>
<p><code>darter.tree</code> is an object of class <code>"phylo"</code>.
</p>
<p><code>eel.tree</code> is an object of class <code>"phylo"</code>. <code>eel.data</code> is a data frame.
</p>
<p><code>elapidae.tree</code> is an object of class <code>"phylo"</code>.
</p>
<p><code>flatworm.tree</code> is an object of class <code>"phylo"</code>. <code>flatworm.data</code> is a data frame.
</p>
<p><code>liolaemid.tree</code> is an object of class <code>"phylo"</code>. <code>liolaemid.data</code> is a data frame.
</p>
<p><code>mammal.tree</code> is an object of class <code>"phylo"</code>. <code>mammal.data</code> is a data frame. <code>mammal.geog</code> is a numerical matrix.
</p>
<p><code>primate.tree</code> is an object of class <code>"phylo"</code>. <code>primate.data</code> is a data frame.
</p>
<p><code>salamanders</code> is an object of class <code>"phylo"</code>.
</p>
<p><code>sunfish.tree</code> is an object of class <code>"simmap"</code>. <code>sunfish.data</code> is a data frame.
</p>
<p><code>tortoise.tree</code> is an object of class <code>"phylo"</code>. <code>tortoise.geog</code> is a data frame.
</p>
<p><code>tropidurid.tree</code> is an object of class <code>"simmap"</code>. <code>tropidurid.data</code> is a data frame.
</p>
<p><code>vertebrate.tree</code> is an object of class <code>"phylo"</code>. <code>vertebrate.data</code> is a data frame.
</p>
<p><code>wasp.trees</code> is an object of class <code>"multiPhylo"</code>. <code>wasp.data</code> is a data frame.
</p>
<p><code>whale.tree</code> is an object of class <code>"phylo"</code>.
</p>


<h3>Details</h3>

<p><code>anoletree</code> is a phylogeny of Greater Antillean anole species with a mapped discrete character - <em>ecomorph class</em>. <code>anole.data</code> is a data frame of morphological characters. Data and tree are from Mahler et al. (2010).
</p>
<p><code>ant.tree</code> is a phylogeny containing 84 species of fungus farming attine ants. <code>ant.geog</code> is a vector containing biogegraphic information for the species of the tree. Both the phylogeny and biogeographic information were originally published by Branstetter et al. (2017).
</p>
<p><code>bat.tree</code> is a ML phylogeny of bats from the families Mormoopidae, Molossidae, and Phyllostomidae, based on GenBank accession numbers reported in Caraballo (2022) and estimated using <span class="pkg">phangorn</span>. <code>betaCoV.tree</code> is a ML phylogenetic tree of betacoronaviruses based on GenBank accession numbers reported in Caraballo (2022) and estimated using <span class="pkg">phangorn</span>. <code>bat_virus.data</code> is an association table of host and virus from Caraballo (2022).
</p>
<p><code>butterfly.tree</code> and <code>butterfly.data</code> are a phylogeny and dataset of habitat use in Mycalesina butterflies from Halali et al. (2020). The phylogeny of that study was adapted from one originally published by Brattstrom et al. (2020).
</p>
<p><code>bonyfish.tree</code> and <code>bonyfish.data</code> are a phylogeny and dataset of spawning mode and parental care in bony fishes from Benun Sutton and Wilson (2019). Their phylogenetic tree was adapted from a tree originally published by Betancur-R et al. (2017)
</p>
<p><code>cordylid.tree</code> and <code>cordylid.data</code> are a phylogeny and dataset of morphological traits for three different principal components axes from Broeckhoven et al. (2016).
</p>
<p><code>darter.tree</code> is a phylogeny of 201 species of darter fish (Etheostomatinae) from Near et al. (2011).
</p>
<p><code>eel.tree</code> and <code>eel.data</code> are a phylogeny and dataset of feeding mode and maximum total length from 61 species of elopomorph eels from Collar et al. (2014).
</p>
<p><code>elapidae.tree</code> is a reconstructed phylogeny containing 175 species of snakes of the family Elapidae from Lee et al. (2016).
</p>
<p><code>flatworm.tree</code> and <code>flatworm.data</code> are a phylogeny and dataset of habitat preferences for flatworms from Benitez-Alvarez et al. (2020). <code>flatworm.tree</code> has been made ultrametric using penalized likelihood.
</p>
<p><code>liolaemid.tree</code> and <code>liolaemid.data</code> are a phylogenetic tree and phenotypic trait dataset of lizards from the family Liolaemidae from Esquerre et al. (2019).
</p>
<p><code>mammal.tree</code> and <code>mammal.data</code> are the phylogeny and dataset for mammal body size and home range size from Garland et al. (1992). <code>mammal.geog</code> is a matrix containing geographic coordinates for various of the species in <code>mammal.tree</code> pulled from citizen science observations registered on the <em>iNaturalist</em> web platform.
</p>
<p><code>primate.tree</code> and <code>primate.data</code> are a phylogeny and phenotypic trait dataset from Kirk and Kay (2004).
</p>
<p><code>salamanders</code> is a phylogeny of <em>Plethodon</em> salamanders from Highton and Larson (1979). According to Wikipedia, the genus <em>Plethodon</em> contains 55 species in total.
</p>
<p><code>sunfish.tree</code> and <code>sunfish.data</code> are the phylogeny and dataset for Centrarchidae and buccal morphology (respectively) from Revell and Collar (2009).
</p>
<p><code>tortoise.tree</code> and <code>tortoise.geog</code> are a phylogeny and dataset of geographic coordinates for Galapagos tortoises. The geographic locality information was approximated from Figure 1 of Poulakakis et al. (2020), and the tree was estimated from nucleotide sequence data published with the same study.
</p>
<p><code>tropidurid.tree</code> and <code>tropidurid.data</code> are the phylogeny and bivariate quantitative trait dataset of tropidurid lizards from Revell et al. (2022).
</p>
<p><code>vertebrate.tree</code> is a time-calibrated phylogeny of vertebrates and <code>vertebrate.data</code> is a dataset of phenotypic traits. The phylogeny is from <a href="http://www.timetree.org/">http://www.timetree.org/</a> (Hedges et al. 2006).
</p>
<p><code>wasp.trees</code> and <code>wasp.data</code> are the phylogeny and host-parasite associations from Lopez-Vaamonde et al. (2001).
</p>
<p><code>whale.tree</code> is a phylogeny of cetaceans originally published in Steeman et al. (2009).
</p>


<h3>Source</h3>

<p>Benitez-Alvarez, L., A. Maria Leal-Zanchet, A. Oceguera-Figueroa, R. Lopes Ferreira, D. de Medeiros Bento, J. Braccini, R. Sluys, and M. Riutort (2020) Phylogeny and biogeography of the Cavernicola (Platyhelminthes: Tricladida): Relicts of an epigean group sheltering in caves? <em>Molecular Phylogenetics and Evolution</em>, <b>145</b>, 106709.
</p>
<p>Benun Sutton, F., and A. B. Wilson (2019) Where are all the moms? External fertilization predicts the rise of male parental care in bony fishes. <em>Evolution</em>, <b>73</b>, 2451-2460.
</p>
<p>Betancur-R, R., E. O. Wiley, G. Arratia, A. Acero, N. Bailly, M. Miya, G. Lecointre, and G. Orti (2017) Phylogenetic classification of bony fishes. <em>BMC Ecology and Evolution</em>, <b>17</b>, 162.
</p>
<p>Branstetter, M. G., A. Jesovnik, J. Sosa-Calvo, M. W. Lloyd, B. C. Faircloth, S. G. Brady, and T. R. Schultz (2017) Dry habitats were crucibles of domestication in the evolution of agriculture in ants. <em>Proceedings of the Royal Society B</em>, <b>284</b>, 20170095.
</p>
<p>Brattstrom, O., K. Aduse-Poku, E. van Bergen, V. French, and P. M. Brakefield (2020) A release from developmental bias accelerates morphological diversification in butterfly eyespots. <em>Proceedings of the National Academy of Sciences</em>, <b>177</b>, 27474-27480.
</p>
<p>Broeckhoven, C., G. Diedericks, C. Hui, B. G. Makhubo, P. le Fras N. Mouton (2016) Enemy at the gates: Rapid defensive trait diversification in an adaptive radiation of lizards. <em>Evolution</em>, <b>70</b>, 2647-2656.
</p>
<p>Caraballo, D. A. (2022) Cross-species transmission of bat coronaviruses in the Americas: Contrasting patterns between alphacoronavirus and betacoronavirus. <em>Microbiology Spectrum</em>, <b>10</b>, e01411-22.
</p>
<p>Collar, D. C., P. C. Wainwright, M. E. Alfaro, L. J. Revell, and R. S. Mehta (2014) Biting disrupts integration to spur skull evolution in eels. <em>Nature Communications</em>, <b>5</b>, 5505.
</p>
<p>Esquerre, D., D. Ramirez-Alvarez, C. J. Pavon-Vazquez, J. Troncoso-Palacios, C. F. Garin, J. S. Keogh, and A. D. Leache (2019) Speciation across mountains: Phylogenomics, species delimitation and taxonomy of the <em>Liolaemus leopardinus</em> clade (Squamata, Liolaemidae). <em>Molecular Phylogenetics and Evolution</em>, <b>139</b>, 106524.
</p>
<p>Garland, T., Jr., P. H. Harvey, and A. R. Ives (1992) Procedures for the analysis of comparative data using phylogenetically independent contrasts. <em>Systematic Biology</em>, <b>41</b>, 18-32.
</p>
<p>Kirk, E. C., and R. F. Kay (2004) The evolution of high visual acuity in the Anthropoidea. In: Ross, C. F., Kay R. F. (Eds), <em>Anthropoid Origins. Developments in Primatology: Progress and Prospects</em>, 539-602. Springer, Boston, MA.
</p>
<p>Halali, S., E. van Bergen, C. J. Breuker, P. M. Brakefield, and O. Brattstrom (2020) Seasonal environments drive convergent evolution of a faster pace-of-life in tropical butterflies. <em>Ecology Letters</em>, <b>24</b>, 102-112.
</p>
<p>Hedges, S. B., J. Dudley, and S. Kumar (2006) TimeTree: A public knowledgebase of divergence times among organisms. <em>Bioinformatics</em>, <b>22</b>, 2971-2972.
</p>
<p>Highton, R., and A. Larson (1979) The genetic relationships of the salamanders of the genus <em>Plethodon</em>. <em>Systematic Zoology</em>, <b>28</b>, 579-599.
</p>
<p>Lee, M. S. Y., K. L. Saunders, B. King, and A. Palci (2016) Diversification rates and phenotypic evolution in venomous snakes (Elapidae). <em>Royal Society Open Science</em>, <b>3</b>, 150277.
</p>
<p>Lopez-Vaamonde, C., J. Y. Rasplus, G. D. Weiblen, and J. M. Cook (2001) Molecular phylogenies of fig wasps: Partial cocladogenesis of pollinators and parasites. <em>Molecular Phylogenetics and Evolution</em>, <b>21</b>, 55-71.
</p>
<p>Mahler, D. L, L. J. Revell, R. E. Glor, and J. B. Losos (2010) Ecological opportunity and the rate of morphological evolution in the diversification of Greater Antillean anoles. <em>Evolution</em>, <b>64</b>, 2731-2745.
</p>
<p>Near, T. J., C. M. Bossu, G. S. Bradburd, R. L. Carlson, R. C. Harrington, P. R. Hollingsworth Jr., B. P. Keck, and D. A. Etnier. (2011) Phylogeny and temporal diversification of darters (Percidae: Etheostomatinae). <em>Systematic Biology</em>, <b>60</b>, 565-595.
</p>
<p>Poulakakis, N., J. M. Miller, E. L. Jensen, L. B. Beheregaray, M. A. Russello, S. Glaberman, J. Boore, and A. Caccone. (2020) Colonization history of Galapagos giant tortoises: Insights from mitogenomes support the progression rule. <em>Journal of Zoological Systematics and Evolutionary Research</em>, <b>58</b>, 1262-1275.
</p>
<p>Revell, L. J., and D. C. Collar (2009) Phylogenetic analysis of the evolutionary correlation using likelihood. <em>Evolution</em>, <b>63</b>, 1090-1100.
</p>
<p>Revell, L. J., K. S. Toyama, and D. L. Mahler (2022) A simple hierarchical model for heterogeneity in the evolutionary correlation a phylogenetic tree. <em>PeerJ</em>, <b>10</b>, e13910.
</p>
<p>Steeman, M. E., M. B. Hebsgaard, R. E. Fordyce, S. Y. W. Ho, D. L. Rabosky, R. Nielsen, C. Rahbek, H. Glenner, M. V. Sorensen, and E. Willerslev (2009) Radiation of extant cetaceans driven by restructuring of the oceans. <em>Systematic Biology</em>, <b>58</b>, 573-585.
</p>


</div>