<div class="container">

<table style="width: 100%;"><tr>
<td>DetectPlasmaMutation</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>function DetectPlasmaMutation</h2>

<h3>Description</h3>

<p>This is the main function of the package that calls mutations by comparing at each genomic position the SNV or INDEL frequencies computed in one tested sample to
the SNV or INDEL Position-Error Rates computed from several control samples by a binomial test. An outlier detection is performed among all intra-sample p-values
to call a mutation.
For users wishing to develop their own analysis for other sequencing panel, it requires recalibrated BAM files control samples to be processed to compute the
Position-Error Rates stored in a file specified in <code>ber.ctrl.file</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">DetectPlasmaMutation(patient.dir = "./", patient.name = NULL,
  pos_ranges.file = NULL, ber.ctrl.file = NULL, bai.ext = ".bai",
  n.trim = 8, cov.min = 0, force = FALSE, show.more = FALSE,
  qcutoff.snv = 0.95, qcutoff.indel = 0.99, cutoff.sb.ref = 0.1,
  cutoff.sb.hotspot = 3.1, cutoff.sb.nonhotspot = 2.5,
  hotspot.indel = "chr7:55227950:55249171", output.dir = patient.dir)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>patient.dir, </code></td>
<td>
<p>char, foldername containing the rBAM folder of the patients. The typical folder hierarchy will consist of 'Plasma/rBAM'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>patient.name, </code></td>
<td>
<p>char, filename of the patient .bam file(s) (default NULL read all patients in folder <code>patient.dir</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos_ranges.file, </code></td>
<td>
<p>char, name of the Rdata file containing the three variables <code>pos_ind</code>, <code>pos_snp</code>, <code>pos_ranges</code> as build by the function <code>PrepareLibrary</code>. Default NULL, use the position_ranges.rda provides that we used for our analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ber.ctrl.file, </code></td>
<td>
<p>char, pathname of the file providing the background error rates obtained from the controls (default NULL use the provided background error rates obtained from our 29 controls). See <code>background_error_rate.txt</code> data and <code>BuildCtrlErrorRate</code> function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bai.ext, </code></td>
<td>
<p>char, filename extension of the bai files (default '.bai')</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.trim, </code></td>
<td>
<p>integer, number of base positions trimmed at the ends of each amplicon (default 8)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cov.min, </code></td>
<td>
<p>integer, minimal coverture required at each position (default 0)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>force, </code></td>
<td>
<p>boolean, (default FALSE) if TRUE force all computations to all files including already processed ones</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.more, </code></td>
<td>
<p>boolean, (default FALSE show only detected positions) if TRUE additional annotations on result plots are given for non-significant mutations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>qcutoff.snv, </code></td>
<td>
<p>numeric, proportion of kept base positions ranged by increasing 95th percentile SNV PER in control samples (default 0.95)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>qcutoff.indel, </code></td>
<td>
<p>numeric, proportion of kept base positions ranged by increasing 95th percentile INDEL PER in control samples (default 0.99)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cutoff.sb.ref, </code></td>
<td>
<p>numeric, exclude reference positions without cutoff &lt; strand bias &lt; 1-cutoff (default 0.1) (see Supplementary Materials in References)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cutoff.sb.hotspot, </code></td>
<td>
<p>numeric, exclude hotspot positions with Symmetric Odds Ratio test &gt; cutoff (default 3.1) (see Supplementary Materials in References)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cutoff.sb.nonhotspot, </code></td>
<td>
<p>numeric, exclude non-hotspot positions with Symmetric Odds Ratio test &gt; cutoff (default 2.5) (see Supplementary Materials in References)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hotspot.indel, </code></td>
<td>
<p>char, a vector containing the known positions of hotspot deletion/insertion defined as chrX:start:end (default 'chr7:55227950:55249171')</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output.dir, </code></td>
<td>
<p>char, name of the folder to save results  (default <code>patient.dir</code>).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>the number of processed patients
</p>


<h3>Author(s)</h3>

<p>N. Pécuchet, P. Laurent-Puig and Y. Rozenholc
</p>


<h3>References</h3>

<p><em>Analysis of base-position error rate of next-generation sequencing to detect tumor mutations in circulating DNA</em> N. Pécuchet, Y. Rozenholc, E. Zonta, D. Pietraz, A. Didelot, P. Combe, L. Gibault, J-B. Bachet, V. Taly, E. Fabre, H. Blons, P. Laurent-Puig in <em>Clinical Chemistry</em>
</p>


<h3>Examples</h3>

<pre><code class="language-R">     patient.dir=system.file("extdata","4test_only/case/",package="PlasmaMutationDetector")
     if (substr(patient.dir,nchar(patient.dir),nchar(patient.dir))!='/')
       patient.dir = paste0(patient.dir,'/') # TO RUN UNDER WINDOWS
     DetectPlasmaMutation(patient.dir,output.dir=paste0(tempdir(),'/'))


</code></pre>


</div>