<div class="container">

<table style="width: 100%;"><tr>
<td>TCGA-ESCA</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Esophageal carcinoma (ESCA) data or lung squamous cell carcinoma (LUSC)
data from The Cancer Genome Atlas (TCGA).
</h2>

<h3>Description</h3>

<p>The <code>TCGA-ESCA</code> dataset contains the objects <code>assemble</code>,
<code>Outcome</code>, and <code>m450info</code> for building the <code>MultiOmics</code>
object. Because its size exceeds the CRAN limits, the data is stored on
a remote server and must be loaded using the function
<code>loadESCAdata</code>.
</p>
<p>The <code>TCGA-LUSC1</code>dataset is a parallel object for lung
squamous cell carcinoma (LUSC) data, whihc must be loaded using the
<code>loadLUSCdata</code> function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">loadESCAdata(env = plasmaEnv)
loadLUSCdata(env = plasmaEnv)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>env</code></td>
<td>
<p>an environment in which to load the data. The default
value is a private environment in the package, accessible as
<code>plasmaEnv</code>. To make access easier, you can use
<code>globalenv()</code> or <code>.GlobalEnv</code>.</p>
</td>
</tr></table>
<h3>Format</h3>

<p>The “TCGA-ESCA” dataset contains the following:
</p>

<dl>
<dt><code>assemble</code></dt>
<dd>
<p>A list of 7 different omics dataframes with
varying numbers of features as rows (D) and varying number of
patients as columns (N). Note that some of these omics dataframes
had been manipulated to contain NAs, where these may be complete
on the GDC Dat Portal from which these data originally came. This
was done to illustrate the capability of the <code>plasma</code> package
on working with missing data.</p>
</dd>
</dl>
<dl>
<dt><code>ClinicalBin</code></dt>
<dd>
<p>a dataframe (53x185) of clinical binary values.</p>
</dd>
<dt><code>ClinicalCont</code></dt>
<dd>
<p>a dataframe (6x185) of clinical continuous values.</p>
</dd>
<dt><code>MAF</code></dt>
<dd>
<p>a dataframe (566x184) of minor allele frequencies
(MAF) that have been converted to binary based on whether they had
a MAF greater than 0.03 (1) or not (0).</p>
</dd>
<dt><code>Meth450</code></dt>
<dd>
<p>a dataframe (1454x185) of continuous beta
values from the Illumina Infinium HumanMethylation450 arrays. The
features in this dataframe have been filtered on mean greater than
0.15 and a standard deviation greater than 0.3.</p>
</dd>
<dt><code>miRSeq</code></dt>
<dd>
<p>a dataframe (926x166) of continuous counts
values from microRNA (miRNA) sequencing. The features in this
dataframe have been filtered on a standard deviation of 0.05.</p>
</dd>
<dt><code>mRNASeq</code></dt>
<dd>
<p>a dataframe (2520x157) of continuous counts
values from mRNA sequencing data. The features in this dataframe
have been filtered on a mean greater than 4 and a standard
deviation greater than 0.7.</p>
</dd>
<dt><code>RPPA</code></dt>
<dd>
<p>a dataframe (192x126) of continuous protein
expression values from reverse phase protein array (RPPA) assays.</p>
</dd>
</dl>
<dl>
<dt><code>Outcome</code></dt>
<dd>
<p>a dataframe (185x5) containing the survival
outcomes for the patients in <code>assemble</code>.</p>
</dd>
<dt><code>m450info</code></dt>
<dd>
<p>a dataframe (1454x3) containing gene symbol,
chromosome number, and genomic coordinate IDs corresponding to the
features (or “probes”) in <code>Meth450</code>.</p>
</dd>
</dl>
<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
Kyoko Yamaguchi <a href="mailto:kyoko.yamaguchi@osumc.edu">kyoko.yamaguchi@osumc.edu</a></p>


<h3>Source</h3>

<p><a href="https://portal.gdc.cancer.gov/projects/TCGA-ESCA">https://portal.gdc.cancer.gov/projects/TCGA-ESCA</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">fls &lt;- try(loadESCAdata())
if (inherits(fls, "try-error")) {
  stop("Unable to load data from remote server.")
}
</code></pre>


</div>