<div class="container">

<table style="width: 100%;"><tr>
<td>pleio_ideogram</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot ideogram from pleio_test results</h2>

<h3>Description</h3>

<p>Plots genomic segments that contain significant pleiotropic SNPs using results of pleio_test(). It also returns a dataframe with segment information.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pleio_ideogram(
  pleio_res,
  alpha = "bonferroni05",
  n_traits = 2,
  bp_positions,
  window_size = 1e+06,
  centromeres = NULL,
  color_bias = 1,
  set_plot = T,
  set_legend = T,
  set_ylim_prop = 1.1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pleio_res</code></td>
<td>
<p>list returned by pleio_test().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>numeric threshold for significance level (Bonferroni correction by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_traits</code></td>
<td>
<p>integer indicating the level of pleiotropy to test (a.k.a. number of traits).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bp_positions</code></td>
<td>
<p>dataframe with colnames 'chr' and 'pos' indicating the chromosome and position for each SNP. Rownames must contain SNP names matching results of pleio_test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>window_size</code></td>
<td>
<p>numeric value indicating the minimum size (in base pairs) of the genomic region that contains significant SNPs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>centromeres</code></td>
<td>
<p>string 'human' or dataframe (or matrix) with chromosome and position (in mbp) of the centromeres in the first and second columns. If NULL (default) does not plot the centromeres.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>color_bias</code></td>
<td>
<p>number for bias of the color scale. See help(colorRampPalette). By default color_bias = 1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>set_plot</code></td>
<td>
<p>logical indicating whether to plot the ideogram (TRUE by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>set_legend</code></td>
<td>
<p>logical indicating whether to plot a legend (TRUE by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>set_ylim_prop</code></td>
<td>
<p>numeric proportion of upper margin to fit the legend (no margin by default). 1 = no margin, 1.1 = 10% left for margin, etc.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>more plot arguments.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Ideogram plot and a dataframe with genomic segments information.
</p>


<h3>See Also</h3>

<p><code>pleio_plot</code>
</p>


</div>