<div class="container">

<table style="width: 100%;"><tr>
<td>readBED</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read BED file</h2>

<h3>Description</h3>

<p>Read a PLINK BED file into a R matrix.
</p>


<h3>Usage</h3>

<pre><code class="language-R">readBED(pfx, iid = 1, vid = 1, vfr = NULL, vto = NULL, quiet = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pfx</code></td>
<td>
<p>prefix of PLINK file set, or the fullname of a BED file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>iid</code></td>
<td>
<p>option to read <code>N</code> IID as row names. (def=1, see <code>readIID()</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vid</code></td>
<td>
<p>option to read <code>P</code> VID as col names. (def=1, see <code>readVID()</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vfr</code></td>
<td>
<p>variant-wise, from where to read? (number/proportion, def=1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vto</code></td>
<td>
<p>varinat-wise, to where then stop? (number/proportion, def=P)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>suppress screen printing? (def=TRUE)</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This is  meant for genotype that  can fit into  system memory; the size  of R
matrix is 16 times  the BED file.  To traverse a huge  BED several varants at
time  without  loading it  entirely  into  the  memoty, see  <code>scanBED()</code>  and
<code>loopBED()</code>.
</p>
<p>A PLINK1 binary fileset has three files,
</p>

<dl>
<dt>
<code>pfx.fam</code>:</dt>
<dd>
<p>text table of <code>N</code> individuals.</p>
</dd>
<dt>
<code>pfx.bim</code>:</dt>
<dd>
<p>text table of <code>P</code> genomic variants (i.e., SNPs).</p>
</dd>
<dt>
<code>pfx.bed</code>:</dt>
<dd>
<p><code>N</code> x <code>P</code> genotype matrix in condensed binary format.</p>
</dd>
</dl>
<p>The three  files comprising a genotype  data are typically referred  by their
common  prefix,  for  example,  the  X  chromosome  genotype  represented  by
<code>chrX.bed</code>, <code>chrX.fam</code>, and <code>chrX.bim</code> are jointly refered by <code>chrX</code>.
</p>


<h3>Value</h3>

<p>genotype matrix with <code>N</code> row individuals and <code>P</code> column variants.
</p>


<h3>See Also</h3>

<p>readBED
</p>


<h3>Examples</h3>

<pre><code class="language-R">## read an entire small data
bed &lt;- system.file("extdata", 'm20.bed', package="plinkFile")
gmx &lt;- readBED(bed, quiet=FALSE)

## read part of a large data
bed &lt;- system.file("extdata", '000.bed', package="plinkFile")
U &lt;- readBED(bed, vfr=01, vto=10, quiet=FALSE)
V &lt;- readBED(bed, vfr=11, vto=20, quiet=FALSE)
W &lt;- cbind(U, V)
X &lt;- readBED(bed, vfr=01, vto=20, quiet=FALSE)
all.equal(W, X)

</code></pre>


</div>