<div class="container">

<table style="width: 100%;"><tr>
<td>scanBED</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>travers variants in a PLINK1 BED fileset</h2>

<h3>Description</h3>

<p>Sequentially visits variants in a PLINK1  BED fileset with a stepping window
matrix, and process each window matrix  with user scripts either in function
or expression form, meant for data to big to fit in the memory.
</p>
<p>To read the entire BED into a R matrix, use <code style="white-space: pre;">⁠[readBED]()⁠</code> instead.
</p>


<h3>Usage</h3>

<pre><code class="language-R">scanBED(
  pfx,
  FUN,
  ...,
  win = 1,
  iid = 1,
  vid = 1,
  vfr = NULL,
  vto = NULL,
  buf = 2^24,
  simplify = TRUE
)

loopBED(
  pfx,
  EXP,
  GVR = "g",
  win = 1,
  iid = 1,
  vid = 1,
  vfr = NULL,
  vto = NULL,
  buf = 2^24,
  simplify = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pfx</code></td>
<td>
<p>prefix of PLINK BED.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FUN</code></td>
<td>
<p>a function to process each window of variants;</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional argument for <em><code>FUN</code></em> when <code>scanBED</code> is used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>win</code></td>
<td>
<p>reading window size (def=100 variants per window)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>iid</code></td>
<td>
<p>option to read <code>N</code> IID as row names (def=1, see <code>readIID()</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vid</code></td>
<td>
<p>option to read <code>P</code> VID as col names (def=1, see <code>readVID()</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vfr</code></td>
<td>
<p>variant-wise, from where to read (number/proportion, def=1)?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vto</code></td>
<td>
<p>varinat-wise, to where then stop (number/proportion, def=P)?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>buf</code></td>
<td>
<p>buffer size in byptes (def=2^24, or 16 MB).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>simplify</code></td>
<td>
<p>try simplifying the results  into an array, or leave them in
a list, or specify a function to simplify the said list.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>EXP</code></td>
<td>
<p>a R expression to evaluate with each window of variants;</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GVR</code></td>
<td>
<p>a R variable name to assign the window to (def="g").</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>results of all windows processed by the user script.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>scanBED()</code>: apply a function to variants in a PLINK1 BED fileset
</p>
<p>Travers <code>P</code>  variants via a sliding  window while calling a  function on each
window of variants without side effects on the calling environment, mimicking
various R <code>apply</code> utilities.
</p>
</li>
<li> <p><code>loopBED()</code>: evaluate an expression on variants in a PLINK1 BED
</p>
<p>Travers <code>P</code> variants via a sliding  window and evaluate an R expression given
each  window of  variants,  with  side effects  on  the calling  environment,
mimicking the syntax of R <code>for</code> loop.
</p>
</li>
</ul>
<h3>BED PLINK1 Binary Pedigree fileset</h3>

<p>A popular format to store biallelic dosage genotype, with three files,
</p>

<ul>
<li> <p><em>pfx</em>.fam: text table for <code>N</code> individuals, detailed in readFAM;
</p>
</li>
<li> <p><em>pfx</em>.bim: text table for <code>P</code> variants, detailed in readBIM;
</p>
</li>
<li> <p><em>pfx</em>.bed: transposed genotype matrix (<code>P</code> x <code>N</code> ) in binary format.
</p>
</li>
</ul>
<p>The triplets are commonly referred by the shared prefix (<code>pfx</code>), e.g., the X
chromosome represented by "chrX.bed", "chrX.fam", and "chrX.bim" are refered
by <code>"chrX"</code>.
</p>
<p>The binary file "<em>pfx</em>.bed" represent each dosage value with two bits - just
enough to encode all four possiblities: 0, 1, or 2 alleles, or missing.
</p>
<p>The number of variants (<code>P</code>) and samples (<code>N</code>) equals to the number of lines
in text file "<em>pfx</em>.bim" and "<em>pfx</em>.fam", respectively.
</p>
<p>For the detailed specification of PLINK1 BED genotype format, see the lagecy
PLINK v1.07 page at: \
<a href="https://zzz.bwh.harvard.edu/plink/binary.shtml">https://zzz.bwh.harvard.edu/plink/binary.shtml</a>. \
For the modern use and management of PLINK1 BED, see the PLINK v1.9 page: \
<a href="https://www.cog-genomics.org/plink/1.9/input#bed">https://www.cog-genomics.org/plink/1.9/input#bed</a>.
</p>


<h3>detailed arguments</h3>


<ul>
<li> <p><code>win</code>: visiting window size.
</p>
<p>the number of variants per window, that  is, the number of columns in each
window matrix passed to the user script.
</p>
<p>For example, a size one window means  the user script will be dealing with
only one variant at  a time, received from in a matrix  of a single column
– a manner similar to  genome wide association analysis (GWAS).  However,
a larger, multi-variant window coupled with R language's vector and matrix
syntax can significantly boost efficiency.
</p>
<p>The default size is 1000 variants / columns per window.
</p>
</li>
<li> <p><code>buf</code>: buffer size in bytes
</p>
<p>a large buffer reduces the frequency of hard disk visits when traversing a
PLINK1 BED file, which in turn reduces non-computation overhead.
</p>
<p>The default size is <code>2^24</code> bytes, or 16 MB.
</p>
</li>
<li> <p><code>simplify</code>:
</p>
<p>when FALSE: resuts  of user script processing each window  of variants are
returned in a list;
</p>
<p>when TRUE,  use <code>simplify2array</code> to put  the results into an  array, if it
fails, fallback and return a list.
</p>
<p>when a function is specified, it is then used to simplify the results, if
an execption is thrown, fallback and return a list.
</p>
<p>e.g., the window script returns a  data frame of estimate, standard error,
t-statistic, and p-value  for each variant, <code>simplify =  rbind</code> to combine
results of all windows into one data frame of <code>P</code> rows and four columns of
statistics.
</p>
</li>
</ul>
<h3>genotype context</h3>

<p>context infomation  such the number of  variants and samples are  updated in
the window processing environment to ease user scripting, which includes:
</p>

<ul>
<li> <p><code>.i</code>: indies of variants in the current visiting window;
</p>
</li>
<li> <p><code>.p</code>: number of variants in the current visiting window.
</p>
</li>
<li> <p><code>.P</code>: total number of variants;
</p>
</li>
<li> <p><code>.w</code>: index of the current window;
</p>
</li>
<li> <p><code>.W</code>: total number of windows to go through;
</p>
</li>
<li> <p><code>.N</code>: number of individuals.
</p>
</li>
<li> <p><code>.b</code>: index of the current buffer.
</p>
</li>
<li> <p><code>.B</code>: number of buffers to be swapped.
</p>
</li>
</ul>
<p>e.g. (1) print percentage progress with <code>print(.w / .W * 100)</code>; \
e.g. (2) use <code>inf &lt;- readBIM(pfx)</code> to  read the table of variants before the
window visits,  later use <code>inf[.i,  ]</code> to  access meta-data for  variants in
each window.
</p>


<h3>See Also</h3>

<p><code style="white-space: pre;">⁠[readBED]⁠</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## traverse genotype, apply R function without side effects
pfx &lt;- file.path(system.file("extdata", package="plinkFile"), "000")
ret &lt;- scanBED(pfx, function(g)
{
    .af &lt;- colMeans(g, na.rm=TRUE) / 2
    maf &lt;- pmin(.af, 1 - .af)
    mis &lt;- colSums(is.na(g)) / .N
    pct &lt;- round(.w / .W * 100, 2)
    cbind(buf=.b, wnd=.w, idx=.i, MAF=maf, MIS=mis, PCT=pct)
},
vfr=NULL, vto=NULL, win=13, simplify=rbind, buf=2^18)
head(ret)
tail(ret)

## traversing genotype, evaluate R expression with side effects
pfx &lt;- file.path(system.file("extdata", package="plinkFile"), "000.bed")
ret &lt;- list() # use side effect to keep the result of each window.
loopBED(pfx,
{
    af &lt;- colMeans(gt, na.rm=TRUE) / 2
    sg &lt;- af * (1 - af)
    ret[[.w]] &lt;- cbind(wnd=.w, alf=af, var=sg)
},
win=13, GVR="gt", vid=3, buf=2^18)
head(ret)
tail(ret)

</code></pre>


</div>