<div class="container">

<table style="width: 100%;"><tr>
<td>check_maf</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Identification of SNPs with low minor allele frequency</h2>

<h3>Description</h3>

<p>Runs and evaluates results from plink –freq. It calculates the minor allele
frequencies for all variants in the individuals that passed the
<code>perIndividualQC</code>. The minor allele frequency distributions is
plotted as a histogram.
</p>


<h3>Usage</h3>

<pre><code class="language-R">check_maf(
  indir,
  name,
  qcdir = indir,
  macTh = 20,
  mafTh = NULL,
  verbose = FALSE,
  interactive = FALSE,
  path2plink = NULL,
  showPlinkOutput = TRUE,
  keep_individuals = NULL,
  remove_individuals = NULL,
  exclude_markers = NULL,
  extract_markers = NULL,
  legend_text_size = 5,
  legend_title_size = 7,
  axis_text_size = 5,
  axis_title_size = 7,
  title_size = 9
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>indir</code></td>
<td>
<p>[character] /path/to/directory containing the basic PLINK data
files name.bim, name.bed, name.fam files.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>name</code></td>
<td>
<p>[character] Prefix of PLINK files, i.e. name.bed, name.bim,
name.fam.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>qcdir</code></td>
<td>
<p>[character] /path/to/directory where results will be written to.
If <code>perIndividualQC</code> was conducted, this directory should be the
same as qcdir specified in <code>perIndividualQC</code>, i.e. it contains
name.fail.IDs with IIDs of individuals that failed QC. User needs writing
permission to qcdir. Per default, qcdir=indir.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>macTh</code></td>
<td>
<p>[double] Threshold for minor allele cut cut-off, if both mafTh
and macTh are specified, macTh is used (macTh = mafTh\*2\*NrSamples).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mafTh</code></td>
<td>
<p>[double] Threshold for minor allele frequency cut-off.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>[logical] If TRUE, progress info is printed to standard out
and specifically, if TRUE, plink log will be displayed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>interactive</code></td>
<td>
<p>[logical] Should plots be shown interactively? When
choosing this option, make sure you have X-forwarding/graphical interface
available for interactive plotting. Alternatively, set interactive=FALSE and
save the returned plot object (p_hwe) via ggplot2::ggsave(p=p_maf,
other_arguments) or pdf(outfile) print(p_maf) dev.off().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>path2plink</code></td>
<td>
<p>[character] Absolute path to PLINK executable
(<a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accessible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code>exec</code>('plink').</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>showPlinkOutput</code></td>
<td>
<p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
retained in the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples not listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remove_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
removed from the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exclude_markers</code></td>
<td>
<p>[character] Path to file with makers to be
removed from the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All listed variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>extract_markers</code></td>
<td>
<p>[character] Path to file with makers to be
included in the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All unlisted variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend_text_size</code></td>
<td>
<p>[integer] Size for legend text.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend_title_size</code></td>
<td>
<p>[integer] Size for legend title.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>axis_text_size</code></td>
<td>
<p>[integer] Size for axis text.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>axis_title_size</code></td>
<td>
<p>[integer] Size for axis title.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title_size</code></td>
<td>
<p>[integer] Size for plot title.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>check_maf</code> uses plink –remove name.fail.IDs –freq to calculate the
minor allele frequencies for all variants in the individuals that passed the
<code>perIndividualQC</code>. It does so without generating a new dataset
but simply removes the IDs when calculating the statistics.
</p>
<p>For details on the output data.frame fail_maf, check the original
description on the PLINK output format page:
<a href="https://www.cog-genomics.org/plink/1.9/formats#frq">https://www.cog-genomics.org/plink/1.9/formats#frq</a>.
</p>


<h3>Value</h3>

<p>Named list with i) fail_maf containing a [data.frame] with CHR
(Chromosome code), SNP (Variant identifier), A1 (Allele 1; usually minor), A2
(Allele 2; usually major), MAF (Allele 1 frequency), NCHROBS (Number of
allele observations) for all SNPs that failed the mafTh/macTh and ii) p_maf,
a ggplot2-object 'containing' the MAF distribution histogram which can be
shown by (print(p_maf)).
</p>


<h3>Examples</h3>

<pre><code class="language-R">indir &lt;- system.file("extdata", package="plinkQC")
qcdir &lt;- tempdir()
name &lt;- "data"
path2plink &lt;- '/path/to/plink'
# the following code is not run on package build, as the path2plink on the
# user system is not known.
## Not run: 
# run on all individuals and markers
fail_maf &lt;- check_maf(indir=indir, qcdir=qcdir, name=name, macTh=15,
interactive=FALSE, verbose=TRUE, path2plink=path2plink)

# run on subset of individuals and markers
keep_individuals_file &lt;- system.file("extdata", "keep_individuals",
package="plinkQC")
exclude_markers_file &lt;- system.file("extdata", "exclude_markers",
package="plinkQC")
fail_maf &lt;- check_maf(qcdir=qcdir, indir=indir,
name=name, interactive=FALSE, verbose=TRUE, path2plink=path2plink,
keep_individuals=keep_individuals_file, exclude_markers=exclude_markers_file)

## End(Not run)
</code></pre>


</div>