<div class="container">

<table style="width: 100%;"><tr>
<td>run_check_ancestry</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Run PLINK principal component analysis</h2>

<h3>Description</h3>

<p>Run plink –pca to calculate the principal components on merged genotypes
of the study and reference dataset.
</p>


<h3>Usage</h3>

<pre><code class="language-R">run_check_ancestry(
  indir,
  prefixMergedDataset,
  qcdir = indir,
  verbose = FALSE,
  path2plink = NULL,
  keep_individuals = NULL,
  remove_individuals = NULL,
  exclude_markers = NULL,
  extract_markers = NULL,
  showPlinkOutput = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>indir</code></td>
<td>
<p>[character] /path/to/directory containing the basic PLINK data
files prefixMergedDataset.bim,prefixMergedDataset.fam and
prefixMergedDataset.bed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prefixMergedDataset</code></td>
<td>
<p>[character] Prefix of merged study and
reference data files, i.e. prefixMergedDataset.bed, prefixMergedDataset.bim,
prefixMergedDataset.fam.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>qcdir</code></td>
<td>
<p>[character] /path/to/directory to save
prefixMergedDataset.eigenvec as returned by plink –pca. User needs writing
permission to qcdir. Per default qcdir=indir.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>[logical] If TRUE, progress info is printed to standard out.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>path2plink</code></td>
<td>
<p>[character] Absolute path to PLINK executable
(<a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accessible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code>exec</code>('plink').</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
retained in the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples not listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remove_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
removed from the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exclude_markers</code></td>
<td>
<p>[character] Path to file with makers to be
removed from the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All listed variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>extract_markers</code></td>
<td>
<p>[character] Path to file with makers to be
included in the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All unlisted variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>showPlinkOutput</code></td>
<td>
<p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Both, <code>run_check_ancestry</code> and its evaluation by
<code>evaluate_check_ancestry</code> can simply be invoked by
<code>check_ancestry</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">indir &lt;- system.file("extdata", package="plinkQC")
qcdir &lt;- tempdir()
prefixMergedDataset &lt;- 'data.HapMapIII'
path2plink &lt;- 'path/to/plink'
# the following code is not run on package build, as the path2plink on the
# user system is not known.
## Not run: 
# ancestry check on all individuals in dataset
run &lt;- run_check_ancestry(indir=indir, qcdir=qcdir, prefixMergedDataset,
path2plink=path2plink)

# ancestry check on subset of dataset
remove_individuals_file &lt;- system.file("extdata", "remove_individuals",
package="plinkQC")
run &lt;- run_check_ancestry(indir=indir, qcdir=qcdir, name=name,
remove_individuals=remove_individuals_file, path2plink=path2plink)

## End(Not run)
</code></pre>


</div>