<div class="container">

<table style="width: 100%;"><tr>
<td>cleanData</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create plink dataset with individuals and markers passing quality control</h2>

<h3>Description</h3>

<p>Individuals that fail per-individual QC and markers that fail
per-marker QC are removed from indir/name.bim/.bed/.fam and a new, dataset
with the remaining individuals and markers is created as
qcdir/name.clean.bim/.bed/.fam.
</p>


<h3>Usage</h3>

<pre><code class="language-R">cleanData(
  indir,
  name,
  qcdir = indir,
  filterSex = TRUE,
  filterHeterozygosity = TRUE,
  filterSampleMissingness = TRUE,
  filterAncestry = TRUE,
  filterRelated = TRUE,
  filterSNPMissingness = TRUE,
  lmissTh = 0.01,
  filterHWE = TRUE,
  hweTh = 1e-05,
  filterMAF = TRUE,
  macTh = 20,
  mafTh = NULL,
  path2plink = NULL,
  verbose = FALSE,
  keep_individuals = NULL,
  remove_individuals = NULL,
  exclude_markers = NULL,
  extract_markers = NULL,
  showPlinkOutput = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>indir</code></td>
<td>
<p>[character] /path/to/directory containing the basic PLINK data
files name.bim, name.bed, name.fam files.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>name</code></td>
<td>
<p>[character] Prefix of PLINK files, i.e. name.bed, name.bim,
name.fam.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>qcdir</code></td>
<td>
<p>[character] /path/to/directory where results will be written to.
If <code>perIndividualQC</code> was conducted, this directory should be the
same as qcdir specified in <code>perIndividualQC</code>, i.e. it contains
name.fail.IDs with IIDs of individuals that failed QC. User needs writing
permission to qcdir. Per default, qcdir=indir.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filterSex</code></td>
<td>
<p>[logical] Set to exclude samples that failed the sex
check (via <code>check_sex</code> or <code>perIndividualQC</code>).
Requires file qcdir/name.fail-sexcheck.IDs (automatically created by
<code>perIndividualQC</code> if do.evaluate_check_sex set to TRUE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filterHeterozygosity</code></td>
<td>
<p>[logical] Set to exclude samples that failed
check for outlying heterozygosity rates (via
<code>check_het_and_miss</code> or
<code>perIndividualQC</code>). Requires file qcdir/name.fail-het.IDs
(automatically created by <code>perIndividualQC</code> if
do.evaluate_check_het_and_miss set to TRUE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filterSampleMissingness</code></td>
<td>
<p>[logical] Set to exclude samples that failed
check for excessive missing genotype rates (via
<code>check_het_and_miss</code> or
<code>perIndividualQC</code>). Requires file qcdir/name.fail-imiss.IDs
(automatically created by <code>perIndividualQC</code> if
do.evaluate_check_het_and_miss set to TRUE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filterAncestry</code></td>
<td>
<p>[logical] Set to exclude samples that failed ancestry
check (via <code>check_ancestry</code> or <code>perIndividualQC</code>).
Requires file qcdir/name.fail-ancestry.IDs (automatically created by
<code>perIndividualQC</code> if do.check_ancestry set to TRUE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filterRelated</code></td>
<td>
<p>[logical] Set to exclude samples that failed relatedness
check (via <code>check_relatedness</code> or <code>perIndividualQC</code>).
Requires file qcdir/name.fail-IBD.IDs (automatically created by
<code>perIndividualQC</code> if do.evaluate_check_relatedness set to TRUE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filterSNPMissingness</code></td>
<td>
<p>[logical] Set to exclude markers that have
excessive missing rates across samples (via
<code>check_snp_missingness</code> or <code>perMarkerQC</code>). Requires
lmissTh to be set.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lmissTh</code></td>
<td>
<p>[double] Threshold for acceptable variant missing rate across
samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filterHWE</code></td>
<td>
<p>[logical] Set to exclude markers that fail HWE exact test
(via <code>check_hwe</code> or <code>perMarkerQC</code>). Requires hweTh to
be set.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hweTh</code></td>
<td>
<p>[double] Significance threshold for deviation from HWE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filterMAF</code></td>
<td>
<p>[logical] Set to exclude markers that fail minor allele
frequency or minor allele count threshold (via <code>check_maf</code> or
<code>perMarkerQC</code>). Requires mafTh or macTh to be set.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>macTh</code></td>
<td>
<p>[double] Threshold for minor allele cut cut-off, if both mafTh
and macTh are specified, macTh is used (macTh = mafTh\*2\*NrSamples).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mafTh</code></td>
<td>
<p>[double] Threshold for minor allele frequency cut-off.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>path2plink</code></td>
<td>
<p>[character] Absolute path to PLINK executable
(<a href="https://www.cog-genomics.org/plink/1.9/">https://www.cog-genomics.org/plink/1.9/</a>) i.e.
plink should be accessible as path2plink -h. The full name of the executable
should be specified: for windows OS, this means path/plink.exe, for unix
platforms this is path/plink. If not provided, assumed that PATH set-up works
and PLINK will be found by <code>exec</code>('plink').</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>[logical] If TRUE, progress info is printed to standard out.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
retained in the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples not listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remove_individuals</code></td>
<td>
<p>[character] Path to file with individuals to be
removed from the analysis. The file has to be a space/tab-delimited text file
with family IDs in the first column and within-family IDs in the second
column. All samples listed in this file will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#indiv">https://www.cog-genomics.org/plink/1.9/filter#indiv</a>.
Default: NULL, i.e. no filtering on individuals.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exclude_markers</code></td>
<td>
<p>[character] Path to file with makers to be
removed from the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All listed variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>extract_markers</code></td>
<td>
<p>[character] Path to file with makers to be
included in the analysis. The file has to be a text file with a list of
variant IDs (usually one per line, but it's okay for them to just be
separated by spaces). All unlisted variants will be removed from the current
analysis. See <a href="https://www.cog-genomics.org/plink/1.9/filter#snp">https://www.cog-genomics.org/plink/1.9/filter#snp</a>.
Default: NULL, i.e. no filtering on markers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>showPlinkOutput</code></td>
<td>
<p>[logical] If TRUE, plink log and error messages are
printed to standard out.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>names [list] with i) passIDs, containing a [data.frame] with family
[FID] and individual [IID] IDs of samples that pass the QC, ii) failIDs,
containing a [data.frame] with family [FID] and individual [IID] IDs of
samples that fail the QC.
</p>


<h3>Examples</h3>

<pre><code class="language-R">package.dir &lt;- find.package('plinkQC')
indir &lt;- file.path(package.dir, 'extdata')
qcdir &lt;- tempdir()
name &lt;- "data"
path2plink &lt;- '/path/to/plink'
# the following code is not run on package build, as the path2plink on the
# user system is not known.
## Not run: 
# Run qc on all samples and markers in the dataset
## Run individual QC checks
fail_individuals &lt;- perIndividualQC(indir=indir, qcdir=qcdir, name=name,
refSamplesFile=paste(qcdir, "/HapMap_ID2Pop.txt",sep=""),
refColorsFile=paste(qcdir, "/HapMap_PopColors.txt", sep=""),
prefixMergedDataset="data.HapMapIII", interactive=FALSE, verbose=FALSE,
path2plink=path2plink)

## Run marker QC checks
fail_markers &lt;- perMarkerQC(indir=indir, qcdir=qcdir, name=name,
path2plink=path2plink)

## Create new dataset of individuals and markers passing QC
ids_all &lt;- cleanData(indir=indir, qcdir=qcdir, name=name, macTh=15,
verbose=TRUE, path2plink=path2plink, filterAncestry=FALSE,
filterRelated=TRUE)

# Run qc on subset of samples and markers in the dataset
highlight_samples &lt;- read.table(system.file("extdata", "keep_individuals",
package="plinkQC"))
remove_individuals_file &lt;- system.file("extdata", "remove_individuals",
package="plinkQC")

fail_individuals &lt;- perIndividualQC(indir=indir, qcdir=qcdir, name=name,
dont.check_ancestry = TRUE, interactive=FALSE, verbose=FALSE,
highlight_samples = highlight_samples[,2], highlight_type = "label",
remove_individuals = remove_individuals_file, path2plink=path2plink)

## Run marker QC checks
fail_markers &lt;- perMarkerQC(indir=indir, qcdir=qcdir, name=name,
path2plink=path2plink)

## Create new dataset of individuals and markers passing QC
ids_all &lt;- cleanData(indir=indir, qcdir=qcdir, name=name, macTh=15,
verbose=TRUE, path2plink=path2plink, filterAncestry=FALSE,
remove_individuals = remove_individuals_file)

## End(Not run)
</code></pre>


</div>