<div class="container">

<table style="width: 100%;"><tr>
<td>prepare_data</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Prepare data for use in PLN models</h2>

<h3>Description</h3>

<p>Prepare data in proper format for use in PLN model and its variants. The function (i) merges a count table and
a covariate data frame in the most comprehensive way and (ii) computes offsets from the count table using one of several normalization schemes (TSS, CSS, RLE, GMPR, Wrench, etc). The function fails with informative messages when the heuristics used for sample matching fail.
</p>


<h3>Usage</h3>

<pre><code class="language-R">prepare_data(counts, covariates, offset = "TSS", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>counts</code></td>
<td>
<p>Required. An abundance count table, preferably with dimensions names and species as columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>covariates</code></td>
<td>
<p>Required. A covariates data frame, preferably with row names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>offset</code></td>
<td>
<p>Optional. Normalization scheme used to compute scaling factors used as offset during PLN inference. Available schemes are "TSS" (Total Sum Scaling, default), "CSS" (Cumulative Sum Scaling, used in metagenomeSeq), "RLE" (Relative Log Expression, used in DESeq2), "GMPR" (Geometric Mean of Pairwise Ratio, introduced in Chen et al., 2018), Wrench (introduced in Kumar et al., 2018) or "none". Alternatively the user can supply its own vector or matrix of offsets (see note for specification of the user-supplied offsets).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional parameters passed on to <code>compute_offset()</code></p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data.frame suited for use in <code>PLN()</code> and its variants with two specials components: an abundance count matrix (in component "Abundance") and an offset vector/matrix (in component "Offset", only if offset is not set to "none")
</p>


<h3>Note</h3>

<p>User supplied offsets should be either vectors/column-matrices or have the same number of column as the original count matrix and either (i) dimension names or (ii) the same dimensions as the count matrix. Samples are trimmed in exactly the same way to remove empty samples.
</p>


<h3>References</h3>

<p>Chen, L., Reeve, J., Zhang, L., Huang, S., Wang, X. and Chen, J. (2018) GMPR: A robust normalization method for zero-inflated count data with application to microbiome sequencing data. PeerJ, 6, e4600 <a href="https://doi.org/10.7717/peerj.4600">doi:10.7717/peerj.4600</a>
</p>
<p>Paulson, J. N., Colin Stine, O., Bravo, H. C. and Pop, M. (2013) Differential abundance analysis for microbial marker-gene surveys. Nature Methods, 10, 1200-1202 <a href="https://doi.org/10.1038/nmeth.2658">doi:10.1038/nmeth.2658</a>
</p>
<p>Anders, S. and Huber, W. (2010) Differential expression analysis for sequence count data. Genome Biology, 11, R106 <a href="https://doi.org/10.1186/gb-2010-11-10-r106">doi:10.1186/gb-2010-11-10-r106</a>
</p>
<p>Kumar, M., Slud, E., Okrah, K. et al. (2018) Analysis and correction of compositional bias in sparse sequencing count data. BMC Genomics 19, 799 <a href="https://doi.org/10.1186/s12864-018-5160-5">doi:10.1186/s12864-018-5160-5</a>
</p>
<p>Robinson, M.D., Oshlack, A. (2010) A scaling normalization method for differential expression analysis of RNA-seq data. Genome Biol 11, R25 <a href="https://doi.org/10.1186/gb-2010-11-3-r25">doi:10.1186/gb-2010-11-3-r25</a>
</p>


<h3>See Also</h3>

<p><code>compute_offset()</code> for details on the different normalization schemes
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(trichoptera)
proper_data &lt;- prepare_data(
 counts     = trichoptera$Abundance,
 covariates = trichoptera$Covariate,
 offset     = "GMPR",
 scale      = "count"
)
proper_data$Abundance
proper_data$Offset
</code></pre>


</div>