<div class="container">

<table style="width: 100%;"><tr>
<td>preprocess</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>preprocess for microarray data</h2>

<h3>Description</h3>

<p>The function <code>preprocess</code> performs a preprocessing of microarray data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">preprocess(Xtrain, Xtest=NULL,Threshold=c(100,16000),Filtering=c(5,500),
		log10.scale=TRUE,row.stand=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Xtrain</code></td>
<td>
<p>a (ntrain x p) data matrix of predictors. <code>Xtrain</code> must be a matrix. 
Each row corresponds to an observation and each column to a predictor variable.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Xtest</code></td>
<td>
<p>a (ntest x p) matrix containing the predictors for the test data
set. <code>Xtest</code> may also be a vector of length p (corresponding to only one 
test observation).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Threshold</code></td>
<td>
<p> a vector of length 2 containing the values (threshmin,threshmax) for 
thresholding data in preprocess. Data is thresholded to value threshmin and ceiled to value 
threshmax. If <code>Threshold</code> is NULL then no thresholding is done. By default, if the value 
given for <code>Threshold</code> is not valid, no thresholding is done.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Filtering</code></td>
<td>
<p> a vector of length 2 containing the values (FiltMin,FiltMax) for filtering genes 
in preprocess. Genes with max/min$&lt;= FiltMin$ and (max-min)$&lt;= FiltMax$ are excluded. 
If <code>Filtering</code> is NULL then no thresholding is done. By default, if the value given for 
<code>Filtering</code> is not valid, no filtering is done.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log10.scale</code></td>
<td>
<p>a logical value equal to TRUE if a log10-transformation has to be done.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>row.stand</code></td>
<td>
<p>a logical value equal to TRUE if a standardisation in row has to be done.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The pre-processing steps recommended by Dudoit et al. (2002) are performed.
The default values are those adapted for Colon data.
</p>


<h3>Value</h3>

<p> A list with the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>pXtrain</code></td>
<td>
<p>the (ntrain x p') matrix containing the preprocessed train data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pXtest</code></td>
<td>
<p>the (ntest x p') matrix containing the preprocessed test data.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Sophie Lambert-Lacroix 
(<a href="http://membres-timc.imag.fr/Sophie.Lambert/">http://membres-timc.imag.fr/Sophie.Lambert/</a>) 
and 
Julie Peyre (<a href="https://membres-ljk.imag.fr/Julie.Peyre/">https://membres-ljk.imag.fr/Julie.Peyre/</a>).
</p>


<h3>References</h3>

<p>Dudoit, S. and Fridlyand, J. and Speed, T. (2002). Comparison of discrimination methods 
for the classification of tumors using gene expression data, 
Journal of the American Statistical Association, 97, 77â€“87.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># load plsgenomics library
library(plsgenomics)

# load Colon data
data(Colon)
IndexLearn &lt;- c(sample(which(Colon$Y==2),27),sample(which(Colon$Y==1),14))

Xtrain &lt;- Colon$X[IndexLearn,]
Ytrain &lt;- Colon$Y[IndexLearn]
Xtest &lt;- Colon$X[-IndexLearn,]

# preprocess data
resP &lt;- preprocess(Xtrain= Xtrain, Xtest=Xtest,Threshold = c(100,16000),Filtering=c(5,500),
				log10.scale=TRUE,row.stand=TRUE)

# how many genes after preprocess ?
dim(resP$pXtrain)[2]
</code></pre>


</div>