<div class="container">

<table style="width: 100%;"><tr>
<td>TFA.estimate</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Prediction of Transcription Factor Activities using PLS</h2>

<h3>Description</h3>

<p>The function <code>TFA.estimate</code> estimates the transcription factor activities from gene 
expression data and ChIP data using the PLS multivariate regression approach described 
in Boulesteix and Strimmer (2005).
</p>


<h3>Usage</h3>

<pre><code class="language-R">TFA.estimate(CONNECdata, GEdata, ncomp=NULL, nruncv=0, alpha=2/3, unit.weights=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>CONNECdata</code></td>
<td>
<p>a (n x p) matrix containing the ChIP data for the n genes and the
p predictors. The n genes must be the same as the n genes of <code>GEdata</code> and the ordering
of the genes must also be the same. Each row of <code>ChIPdata</code> corresponds to a gene, each
column to a transcription factor. <code>CONNECdata</code> might have either binary (e.g. 0-1) or 
numeric entries.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GEdata</code></td>
<td>
<p>a (n x m) matrix containing the gene expression levels of the n
considered genes for m samples. Each row of <code>GEdata</code> corresponds to a gene, each
column to a sample.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncomp</code></td>
<td>
<p>if <code>nruncv=0</code>, <code>ncomp</code> is the number of latent 
components to be  constructed. If <code>nruncv&gt;0</code>, the  number of 
latent components to be used for PLS
regression is chosen from 1,...,<code>ncomp</code> using the cross-validation procedure described
in Boulesteix and Strimmer (2005). If <code>ncomp=NULL</code>, <code>ncomp</code>
is set to min(n,p).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nruncv</code></td>
<td>
<p>the number of cross-validation iterations to be performed for the choice of
the number of latent components. If <code>nruncv=0</code>, cross-validation is not performed and
<code>ncomp</code> latent components are used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>the proportion of genes to be included in the training set for
the cross-validation procedure.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>unit.weights</code></td>
<td>
<p>If <code>TRUE</code> then the latent components
will be constructed from weight vectors that are standardized to length 1, 
otherwise the weight vectors do not have length 1 but the latent components have
norm 1.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The gene expression data as well as the ChIP data are assumed to have been
properly normalized. However, they do not have to be centered or scaled, since
centering and scaling are performed by the function <code>TFA.estimate</code> as a
preliminary step. 
</p>
<p>The matrix <code>ChIPdata</code> containing the ChIP data for the n genes and p transcription 
factors might be replaced by  any 'connectivity' matrix whose entries give the strength
of the interactions between the genes and transcription factors. For instance, a connectivity
matrix obtained by aggregating qualitative information from various genomic databases 
might be used as argument  in place of ChIP data.
</p>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>TFA</code></td>
<td>
<p>a (p x m) matrix containing the estimated transcription factor
activities for the p transcription factors and the m samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>metafactor</code></td>
<td>
<p>a (m x <code>ncomp</code>) matrix containing the metafactors for the m
samples. Each row corresponds to a sample, each column to a metafactor.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncomp</code></td>
<td>
<p>the number of latent components used in the PLS regression.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Anne-Laure Boulesteix 
(<a href="https://www.ibe.med.uni-muenchen.de/mitarbeiter/professoren/boulesteix/index.html">https://www.ibe.med.uni-muenchen.de/mitarbeiter/professoren/boulesteix/index.html</a>) 
and 
Korbinian Strimmer (<a href="https://strimmerlab.github.io/korbinian.html">https://strimmerlab.github.io/korbinian.html</a>).
</p>


<h3>References</h3>

<p>A. L. Boulesteix and K. Strimmer (2005). Predicting Transcription Factor Activities 
from Combined Analysis of Microarray and ChIP Data: A Partial Least Squares Approach.
</p>
<p>A. L. Boulesteix, K. Strimmer (2007). Partial least squares: a versatile tool for the analysis 
of high-dimensional genomic data. Briefings in Bioinformatics 7:32-44.
</p>
<p>S. de Jong (1993). SIMPLS: an alternative approach to partial least squares 
regression, Chemometrics Intell. Lab. Syst. <b>18</b>, 251â€“263.
</p>


<h3>See Also</h3>

<p><code>pls.regression</code>, <code>pls.regression.cv</code>.</p>


<h3>Examples</h3>

<pre><code class="language-R"># load plsgenomics library
library(plsgenomics)

# load Ecoli data
data(Ecoli)

# estimate TFAs based on 3 latent components
TFA.estimate(Ecoli$CONNECdata,Ecoli$GEdata,ncomp=3,nruncv=0)

# estimate TFAs and determine the best number of latent components simultaneously
TFA.estimate(Ecoli$CONNECdata,Ecoli$GEdata,ncomp=1:5,nruncv=20)

</code></pre>


</div>