<div class="container">

<table style="width: 100%;"><tr>
<td>total_count_filter</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Total Count Filter Object</h2>

<h3>Description</h3>

<p>This function returns a totalcountFilt object for use with
<code>applyFilt</code>
</p>


<h3>Usage</h3>

<pre><code class="language-R">total_count_filter(omicsData)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>omicsData</code></td>
<td>
<p>an object of the class 'seqData', created by
<code>as.seqData</code></p>
</td>
</tr></table>
<h3>Details</h3>

<p>Filter is based off of recommendations in edgeR processing, where
the low-observed biomolecules are removed from processing. Default
recommendation in edgeR is at least 15 total counts observed across samples
(i.e., if the sum of counts in a row of e_data is &lt; 15, default edgeR
filtering would remove this biomolecule).
</p>


<h3>Value</h3>

<p>An S3 object of class 'totalcountFilt' (data.frame) that contains the
molecule identifier and the total count of observed reads for that molecule
across all samples.
</p>


<h3>Author(s)</h3>

<p>Rachel Richardson
</p>


<h3>References</h3>

<p>Chen Y, Lun ATL, and Smyth, GK (2016). From reads to genes to pathways:
differential expression analysis of RNA-Seq experiments using Rsubread and
the edgeR quasi-likelihood pipeline. F1000Research 5, 1438.
http://f1000research.com/articles/5-1438
</p>


<h3>Examples</h3>

<pre><code class="language-R">

library(pmartRdata)
to_filter &lt;- total_count_filter(omicsData = rnaseq_object)
summary(to_filter, min_count = 15)


</code></pre>


</div>