<div class="container">

<table style="width: 100%;"><tr>
<td>dim_reduction</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Reduce Dimension of Data for Exploratory Data Analysis</h2>

<h3>Description</h3>

<p>For data types other than seqData, this function calculates principal
components using projection pursuit estimation, which implements an
expectation-maximization (EM) estimation algorithm when data is missing. For
seqData counts, a generalized version of principal components analysis for
non-normally distributed data is calculated under the assumption of a
negative binomial distribution with global dispersion.
</p>


<h3>Usage</h3>

<pre><code class="language-R">dim_reduction(omicsData, k = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>omicsData</code></td>
<td>
<p>an object of the class 'pepdata', 'prodata', 'metabData',
'lipidData', 'nmrData', or 'seqData', created by <code>as.pepData</code>,
<code>as.proData</code>, <code>as.metabData</code>,
<code>as.lipidData</code>, <code>as.nmrData</code>, or
<code>as.seqData</code>, respectively.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>integer number of principal components to return. Defaults to 2.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Any biomolecules seen in only one sample or with a variance less
than 1E-6 across all samples are not included in the PCA calculations. This
function leverages code from <code>pca</code> and
<code>glmpca</code> .
</p>


<h3>Value</h3>

<p>a data.frame with first <code>k</code> principal component scores, sample
identifiers, and group membership for each sample (if group designation was
previously run on the data). The object is of class dimRes (dimension
reduction Result).
</p>


<h3>References</h3>

<p>Redestig H, Stacklies W, Scholz M, Selbig J, &amp; Walther
D (2007). <em>pcaMethods - a bioconductor package providing PCA methods
for incomplete data</em>. Bioinformatics. 23(9): 1164-7.
</p>
<p>Townes FW, Hicks SC, Aryee MJ, Irizarry RA (2019). <em>Feature selection
and dimension reduction for single-cell RNA-seq based on a multinomial
model.</em> Genome Biol. 20, 1â€“16.
</p>
<p>Huang H, Wang Y, Rudin C, Browne EP (2022). <em>Towards a comprehensive
evaluation of dimension reduction methods for transcriptomic data
visualization.</em> Communications Biology 5, 719.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
library(pmartRdata)

mylipid &lt;- edata_transform(omicsData = lipid_neg_object, data_scale = "log2")
mylipid &lt;- group_designation(omicsData = mylipid, main_effects = "Virus")
pca_lipids &lt;- dim_reduction(omicsData = mylipid)


myseq &lt;- group_designation(omicsData = rnaseq_object, main_effects = "Virus")
pca_seq &lt;- dim_reduction(omicsData = myseq)


</code></pre>


</div>