<div class="container">

<table style="width: 100%;"><tr>
<td>dispersion_est</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Diagnostic plot for seqData</h2>

<h3>Description</h3>

<p>For generating statistics for 'seqData' objects
</p>


<h3>Usage</h3>

<pre><code class="language-R">dispersion_est(
  omicsData,
  method,
  interactive = FALSE,
  x_lab = NULL,
  x_lab_size = 11,
  x_lab_angle = NULL,
  y_lab = NULL,
  y_lab_size = 11,
  title_lab = NULL,
  title_lab_size = 14,
  legend_lab = NULL,
  legend_position = "right",
  bw_theme = TRUE,
  palette = NULL,
  point_size = 0.2,
  custom_theme = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>omicsData</code></td>
<td>
<p>seqData object used to terst dispersions</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>either "DESeq2", "edgeR", or "voom" for testing dispersion</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>interactive</code></td>
<td>
<p>Logical. If TRUE produces an interactive plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x_lab</code></td>
<td>
<p>A character string specifying the x-axis label when the metric
argument is NULL. The default is NULL in which case the x-axis label will
be "count".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x_lab_size</code></td>
<td>
<p>An integer value indicating the font size for the x-axis.
The default is 11.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x_lab_angle</code></td>
<td>
<p>An integer value indicating the angle of x-axis labels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y_lab</code></td>
<td>
<p>A character string specifying the y-axis label. The default is
NULL in which case the y-axis label will be the metric selected for the
<code>metric</code> argument.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y_lab_size</code></td>
<td>
<p>An integer value indicating the font size for the y-axis.
The default is 11.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title_lab</code></td>
<td>
<p>A character string specifying the plot title when the
<code>metric</code> argument is NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title_lab_size</code></td>
<td>
<p>An integer value indicating the font size of the plot
title. The default is 14.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend_lab</code></td>
<td>
<p>A character string specifying the legend title.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend_position</code></td>
<td>
<p>A character string specifying the position of the
legend. Can be one of "right", "left", "top", or "bottom". The default is
"right".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bw_theme</code></td>
<td>
<p>Logical. If TRUE uses the ggplot2 black and white theme.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>palette</code></td>
<td>
<p>A character string indicating the name of the RColorBrewer
palette to use. For a list of available options see the details section in
<code>RColorBrewer</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>point_size</code></td>
<td>
<p>An integer specifying the size of the points. The default
is 0.2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>custom_theme</code></td>
<td>
<p>a ggplot 'theme' object to be applied to non-interactive
plots, or those converted by plotly::ggplotly().</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>DESeq2 option requires package "survival" to be available.
</p>


<h3>Value</h3>

<p>plot result
</p>


<h3>References</h3>

<p>Robinson MD, McCarthy DJ, Smyth GK (2010). “edgeR: a Bioconductor package
for differential expression analysis of digital gene expression data.”
Bioinformatics, 26(1), 139-140. doi: 10.1093/bioinformatics/btp616.
</p>
<p>Love, M.I., Huber, W., Anders, S. Moderated estimation of fold change and
dispersion for RNA-seq data with DESeq2 Genome Biology 15(12):550 (2014)
</p>
<p>Ritchie, M.E., Phipson, B., Wu, D., Hu, Y., Law, C.W., Shi, W., and Smyth,
G.K. (2015). limma powers differential expression analyses for
RNA-sequencing and microarray studies. Nucleic Acids Research 43(7), e47.
</p>


<h3>Examples</h3>

<pre><code class="language-R">

library(pmartRdata)
myseqData &lt;- group_designation(omicsData = rnaseq_object, main_effects = "Virus")
dispersion_est(omicsData = myseqData, method = "edgeR")
dispersion_est(omicsData = myseqData, method = "DESeq2")
dispersion_est(omicsData = myseqData, method = "voom")


</code></pre>


</div>