<div class="container">

<table style="width: 100%;"><tr>
<td>trelli_rnaseq_nonzero_bar</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Bar chart trelliscope building function for Non-Zero counts in RNA-seq data</h2>

<h3>Description</h3>

<p>Specify a plot design and cognostics for the Non-Zero barchart
trelliscope. Non-Zeroes are displayed per panel_by variable. Main_effects
data is used to split samples when applicable. For MS/NMR data, use 
"trelli missingness bar".
</p>


<h3>Usage</h3>

<pre><code class="language-R">trelli_rnaseq_nonzero_bar(
  trelliData,
  cognostics = c("total count", "non-zero count", "non-zero proportion"),
  proportion = TRUE,
  ggplot_params = NULL,
  interactive = FALSE,
  path = .getDownloadsFolder(),
  name = "Trelliscope",
  test_mode = FALSE,
  test_example = 1,
  single_plot = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>trelliData</code></td>
<td>
<p>A trelliscope data object made by as.trelliData.edata or
as.trelliData. Must be built using seqData. Required.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cognostics</code></td>
<td>
<p>A vector of cognostic options for each plot. Defaults are "total count",
"non-zero count", and "non-zero proportion". If grouping
data is included, all cognostics will be reported per group. If the 
trelliData is paneled by a biomolecule, the counts and proportion we be 
samples. If paneled by a sample or biomolecule class, the counts and proportions
will be biomolecules.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>proportion</code></td>
<td>
<p>A logical to determine whether plots should display counts
or proportions. Default is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ggplot_params</code></td>
<td>
<p>An optional vector of strings of ggplot parameters to
the backend ggplot function. For example, c("ylab(”)", "xlab(”)").
Default is NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>interactive</code></td>
<td>
<p>A logical argument indicating whether the plots should be
interactive or not. Interactive plots are ggplots piped to ggplotly (for
now). Default is FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>path</code></td>
<td>
<p>The base directory of the trelliscope application. Default is
Downloads.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>name</code></td>
<td>
<p>The name of the display. Default is Trelliscope.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>test_mode</code></td>
<td>
<p>A logical to return a smaller trelliscope to confirm plot
and design. Default is FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>test_example</code></td>
<td>
<p>A vector of plot indices to return for test_mode. Default
is 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>single_plot</code></td>
<td>
<p>A TRUE/FALSE to indicate whether 1 plot (not a
trelliscope) should be returned. Default is FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional arguments to be passed on to the trelli builder</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>No return value, builds a trelliscope display of bar charts that is stored in 'path'
</p>


<h3>Author(s)</h3>

<p>David Degnan, Lisa Bramer
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## Not run: 
library(pmartRdata)

trelliData_seq1 &lt;- as.trelliData.edata(e_data = rnaseq_edata, 
                                      edata_cname = "Transcript",
                                      omics_type = "seqData")
omicsData_seq &lt;- group_designation(omicsData = rnaseq_object, main_effects = c("Virus"))

# Filter low transcript counts
omicsData_seq &lt;- applyFilt(filter_object = total_count_filter(omicsData = omicsData_seq), 
 omicsData = omicsData_seq, min_count = 15)

# Select a normalization and statistics method (options are 'edgeR', 'DESeq2', and 'voom').
# See ?difexp_seq for more details
statRes_seq &lt;- diffexp_seq(omicsData = omicsData_seq, method = "voom")

# Generate the trelliData object
trelliData_seq2 &lt;- as.trelliData(omicsData = omicsData_seq)
trelliData_seq3 &lt;- as.trelliData(statRes = statRes_seq)
trelliData_seq4 &lt;- as.trelliData(omicsData = omicsData_seq, statRes = statRes_seq)

# Build the non-zero bar plot with an edata file. Generate trelliData in as.trelliData.edata
trelli_panel_by(trelliData = trelliData_seq1, panel = "Transcript") %&gt;% 
  trelli_rnaseq_nonzero_bar(test_mode = TRUE, test_example = 1:10, path = tempdir())
trelli_panel_by(trelliData = trelliData_seq1, panel = "Sample") %&gt;% 
  trelli_rnaseq_nonzero_bar(test_mode = TRUE, test_example = 1:10, 
   cognostics = "non-zero proportion", path = tempdir())

# Build the non-zero bar plot with an omicsData object. Generate trelliData in as.trelliData
trelli_panel_by(trelliData = trelliData_seq2, panel = "Transcript") %&gt;% 
  trelli_rnaseq_nonzero_bar(test_mode = TRUE, test_example = 1:10, path = tempdir())

# Build the non-zero bar plot with a statRes object. Generate trelliData in as.trelliData
trelli_panel_by(trelliData = trelliData_seq3, panel = "Transcript") %&gt;%
  trelli_rnaseq_nonzero_bar(test_mode = TRUE, test_example = 1:10,
                         cognostics = c("non-zero proportion"), path = tempdir())

# Build the non-zero bar plot with an omicsData and statRes object. 
# Generate trelliData in as.trelliData.
trelli_panel_by(trelliData = trelliData_seq4, panel = "Gene") %&gt;%
  trelli_rnaseq_nonzero_bar(test_mode = TRUE, test_example = 1:10, path = tempdir()) 

# Or making the plot interactive 
trelli_panel_by(trelliData = trelliData_seq2, panel = "Transcript") %&gt;% 
   trelli_rnaseq_nonzero_bar(test_mode = TRUE, test_example = 1:5, 
     interactive = TRUE, path = tempdir())
   
# Or visualize only count data 
trelli_panel_by(trelliData = trelliData_seq2, panel = "Transcript") %&gt;% 
   trelli_rnaseq_nonzero_bar(test_mode = TRUE, test_example = 1:5, 
     cognostics = "non-zero count", proportion = FALSE, path = tempdir())

\dontshow{closeAllConnections()}   

## End(Not run)

</code></pre>


</div>