<div class="container">

<table style="width: 100%;"><tr>
<td>diffexp_seq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Differential Expression for seqData</h2>

<h3>Description</h3>

<p>Performs statistical analysis for differential expression of seqData objects,
using methods from one of: edgeR, DESeq2, or limma-voom
</p>


<h3>Usage</h3>

<pre><code class="language-R">diffexp_seq(
  omicsData,
  method = "edgeR",
  p_adjust = "BH",
  comparisons = NULL,
  p_cutoff = 0.05,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>omicsData</code></td>
<td>
<p>object of type 'seqData' created by
<code>as.seqData</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>character string of length one specifying which wrapper to
use. Can be 'edgeR', 'DESeq2', or 'voom'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p_adjust</code></td>
<td>
<p>character string for p-value correction method, refer to
?p.adjust() for valid options. Defaults to "BH" (Benjamini &amp; Hochberg).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>comparisons</code></td>
<td>
<p>data frame with columns for "Control" and "Test"
containing the different comparisons of interest. Comparisons will be made
between the Test and the corresponding Control  If left NULL, then all
pairwise comparisons are executed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p_cutoff</code></td>
<td>
<p>numeric value between 0 and 1 for setting p-value
significance threshold</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional arguments passed to methods functions. Note, formatting
option changes will interfere with wrapping functionality.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Runs default differential expression workflows.
</p>
<p>Flags (signatures) -	Indicator of statistical significance.
Zeroes indicate no significance, while +/- 1 indicates direction of
significance.
</p>
<p>Method "edgeR" - Runs default edgeR workflow with empirical Bayes
quasi-likelihood F-tests. Additional arguments can be passed for use in the
function. Refer to calcNormFactors() and glmQLFit() in edgeR package.
Requires the 'edgeR' and 'limma' packages to run.
</p>
<p>Method "DESeq2" - Runs default DESeq workflow. Defaults to Wald test, no
independent filtering, and running in parallel. Additional arguments can be
passed for use in the function. Refer to DESeq() and results() in DESeq2
package. Requires 'survival' package to run.
</p>
<p>Method "voom" - Runs default limma-voom workflow using empirical Bayes
moderated t-statistics. Additional arguments can be passed for use in the
function. Refer to calcNormFactors() in edgeR package.
Requires the 'edgeR' and 'limma' packages to run.
</p>


<h3>Value</h3>

<p>object of class statRes
</p>


<h3>References</h3>

<p>Robinson MD, McCarthy DJ, Smyth GK (2010). “edgeR: a Bioconductor package
for differential expression analysis of digital gene expression data.”
Bioinformatics, 26(1), 139-140. doi: 10.1093/bioinformatics/btp616.
</p>
<p>Love, M.I., Huber, W., Anders, S. Moderated estimation of fold change and
dispersion for RNA-seq data with DESeq2 Genome Biology 15(12):550 (2014)
</p>
<p>Ritchie, M.E., Phipson, B., Wu, D., Hu, Y., Law, C.W., Shi, W., and Smyth,
G.K. (2015). limma powers differential expression analyses for
RNA-sequencing and microarray studies. Nucleic Acids Research 43(7), e47.
</p>


<h3>Examples</h3>

<pre><code class="language-R">

library(pmartRdata)
myseqData &lt;- group_designation(omicsData = rnaseq_object, main_effects = "Virus")
edger_results &lt;- diffexp_seq(omicsData = myseqData, method = "edgeR")
deseq_results &lt;- diffexp_seq(omicsData = myseqData, method = "DESeq2")
voom_results &lt;- diffexp_seq(omicsData = myseqData, method = "voom")


</code></pre>


</div>