<div class="container">

<table style="width: 100%;"><tr>
<td>genbinary-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Class "genbinary"</h2>

<h3>Description</h3>

<p>This is a subclass of <code>gendata</code> that allows genotypes to be stored
as a matrix indicating the presence and absence of alleles.
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("genbinary",
  samples, loci, ...)</code>.  After objects are initialized with sample and
locus names, data can be added to slots using the replacement functions.
</p>


<h3>Slots</h3>


<dl>
<dt>
<code>Genotypes</code>:</dt>
<dd>
<p>Object of class <code>"matrix"</code>.  Row names
of the matrix are sample names.  Each column name is a locus name
and an allele separated by a period (<em>e.g.</em>
<code>"loc1.124"</code>); each column represents on allele.  The number of
alleles per locus is not limited and can be expanded even after
entering initial data.  Each element of the matrix must be equal to
either <code>Present(object)</code>, <code>Absent(object)</code>, or
<code>Missing(object)</code>.  These symbols indicate, respectively, that
a sample has an allele, that a sample does not have an allele, or
that data for the sample at that locus are missing.</p>
</dd>
<dt>
<code>Present</code>:</dt>
<dd>
<p>Object of class <code>"ANY"</code>.  The integer
<code>1</code> by default.  This symbol is used in the <code>Genotypes</code>
slot to indicate the presence of an allele in a sample. </p>
</dd>
<dt>
<code>Absent</code>:</dt>
<dd>
<p>Object of class <code>"ANY"</code>.  The integer
<code>0</code> by default.  This symbol is used in the <code>Genotypes</code>
slot to indicate the absence of an allele in a sample. </p>
</dd>
<dt>
<code>Description</code>:</dt>
<dd>
<p>Object of class <code>"character"</code>.  A
character string or vector describing the dataset, for the
convenience of the user. </p>
</dd>
<dt>
<code>Missing</code>:</dt>
<dd>
<p>Object of class <code>"ANY"</code>.  The integer
<code>-9</code> by default.  This symbol is used in the <code>Genotypes</code>
slot to indicate that data are missing for a given sample and locus.</p>
</dd>
<dt>
<code>Usatnts</code>:</dt>
<dd>
<p>Object of class <code>"integer"</code>.  A vector,
named by loci.  This indicates the repeat length of each locus.
<code>2</code> indicates dinucleotide repeats, <code>3</code>
indicates trinucleotide repeats, and so on.  If the alleles stored
in the column names of the <code>Genotypes</code> slot for a given locus
are already written
in terms of repeat number, the <code>Usatnts</code> value for that locus
should be <code>1</code>.  In other words, all alleles for a locus can be
divided by the number in <code>Usatnts</code> to give alleles expressed in
terms of relative repeat number. </p>
</dd>
<dt>
<code>Ploidies</code>:</dt>
<dd>
<p>Object of class <code>"integer"</code>.  A vector,
named by samples.  This indicates the ploidy of each sample.</p>
</dd>
<dt>
<code>PopInfo</code>:</dt>
<dd>
<p>Object of class <code>"integer"</code>.  A vector,
named by samples.  This indicates the population identity of each
sample. </p>
</dd>
<dt>
<code>PopNames</code>:</dt>
<dd>
<p>Object of class <code>"character"</code>.  Names
of each population.  The position of the population name in the
vector corresponds to the number used to represent that population
in the <code>PopInfo</code> slot.</p>
</dd>
</dl>
<h3>Extends</h3>

<p>Class <code>"gendata"</code>, directly.
</p>


<h3>Methods</h3>


<dl>
<dt>Absent</dt>
<dd>
<p><code>signature(object = "genbinary")</code>: Returns the
symbol used to indicate that a given allele is absent in a given
sample. </p>
</dd>
<dt>Absent&lt;-</dt>
<dd>
<p><code>signature(object = "genbinary")</code>: Changes the
symbol used to indicate that a given allele is absent in a given
sample.  The matrix in the <code>Genotypes</code> slot is searched for
the old symbol, which is replaced by the new.  The new symbol is
then written to the <code>Absent</code> slot. </p>
</dd>
<dt>Genotype</dt>
<dd>
<p><code>signature(object = "genbinary")</code>: Returns a
matrix containing the genotype for a given sample and locus (by a
call to <code>Genotypes</code>). </p>
</dd>
<dt>Genotypes</dt>
<dd>
<p><code>signature(object = "genbinary")</code>: Returns the
matrix stored in the <code>Genotypes</code> slot, or a subset as specified
by the <code>samples</code> and <code>loci</code> arguments.</p>
</dd>
<dt>Genotypes&lt;-</dt>
<dd>
<p><code>signature(object = "genbinary")</code>: A method
for adding or replacing genotype data in the object.  Note that
allele columns cannot be removed from the matrix in the
<code>Genotypes</code> slot using this method, although an entire column
could be filled with zeros in order to effectively remove an
allele from the dataset.  If the order of rows in <code>value</code>
(the matrix containing values to be assigned to the
<code>Genotypes</code> slot) is not identical to <code>Samples(object)</code>,
the <code>samples</code> argument should be used to indicate row order.
Row names in <code>value</code> are ignored.  The <code>loci</code> argument can
be left at the default, even if only a subset of loci are being
assigned.  Column names of <code>value</code> are important, and should be
the locus name and allele name separated by a period, as they are in
the <code>Genotypes</code> slot.  After checking that the column name is
valid, the method checks for whether the column name already exists or
not in the <code>Genotypes</code> slot.  If it does exist, data from that
column are replaced with data from <code>value</code>.  If not, a column is
added to the matrix in the <code>Genotypes</code> slot for the new allele.  If
the column is new and data are not being written for samples, the method
automatically fills in <code>Missing</code> or <code>Absent</code> symbols for
additional samples, depending on whether or not data for the locus
appear to be missing for the sample or not.</p>
</dd>
<dt>initialize</dt>
<dd>
<p><code>signature(.Object = "genbinary")</code>: Sets up a
<code>genbinary</code> object when <code>new("genbinary")</code> is called.  If
<code>samples</code> or <code>loci</code> arguments are missing, these are
filled in with dummy values (<code>"ind1", "ind2", "loc1",
      "loc2"</code>).  The matrix is then set up in the <code>Genotypes</code>
slot.  Sample names are used for row names, and there are zero
columns.  The <code>initialize</code> method for <code>gendata</code> is then
called.</p>
</dd>
<dt>Missing&lt;-</dt>
<dd>
<p><code>signature(object = "genbinary")</code>: Replaces
all elements in matrix in the <code>Genotypes</code> slot containing the
old <code>Missing</code> symbol with the new <code>Missing</code> symbol.  The
method for <code>gendata</code> is then called to replace the value in the
<code>Missing</code> slot.</p>
</dd>
<dt>Present</dt>
<dd>
<p><code>signature(object = "genbinary")</code>: Returns the
symbol used to indicate that a given allele is present in a given
sample. </p>
</dd>
<dt>Present&lt;-</dt>
<dd>
<p><code>signature(object = "genbinary")</code>: Changes the
symbol used for indicating that a given allele is present in a given
sample.  The symbol is first replaced in the <code>Genotypes</code> slot,
and then in the <code>Present</code> slot.</p>
</dd>
<dt>Samples&lt;-</dt>
<dd>
<p><code>signature(object = "genbinary")</code>: Changes sample
names in the dataset.  Changes the row names in the <code>Genotypes</code>
slot, then calls the method for <code>gendata</code> to change the names in
the <code>PopInfo</code> and <code>Ploidies</code> slots.</p>
</dd>
<dt>Loci&lt;-</dt>
<dd>
<p><code>signature(object = "genbinary")</code>: Changes locus
names in the dataset.  Replaces the locus portion of the column names
in the <code>Genotypes</code> slot, then calls the method for <code>gendata</code>
to change the names in the <code>Usatnts</code> slot.</p>
</dd>
<dt>isMissing</dt>
<dd>
<p><code>signature(object = "genbinary")</code>: Returns Boolean
values, by sample and locus, indicating whether genotypes are missing.
If there are any missing data symbols within the genotype, it is
considered missing.</p>
</dd>
<dt>summary</dt>
<dd>
<p><code>signature(object = "genbinary")</code>: Prints
description of dataset and number of missing genotypes, then calls the
method for <code>gendata</code> to print additional information.</p>
</dd>
<dt>editGenotypes</dt>
<dd>
<p><code>signature(object = "genbinary")</code>: Opens the
genotype matrix in the Data Editor for editing.  Useful for making
minor changes, although allele columns cannot be added using this method.</p>
</dd>
<dt>viewGenotypes</dt>
<dd>
<p><code>signature(object = "genbinary")</code>: Prints the
genotype matrix to the console, one locus at a time.</p>
</dd>
<dt>deleteSamples</dt>
<dd>
<p><code>signature(object = "genbinary")</code>: Removes the
specified samples from the genotypes matrix, then calls the method for
<code>gendata</code>.</p>
</dd>
<dt>deleteLoci</dt>
<dd>
<p><code>signature(object = "genbinary")</code>: Removes the
specified loci from the genotypes matrix, then calls the method for
<code>gendata</code>.</p>
</dd>
<dt>"["</dt>
<dd>
<p><code>signature(x = "genbinary", i = "ANY", j = "ANY")</code>:
Subscripting method.  Returns a <code>genbinary</code> object with a subset
of the samples and/or loci from <code>x</code>.  Usage:
<code>genobject[samples,loci]</code>.</p>
</dd>
<dt>estimatePloidy</dt>
<dd>
<p><code>signature(object = "genbinary")</code>: Creates a
data frame of mean and maximum number of alleles per sample, which is
then opened in the Data Editor so that the user can manually specify
the ploidy of each sample.  Ploidies are then written to the
<code>Ploidies</code> slot of the object.</p>
</dd>
<dt>merge</dt>
<dd>
<p><code>signature(x = "genbinary", y = "genbinary")</code>: Merges
two genotype objects together.  See
<code>merge,genbinary,genbinary-method</code>.</p>
</dd>
</dl>
<h3>Author(s)</h3>

<p>Lindsay V. Clark
</p>


<h3>See Also</h3>

<p><code>gendata</code>, <code>Accessors</code>,
<code>genambig</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># show the class definition
showClass("genbinary")

# create a genbinary object
mygen &lt;- new("genbinary", samples = c("indA", "indB", "indC", "indD"),
             loci = c("loc1", "loc2"))
Description(mygen) &lt;- "Example genbinary object for the documentation."
Usatnts(mygen) &lt;- c(2,3)
PopNames(mygen) &lt;- c("Maine", "Indiana")
PopInfo(mygen) &lt;- c(1,1,2,2)
Genotypes(mygen) &lt;- matrix(c(1,1,0,0, 1,0,0,1, 0,0,1,1,
                             1,-9,1,0, 0,-9,0,1, 1,-9,0,1, 0,-9,1,1),
   nrow=4, ncol=7, dimnames = list(NULL,
   c("loc1.140", "loc1.144", "loc1.150",
     "loc2.97", "loc2.100", "loc2.106", "loc2.109")))

# view all of the data in the object
mygen
</code></pre>


</div>