<div class="container">

<table style="width: 100%;"><tr>
<td>Bruvo.distance</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Genetic Distance Metric of Bruvo et al.</h2>

<h3>Description</h3>

<p>This function calculates the distance between two individuals at one
microsatellite locus using a method based on that of Bruvo
<em>et al.</em> (2004).
</p>


<h3>Usage</h3>

<pre><code class="language-R">Bruvo.distance(genotype1, genotype2, maxl=10, usatnt=2, missing=-9)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genotype1</code></td>
<td>
<p>A vector of alleles for one individual at one
locus.  Allele length is in nucleotides or repeat count.  Each
unique allele corresponds to one element in the vector, and the
vector is no longer than it needs to be to contain all unique
alleles for this individual at this locus.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotype2</code></td>
<td>
<p>A vector of alleles for another individual at the
same locus.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxl</code></td>
<td>
<p>If both individuals have more than this number of
alleles at this locus, <code>NA</code> is returned instead of a
numerical distance.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>usatnt</code></td>
<td>
<p>Length of the repeat at this locus.  For example
<code>usatnt=2</code> for dinucleotide repeats, and <code>usatnt=3</code>
for trinucleotide repeats.  If the alleles in <code>genotype1</code>
and <code>genotype2</code> are expressed in repeat count instead of
nucleotides, set <code>usatnt=1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>missing</code></td>
<td>
<p>A numerical value that, when in the first allele
position, indicates missing data. <code>NA</code> is returned if this
value is found in either genotype.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Since allele copy number is frequently unknown in polyploid
microsatellite data, Bruvo <em>et al.</em> developed a measure of genetic
distance similar to band-sharing indices used with dominant data,
but taking into account mutational distances between alleles.  A
matrix is created containing all differences in repeat count between
the alleles of two individuals at one locus.  These differences are
then geometrically transformed to reflect the probabilities of
mutation from one allele to another.  The matrix is then searched to
find the minimum sum if each allele from one individual is paired to
one allele from the other individual.  This sum is divided by the
number of alleles per individual.
</p>
<p>If one genotype has more alleles than the other, ‘virtual alleles’ must
be created so that both genotypes are the same length.  There are
three options for the value of these virtual alleles, but
<code>Bruvo.distance</code> only implements the simplest one, assuming that it is
not known whether differences in ploidy arose from genome addition or
genome loss.  Virtual alleles are set to infinity, such that the
geometric distance between any allele and a virtual allele is 1.
</p>
<p>In the original publication by Bruvo <em>et al.</em> (2004), ambiguous
genotypes were dealt with by calculating the distance for all possible
unambiguous genotype combinations and averaging across all of them
equally.  When <code>Bruvo.distance</code> is called from
<code>meandistance.matrix</code>, ploidy is unknown and all genotypes are
simply treated as if they had one copy of each allele.  When
<code>Bruvo.distance</code> is called from <code>meandistance.matrix2</code>, the
analysis is truer to the original, in that ploidy is known and all
possible unambiguous genotype combinations are considered.  However,
instead of all possible unambiguous genotypes being weighted equally,
in <code>meandistance.matrix2</code> they are weighted based on allele
frequencies and selfing rate, since some unambiguous genotypes are
more likely than others.
</p>


<h3>Value</h3>

<p>A number ranging from 0 to 1, with 0 indicating identical
genotypes, and 1 being a theoretical maximum distance if all alleles
from <code>genotype1</code> differed by an infinite number of repeats from all
alleles in <code>genotype2</code>.  <code>NA</code> is returned if both genotypes have
more than <code>maxl</code> alleles or if either genotype has the symbol for
missing data as its first allele.
</p>


<h3>Note</h3>

<p>The processing time is a function of the factorial of the number of
alleles, since each possible combination of allele pairs must be
evaluated.  For genotypes with a sufficiently large number of alleles,
it may be more efficient to estimate distances manually by creating
the matrix in Excel and visually picking out the shortest distances
between alleles.  This is the purpose of the <code>maxl</code> argument.  On my
personal computer, if both genotypes had more than nine alleles, the
calculation could take an hour or more, and so this is the default
limit.  In this case, <code>Bruvo.distance</code> returns <code>NA</code>.
</p>


<h3>Author(s)</h3>

<p>Lindsay V. Clark</p>


<h3>References</h3>

<p>Bruvo, R., Michiels, N. K., D'Sousa, T. G., and Schulenberg, H. (2004)
A simple method for calculation of microsatellite genotypes
irrespective of ploidy level. <em>Molecular Ecology</em> <b>13</b>, 2101-2106.
</p>


<h3>See Also</h3>

<p><code>meandistance.matrix</code>, <code>Lynch.distance</code>,
<code>Bruvo2.distance</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">  Bruvo.distance(c(202,206,210,220),c(204,206,216,222))
  Bruvo.distance(c(202,206,210,220),c(204,206,216,222),usatnt=4)
  Bruvo.distance(c(202,206,210,220),c(204,206,222))
  Bruvo.distance(c(202,206,210,220),c(204,206,216,222),maxl=3)
  Bruvo.distance(c(202,206,210,220),c(-9))
</code></pre>


</div>