<div class="container">

<table style="width: 100%;"><tr>
<td>MergeIdenticalHaplotypes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Merge Alleles with Identical DNA Sequences
</h2>

<h3>Description</h3>

<p>If any alleles within a locus have identical <code>alleleNucleotides</code> values
(including those identical based on IUPAC ambiguity codes),
this function merges those alleles, summing their read depths.  This function is
primarily intended to be used internally in cases where tags vary in length
within a locus, resulting in truncated <code>alleleNucleotides</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">MergeIdenticalHaplotypes(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>

<p>A <code>RADdata</code> object.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>Additional arguments (none implemented).
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A <code>RADdata</code> object identical to <code>object</code>, but with alleles merged.
</p>


<h3>Author(s)</h3>

<p>Lindsay V. Clark
</p>


<h3>See Also</h3>

<p><code>MergeRareHaplotypes</code>, <code>readProcessIsoloci</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(exampleRAD)
# change a haplotype for this example
exampleRAD$alleleNucleotides[5] &lt;- "GY"

nAlleles(exampleRAD)
exampleRAD &lt;- MergeIdenticalHaplotypes(exampleRAD)
nAlleles(exampleRAD)
</code></pre>


</div>