<div class="container">

<table style="width: 100%;"><tr>
<td>ExportGAPIT</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Export RADdata Object for Use by Other R Packages
</h2>

<h3>Description</h3>

<p>After a <code>"RADdata"</code> object has been run through a pipeline such as
<code>IteratePopStruct</code>, these functions can be used to export
the genotypes to R packages and other software that can
perform genome-wide association and genomic prediction.  <code>ExportGAPIT</code>,
<code>Export_rrBLUP_Amat</code>, <code>Export_rrBLUP_GWAS</code>, <code>Export_GWASpoly</code>,
and <code>Export_TASSEL_Numeric</code> all export continuous numerical genotypes 
generated by <code>GetWeightedMeanGenotypes</code>.  <code>Export_polymapR</code>,
<code>Export_Structure</code>, and <code>Export_adegenet_genind</code> use 
<code>GetProbableGenotypes</code> to export discrete 
genotypes.  <code>Export_MAPpoly</code> and <code>Export_polymapR_probs</code> export
genotype posterior probabilities.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ExportGAPIT(object, onePloidyPerAllele = FALSE)

Export_rrBLUP_Amat(object, naIfZeroReads = FALSE, 
                   onePloidyPerAllele = FALSE)

Export_rrBLUP_GWAS(object, naIfZeroReads = FALSE, 
                   onePloidyPerAllele = FALSE)

Export_TASSEL_Numeric(object, file, naIfZeroReads = FALSE,
                      onePloidyPerAllele = FALSE)

Export_polymapR(object, naIfZeroReads = TRUE,
                progeny = GetTaxa(object)[!GetTaxa(object) %in% 
                  c(GetDonorParent(object), GetRecurrentParent(object),
                    GetBlankTaxa(object))])

Export_polymapR_probs(object, maxPcutoff = 0.9,
                      correctParentalGenos = TRUE,
                      multiallelic = "correct")

Export_MAPpoly(object, file, pheno = NULL, ploidyIndex = 1,
               progeny = GetTaxa(object)[!GetTaxa(object) %in% 
                 c(GetDonorParent(object), GetRecurrentParent(object),
                   GetBlankTaxa(object))],
               digits = 3)

Export_GWASpoly(object, file, naIfZeroReads = TRUE, postmean = TRUE, digits = 3,
                splitByPloidy = TRUE)

Export_Structure(object, file, includeDistances = FALSE, extraCols = NULL,
                 missingIfZeroReads = TRUE)

Export_adegenet_genind(object, ploidyIndex = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>

<p>A <code>"RADdata"</code> object with posterior genotype probabilities
already estimated.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>onePloidyPerAllele</code></td>
<td>

<p>Logical.  If <code>TRUE</code>, for each allele the inheritance mode with the
lowest <code class="reqn">\chi ^ 2</code> value is selected and is assumed to be
the true inheritance mode.  If <code>FALSE</code>, inheritance modes are weighted
by inverse <code class="reqn">\chi ^ 2</code> values for each allele, and mean
genotypes that have been weighted across inheritance modes are returned.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>naIfZeroReads</code></td>
<td>

<p>A logical indicating whether <code>NA</code> should be inserted into the 
output matrix for any taxa and loci where the total read depth for
the locus is zero.  If <code>FALSE</code>, the output for these genotypes is
essentially the mode (for <code>Export_polymapR</code> and <code>Export_GWASpoly</code>)
or mean (for others) across prior genotype probabilities, since
prior and posterior genotype probabilities are equal when there are no
reads.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>

<p>A character string indicating a file path to which to write.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno</code></td>
<td>

<p>A data frame or matrix of phenotypic values, with progeny in rows and traits
in columns.  Columns should be named.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ploidyIndex</code></td>
<td>

<p>Index, within <code>object$possiblePloidies</code>, of the ploidy to be exported.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>progeny</code></td>
<td>

<p>A character vector indicating which individuals to export as progeny of the
cross.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxPcutoff</code></td>
<td>

<p>A cutoff for posterior probabilities, below which genotypes will be reported
as 'NA' in the 'geno' column.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>correctParentalGenos</code></td>
<td>

<p>Passed to <code>GetProbableGenotypes</code>. If <code>TRUE</code>, parental genotypes
are corrected based on progeny allele frequencies.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>multiallelic</code></td>
<td>

<p>Passed to <code>GetProbableGenotypes</code>. Under the default, genotypes at
multiallelic loci will be corrected to sum to the ploidy.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>digits</code></td>
<td>

<p>Number of decimal places to which to round genotype probabilities or posterior
mean genotypes in the output file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>postmean</code></td>
<td>

<p>Logical. If <code>TRUE</code>, posterior mean genotypes will be output.  If
<code>FALSE</code>, discrete genotypes will be output.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>splitByPloidy</code></td>
<td>

<p>Logical. If <code>TRUE</code> and there are multiple <code>taxaPloidy</code> values in the
dataset, multiple files are written, one per ploidy.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>includeDistances</code></td>
<td>

<p>Logical. If <code>TRUE</code>, the second row of the Structure file will contain
distances between markers, which can be used by the linkage model in Structure.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>extraCols</code></td>
<td>

<p>An optional data frame, with one row per taxon, containing columns of data
to output to the left of the genotypes in the Structure file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>missingIfZeroReads</code></td>
<td>

<p>See <code>naIfZeroReads</code>.  If <code>TRUE</code>, a value of <code>-9</code> will be output
for any genotypes with zero reads, indicating that those genotypes are missing.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><span class="pkg">GAPIT</span>, <span class="pkg">FarmCPU</span>, <span class="pkg">rrBLUP</span>, <span class="pkg">TASSEL</span>, and <span class="pkg">GWASpoly</span> allow
genotypes to be a continuous numeric variable.  <span class="pkg">MAPpoly</span> and <span class="pkg">polymapR</span>
allow for import of genotype probabilities.
<span class="pkg">GAPIT</span> does not allow missing data, hence there is no <code>naIfZeroReads</code>
argument for <code>ExportGAPIT</code>.  Genotypes are exported on a scale of -1
to 1 for <span class="pkg">rrBLUP</span>, on a scale of 0 to 2 for <span class="pkg">GAPIT</span> and <span class="pkg">FarmCPU</span>,
and on a scale of 0 to 1 for <span class="pkg">TASSEL</span>.
</p>
<p>For all functions except <code>Export_Structure</code> and <code>Export_adegenet_genind</code>,
one allele per marker is dropped.  <code>Export_MAPpoly</code>
also drops alleles where one or both parental genotypes could not be determined,
and where both parents are homozygotes.
</p>
<p>For <code>ExportGAPIT</code>
and <code>Export_rrBLUP_GWAS</code>, chromosome and position are filled with dummy
data if they do not exist in <code>object$locTable</code>.  For <code>Export_TASSEL_Numeric</code>,
allele names are exported, but no chromosome or position information per se.
</p>
<p>If the chromosomes in <code>object$locTable</code> are in character format,
<code>ExportGAPIT</code>, <code>Export_MAPpoly</code>, and <code>Export_GWASpoly</code> will
attempt to extract chromosome numbers.
</p>
<p>For <span class="pkg">polymapR</span> there must only be one possible inheritance mode across loci
(one value in <code>object$possiblePloidies</code>) in the <code>RADdata</code> object, although
triploid F1 populations derived from diploid and tetraploid parents are allowed.
See <code>SubsetByPloidy</code> for help reducing a <code>RADdata</code> object to a
single inheritance mode.
</p>
<p><span class="pkg">MAPpoly</span> only
allows one ploidy, but <code>Export_MAPpoly</code> allows the user to select which
inheritance mode from the <code>RADdata</code> object to use.  (This is due to how internal
<span class="pkg">polyRAD</span> functions are coded.)
</p>


<h3>Value</h3>

<p>For <code>ExportGAPIT</code>, a list:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>GD </code></td>
<td>
<p>A data frame with taxa in the first column and alleles (markers)
in subsequent columns, containing the genotypes.  To be passed to the <code>GD</code>
argument for <code>GAPIT</code> or <code>FarmCPU</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GM </code></td>
<td>
<p>A data frame with the name, chromosome number, and position of
every allele (marker).  To be passed to the <code>GM</code> argument for <code>GAPIT</code>
or <code>FarmCPU</code>.</p>
</td>
</tr>
</table>
<p>For <code>Export_rrBLUP_Amat</code>, a matrix with taxa in rows and alleles (markers)
in columns, containing genotype data.  This can be passed to <code>A.mat</code> in
<span class="pkg">rrBLUP</span>.
</p>
<p>For <code>Export_rrBLUP_GWAS</code>, a data frame with alleles (markers) in rows.
The first three columns contain the marker names, chromosomes, and positions,
and the remaining columns each represent one taxon and contain the genotype
data.  This can be passed to the <code>GWAS</code> function in <span class="pkg">rrBLUP</span>.
</p>
<p><code>Export_TASSEL_Numeric</code> and <code>Export_MAPpoly</code> write a file but does
not return an object.
</p>
<p>For <code>Export_polymapR</code>, a matrix of integers indicating the most probable 
allele copy number, with markers in rows and individuals in columns.  The 
parents are listed first, followed by all progeny.
</p>
<p>For <code>Export_polymapR_probs</code>, a data frame suitable to pass to the
<code>probgeno_df</code> argument of checkF1.  Note that under
default parameters, in some cases the <code>maxP</code>, <code>maxgeno</code>, and
<code>geno</code> columns may not actually reflect the maximum posterior probability
if genotype correction was performed.
</p>
<p>For <code>Export_adegenet_genind</code>, a <code>"genind"</code> object.
</p>
<p><code>Export_MAPpoly</code>, <code>Export_GWASpoly</code>, and <code>Export_Structure</code>
write files but do not return
an object.  Files output by <code>Export_GWASpoly</code> are comma delimited and
in numeric format.  Sample and locus names are included in the file output
by <code>Export_Structure</code>, and the number of rows for each sample is
equal to the highest ploidy as determined by the <code>taxaPloidy</code> slot and the
output of <code>GetProbableGenotypes</code>.
</p>


<h3>Note</h3>

<p><span class="pkg">rrBLUP</span> and <span class="pkg">polymapR</span> are available through CRAN, and <span class="pkg">GAPIT</span>
and <span class="pkg">FarmCPU</span> must be downloaded from the Zhang lab website.  <span class="pkg">MAPpoly</span>
is available on GitHub but not yet on CRAN.  <span class="pkg">GWASpoly</span> is available from
GitHub.
</p>
<p>In my experience with <span class="pkg">TASSEL 5</span>, numerical genotype files that are too large do 
not load/display properly.  If you run into this problem I recommend using
<code>SplitByChromosome</code> to split your <code>RADdata</code> object into 
multiple smaller objects, which can then be exported to separate files using
<code>Export_TASSEL_Numeric</code>.  If performing GWAS, you may also need to compute 
a kinship matrix using separate software such as <span class="pkg">rrBLUP</span>.
</p>


<h3>Author(s)</h3>

<p>Lindsay V. Clark
</p>


<h3>References</h3>

<p><b>GAPIT and FarmCPU:</b>
</p>
<p><a href="https://zzlab.net/GAPIT/">https://zzlab.net/GAPIT/</a>
</p>
<p>Lipka, A. E., Tian, F., Wang, Q., Peiffer, J., Li, M., Bradbury, P. J., 
Gore, M. A., Buckler, E. S. and Zhang, Z. (2012) GAPIT: genome association 
and prediction integrated tool.  <em>Bioinformatics</em> <b>28</b>, 2397–2399.
</p>
<p><a href="https://zzlab.net/FarmCPU/">https://zzlab.net/FarmCPU/</a>
</p>
<p>Liu, X., Huang, M., Fan, B., Buckler, E. S., Zhang, Z. (2016) Iterative usage
of fixed and random effects models for powerful and efficient genome-wide
association studies.  <em>PLoS Genetics</em> <b>12</b>, e1005767.
</p>
<p><b>rrBLUP:</b>
</p>
<p>Endelman, J.B. (2011) Ridge Regression and Other Kernels for Genomic 
Selection with R Package rrBLUP.  <em>The Plant Genome</em> <b>4</b>, 250–255.
</p>
<p><b>TASSEL:</b>
</p>
<p><a href="https://www.maizegenetics.net/tassel">https://www.maizegenetics.net/tassel</a>
</p>
<p>Bradbury, P. J., Zhang, Z., Kroon, D. E., Casstevens, T. M., Ramdoss, Y. 
and Buckler, E. S. (2007) TASSEL: Software for association mapping of complex 
traits in diverse samples. <em>Bioinformatics</em> <b>23</b>, 2633–2635.
</p>
<p><b>polymapR:</b>
</p>
<p>Bourke, P., van Geest, G., Voorrips, R. E., Jansen, J., Kranenberg, T.,
Shahin, A., Visser, R. G. F., Arens, P., Smulders, M. J. M. and Maliepaard, C.
(2018) polymapR: linkage analysis and genetic map construction from F1 
populations of outcrossing polyploids. <em>Bioinformatics</em> <b>34</b>, 
3496–3502.
</p>
<p><b>MAPpoly:</b>
</p>
<p><a href="https://github.com/mmollina/MAPpoly">https://github.com/mmollina/MAPpoly</a>
</p>
<p>Mollinari, M. and Garcia, A. A. F. (2018) Linkage analysis and haplotype
phasing in experimental autopolyploid populations with high ploidy level using
hidden Markov models.  <em>bioRxiv</em> doi: https://doi.org/10.1101/415232.
</p>
<p><b>GWASpoly:</b>
</p>
<p><a href="https://github.com/jendelman/GWASpoly">https://github.com/jendelman/GWASpoly</a>
</p>
<p>Rosyara, U. R., De Jong, W. S., Douches, D. S., and Endelman, J. B.  (2016)
Software for Genome-Wide Association Studies in Autopolyploids and Its 
Application to Potato.  <em>Plant Genome</em> <b>9</b>.
</p>
<p><b>Structure:</b>
</p>
<p><a href="https://web.stanford.edu/group/pritchardlab/structure.html">https://web.stanford.edu/group/pritchardlab/structure.html</a>
</p>
<p>Hubisz, M. J., Falush, D., Stephens, M. and Pritchard, J. K. (2009)
Inferring weak population structure with the assistance of sample
group information.  <em>Molecular Ecology Resources</em> <b>9</b>,
1322–1332.
</p>
<p>Falush, D., Stephens, M. and Pritchard, J. K.  (2007)
Inferences of population structure using multilocus genotype data:
dominant markers and null alleles.  <em>Molecular Ecology Notes</em>
<b>7</b>, 574–578
</p>
<p>Falush, D., Stephens, M. and Pritchard, J. K.  (2003)
Inferences of population structure using multilocus genotype data:
linked loci and correlated allele frequencies.
<em>Genetics</em> <b>164</b>, 1567–1587.
</p>
<p>Pritchard, J. K., Stephens, M. and Donnelly, P.  (2000)
Inference of population structure using multilocus genotype data.
<em>Genetics</em> <b>155</b>, 945–959.
</p>


<h3>See Also</h3>

<p><code>GetWeightedMeanGenotypes</code>, <code>RADdata2VCF</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># load example dataset
data(exampleRAD)
# get genotype posterior probabilities
exampleRAD &lt;- IterateHWE(exampleRAD)

# export to GAPIT
exampleGAPIT &lt;- ExportGAPIT(exampleRAD)

# export to rrBLUP
example_rrBLUP_A &lt;- Export_rrBLUP_Amat(exampleRAD)
example_rrBLUP_GWAS &lt;- Export_rrBLUP_GWAS(exampleRAD)

# export to TASSEL
outfile &lt;- tempfile() # temporary file for example
Export_TASSEL_Numeric(exampleRAD, outfile)

# for mapping populations
data(exampleRAD_mapping)

# specify donor and recurrent parents
exampleRAD_mapping &lt;- SetDonorParent(exampleRAD_mapping, "parent1")
exampleRAD_mapping &lt;- SetRecurrentParent(exampleRAD_mapping, "parent2")

# run the pipeline
exampleRAD_mapping &lt;- PipelineMapping2Parents(exampleRAD_mapping)

# convert to polymapR format
examplePMR &lt;- Export_polymapR(exampleRAD_mapping)

examplePMR2 &lt;- Export_polymapR_probs(exampleRAD_mapping)

# export to MAPpoly
outfile2 &lt;- tempfile() # temporary file for example
# generate a dummy phenotype matrix containing random numbers
mypheno &lt;- matrix(rnorm(200), nrow = 100, ncol = 2,
                  dimnames = list(GetTaxa(exampleRAD_mapping)[-(1:2)],
                                  c("Height", "Yield")))
Export_MAPpoly(exampleRAD_mapping, file = outfile2, pheno = mypheno)

# load data into MAPpoly
# require(mappoly)
# mydata &lt;- read_geno_prob(outfile2)

# export to GWASpoly
outfile3 &lt;- tempfile() # temporary file for example
Export_GWASpoly(SubsetByPloidy(exampleRAD, list(2)), outfile3)

# export to Structure
outfile4 &lt;- tempfile() # temporary file for example
Export_Structure(exampleRAD, outfile4)

# export to adegenet
if(requireNamespace("adegenet", quietly = TRUE)){
  mygenind &lt;- Export_adegenet_genind(exampleRAD)
}
</code></pre>


</div>