<div class="container">

<table style="width: 100%;"><tr>
<td>visualiseQTLeffects</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Visualise QTL homologue effects around a QTL position</h2>

<h3>Description</h3>

<p>Function to visualise the effect of parental homologues around a QTL peak across the population.
</p>


<h3>Usage</h3>

<pre><code class="language-R">visualiseQTLeffects(
  IBD_list,
  Phenotype.df,
  genotype.ID,
  trait.ID,
  linkage_group,
  LOD_data,
  cM_range = NULL,
  col.pal = c("purple4", "white", "seagreen"),
  point.density = 50,
  zero.sum = FALSE,
  allelic_interaction = FALSE,
  exploreQTL_output = NULL,
  return_plotData = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>IBD_list</code></td>
<td>
<p>List of IBD probabilities</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Phenotype.df</code></td>
<td>
<p>A data.frame containing phenotypic values</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotype.ID</code></td>
<td>
<p>The colname of <code>Phenotype.df</code> that contains the population identifiers (F1 names) (must be a colname of <code>Phenotype.df</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trait.ID</code></td>
<td>
<p>The colname of <code>Phenotype.df</code> that contains the response variable to use in the model (must be a colname of <code>Phenotype.df</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>linkage_group</code></td>
<td>
<p>Numeric identifier of the linkage group being tested, based on the order of <code>IBD_list</code>.
Only a single linkage group is allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LOD_data</code></td>
<td>
<p>Output of <code>QTLscan</code> function</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cM_range</code></td>
<td>
<p>If required, the plotting region can be restricted to a specified range of centiMorgan positions
(provided as a vector of start and end positions).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col.pal</code></td>
<td>
<p>Vector of colours to use in the visualisations (it is best to provide two or three colours for simplicity). By default,
effects will be coloured from purple to green through white.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>point.density</code></td>
<td>
<p>Parameter to increase the smoothing of homologue effect tracks</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>zero.sum</code></td>
<td>
<p>How allele substitution effect should be defined. If <code>FALSE</code> (by default), the effect of each homologue
is computed relative to the overall phenotypic mean, otherwise contrasts (against offspring without the inherited homologue) are used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allelic_interaction</code></td>
<td>
<p>By default <code>FALSE</code>, in which case the additive effects of parental alleles are visualised. If <code>TRUE</code>, a plot of
the mean effect of combinations of parental alleles is visualised instead. <code>exploreQTL_output</code> is required in this case.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exploreQTL_output</code></td>
<td>
<p>If <code>allelic_interaction = TRUE</code>, the output of the function <code>exploreQTL</code> must be provided.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return_plotData</code></td>
<td>
<p>Logical, by default <code>FALSE</code>. If <code>TRUE</code>, plot data is returned, otherwise <code>NULL</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The estimated effects of the homologues, used in the visualisation
</p>


<h3>Examples</h3>

<pre><code class="language-R">data("IBD_4x","BLUEs.pheno","qtl_LODs.4x")
visualiseQTLeffects(IBD_list = IBD_4x,
                    Phenotype.df = BLUEs.pheno,
                    genotype.ID = "Geno",
                    trait.ID = "BLUE",
                    linkage_group = 2,
                    LOD_data = qtl_LODs.4x)
</code></pre>


</div>