<div class="container">

<table style="width: 100%;"><tr>
<td>sim.autoCross</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate dominant markers for an autopolyploid cross for all
parental types </h2>

<h3>Description</h3>

<p>Simulates dominant markers from an autopolyploid cross given the ploidy
level and/or expected segregation ratios and the proportions in each
dosage marker class. This is a wrapper to <code>sim.autoMarkers</code> to
generate markers for ‘10’, ‘01’ and ‘11’ parents
</p>


<h3>Usage</h3>

<pre><code class="language-R">sim.autoCross(ploidy.level, prop.par.type = structure(c(0.4, 0.4, 0.2),
names = c("p10", "p01", "p11")), n.markers = 500, n.individuals = 200,
dose.proportion, true.seg.ratios, no.dosage.classes,
marker.names = paste("M", 1:n.markers, sep = "."),
individual.names = paste("X", 1:n.individuals, sep = "."),
parent.names = c("P.1", "P.2"), seed)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ploidy.level</code></td>
<td>
<p>the number of homologous chromosomes, either as
numeric (single value) or as a character string containing
type tetraploid, hexaploid, octoploid, ...</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prop.par.type</code></td>
<td>
<p> the proportion of markers generated from each
parental type '10', '01' and '11'. Note that the exact number will
be randomly generated from the multinomial distribution
(Default: c(0.4,0.4,0.2))</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.markers</code></td>
<td>
<p> number of markers (Default: 500)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.individuals</code></td>
<td>
<p> number of individuals in the cross (Default: 200)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dose.proportion</code></td>
<td>
<p> the proportion of
markers to be simulated in each dosage class. Note that the exact
number will be randomly generated from the multinomial distribution
NB: If a vector is supplied the dose.proportion is same for each
parental type otherwise as list with components ‘p01’,
‘p10’ and ‘p11’</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>true.seg.ratios</code></td>
<td>
<p> numeric vector containing segregation
proportion to be supplied if you wish to overide automatic
calculations using ploidy.level</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>no.dosage.classes</code></td>
<td>
<p>numeric vector containing the number of
dosage classes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>marker.names</code></td>
<td>
<p> labels for markers (Default: M.1 ... M.n.markers)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>individual.names</code></td>
<td>
<p> labels for offspring (Default: ... X.j ... )</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parent.names</code></td>
<td>
<p> numeric vector of length 2 containing columns
of marker matrix containing parental markers
(Default: first 2 columns) </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p> integer used to set seed for random number generator
(RNG) which (if set) may be used to reproduce results</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns an object of class <code>simAutoCross</code> containing
</p>
<table>
<tr style="vertical-align: top;">
<td><code>markers</code></td>
<td>
<p> matrix of 0,1 dominant markers with individuals as
cols and rows as markers</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>true.dosage</code></td>
<td>
<p><em>true</em> doses for each marker</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>name.true.dose</code></td>
<td>
<p>names of <em>true</em> doses for each marker</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p10</code></td>
<td>
<p>object of class <code>simAutoMarkers</code> for parental
type ‘10’ </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p01</code></td>
<td>
<p>object of class <code>simAutoMarkers</code> for parental
type ‘01’ </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p11</code></td>
<td>
<p>object of class <code>simAutoMarkers</code> for parental
type ‘11’ </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ploidy.level</code></td>
<td>
<p>the number of homologous chromosomes as numeric
(single value)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prop.par.type</code></td>
<td>
<p>proportion of markers for each parental type
‘p01’, ‘p10’ and ‘p11’</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.markers</code></td>
<td>
<p> number of markers (Default: 500)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.individuals</code></td>
<td>
<p> number of individuals in the cross (Default: 200)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dose.proportion</code></td>
<td>
<p>proportion in each dose – if numeric vector is
the same for ‘p01’, ‘p10’ and ‘p11’ else a list
with components ‘p01’, ‘p10’ and ‘p11’</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>no.dosage.classes</code></td>
<td>
<p>number in each dosage class</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>no.parType</code></td>
<td>
<p>number in each parental type</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>time.generated</code></td>
<td>
<p>time/date when data set generated</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>seed for random number generator seed which could be used to
reproduce results (I hope)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>call</code></td>
<td>
<p>matches arguments when function called</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>All parameters
except the proportions of marker dosage types can be left at the
default. If only one value is set, then individual list
components will be assumed to be equal. The marker matrix is
prepended with parental marker alleles. An alternative is to
simply create each group using sim.automarkers and cbind them.
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>See Also</h3>

<p><code>simAutoCross</code>,<code>simAutoMarkers</code>,
<code>sim.autoMarkers</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">p1 &lt;- sim.autoCross(4, dose.proportion=c(0.7,0.3), n.markers=20, n.indiv=10)
print(p1)

p2 &lt;- sim.autoCross(4, dose.proportion=list(p01=c(0.7,0.3),p10=c(0.7,0.3),p11=c(0.6,0.2,0.2)))
print(p2)
</code></pre>


</div>