<div class="container">

<table style="width: 100%;"><tr>
<td>exploreQTL</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Explore the possible segregation type of a QTL peak using Schwarz Information Criterion</h2>

<h3>Description</h3>

<p>Function to explore the possible segregation type at a QTL position using the Schwarz Information Criterion
</p>


<h3>Usage</h3>

<pre><code class="language-R">exploreQTL(
  IBD_list,
  Phenotype.df,
  genotype.ID,
  trait.ID,
  linkage_group,
  LOD_data,
  cM = NULL,
  QTLconfig = NULL,
  plotBIC = TRUE,
  deltaBIC = 6,
  testAllele_Effects = TRUE,
  log = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>IBD_list</code></td>
<td>
<p>List of IBD probabilities</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Phenotype.df</code></td>
<td>
<p>A data.frame containing phenotypic values</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotype.ID</code></td>
<td>
<p>The colname of <code>Phenotype.df</code> that contains the population identifiers (F1 names) (must be a colname of <code>Phenotype.df</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trait.ID</code></td>
<td>
<p>The colname of <code>Phenotype.df</code> that contains the response variable to use in the model (must be a colname of <code>Phenotype.df</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>linkage_group</code></td>
<td>
<p>Numeric identifier of the linkage group being tested, based on the order of <code>IBD_list</code>.
Only a single linkage group is allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LOD_data</code></td>
<td>
<p>Output of <code>QTLscan</code> function</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cM</code></td>
<td>
<p>By default <code>NULL</code>, in which case the position of maximum LOD score is taken as the position of interest. 
Otherwise, the cM position to be explored.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>QTLconfig</code></td>
<td>
<p>Nested list of homologue configurations and modes of action of QTL to be explored and compared, the output of <code>segMaker</code>.
Note that a default List is available of all possible bi-allelic QTL if none is provided.
Each list element is itself a list with components
</p>

<dl>
<dt>homs : </dt>
<dd>
<p> a vector of length at least 1, describing the proposed homologues the functional allele Q is on</p>
</dd>
<dt>mode : </dt>
<dd>
<p> Vector of same length as <code>homs</code> with codes "a" for additive and "d" for dominant.</p>
</dd>
</dl>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plotBIC</code></td>
<td>
<p>Logical, with default <code>TRUE</code> - should the calculated BIC values be plotted?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>deltaBIC</code></td>
<td>
<p>Numeric, by default 6. Configurations within this distance of the minimum BIC are considered plausible.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>testAllele_Effects</code></td>
<td>
<p>Logical, with default <code>TRUE</code> - should the effects of the different alleles be tested
using the most likely QTL configuration?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log</code></td>
<td>
<p>Character string specifying the log filename to which standard output should be written. If <code>NULL</code> log is send to stdout.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>List with the following items:
</p>

<dl>
<dt>linkage_group</dt>
<dd>
<p>Linkage group of the QTL peak being explored</p>
</dd>
<dt>cM</dt>
<dd>
<p>CentiMorgan position of the locus being explored</p>
</dd>
<dt>BIC</dt>
<dd>
<p>Vector of BIC values corresponding to elements of <code>QTLconfig</code> provided for testing</p>
</dd>
<dt>Allele.effects</dt>
<dd>
<p>Summary of the means and standard errors of groups with (+)
and without(-) the specified allele combinations for the most likely QTLconfig
if <code>testAllele_Effects</code> = <code>TRUE</code> (<code>NULL</code> otherwise).</p>
</dd>
<dt>genotype.means</dt>
<dd>
<p>A one-column matrix of mean phenotype values of offspring classes, with rownames
corresponding to the genotype class. If the probability of certain genotype classes is 0 (e.g. double reduction
classes where no double reduction occurred), then the genotype mean for that class will be <code>NA</code></p>
</dd>
</dl>
<h3>Examples</h3>

<pre><code class="language-R">data("IBD_4x","BLUEs.pheno","qtl_LODs.4x")
exploreQTL(IBD_list = IBD_4x,
           Phenotype.df = BLUEs.pheno,
           genotype.ID = "Geno",
           trait.ID = "BLUE",
           linkage_group = 1,
           LOD_data = qtl_LODs.4x)
</code></pre>


</div>