<div class="container">

<table style="width: 100%;"><tr>
<td>gendata-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Class "gendata"</h2>

<h3>Description</h3>

<p>This is a superclass for other classes that contain population
genetic datasets.  It has slots for population information, ploidy,
microsatellite repeat lengths, and a missing data symbol, but does not
have a slot to store genotypes.  Sample and locus names are stored as
the <code>names</code> of vectors in the slots.
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("gendata", samples, loci, ...)</code>.
The missing data symbol will be set to <code>-9</code> by default.  The
default initial value for <code>PopNames</code> is a
character vector of length 0, and for <code>Description</code> is the string
<code>"Insert dataset description here"</code>.  The default initial value for
the <code>Ploidies</code> slot is a <code>"ploidymatrix"</code>
object, containing a matrix filled with <code>NA</code> and named by samples
in the first dimension and loci in the second dimension.  For
other slots, vectors filled with <code>NA</code> will be generated and will be
named by samples (for <code>PopInfo</code>) or loci (for
<code>Usatnts</code>).  The slots can then be edited
using the methods described below.
</p>
<p>Note that in most cases you will want to instead create an
object from one of <code>gendata</code>'s subclasses, such as <code>genambig</code>.
</p>


<h3>Slots</h3>


<dl>
<dt>
<code>Description</code>:</dt>
<dd>
<p>Object of class <code>"character"</code>.  One
or more character strings to name or describe the dataset. </p>
</dd>
<dt>
<code>Missing</code>:</dt>
<dd>
<p>Object of class <code>"ANY"</code>.  A value to
indicate missing data in the genotypes of the dataset.  <code>-9</code>
by default. </p>
</dd>
<dt>
<code>Usatnts</code>:</dt>
<dd>
<p>Object of class <code>"integer"</code>.  This
vector must be named by locus names.  Each element should be the
length of the microsatellite repeat for that locus, given in nucleotides.
For example, <code>2</code> would indicate a locus with dinucleotide
repeats, and <code>3</code> would indicate a locus with trinucleotide
repeats.  <code>1</code> should be used for mononucleotide repeats OR if
alleles for that locus are already expressed in terms of repeat
number rather than nucleotides.  To put it another way, if you
divided the number used to represent an allele by the corresponding
number in <code>Usatnts</code> (and rounded if necessary), the result would
be the number of repeats (plus some additional length for flanking regions). </p>
</dd>
<dt>
<code>Ploidies</code>:</dt>
<dd>
<p>Object of class <code>"ploidysuper"</code>.  This
object will contain a <code>pld</code> slot that is a matrix named by
samples and loci, a vector named by samples or loci, or a single
value, depending on the subclass.  Each element is an integer that
represents ploidy.  <code>NA</code> indicates unknown ploidy. </p>
</dd>
<dt>
<code>PopInfo</code>:</dt>
<dd>
<p>Object of class <code>"integer"</code>.  This
vector also must be named by sample names.  Each element
represents the number of the population to which each sample belongs. </p>
</dd>
<dt>
<code>PopNames</code>:</dt>
<dd>
<p>Object of class <code>"character"</code>.  An
unnamed vector containing the name of each population.  If a
number from <code>PopInfo</code> is used to index <code>PopNames</code>, it
should find the correct population name.  For example, if the
first element of <code>PopNames</code> is <code>"ABC"</code>, then any samples
with <code>1</code> as their <code>PopInfo</code> value belong to population
<code>"ABC"</code>.</p>
</dd>
</dl>
<h3>Methods</h3>


<dl>
<dt>deleteLoci</dt>
<dd>
<p><code>signature(object = "gendata")</code>: Permanently
remove loci from the dataset.  This removes elements from <code>Usatnts</code>. </p>
</dd>
<dt>deleteSamples</dt>
<dd>
<p><code>signature(object = "gendata")</code>:
Permanently remove samples from the dataset.  This removes
elements from <code>PopInfo</code> and <code>Ploidies</code>. </p>
</dd>
<dt>Description</dt>
<dd>
<p><code>signature(object = "gendata")</code>: Returns the
character vector in the <code>Description</code> slot.</p>
</dd>
<dt>Description&lt;-</dt>
<dd>
<p><code>signature(object = "gendata")</code>: Assigns a
new value to the character vector in the <code>Description</code> slot.</p>
</dd>
<dt>initialize</dt>
<dd>
<p><code>signature(.Object = "gendata")</code>: This is
called when the <code>new("gendata")</code> function is used.  A new
<code>gendata</code> object is created with sample and locus names used
to index the appropriate slots. </p>
</dd>
<dt>Loci</dt>
<dd>
<p><code>signature(object = "gendata", usatnts =
	"missing", ploidies="missing")</code>: Returns a character vector
containing all locus
names for the object.  The method accomplishes this by returning
<code>names(object@Usatnts)</code>. </p>
</dd>
<dt>Loci</dt>
<dd>
<p><code>signature(object = "gendata", usatnts =
	"numeric", ploidies = "missing")</code>: Returns a character vector
of all loci for a given
set of repeat lengths.  For example, if <code>usatnts = 2</code> all
loci with dinucleotide repeats will be returned.</p>
</dd>
<dt>Loci</dt>
<dd>
<p><code>signature(object= "gendata", usatnts = "missing",
    ploidies = "numeric")</code>: Returns a character vector of all loci for a
given set of ploidies. Only works if <code>object@Ploidies</code> is a
<code>"ploidylocus"</code> object.</p>
</dd>
<dt>Loci</dt>
<dd>
<p><code>signature(object = "gendata", usatnts = "numeric",
ploidies = "numeric")</code>: Returns a character vector of all loci that have
one of the indicated repeat types and one of the indicated ploidies.
Only works if <code>object@Ploidies</code> is a <code>"ploidylocus"</code> object.</p>
</dd>
<dt>Loci&lt;-</dt>
<dd>
<p><code>signature(object = "gendata")</code>: Assigns new
names to loci in the dataset (changes <code>names(object@Usants)</code>.
Should not be used for adding or removing loci.</p>
</dd>
<dt>Missing</dt>
<dd>
<p><code>signature(object = "gendata")</code>: Returns the
missing data symbol from <code>object@Missing</code>. </p>
</dd>
<dt>Missing&lt;-</dt>
<dd>
<p><code>signature(object = "gendata")</code>: Assigns a new
value to <code>object@Missing</code> (changes the missing data symbol).</p>
</dd>
<dt>Ploidies</dt>
<dd>
<p><code>signature(object = "gendata", samples = "ANY",
    loci = "ANY")</code>: Returns the
ploidies in the dataset (<code>object@Ploidies</code>), indexed by
sample and locus if applicable.. </p>
</dd>
<dt>Ploidies&lt;-</dt>
<dd>
<p><code>signature(object = "gendata")</code>: Assigns new
values to ploidies of samples in the dataset. The assigned values
are coerced to integers by the method.  Names in the assigned
vector or matrix are ignored; sample and/or locus names already
present in the <code>gendata</code> object are used instead.</p>
</dd>
<dt>PopInfo</dt>
<dd>
<p><code>signature(object = "gendata")</code>: Returns the
population numbers of samples in the dataset  (<code>object@PopInfo</code>). </p>
</dd>
<dt>PopInfo&lt;-</dt>
<dd>
<p><code>signature(object = "gendata")</code>: Assigns new
population numbers to samples in the dataset.  The assigned values
are coerced to integers by the method.  Names in the assigned
vector are ignored; sample names already present in the
<code>gendata</code> object are used instead. </p>
</dd>
<dt>PopNames</dt>
<dd>
<p><code>signature(object = "gendata")</code>: Returns a
character vector of population names (<code>object@PopNames</code>). </p>
</dd>
<dt>PopNames&lt;-</dt>
<dd>
<p><code>signature(object = "gendata")</code>: Assigns new
names to populations. </p>
</dd>
<dt>PopNum</dt>
<dd>
<p><code>signature(object = "gendata",
	popname="character")</code>: Returns the number corresponding to a
population name.</p>
</dd>
<dt>PopNum&lt;-</dt>
<dd>
<p><code>signature(object = "gendata", popname =
	"character")</code>: Changes the population number for a given
population name, merging it with an existing population of that
number if applicable.</p>
</dd>
<dt>Samples</dt>
<dd>
<p><code>signature(object = "gendata", populations =
	"character", ploidies = "missing")</code>: Returns all sample names
for a given set of population names. </p>
</dd>
<dt>Samples</dt>
<dd>
<p><code>signature(object = "gendata", populations =
	"character", ploidies = "numeric")</code>: Returns all sample names
for a given set of population names and ploidies.  Only samples
that fit both criteria will be returned. </p>
</dd>
<dt>Samples</dt>
<dd>
<p><code>signature(object = "gendata", populations =
	"missing", ploidies = "missing")</code>: Returns all sample names. </p>
</dd>
<dt>Samples</dt>
<dd>
<p><code>signature(object = "gendata", populations =
	"missing", ploidies = "numeric")</code>: Returns all sample names for
a given set of ploidies. Only works if <code>object@Ploidies</code> is a
<code>"ploidysample"</code> object.</p>
</dd>
<dt>Samples</dt>
<dd>
<p><code>signature(object = "gendata", populations =
	"numeric", ploidies = "missing")</code>: Returns all sample names for
a given set of population numbers. </p>
</dd>
<dt>Samples</dt>
<dd>
<p><code>signature(object = "gendata", populations =
	"numeric", ploidies = "numeric")</code>: Returns all sample names for
a given set of population numbers and ploidies.  Only samples that
fit both criteria will be returned.  Only works if
<code>object@Ploidies</code> is a <code>"ploidysample"</code> object.</p>
</dd>
<dt>Samples&lt;-</dt>
<dd>
<p><code>signature(object = "gendata")</code>: Assigns new
names to samples.  This edits both <code>names(object@PopInfo)</code> and
<code>names(object@Ploidies)</code>.  It should not be used for adding
or removing samples from the dataset.</p>
</dd>
<dt>summary</dt>
<dd>
<p><code>signature(object = "gendata")</code>: Prints some
informaton to the console, including the numbers of samples, loci,
and populations, the ploidies present, and the types of
microsatellite repeats present.</p>
</dd>
<dt>Usatnts</dt>
<dd>
<p><code>signature(object = "gendata")</code>: Returns
microsatellite repeat lengths for loci in the dataset
(<code>object@Usatnts</code>). </p>
</dd>
<dt>Usatnts&lt;-</dt>
<dd>
<p><code>signature(object = "gendata")</code>: Assigns new
values to microsatellite repeat lengths of loci
(<code>object@Usatnts</code>).  The assigned values
are coerced to integers by the method.  Names in the assigned
vector are ignored; locus names already present in the
<code>gendata</code> object are used instead. </p>
</dd>
<dt>"["</dt>
<dd>
<p><code>signature(x = "gendata", i = "ANY", j = "ANY")</code>:
Subscripts the data by a subset of samples and/or loci.  Should be
used in the format <code>mygendata[mysamples, myloci]</code>.  Returns a
<code>gendata</code> object with <code>PopInfo</code>, <code>Ploidies</code>, and
<code>Usatnts</code> truncated to only contain the samples and loci
listed in <code>i</code> and <code>j</code>, respectively.
<code>Description</code>, <code>Missing</code>, and <code>PopNames</code> are left
unaltered.</p>
</dd>
<dt>merge</dt>
<dd>
<p><code>signature(x = "gendata", y = "gendata")</code>: Merges
two genotype objects.  See <code>merge,gendata,gendata-method</code>.</p>
</dd>
</dl>
<h3>Author(s)</h3>

<p>Lindsay V. Clark
</p>


<h3>See Also</h3>

<p><code>genambig</code>, <code>genbinary</code>,
<code>Accessors</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># show class definition
showClass("gendata")

# create an object of the class gendata
# (in reality you would want to create an object belonging to one of the
# subclasses, but the procedure is the same)
mygen &lt;- new("gendata", samples = c("a", "b", "c"),
             loci = c("loc1", "loc2"))
Description(mygen) &lt;- "An example for the documentation"
Usatnts(mygen) &lt;- c(2,3)
PopNames(mygen) &lt;- c("PopV", "PopX")
PopInfo(mygen) &lt;- c(2,1,2)
Ploidies(mygen) &lt;- c(2,2,4,2,2,2)

# view a summary of the object
summary(mygen)
</code></pre>


</div>