<div class="container">

<table style="width: 100%;"><tr>
<td>genambig-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Class "genambig"</h2>

<h3>Description</h3>

<p>Objects of this class store microsatellite datasets in which allele copy
number is ambiguous.  Genotypes are stored as a two-dimensional list of
vectors, each vector containing all unique alleles for a given sample at
a given locus.  <code>genambig</code> is a subclass of <code>gendata</code>.
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("genambig",
  samples, loci, ...)</code>.  This automatically sets up a two-dimensional
list in the <code>Genotypes</code> slot, with <code>dimnames=list(samples,
  loci)</code>.  This array-list is initially populated with the missing data
symbol.  All other slots are given initial values according to the
<code>initialize</code> method for <code>gendata</code>.  Data can then be inserted
into the slots using the replacement functions (see
<code>Accessors</code>).
</p>


<h3>Slots</h3>


<dl>
<dt>
<code>Genotypes</code>:</dt>
<dd>
<p>Object of class <code>"array"</code>.  The first
dimension of the array represents and is named by samples, while
the second dimension represents and is named by loci.  Each
element of the array can contain a vector.  Each vector should
contain each unique allele for the genotype once.  If an array
element contains a vector of length 1 containing only the symbol
that is in the <code>Missing</code> slot, this indicates missing data
for that sample and locus.</p>
</dd>
<dt>
<code>Description</code>:</dt>
<dd>
<p>Object of class <code>"character"</code>.  This
stores a description of the dataset for the user's convenience.</p>
</dd>
<dt>
<code>Missing</code>:</dt>
<dd>
<p>Object of class <code>"ANY"</code>.  A symbol to be
used to indicate missing data in the <code>Genotypes</code> slot.  This is
the integer <code>-9</code> by default.</p>
</dd>
<dt>
<code>Usatnts</code>:</dt>
<dd>
<p>Object of class <code>"integer"</code>.  A vector,
named by loci.  Each element indicates the repeat type of the
locus.  <code>2</code> indicates dinucleotide repeats, <code>3</code>
indicates trinucleotide repeats, and so on.  If the alleles stored
in the <code>Genotypes</code> slot for a given locus are already written
in terms of repeat number, the <code>Usatnts</code> value for that locus
should be <code>1</code>.  In other words, all alleles for a locus can be
divided by the number in <code>Usatnts</code> to give alleles expressed in
terms of relative repeat number.</p>
</dd>
<dt>
<code>Ploidies</code>:</dt>
<dd>
<p>Object of class <code>"integer"</code>.  A vector,
named by samples.  This stores the ploidy of each sample.  <code>NA</code>
indicates unknown ploidy.  See <code>Ploidies&lt;-</code> and
<code>estimatePloidy</code> for ways to fill this slot.</p>
</dd>
<dt>
<code>PopInfo</code>:</dt>
<dd>
<p>Object of class <code>"integer"</code>.  A vector,
named by samples, containing the population identity of each sample. </p>
</dd>
<dt>
<code>PopNames</code>:</dt>
<dd>
<p>Object of class <code>"character"</code>.  A
vector containing names for all populations.  The position of a
population name in the vector indicates the integer used to
represent that population in <code>PopInfo</code>. </p>
</dd>
</dl>
<h3>Extends</h3>

<p>Class <code>"gendata"</code>, directly.
</p>


<h3>Methods</h3>

<p>For more information on any of these methods, see the help files of
their respective generic functions.
</p>

<dl>
<dt>deleteLoci</dt>
<dd>
<p><code>signature(object = "genambig")</code>: Removes
columns in the array in the <code>Genotypes</code> slot corresponding to
the locus names supplied, then passes the arguments to the method
for <code>gendata</code>.</p>
</dd>
<dt>deleteSamples</dt>
<dd>
<p><code>signature(object = "genambig")</code>: Removes
rows in the array in the <code>Genotypes</code> slot corresponding to
the sample names supplied, then passes the arguments to the method
for <code>gendata</code>.</p>
</dd>
<dt>editGenotypes</dt>
<dd>
<p><code>signature(object = "genambig")</code>: Each
vector in the <code>Genotypes</code> slot is placed into the row of a
data frame, along with the sample and locus name for this vector.
The data frame is then opened in the Data Editor so that the user
can make changes.  When the Data Editor window is closed, vectors
are extracted back out of the data frame and written to the
<code>Genotypes</code> slot.</p>
</dd>
<dt>estimatePloidy</dt>
<dd>
<p><code>signature(object = "genambig")</code>:
Calculates the length of each genotype vector (excluding those
with the missing data symbol), and creates a data frame showing
the maximum and mean number of alleles per locus for each sample.
This data frame is then opened in the Data Editor, where the user
may edit ploidy levels.  Once the Data Editor is closed, the
<code>genambig</code> object is returned with the new values written to
the <code>Ploidies</code> slot.</p>
</dd>
<dt>Genotype</dt>
<dd>
<p><code>signature(object = "genambig")</code>: Retrieves a
single genotype vector, as specified by <code>sample</code> and
<code>locus</code> arguments. </p>
</dd>
<dt>Genotype&lt;-</dt>
<dd>
<p><code>signature(object = "genambig")</code>: Replaces a
single genotype vector. </p>
</dd>
<dt>Genotypes</dt>
<dd>
<p><code>signature(object = "genambig")</code>: Retrieves a
two-dimensional list of genotype vectors. </p>
</dd>
<dt>Genotypes&lt;-</dt>
<dd>
<p><code>signature(object = "genambig")</code>: Replaces a
one- or two-dimensional list of genotype vectors.</p>
</dd>
<dt>initialize</dt>
<dd>
<p><code>signature(.Object = "genambig")</code>: When
<code>new</code> is called to create a new <code>genambig</code> object, the
<code>initialize</code> method sets up a two dimensional list in the
<code>Genotypes</code> slot indexed by sample and locus, and fills this
list with the missing data symbol.  The <code>initialize</code> method
for <code>gendata</code> is then called. </p>
</dd>
<dt>isMissing</dt>
<dd>
<p><code>signature(object = "genambig")</code>: Given a set of
samples and loci, each position in the array in the <code>Genotypes</code>
slot is checked to see if it matches the missing data value.  A single
Boolean value or an array of Boolean values is returned.</p>
</dd>
<dt>Loci&lt;-</dt>
<dd>
<p><code>signature(object = "genambig")</code>: For changing
the names of loci.  The names are changed in the second dimension
of the array in the <code>Genotypes</code> slot, and then the
<code>Loci&lt;-</code> method for <code>gendata</code> is called. </p>
</dd>
<dt>Missing&lt;-</dt>
<dd>
<p><code>signature(object = "genambig")</code>: For changing
the missing data symbol.  All elements of the <code>Genotypes</code> array
that match the current missing data symbol are changed to the new
missing data symbol.  The <code>Missing&lt;-</code> method for <code>gendata</code>
is then called.</p>
</dd>
<dt>Samples&lt;-</dt>
<dd>
<p><code>signature(object = "genambig")</code>: For changing
the names of samples.  The names are changed in the first dimension
of the array in the <code>Genotypes</code> slot, and then the
<code>Samples&lt;-</code> method for <code>gendata</code> is called.</p>
</dd>
<dt>summary</dt>
<dd>
<p><code>signature(object = "genambig")</code>: Prints the
dataset description (<code>Description</code> slot) to the console as
well as the number of missing genotypes, then calls the
<code>summary</code> method for <code>gendata</code>. </p>
</dd>
<dt>show</dt>
<dd>
<p><code>signature(object = "genambig")</code>: Prints the
data to the console, formatted to make it more legible. The
genotype for each locus is shown as the size of each allele,
separated by a '/'. The SSR motif length ('Usatnts'), ploidies,
population names, and population membership ('PopInfo') are
displayed if they exist.</p>
</dd>
<dt>viewGenotypes</dt>
<dd>
<p><code>signature(object = "genambig")</code>: Prints a
tab-delimited table of samples, loci, and genotype vectors to the console.</p>
</dd>
<dt>"["</dt>
<dd>
<p><code>signature(x = "genambig", i = "ANY", j = "ANY")</code>:
For subscipting <code>genambig</code> objects.  Should be of the form
<code>mygenambig[mysamples, myloci]</code>.  Returns a <code>genambig</code>
object.  The <code>Genotypes</code> slot
is replaced by one containing only samples <code>i</code> and loci
<code>j</code>.  Likewise, the <code>PopInfo</code> and <code>Ploidies</code> slots
are truncated to contain only samples <code>i</code>, and the
<code>Usatnts</code> slot is truncated to contain only loci
<code>j</code>.  Other slots are left unaltered.</p>
</dd>
<dt>merge</dt>
<dd>
<p><code>signature(x = "genambig", y = "genambig")</code>:
Merges two genotypes objects together.  See
<code>merge,genambig,genambig-method</code>.</p>
</dd>
</dl>
<h3>Author(s)</h3>

<p>Lindsay V. Clark, Tyler W. Smith
</p>


<h3>See Also</h3>

<p><code>gendata</code>, <code>Accessors</code>,
<code>merge,genambig,genambig-method</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># display class definition
showClass("genambig")

# create a genambig object
mygen &lt;- new("genambig", samples=c("a", "b", "c", "d"),
             loci=c("L1", "L2", "L3"))
# add some genotypes
Genotypes(mygen)[,"L1"] &lt;- list(c(133, 139, 145), c(142, 154),
                                c(130, 142, 148), Missing(mygen))
Genotypes(mygen, loci="L2") &lt;- list(c(105, 109, 113), c(111, 117),
                                    c(103, 115), c(105, 109, 113))
Genotypes(mygen, loci="L3") &lt;- list(c(254, 258), Missing(mygen),
                                    c(246, 250, 262), c(250, 258))

# see a summary of the object
summary(mygen)
# display some of the genotypes
viewGenotypes(mygen[c("a", "b", "c"),])
</code></pre>


</div>