<div class="container">

<table style="width: 100%;"><tr>
<td>readProcessIsoloci</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Import Read Depth from Output of process_isoloci.py
</h2>

<h3>Description</h3>

<p>After process_isoloci.py is used to assign RAD tags to alignment locations
within a highly duplicated genome, <code>readProcessIsoloci</code> imports the
resulting CSV to a <code>"RADdata"</code> object.
</p>


<h3>Usage</h3>

<pre><code class="language-R">readProcessIsoloci(sortedfile, min.ind.with.reads = 200,
                   min.ind.with.minor.allele = 10,
                   min.median.read.depth = 10,
                   possiblePloidies = list(2), taxaPloidy = 2L,
                   contamRate = 0.001,
                   nameFromTagStart = TRUE, mergeRareHap = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sortedfile</code></td>
<td>

<p>File path to a CSV output by process_isoloci.py.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.ind.with.reads</code></td>
<td>

<p>Minimum number of individuals with reads needed to retain a locus.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.ind.with.minor.allele</code></td>
<td>

<p>Minimum number of individuals with reads in a minor allele needed to retain
a locus.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.median.read.depth</code></td>
<td>

<p>Minimum median read depth across individuals (including individuals with depth 0)
needed to retain a locus.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>possiblePloidies</code></td>
<td>

<p>A list indicating possible inheritance modes of loci.  See <code>RADdata</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>taxaPloidy</code></td>
<td>

<p>A single integer, or an integer vector with one value per taxon, indicating
ploidy.  See <code>RADdata</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>contamRate</code></td>
<td>

<p>Approximate rate of cross-contamination among samples.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nameFromTagStart</code></td>
<td>

<p>If <code>TRUE</code> loci will be named based on the alignment position and strand
of the RAD tag itself.  If <code>FALSE</code>, loci will be named based on the leftmost
position of the variable region of the RAD tag.  In either case,
<code>locTable$Pos</code> within the output will indicate the position of the variable
region of the tag.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mergeRareHap</code></td>
<td>

<p>Boolean indicating whether to run <code>MergeRareHaplotypes</code> after
building the <code>"RADdata"</code> object.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>MergeIdenticalHaplotypes</code> is used internally by this function to
merge alleles with identical sequence for the region shared by all tags, in
cases where tags vary in length within a locus.
</p>


<h3>Value</h3>

<p>A <code>"RADdata"</code> object containing read depth and alignment positions
from <code>sortedfile</code>.
</p>


<h3>Author(s)</h3>

<p>Lindsay V. Clark
</p>


<h3>See Also</h3>

<p><code>readProcessSamMulti</code>
</p>


</div>