<div class="container">

<table style="width: 100%;"><tr>
<td>RADdata</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
RADdata object constructor
</h2>

<h3>Description</h3>

<p><code>RADdata</code> is used internally to generate objects of the S3 class 
“RADdata” by <span class="pkg">polyRAD</span> functions for importing read depth data.
It is also available at the user level for cases where the data for import are
not already in a format supported by <span class="pkg">polyRAD</span>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">RADdata(alleleDepth, alleles2loc, locTable, possiblePloidies, contamRate,
        alleleNucleotides, taxaPloidy)
        
## S3 method for class 'RADdata'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>alleleDepth</code></td>
<td>

<p>An integer matrix, with taxa in rows and alleles in columns.  Taxa names should
be included as row names.  Each value indicates the number of reads for a given
allele in a given taxon.  There should be no <code>NA</code> values; use zero to 
indicate no reads.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleles2loc</code></td>
<td>

<p>An integer vector with one value for each column of <code>alleleDepth</code>.  The 
number indicates the identity of the locus to which the allele belongs.  A 
locus can have any number of alleles assigned to it (including zero).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>locTable</code></td>
<td>

<p>A data frame, where locus names are row names.  There must be at least as 
many rows as the highest value of <code>alleles2loc</code>; each number in 
<code>alleles2loc</code> corresponds to a row index in <code>locTable</code>.  No columns
are required, although if provided a column named “Chr” will be used for
indicating chromosome identities, a column named “Pos” will be used
for indicating physical position, and a column named “Ref” will be used
to indicate the reference sequence.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>possiblePloidies</code></td>
<td>

<p>A list, where each item in the list is an integer vector (or a numeric vector
that can be converted to integer).  Each vector indicates an inheritance 
pattern that markers in the dataset might obey.  <code>2</code> indicates diploid,
<code>4</code> indicates autotetraploid, <code>c(2, 2)</code> indicates allotetraploid,
<em>etc.</em>
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>contamRate</code></td>
<td>

<p>A number ranging from zero to one (although in practice probably less than 
0.01) indicating the expected sample cross-contamination rate.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleleNucleotides</code></td>
<td>

<p>A character vector with one value for each column of 
<code>alleleDepth</code>, indicating the DNA sequence for that allele.  Typically 
only the sequence at variable sites is provided, although intervening 
non-variable sequence can also be provided.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>taxaPloidy</code></td>
<td>

<p>An integer vector indicating ploidies of taxa.  If a single value is provided,
it will be assumed that all taxa are the same ploidy.  Otherwise, one value
must be provided for each taxon.  If unnamed, it is assumed that taxa are in
the same order as the rows of <code>alleleDepth</code>.  If named, names must match
the row names of <code>alleleDepth</code> but do not need to be in the same order.
This value is used as a multiplier with <code>possiblePloidies</code>; see Details.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>

<p>A “RADdata” object.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>Additional arguments to pass to <code>plot</code>, for example <code>col</code> or 
<code>pch</code>.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For a single locus, ideally the string provided in <code>locTable$Ref</code> and all
strings in <code>alleleNucleotides</code> are the same length, so that SNPs and indels
may be matched by position.  The character “-” indicates a deletion with
respect to the reference, and can be used within <code>alleleNucleotides</code>.  The
character “.” is a placeholder where other alleles have an insertion with
respect to the reference, and may be used in <code>locTable$Ref</code> and
<code>alleleNucleotides</code>.  Note that it is possible for the sequence in
<code>locTable$Ref</code> to be absent from <code>alleleNucleotides</code> if the reference
haplotype is absent from the dataset, as may occur if the reference genome is that
of a related species and not the actual study species.  For the
<code>alleleNucleotides</code> vector, the attribute <code>"Variable_sites_only"</code>
indicates whether non-variable sequence in between variants is included; this
needs to be <code>FALSE</code> for other functions to determine the position of each
variant within the set of tags.
</p>
<p>Inheritance mode is determined by multiplying the values in
<code>possiblePloidies</code> by the values in <code>taxaPloidy</code> and dividing by two.
For example, if you wanted to assume autotetraploid inheritance across the
entire dataset, you could set <code>possiblePloidies = list(4)</code> and
<code>taxaPloidy = 2</code>, or alternatively <code>possiblePloidies = list(2)</code> and
<code>taxaPloidy = 4</code>.  To indicate a mix of diploid and allotetraploid
inheritance across loci, set <code>possiblePloidies = list(2, c(2, 2))</code> and
<code>taxaPloidy = 2</code>.  If taxa themselves vary in ploidy, provide one
value of <code>taxaPloidy</code> for each taxon.  All inheritance modes listed in
<code>possiblePloidies</code> apply equally to all taxa, even when ploidy varies
by taxon.
</p>


<h3>Value</h3>

<p>An object of the S3 class “RADdata”.  The following slots are available
using the <code>$</code> operator:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>alleleDepth </code></td>
<td>
<p>Identical to the argument provided to the function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleles2loc </code></td>
<td>
<p>Identical to the argument provided to the function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>locTable </code></td>
<td>
<p>Identical to the argument provided to the function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>possiblePloidies </code></td>
<td>
<p>The <code>possiblePloidies</code> argument, converted to
integer.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>locDepth </code></td>
<td>
<p>A matrix with taxa in rows and loci in columns, with read 
depth summed across all alleles for each locus.  Column names are locus 
numbers rather than locus names.  See <code>GetLocDepth</code> for retrieving
the same matrix but with locus names as column names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>depthSamplingPermutations </code></td>
<td>
<p>A numeric matrix with taxa in rows and 
alleles in columns.  It is calculated as <code class="reqn">log(locDepth choose alleleDepth)</code>.
This is used as a coefficient for likelihood estimations done by other
<span class="pkg">polyRAD</span> functions (<em>i.e.</em> <code>AddGenotypeLikelihood</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>depthRatio </code></td>
<td>
<p>A numeric matrix with taxa in rows and alleles in columns.
Calculated as <code class="reqn">alleleDepth / locDepth</code>.  Used by other <span class="pkg">polyRAD</span>
functions for rough estimation of genotypes and allele frequency.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>antiAlleleDepth </code></td>
<td>
<p>An integer matrix with taxa in rows and alleles in
columns.  For each allele, the number of reads from the locus that do NOT
belong to that allele.  Calculated as <code class="reqn">locDepth - alleleDepth</code>. Used for
likelihood estimations by other <span class="pkg">polyRAD</span> functions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleleNucleotides </code></td>
<td>
<p>Identical to the argument provided to the 
function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>taxaPloidy </code></td>
<td>
<p>A named integer vector with one value per taxon, indicating
the ploidy of taxa.</p>
</td>
</tr>
</table>
<p>The object additionally has several attributes (see <code>attr</code>):
</p>
<table>
<tr style="vertical-align: top;">
<td><code>taxa </code></td>
<td>
<p>A character vector listing all taxa names, in the same order as
the rows of <code>alleleDepth</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nTaxa </code></td>
<td>
<p>An integer indicating the number of taxa.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nLoc </code></td>
<td>
<p>An integer indicating the number of loci in <code>locTable</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>contamRate </code></td>
<td>
<p>Identical to the argument provided to the function.</p>
</td>
</tr>
</table>
<p>The <code>plot</code> method performs a principal components analysis with 
<code>AddPCA</code> if not already done, then plots the first two axes.
Points represent individuals (taxa).  If mapping population parents have been
noted in the object (see <code>SetDonorParent</code>), they are indicated in
the plot.
</p>


<h3>Author(s)</h3>

<p>Lindsay V. Clark
</p>


<h3>See Also</h3>

<p>Data import functions that internally call <code>RADdata</code>:
</p>
<p><code>readHMC</code>, <code>readTagDigger</code>, 
<code>VCF2RADdata</code>, <code>readStacks</code>,
<code>readTASSELGBSv2</code>, <code>readProcessSamMulti</code>,
<code>readProcessIsoloci</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># create the dataset
mydepth &lt;- matrix(sample(100, 16), nrow = 4, ncol = 4,
                  dimnames = list(paste("taxon", 1:4, sep = ""),
                  paste("loc", c(1,1,2,2), "_", c(0,1,0,1), sep = "")))
mydata &lt;- RADdata(mydepth, c(1L,1L,2L,2L), 
                  data.frame(row.names = c("loc1", "loc2"), Chr = c(1,1),
                             Pos = c(2000456, 5479880)),
                  list(2, c(2,2)), 0.001, c("A", "G", "G", "T"), 6)

# inspect the dataset
mydata
mydata$alleleDepth
mydata$locDepth
mydata$depthRatio
mydata$taxaPloidy

# the S3 class structure is flexible; other data can be added
mydata$GPS &lt;- data.frame(row.names = attr(mydata, "taxa"),
                         Lat = c(43.12, 43.40, 43.05, 43.27),
                         Long = -c(70.85, 70.77, 70.91, 70.95))
mydata$GPS

# If you have NA in your alleleDepth matrix to indicate zero reads,
# perform the following before running the RADdata constructor:
mydepth[is.na(mydepth)] &lt;- 0L

# plotting a RADdata object
plot(mydata)
</code></pre>


</div>