<div class="container">

<table style="width: 100%;"><tr>
<td>polySegratioMM-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Bayesian Mixture Models for Marker Dosage in Autopolyploids
</h2>

<h3>Description</h3>

<p>These functions provide tools for estimating marker dosage for
dominant markers in regular autopolyploids via Bayesian mixture
model. Wrappers are provided for generating MCMC samples
using the <code>JAGS</code> software. Convergence diagnostics and posterior
distribution densities are provided by the <code>coda</code>
package. 


</p>


<h3>Details</h3>


<table>
<tr>
<td style="text-align: left;">
Package: </td>
<td style="text-align: left;"> polySegratioMM</td>
</tr>
<tr>
<td style="text-align: left;">
Type: </td>
<td style="text-align: left;"> Package</td>
</tr>
<tr>
<td style="text-align: left;">
Version: </td>
<td style="text-align: left;"> 0.6-4</td>
</tr>
<tr>
<td style="text-align: left;">
Date: </td>
<td style="text-align: left;"> 2018-03-22</td>
</tr>
<tr>
<td style="text-align: left;">
License: </td>
<td style="text-align: left;">  GPL-3
</td>
</tr>
</table>
<p>The simplest way to fit a model is to use <code>runSegratioMM</code>.
Given segregation ratios and a ploidy level, a mixture model is
constructed with default priors and initial values and <code>JAGS</code> run
to produce an MCMC sample for statistical inference.
</p>
<p>A standard model may be set up with <code>setModel</code> where two
parameters are set, namely <code>ploidy.level</code> or the number of
homologous chromosomes set either as a numeric or as a character
string and also <code>n.components</code> or the number of components for
mixture model (less than or equal to maximum number of possible
dosages).
</p>
<p>Vague or strong priors may be constructed automatically using
<code>setPriors</code>. Plots of standard conjugate distributions may
be obtained using <code>DistributionPlotBinomial</code>
<code>DistributionPlotGamma</code> and
<code>DistributionPlotNorm</code>.
</p>
<p>If necessary, other operations like setting up initial values or the
control files for <code>JAGS</code> may be set using <code>setInits</code>
<code>setControl</code> <code>dumpData</code> <code>dumpInits</code>
<code>writeControlFile</code> <code>writeJagsFile</code>. Once the
<code>BUGS</code> files and <code>JAGS</code> control files are set up then
<code>JAGS</code> may be run using <code>runJags</code> and results read
using <code>readJags</code>.
</p>
<p>Convergence diagnostics may be carried out using <code>coda</code> or the
convenience wrapper <code>diagnosticsJagsMix</code>.
</p>
<p>Dose allocation can be carried out using <code>dosagesJagsMix</code>.
</p>
<p>Plots may be produced and objects printed and summarised using standard
<code>print</code> and <code>plot</code> methods. Plots of theoretical
binomial distributions with different ploidy levels and sample sizes may
be obtained with <code>plotFitted</code>. In addition,
<code>plotFitted</code> produces a lattice plot of the observed
segregation ratios and fitted mixture model on the logit scale.
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>References</h3>


<ul>
<li>
<p>Baker P, Jackson P, and Aitken K. (2010) Bayesian estimation
of marker dosage in sugarcane and other autopolyploids.
<em>TAG Theoretical and Applied Genetics</em> <b>120</b> (8): 1653-1672.
</p>
</li>
<li>
<p>J B S Haldane   (1930) Theoretical genetics of
autopolyploids. <em>Journal of genetics</em> <b>22</b> 359–372
</p>
</li>
<li>
<p>Ripol, M I et al   (1999) Statistical aspects of genetic mapping in
autopolyploids. <em>Gene</em> <b>235</b> 31–41 
</p>
</li>
<li>
<p>JAGS  <a href="http://mcmc-jags.sourceforge.net/">http://mcmc-jags.sourceforge.net/</a>
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">## simulate small autooctaploid data set of 100 markers for 50 individuals
## with %70 Single, %20 Double and %10 Triple Dose markers 
a1 &lt;- sim.autoMarkers(8,c(0.7,0.2,0.1),n.markers=400,n.individuals=275)
##print(a1)
sr &lt;-  segregationRatios(a1$markers)
x &lt;- setModel(3,8)  # autooctapolid mode with 3 components

## Not run: 
## fit simple model in one hit with default priors, inits etc
## warning: this is too small an MCMC sample so should give inaccurate
## answers but it could still take quite a while
x.run &lt;- runSegratioMM(sr, x, burn.in=2000, sample=5000)
print(x.run)

## plot observed segregation ratios, fitted model and expected distribution
plot(x.run, theoretical=TRUE)

## End(Not run)
</code></pre>


</div>