<div class="container">

<table style="width: 100%;"><tr>
<td>Accessors</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Accessor and Replacement Functions for "gendata" Objects
</h2>

<h3>Description</h3>

<p>The accessor functions return information that is contained, either
directly or indirectly, in the slots of a <code>gendata</code> object.  The
replacement functions alter information in one or more slots as appropriate.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Samples(object, populations, ploidies)
Samples(object) &lt;- value
Loci(object, usatnts, ploidies)
Loci(object) &lt;- value
PopInfo(object)
PopInfo(object) &lt;- value
PopNames(object)
PopNames(object) &lt;- value
PopNum(object, popname)
PopNum(object, popname) &lt;- value
Ploidies(object, samples, loci)
Ploidies(object) &lt;- value
Usatnts(object)
Usatnts(object) &lt;- value
Description(object)
Description(object) &lt;- value
Missing(object)
Missing(object) &lt;- value
Present(object)
Present(object) &lt;- value
Absent(object)
Absent(object) &lt;- value
Genotype(object, sample, locus)
Genotype(object, sample, locus) &lt;- value
Genotypes(object, samples = Samples(object), loci = Loci(object))
Genotypes(object, samples = Samples(object), loci = Loci(object)) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>

<p>An object of the class <code>gendata</code> or one of its subclasses.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>populations</code></td>
<td>

<p>A character or numeric vector indicating from which populations to return
samples. (optional)
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ploidies</code></td>
<td>

<p>A numeric vector indicating ploidies, if only samples or loci with a certain
ploidy should be returned.  (optional)
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample</code></td>
<td>
<p>A character string or number indicating the name or number
of the sample whose genotype should be returned.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>locus</code></td>
<td>
<p>A character string or number indicating the name or number
of the locus whose genotype should be returned.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>samples</code></td>
<td>
<p>A character or numeric vector indicating samples for
which to return genotypes or ploidies. (optional)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loci</code></td>
<td>
<p>A character or numeric vector indicating loci for which to
return genotypes or ploidies.  (optional)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>usatnts</code></td>
<td>
<p>A numeric vector indicating microsatellite repeat
lengths, where only loci of those repeat lengths should be returned.
(optional)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>popname</code></td>
<td>
<p>Chacter string or vector.  The name(s) of the
population(s) for which to retrieve or replace the corresponding
<code>PopInfo</code> number(s).  The replacement function should only be
used for one population at a time.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>value</code></td>
<td>

<ul>
<li>
<p> For <code>Samples</code>: a character vector of sample names.
</p>
</li>
<li>
<p> For <code>Loci</code>: a character vector of locus names.
</p>
</li>
<li>
<p> For <code>PopInfo</code>: A numeric vector (integer or can be coerced to
integer) indicating the population identities of samples.
</p>
</li>
<li>
<p> For <code>PopNames</code>: A character vector indicating the names of
populations.
</p>
</li>
<li>
<p> For <code>PopNum</code>: A number (integer or can be coerced to integer)
that should be the new population number associated with
<code>popname</code>.
</p>
</li>
<li>
<p> For <code>Ploidies</code>: A numeric vector or matrix (integer or can be coerced to
integer) indicating the ploidy of each sample, locus, and/or the
dataset.  See <code>reformatPloidies</code> and
<code>"ploidysuper"</code>.
</p>
</li>
<li>
<p> For <code>Usatnts</code>: A numeric vector (integer or can be coerced to
integer) indicating the repeat type of each microsatellite locus.
Dinucleotide repeats should be represented with <code>2</code>,
trinucleotide repeat with <code>3</code>, and so on.  If the alleles for a
given locus are already stored in terms of repeat number rather than
fragment length in nucleotides, the <code>Usatnts</code> value for that
locus should be <code>1</code>.
</p>
</li>
<li>
<p> For <code>Description</code>: A character string or character vector
describing the dataset.
</p>
</li>
<li>
<p> For <code>Missing</code>: A symbol (usually an integer) to be used to
indicate missing data.
</p>
</li>
<li>
<p> For <code>Present</code>: A symbol (usually an integer) to be used to
indicate the presence of an allele.
</p>
</li>
<li>
<p> For <code>Absent</code>: A symbol (usually an integer) to be used to
indicate the absence of an allele.
</p>
</li>
<li>
<p> For <code>Genotype</code>: a vector of alleles, if the object is of class
<code>genambig</code>.
</p>
</li>
<li>
<p> For <code>Genotypes</code>: A list of vectors (genotypes), of the same
dimensionality as <code>c(length(samples), length(loci))</code>, if the
object is of class <code>genambig</code>.  If the object is of class
<code>genbinary</code>, <code>value</code> should be a matrix, with column names
of the form <code>"locus.allele"</code>.  See
<code>Genotypes&lt;-,genbinary-method</code> for more information.
</p>
</li>
</ul>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>Samples&lt;-</code> and <code>Loci&lt;-</code> can only be used to change sample
and locus names, not to add or remove samples and loci from the
dataset.
</p>
<p>For slots that require integer values, numerical values used in
replacement functions will be coerced to integers.  The replacement
functions also ensure that all slots remain properly indexed.
</p>
<p>The <code>Missing&lt;-</code> function finds any genotypes with the old missing
data symbol and changes them to the new missing data symbol, then
assigns the new symbol to the slot that indicates what the missing
data symbol is.  <code>Present&lt;-</code> and <code>Absent&lt;-</code> work similarly for
the <code>genbinary</code> class.
</p>
<p>The <code>Genotype</code> access and replacement functions deal with
individual genotypes, which are vectors in the <code>genambig</code> class.
The <code>Genotypes</code> access and replacement functions deal with lists
of genotypes.
</p>
<p>The <code>PopInfo&lt;-</code> replacement function also adds elements to
<code>PopNames(object)</code> if necessary in order to have names for all of
the populations.  These will be of the form <code>"Pop"</code> followed by
the population number, and can be later edited using
<code>PopNames&lt;-</code>.
</p>
<p>The <code>PopNum&lt;-</code> replacement function first finds all samples in
the population <code>popname</code>, and replaces the number in
<code>PopInfo(object)</code> for those samples with <code>value</code>.  It then
inserts <code>NA</code> into the original <code>PopNames</code> slot that
contained <code>popname</code>, and inserts <code>popname</code> into
<code>PopNames(object)[value]</code>.  If this results in two populations
being merged into one, a message is printed to the console.
</p>


<h3>Value</h3>

<p><code>PopInfo</code>, <code>PopNames</code>, <code>Missing</code>, <code>Description</code>,
<code>Usatnts</code>, <code>Ploidies</code> and <code>Genotypes</code> simply return the
contents of the slots of the same names (or in the case of
<code>Ploidies</code>, <code>object@Ploidies@pld</code> is returned). <code>Samples</code> and
<code>Loci</code> return character vectors taken from the <code>names</code> of
other slots (<code>PopInfo</code> and <code>Usatnts</code>,
respectively; the initialization and replacement methods ensure that
these slots are always named according to samples and
loci). <code>PopNum</code> returns an integer vector indicating the
population number(s) of the population(s) named in <code>popname</code>.
<code>Genotype</code> returns a single genotype for a given sample and
locus, which is a vector whose exact form will depend on the class of
<code>object</code>.
</p>


<h3>Author(s)</h3>

<p>Lindsay V. Clark
</p>


<h3>See Also</h3>

<p><code>deleteSamples</code>, <code>deleteLoci</code>,
<code>viewGenotypes</code>, <code>editGenotypes</code>,
<code>isMissing</code>, <code>estimatePloidy</code>,
<code>merge,gendata,gendata-method</code>, <code>gendata</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># create a new genambig (subclass of gendata) object to manipulate
mygen &lt;- new("genambig", samples=c("a", "b", "c"), loci=c("locG",
"locH"))

# retrieve the sample and locus names
Samples(mygen)
Loci(mygen)

# change some of the sample and locus names
Loci(mygen) &lt;- c("lG", "lH")
Samples(mygen)[2] &lt;- "b1"

# describe the dataset
Description(mygen) &lt;- "Example dataset for documentation."

# name some populations and assign samples to them
PopNames(mygen) &lt;- c("PopL", "PopK")
PopInfo(mygen) &lt;- c(1,1,2)
# now we can retrieve samples by population
Samples(mygen, populations="PopL")
# we can also adjust the numbers if we want to make them
# match another dataset
PopNum(mygen, "PopK") &lt;- 3
PopNames(mygen)
PopInfo(mygen)
# change the population identity of just one sample
PopInfo(mygen)["b1"] &lt;- 3

# indicate that both loci are dinucleotide repeats
Usatnts(mygen) &lt;- c(2,2)

# indicate that all samples are tetraploid
Ploidies(mygen) &lt;- 4
# or
Ploidies(mygen) &lt;- rep(4, times = length(Samples(mygen)) * length(Loci(mygen)))
# actually, one sample is triploid
Ploidies(mygen)["c",] &lt;- 3
# view ploidies
Ploidies(mygen)

# view the genotype array as it currently is: filled with missing
# values
Genotypes(mygen)
# fill in the genotypes
Genotypes(mygen, loci="lG") &lt;- list(c(120, 124, 130, 136), c(122, 120),
                                    c(128, 130, 134))
Genotypes(mygen, loci="lH") &lt;- list(c(200, 202, 210), c(206, 208, 210,
                                                        214),
                                    c(208))
# genotypes can also be edited or retrieved by sample
Genotypes(mygen, samples="a")
# fix a single genotype
Genotype(mygen, "a", "lH") &lt;- c(200, 204, 210)
# retrieve a single genotype
Genotype(mygen, "c", "lG")

# change a genotype to being missing
Genotype(mygen, "c", "lH") &lt;- Missing(mygen)
# show the current missing data symbol
Missing(mygen)
# an example of genotypes where one contains the missing data symbol
Genotypes(mygen, samples="c")
# change the missing data symbol
Missing(mygen) &lt;- as.integer(-1)
# now look at the genotypes
Genotypes(mygen, samples="c")

</code></pre>


</div>