<div class="container">

<table style="width: 100%;"><tr>
<td>ExamineGenotype</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Plots to Examine Genotype Calling at a Single Taxon and Allele
</h2>

<h3>Description</h3>

<p>For a given taxon and allele, this function generates barplots showing read
depth ratio, posterior mean genotype, genotype prior probabilities, genotype
likelihoods, and genotype posterior probabilities. It is intended as a sanity
check on genotype calling, as well as a means to visually demonstrate the
concept of Bayesian genotype calling.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ExamineGenotype(object, ...)

## S3 method for class 'RADdata'
ExamineGenotype(object, taxon, allele, pldindex = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>

<p>A <code>RADdata</code> object for which genotype calling has already been performed.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>taxon</code></td>
<td>

<p>A single character string indicating the taxon to show.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele</code></td>
<td>

<p>A single character string indicating the allele to show.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pldindex</code></td>
<td>

<p>An index of which inheritance mode to use within <code>object$possiblePloidies</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>Other arguments (none implemented).
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A barplot is generated. Invisibly, a list is returned:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>alleleDepth </code></td>
<td>
<p>Sequence read depth for the selected allele.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>antiAlleleDepth </code></td>
<td>
<p>Sequence read depth for all other alleles at the locus.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>depthRatio </code></td>
<td>
<p>Proportion of reads at this taxon and locus belonging to this allele.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>priorProb </code></td>
<td>
<p>A vector of genotype prior probabilities.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotypeLikelhood </code></td>
<td>
<p>A vector of genotype likelihoods.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>posteriorProb </code></td>
<td>
<p>A vector of genotype posterior probabilities.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>postMean </code></td>
<td>
<p>The posterior mean genotype on a scale of 0 to 1.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Lindsay V. Clark
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(exampleRAD)

exampleRAD &lt;- IterateHWE(exampleRAD)

eg &lt;- ExamineGenotype(exampleRAD, "sample088", "loc1_T")
</code></pre>


</div>