<div class="container">

<table style="width: 100%;"><tr>
<td>recodeAllopoly</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Create a New <code>genambig</code> Dataset with Loci Split into Isoloci
</h2>

<h3>Description</h3>

<p>Given a <code>"genambig"</code> object and a list of allele
assignments such as those produced by <code>testAlGroups</code> or
<code>catalanAlleles</code>, <code>recodeAllopoly</code> will generate a
new <code>"genambig"</code> object, with genotypes split according to which
alleles belong to which isoloci.
</p>


<h3>Usage</h3>

<pre><code class="language-R">recodeAllopoly(object, x, allowAneuploidy = TRUE,
               samples = Samples(object), loci = Loci(object))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>

<p>A <code>"genambig"</code> object containing the dataset that needs to be re-coded.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>

<p>A list.  Each item in the list should itself be a list, in the format
output by <code>testAlGroups</code>, <code>catalanAlleles</code>, or
<code>mergeAlleleAssignments</code>.  Each sub-list
has three items: <code>$locus</code> is the name of the locus,
<code>$SGploidy</code> is an integer indicating the ploidy of each subgenome
(<em>e.g.</em> <code>2</code> for an allotetraploid), and <code>$assignments</code> is
a matrix of ones and zeros indicating which alleles belong to which isoloci.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allowAneuploidy</code></td>
<td>

<p>Boolean.  This controls what happens when the function encounters
genotypes that have more alleles than are possible for a given
isolocus.  (For example, the genotype has four alleles, but three belong
to isolocus 1 and one belongs to isolocus 2.)  If <code>TRUE</code>, the
individual is assumed to be aneuploid at that locus, and its ploidy is
adjusted only for that locus.  If <code>FALSE</code>, missing data are recorded.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>samples</code></td>
<td>

<p>An optional character vector indicating which samples to analyze and output.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loci</code></td>
<td>

<p>An optional character vector indicating which loci to analyze and output.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The same locus may appear more than once in <code>x</code>, for example if
distinct populations were analyzed separately to produce the
allele assignments.  If this is the case, <code>recodeAllopoly</code> will
internally use <code>mergeAlleleAssignments</code> to consolidate items in
<code>x</code> with the same locus name.  Loci that are in <code>x</code> but not
<code>object</code> are ignored with a warning.  Loci that are in
<code>object</code> but not <code>x</code> are retained in the output of the
function, but not re-coded.
</p>
<p>This function allows homoplasy, and uses process-of-elimination to try
to determine which isoloci the homoplasious alleles belong to.  In cases
where genotypes cannot be determined for certain due to homoplasy,
missing data are inserted.
</p>
<p>If a genotype has more alleles than should be possible (<em>e.g.</em> five
alleles in an allotetraploid), the genotype is skipped and will be
output as missing data for all corresponding isoloci.
</p>


<h3>Value</h3>

<p>A <code>"genambig"</code> object, with loci that are in <code>x</code> split into
the appropriate number of isoloci.
</p>


<h3>Author(s)</h3>

<p>Lindsay V. Clark
</p>


<h3>References</h3>

<p>Clark, L. V. and Drauch Schreier, A. (2017) Resolving microsatellite genotype ambiguity in
populations of allopolyploid and diploidized autopolyploid organisms
using negative correlations between alleles. <em>Molecular Ecology Resources</em>, 
<b>17</b>, 1090â€“1103.  DOI: 10.1111/1755-0998.12639.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># generate a dataset for this example
testdata &lt;- new("genambig", samples = paste("S", 1:9, sep = ""),
                loci = c("L1", "L2","L3"))
Genotypes(testdata, loci="L1") &lt;-
    list(c(120,124),c(124,126,130),c(120,126),c(126,132,134),
         c(120,124,130,132),c(120,126,130),c(120,132,134),
         c(120,124,126,130),c(120,132,138))
Genotypes(testdata, loci="L2") &lt;-
    list(c(210,219,222,225),c(216,228),c(210,213,219,222),c(213,222,225,228),
         c(210,213,216,219),c(222,228),c(213),c(210,216),c(219,222,228))
Genotypes(testdata, loci="L3") &lt;-
    list(c(155,145,153),c(157,155),c(151,157,159,165),c(147,151),c(149,153,157),
         c(149,157),c(153,159,161),c(163,165),c(147,163,167))
viewGenotypes(testdata)

# generate allele assignments for this example
myAssign &lt;- list(list(locus="L1", SGploidy=2,
             assignments=matrix(c(1,0,0,1,1,1,0,1,1,0,1,1), nrow=2,
                                ncol=6, dimnames=list(NULL,
                                  c("120","124","126","130","132","134")))),
    list(locus="L2", SGploidy=2,
         assignments=matrix(c(1,1,1,1,1,1,1,0,1,0,1,0,0,1), nrow=2, ncol=7,
    dimnames=list(NULL,c("210","213","216","219","222","225","228")))),
    list(locus="L3", SGploidy=2, assignments="No assignment"))
myAssign

# recode the dataset
splitdata &lt;- recodeAllopoly(testdata, myAssign)

# view results
viewGenotypes(splitdata)
Ploidies(splitdata)

</code></pre>


</div>