<div class="container">

<table style="width: 100%;"><tr>
<td>expected.segRatio</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Compute theoretical segregation proportions for regular autopolyploids</h2>

<h3>Description</h3>

<p>Expected segregation proportions for various dosages of dominant
markers for regular autopolyploids are calculated
using the formula of Ripol et al (1999) based on Haldane (1930) for
single dose and multiple dose parents cross nulliplex
("homozygous") and an unpublished formula where both parents
possess at least single dose markers ("heterogeneous") 
</p>


<h3>Usage</h3>

<pre><code class="language-R">expected.segRatio(ploidy.level = stop("No ploidy level set"),
type.parents = c("heterogeneous", "homozygous"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ploidy.level</code></td>
<td>
<p>the number of homologous chromosomes, either as
numeric or as a character string</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type.parents</code></td>
<td>
<p> "heterogeneous" if parental markers are 0,1 or
"homogeneous" if parental markers are both 1</p>
</td>
</tr>
</table>
<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ratio</code></td>
<td>
<p>vector of proportions for each dosage</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ploidy.level</code></td>
<td>
<p>numeric value of ploidy level 2,4,6,8, ...</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ploidy.name</code></td>
<td>
<p>name of ploidy</p>
</td>
</tr>
</table>
<h3>Warning </h3>

<p>While results will be returned if the ploidy level is set as an odd
number, the formula used are only for even numbers. 
</p>


<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>References</h3>


<ul>
<li>
<p>J B S Haldane (1930) Theoretical genetics of
autopolyploids. <em>Journal of genetics</em> <b>22</b> 359–372
</p>
</li>
<li>
<p>Ripol, M I et al(1999) Statistical aspects of genetic mapping in
autopolyploids. <em>Gene</em> <b>235</b> 31–41 
</p>
</li>
</ul>
<h3>See Also</h3>

 <p><code>segRatio</code>, <code>test.segRatio</code></p>


<h3>Examples</h3>

<pre><code class="language-R">## heterogeneous parents

expected.segRatio(2)
expected.segRatio("Tetraploid")
expected.segRatio("tEtR")
expected.segRatio("octo")
expected.segRatio("Octa")
expected.segRatio(14)
## warning
expected.segRatio(9)

## errors - not run
## expected.segRatio("abcd")
## expected.segRatio(-1)

## homogeneous parents

expected.segRatio("Octa", type.parents="heter")
expected.segRatio("Octa", type.parents="homo")
expected.segRatio("tetra", type.parents="homo")
expected.segRatio(6, type.parents="homo")
expected.segRatio(9, type.parents="homo")

</code></pre>


</div>