<div class="container">

<table style="width: 100%;"><tr>
<td>gendata.to.genind</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Convert Data to genind Format
</h2>

<h3>Description</h3>

<p>This is a function for exporting data to the package <span class="pkg">adegenet</span>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gendata.to.genind(object, samples = Samples(object), loci = Loci(object))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>

<p>A <code>"genambig"</code> or (preferably) a <code>"genbinary"</code> object.  
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>samples</code></td>
<td>

<p>A character vector indicating the samples to include in the output.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loci</code></td>
<td>

<p>A character vector indicating the loci to include in the output.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>gendata.to.genind</code> converts a <code>"genambig"</code> or
<code>"genbinary"</code> object to a <code>"genind"</code> object using the package
<span class="pkg">adegenet</span>.  Each individual must have a single ploidy.  
Ploidy and population information are carried over to the new
object.  Data will be coded as presence/absence in the new object.  The
locus names in the new object are locus and allele names
seperated by a hyphen.
</p>
<p><span class="pkg">adegenet</span> must be installed in order to use this function.
</p>


<h3>Value</h3>

<p>A genetic dataset in the <code>"genind"</code> class, ready for use in
<span class="pkg">adegenet</span>. 
</p>


<h3>Author(s)</h3>

<p>Lindsay V. Clark
</p>


<h3>References</h3>

<p><a href="http://adegenet.r-forge.r-project.org/">http://adegenet.r-forge.r-project.org/</a>
</p>
<p>Jombart, T.  (2008)  adegenet: a R package for the multivariate analysis
of genetic markers.  <em>Bioinformatics</em> <b>24</b>, 1403â€“1405.
</p>


<h3>See Also</h3>

<p><code>freq.to.genpop</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># create a "genambig" object
mydata &lt;- new("genambig", samples=c("a","b","c","d"), loci=c("e","f"))
PopNames(mydata) &lt;- c("G","H")
PopInfo(mydata) &lt;- c(1,1,2,2)
mydata &lt;- reformatPloidies(mydata, output="one")
Ploidies(mydata) &lt;- 3
Genotypes(mydata, loci="e") &lt;- list(c(100),c(100,102),
                                    c(98,102,104),c(102,106))
Genotypes(mydata, loci="f") &lt;- list(c(200,202,204),Missing(mydata),
                                    c(210,212),c(204,210,214))

# convert to "genind"; not tested as it takes several seconds to load adegenet

if(require("adegenet")){
  mydata2 &lt;- gendata.to.genind(mydata)
  mydata2@tab
  locNames(mydata2)
  indNames(mydata2)
  popNames(mydata2)
  pop(mydata2)
}

</code></pre>


</div>