<div class="container">

<table style="width: 100%;"><tr>
<td>inputData</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read in, check and preprocess the allele dataset</h2>

<h3>Description</h3>

<p>Read in an allele dataset from file, and return a checked and
preprocessed data frame.
</p>


<h3>Usage</h3>

<pre><code class="language-R">inputData(file, numLoci, ploidy, dataType, dioecious, selfCompatible = NULL,
  mothersOnly = NULL, lociMin = 1, matMismatches = 0, skip = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>character: the name of the allele data file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>numLoci</code></td>
<td>
<p>integer: the number of loci in the allele dataset.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ploidy</code></td>
<td>
<p>integer: the species' ploidy, one of <code>2</code>,
<code>4</code>, <code>6</code>, or <code>8</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dataType</code></td>
<td>
<p>character: either <code>"genotype"</code> or
<code>"phenotype"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dioecious</code></td>
<td>
<p>logical: is the species dioecious (<code>TRUE</code>)
or monoecious (<code>FALSE</code>)?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>selfCompatible</code></td>
<td>
<p>logical: In monoecious species
(<code>dioecious=FALSE</code>), can individuals self-fertilise?  When
<code>dioecious=FALSE</code>, this argument may be left at its
default value of NULL - it will be set to <code>FALSE</code> by
<code>preprocessData</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mothersOnly</code></td>
<td>
<p>logical: in dioecious species, should females
without progeny present be removed from the dataset?  If
<code>dioecious=TRUE</code>, then <code>mothersOnly</code> must be set to
either <code>TRUE</code> or <code>FALSE</code>.  If <code>dioecious=FALSE</code>,
argument <code>mothersOnly</code> should be left at its default value of
<code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lociMin</code></td>
<td>
<p>integer: the minimum number of loci in an
individual that must have alleles present for the individual (and
its progeny, if the individual is a mother) to be retained in the
dataset.  See the help for <code>preprocessData</code> for more
on this parameter.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>matMismatches</code></td>
<td>
<p>an integer between 0 and <code>numLoci</code>-1,
being the maximum number of mismatching loci between mother and
offspring that are allowed before the offspring is removed from
the dataset.  The default value is 0.  If an offspring has fewer
than <code>matMismatches</code> loci that mismatch with its mother, the
offending loci are set to contain no alleles.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>skip</code></td>
<td>
<p>integer: the number of lines in the CSV to skip
before the header row of the table.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>inputData</code> reads in an allele dataset from the specified
file, then calls <code>preprocessData</code> to perform a series
of data format checks and preprocessing steps before returning the
checked and preprocessed dataset as an R data frame.  The
reference information for <code>preprocessData</code> contains
further information on the checks and preprocessing - it is
strongly recommended you read that information in addition to the
information below.
</p>
<p>The use of <code>inputData</code> is optional, if you wish to create or
load the allele dataset into R by other means. However, it is then
necessary to call <code>preprocessData</code> on the data frame
prior to using any other analysis functions in this package.
Similarly, if you decide to change or manipulate the data frame
contents within R, you should call <code>preprocessData</code>
again on the data frame prior to using any of the <span class="pkg">PolyPatEx</span>
analysis functions.  See the help for <code>preprocessData</code>
for further details.
</p>
<p>Note that <code>inputData</code> strips leading or trailing spaces
(whitespace) from each entry in the allele dataset as it is read
in.  If you load your data by a means other than <code>inputData</code>,
you should ensure that you perform this step yourself, as
<code>preprocessData</code> will not carry out this necessary
step.
</p>
<p>Note also that you should not use spaces in any of your allele
codes - PolyPatEx functions use spaces to separate allele codes as
they process the data - if allele codes already contains spaces,
errors will occur in this processing. If you need a separator, I
recommend using either ‘code.’ (a period) or
‘code_’ (an underscore) rather than a space.
</p>
<p>Neither <code>inputData</code> (nor <code>preprocessData</code>) will
alter the CSV file from which the data is loaded - they merely
return a checked and preprocessed version of your allele dataset
(in the form of an R data frame) within the R environment, ready
for use by other <span class="pkg">PolyPatEx</span> functions.
</p>
<p>To load the allele dataset into R, <code>inputData</code> calls R's
<code>read.csv</code> function with certain arguments specified.
These arguments make <code>read.csv</code> more stringent about
the precise format of the input datafile, requiring in particular
that each row of the CSV-formatted data file contain the correct
number of commas.  This is not always guaranteed when the CSV file
has been exported from spreadsheet software.  Should you get
‘Error in scan’ messages complaining about the number of
elements in a line of the input file, consider calling
<code>fixCSV</code> on the data file, before calling
<code>inputData</code> again.  <code>fixCSV</code> attempts to find and
correct such errors in a CSV file - see the help for this
function.  Note that if you specify the <code>skip</code> parameter in
a call to <code>fixCSV</code>, you should use the same value for
this parameter in <code>inputData</code> to avoid an error.
</p>
<p>The various <span class="pkg">PolyPatEx</span> functions need to know the characteristics
of the dataset being analysed - these are specified in the
<code>inputData</code> or <code>preprocessData</code> calls and are
invisibly attached to the allele data frame that is returned, for
use by other <span class="pkg">PolyPatEx</span> functions. The required characteristics
are:
</p>

<ul>
<li> <p><code>numLoci</code>: the number of loci in the dataset.
</p>
</li>
<li> <p><code>ploidy</code>: the ploidy <code class="reqn">p</code> of the species (currently
allowed to be 4, 6, or 8.  <code>ploidy</code> can also be 2, provided
<code>dataType="genotype"</code>).
</p>
</li>
<li> <p><code>dataType</code>: whether the data is genotypic (all <code class="reqn">p</code>
alleles at each locus are observed) or phenotypic (only the
distinct allele states at a locus are observed - alleles that
appear more than once in the genotype of a locus only appear once
in the phenotype).
</p>
</li>
<li> <p><code>dioecious</code>: whether the species is dioecious or
monoecious.
</p>
</li>
<li> <p><code>selfCompatible</code> whether a monoecious species is self
compatible (i.e., whether an individual can fertilise itself).
</p>
</li>
<li> <p><code>mothersOnly</code>: whether a dioecious dataset should
retain only adult females that are mothers of progeny in the
dataset.  If <code>dioecious=TRUE</code>, then <code>mothersOnly</code> must
be set to either <code>TRUE</code> or <code>FALSE</code>.
</p>
</li>
</ul>
<h3>Value</h3>

<p>A data frame, containing the checked and pre-processed
allele data, ready for further analysis by other <span class="pkg">PolyPatEx</span>
functions.  All columns in the output data frame will be of mode
<code>character</code>.
</p>


<h3>Author(s)</h3>

<p>Alexander Zwart (alec.zwart at csiro.au)
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 

## Obtain path to the example genotype data file
## 'FR_Genotype.csv'
gDataFile &lt;- system.file("extdata/FR_Genotype.csv",
                         package="PolyPatEx")
print(gDataFile)

gData &lt;- inputData(gDataFile,
                   numLoci=7,
                   ploidy=4,
                   dataType="genotype",
                   dioecious=TRUE,
                   mothersOnly=TRUE)

## ...or use 'mothersOnly=FALSE' if you wish to retain
## non-maternal females in the dataset.

## gData now contains the checked and preprocessed allele dataset,
## ready to be passed to other PolyPatEx analysis functions.

## In your own workflow, you would typically specify the path to
## your allele dataset directly - e.g. if the dataset
## myAlleleData.csv is on the Data subdirectory of the current R
## working directory (see R function setwd()), then:
##
## gData &lt;- inputData("Data/myAlleleData.csv",
##                    numLoci= etc etc etc...,


## End(Not run)
</code></pre>


</div>