<div class="container">

<table style="width: 100%;"><tr>
<td>LocusInfo</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Get Information about a Single Locus
</h2>

<h3>Description</h3>

<p>This function returns, and optionally prints, information about a single locus
with a <code>RADdata</code> object, including alignment position, allele
sequences, and genes overlapping the site.
</p>


<h3>Usage</h3>

<pre><code class="language-R">LocusInfo(object, ...)
## S3 method for class 'RADdata'
LocusInfo(object, locus, genome = NULL,
          annotation = NULL, verbose = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>

<p>A <code>RADdata</code> object.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>locus</code></td>
<td>

<p>A character string indicating the name of the locus to display.  Alternatively,
a character string indicating the name of an allele, for which the 
corresponding locus will be identified.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome</code></td>
<td>

<p>An optional <code>FaFile</code> or 
<code>BSgenome</code>
object containing the reference genome sequence.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotation</code></td>
<td>

<p>An optional <code>TxDb</code> object containing the genome annotation.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p>If <code>TRUE</code>, results will be printed to the console.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>Additional arguments (none implemented).
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The locus name, allele names, and allele sequences are always returned (although
allele names are not printed with <code>verbose</code>).  If the chromosome and 
position are known, those are also returned and printed.  If <code>annotation</code>
is provided, the function will return and print genes that overlap the locus.
If <code>annotation</code> and <code>genome</code> are provided, the function will attempt
to identify any amino acid changes caused by the alleles, using
<code>predictCoding</code> 
internally.  Identification of
amino acid changes will work if the <code>RADdata</code> object was created with
<code>VCF2RADdata</code> using the <code>refgenome</code> argument to fill in non-variable
sites, and/or if the alleles are only one nucleotide long.
</p>


<h3>Value</h3>

<p>A list containing:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>Locus </code></td>
<td>
<p>The name of the locus.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Chromosome </code></td>
<td>
<p>The chromosome name, if present.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Position </code></td>
<td>
<p>The position in base pairs on the chromosome, if present.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Alleles </code></td>
<td>
<p>Allele names for the locus.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Haplotypes </code></td>
<td>
<p>Allele sequences for the locus, in the same order.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Frequencies </code></td>
<td>
<p>Allele frequencies, if present, in the same order.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Transcripts </code></td>
<td>
<p>Transcripts overlapping the locus, if an annotation was
provided but it wasn't possible to predict amino acid changes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>PredictCoding </code></td>
<td>
<p>The output of <code>predictCoding</code>, if it was run.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Lindsay V. Clark
</p>


<h3>See Also</h3>

<p><code>makeTxDbFromGFF</code>, <code>GetLoci</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(exampleRAD)
exampleRAD &lt;- AddAlleleFreqHWE(exampleRAD)
loc2info &lt;- LocusInfo(exampleRAD, "loc2")
</code></pre>


</div>