<div class="container">

<table style="width: 100%;"><tr>
<td>StripDown</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Remove Unneeded Slots to Conserve Memory
</h2>

<h3>Description</h3>

<p>This function is designed to be used after a <code>RADdata</code> object
has been processed by one of the pipeline functions.  
Slots that are no longer needed are removed in order to conserve memory.
</p>


<h3>Usage</h3>

<pre><code class="language-R">StripDown(object, ...)
## S3 method for class 'RADdata'
StripDown(object, 
          remove.slots = c("depthSamplingPermutations",
                           "depthRatio", "antiAlleleDepth",
                           "genotypeLikelihood", "priorProb",
                           "priorProbLD"),
          ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>

<p>A <code>RADdata</code> object.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remove.slots</code></td>
<td>

<p>A character vector listing slots that will be removed.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>Additional arguments (none implemented).
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The default slots that are removed take up a lot of memory but are not used
by the export functions.  Other slots to consider removing are 
<code>alleleFreq</code>, <code>alleleFreqByTaxa</code>, <code>PCA</code>, <code>locDepth</code>, 
<code>alleleDepth</code>, and <code>alleleLinkages</code>.  Of course, if you
have custom uses for some of the slots that are removed by default, you can 
change the <code>remove.slots</code> vector to not include them.
</p>
<p>The function will throw an error if the user attempts to remove key slots that
are needed for export and downstream analysis, including:
</p>

<ul>
<li>
<p> alleles2loc
</p>
</li>
<li>
<p> alleleNucleotides
</p>
</li>
<li>
<p> locTable
</p>
</li>
<li>
<p> possiblePloidies
</p>
</li>
<li>
<p> ploidyChiSq
</p>
</li>
<li>
<p> posteriorProb
</p>
</li>
</ul>
<h3>Value</h3>

<p>A <code>RADdata</code> object 
</p>


<h3>Author(s)</h3>

<p>Lindsay V. Clark
</p>


<h3>See Also</h3>

<p><code>SubsetByTaxon</code>, <code>SubsetByLocus</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># load a dataset for this example
data(exampleRAD)

# run a pipeline
exampleRAD &lt;- IterateHWE(exampleRAD)

# check the size of the resulting object
object.size(exampleRAD)

# remove unneeded slots
exampleRAD &lt;- StripDown(exampleRAD)

# check object size again
object.size(exampleRAD)
</code></pre>


</div>