<div class="container">

<table style="width: 100%;"><tr>
<td>read.SATlotyper.output</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>read the haplotyping results from the SATlotyper output</h2>

<h3>Description</h3>

<p>read the haplotyping results from the SATlotyper output
</p>


<h3>Usage</h3>

<pre><code class="language-R">read.SATlotyper.output(fname, output="", allelecodes=c("A", "B"),
sep, haploblockname="")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>fname</code></td>
<td>
<p>filename of an xml file produced by SATlotyper</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output</code></td>
<td>
<p>character vector, console output of SATlotyper;
this contains a table of calculated haplotypes with more info
than the xml file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allelecodes</code></td>
<td>
<p>the codes used for the ref and alt SNP alleles in
make.SATlotyper.input, default "A" and "B"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep</code></td>
<td>
<p>the separator used in make.SATlotyper.input (SATlotyper cleverly
identifies this separator and uses it for its output), default tab</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>haploblockname</code></td>
<td>
<p>if not "" (default) the haplotype IDs are given as
&lt;haploblockname&gt;_&lt;hapnr&gt;, else just as &lt;hapnr&gt;, left_padded with zeroes</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The xml file is parsed using the package XML. The resulting list has
3 items named source, bootstrapping and haplotyping. In this function source is
ignored and the results are obtained from haplotypings[[1]] (i.e. even if multiple
haplotypings were done only the first is extracted), with the haplotype
score from the bootstrapping item added to the haplotype.info.
</p>


<h3>Value</h3>

<p>a list with 3 items: <br>
$hapdos is a matrix with individuals in columns and haplotypes in rows,
giving the dosages of the haplotypes in each individual (summing to ploidy).
This is the same format as the hapdos components of the inferHaplotypes
results of PolyHaplotyper <br>
$haplotype.info is a data.frame that is a combination of information from two
or three tables:<br>
columns Haplotype and necessity are from element
haplotypings[[1]]$haplotypes in the xml file, <br>
column hapnr gives the PolyHaplotyper equivalents of the Haplotypes, <br>
column score is from element bootstrapping in the xml file, <br>
columns id, number, frequency, homozygous, necessary, distance and
neighbours are from (the console) output if present. <br>
The haplotype.info data.frame is ordered by hapnr. <br>
$HaplotypingScore is the single number in $haplotyping[[1]]$score.
</p>


</div>