<div class="container">

<table style="width: 100%;"><tr>
<td>MergeTaxaDepth</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Combine Read Depths from Multiple Taxa into One Taxon
</h2>

<h3>Description</h3>

<p>This function should be used in situations where data that were imported as
separate taxa should be merged into a single taxon.  The function should be
used before any of the pipeline functions for genotype calling.  Read depths
are summed across duplicate taxa and output as a single taxon.
</p>


<h3>Usage</h3>

<pre><code class="language-R">MergeTaxaDepth(object, ...)

## S3 method for class 'RADdata'
MergeTaxaDepth(object, taxa, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>

<p>A <code>RADdata</code> object.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>taxa</code></td>
<td>

<p>A character vector indicating taxa to be merged.  The first taxon in the vector
will be used to name the combined taxon in the output.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>Additional arguments (none implemented).
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Examples of reasons to use this function:
</p>

<ul>
<li>
<p> Duplicate samples across different libraries were given different
names so that preliminary analysis could confirm that they were truly the same
(<em>i.e.</em> no mix-ups) before combining them.
</p>
</li>
<li>
<p> Typos in the key file for the SNP mining software (TASSEL, Stacks, etc.)
caused duplicate samples to have different names when they really should have
had the same name.
</p>
</li>
</ul>
<p>To merge multiple sets of taxa into multiple combined taxa, this function can
be run multiple times or in a loop.
</p>


<h3>Value</h3>

<p>A <code>RADdata</code> object derived from <code>object</code>.  The <code>alleleDepth</code>,
<code>antiAlleleDepth</code>, <code>locDepth</code>, <code>depthRatio</code>, and
<code>depthSamplingPermutation</code> slots, and <code>"taxa"</code> and <code>"nTaxa"</code>
attributes, have been changed accordingly to reflect the merge.
</p>


<h3>Author(s)</h3>

<p>Lindsay V. Clark
</p>


<h3>See Also</h3>

<p><code>SubsetByTaxon</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># dataset for this example
data(exampleRAD)

# merge the first three taxa into one
exampleRADm &lt;- MergeTaxaDepth(exampleRAD, c("sample001", "sample002", "sample003"))

# inspect read depth
exampleRAD$alleleDepth[1:3,]
exampleRADm$alleleDepth[1:3,]
</code></pre>


</div>