<div class="container">

<table style="width: 100%;"><tr>
<td>test.segRatio</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Classic tests for assessing marker dosage in autopolyploids</h2>

<h3>Description</h3>

<p>Perform chi–squared tests or binomial CIs to obtain expected marker dosage
in autopolyploids
</p>


<h3>Usage</h3>

<pre><code class="language-R">test.segRatio(seg.ratio, ploidy.level = 4,
  type.parents = c("heterogeneous", "homozygous"),
  method = c("chi.squared", "binomial"), alpha = 0.05, expected.ratio)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seg.ratio</code></td>
<td>
<p> object of class <code>segRatio</code> containing
segregation proportions</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ploidy.level</code></td>
<td>
<p>the number of homologous chromosomes, either as
numeric or as a character string</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type.parents</code></td>
<td>
<p> "heterogeneous" if parental markers are 0,1 or
"homozygous" if parental markers are both 1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p> specify which method ‘chi.squared’ or
‘binomial’ </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p> significance level for tests/CIs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>expected.ratio</code></td>
<td>
<p> vector of expected segregation proportions
Default: determined by using function <code>expected.segRatio</code> given
the <code>ploidy.level</code>
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns object of class <code>testSegRatio</code> with components
</p>
<table>
<tr style="vertical-align: top;">
<td><code>probability</code></td>
<td>
<p>matrix of probabilities under the test for each
dosage where columns are doses  and rows are markers</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dosage</code></td>
<td>
<p> vector of allocated dosages where allocation unique
otherwise <code>NA</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allocated</code></td>
<td>
<p> matrix of 0's and 1's where 1 indicates dosage
allocation where columns are doses  and rows are markers</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>alpha level for significance test or CI construction</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>expected.ratios</code></td>
<td>
<p>expected segregation ratios under null hypotheses</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>call</code></td>
<td>
<p>call to test.segRatio</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Peter Baker <a href="mailto:p.baker1@uq.edu.au">p.baker1@uq.edu.au</a></p>


<h3>References</h3>


<ul>
<li>
<p>K Mather(1951) The measurement of linkage in heredity.
<em>Methuen</em> London
</p>
</li>
<li>
<p>Ripol, M I et al(1999) Statistical aspects of genetic mapping in
autopolyploids. <em>Gene</em> <b>235</b> 31–41 
</p>
</li>
</ul>
<h3>See Also</h3>

 <p><code>segregationRatios</code> for computing segregation
ratios and <code>segRatio</code>, <code>expected.segRatio</code></p>


<h3>Examples</h3>

<pre><code class="language-R">
  ## simulated data
  a &lt;- sim.autoMarkers(ploidy = 8, c(0.7,0.2,0.09,0.01))
  print(a)

  ## summarise chi-squared test vs true
  ac &lt;- test.segRatio(a$seg.ratios, ploidy=8, method="chi.squared")
  print(addmargins(table(a$true.doses$dosage, ac$dosage, exclude=NULL)))

  ## summarise binomial CI vs true
  ab &lt;- test.segRatio(a$seg.ratios, ploidy=8, method="bin")
  print(addmargins(table(a$true.doses$dosage, ab$dosage, exclude=NULL)))
</code></pre>


</div>