<div class="container">

<table style="width: 100%;"><tr>
<td>freq.to.genpop</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Convert Allele Frequencies for Adegenet
</h2>

<h3>Description</h3>

<p>Given a data frame of allele frequencies such as that produced by
<code>simpleFreq</code> or <code>deSilvaFreq</code>, <code>freq.to.genpop</code> creates a
data frame of allele counts that can be read by the <code>as.genpop</code>
function in the package <span class="pkg">adegenet</span>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">freq.to.genpop(freqs, pops = row.names(freqs),
               loci =
                unique(as.matrix(as.data.frame(strsplit(names(freqs),
                split = ".", fixed = TRUE), stringsAsFactors = FALSE))[1, ]))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>freqs</code></td>
<td>

<p>A data frame of allele frequencies.  Row names are population names.
The first column is called <code>"Genomes"</code> and indicates the size of
each population in terms of number of haploid genomes.  All other column
names are the locus and allele separated by a period.  These columns
contain the frequencies of each allele in each population.  For each
locus and population, all frequencies should total to 1.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pops</code></td>
<td>

<p>An optional character vector indicating the names of populations to use.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loci</code></td>
<td>

<p>An optional character vector indicating the names of loci to use.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><span class="pkg">adegenet</span> expects one ploidy for the entire dataset.  Therefore, data
frames of allele frequencies with multiple “Genomes” columns,
such as those produced when ploidy varies by locus, are not allowed as
the <code>freqs</code> argument.
</p>


<h3>Value</h3>

<p>A data frame with row and column names identical to those in
<code>freqs</code>, minus the <code>"Genomes"</code> column and any columns for loci
not included in <code>loci</code>.  Allele frequencies are converted to counts
by multiplying by the values in the <code>"Genomes"</code> column and rounding
to the nearest integer.
</p>


<h3>Author(s)</h3>

<p>Lindsay V. Clark
</p>


<h3>References</h3>

<p>Jombart, T.  (2008)  adegenet: a R package for the multivariate analysis
of genetic markers.  <em>Bioinformatics</em> <b>24</b>, 1403-1405.
</p>


<h3>See Also</h3>

<p><code>simpleFreq</code>, <code>deSilvaFreq</code>,
<code>write.freq.SPAGeDi</code>, <code>gendata.to.genind</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># create a simple allele frequency table
# (usually done with simpleFreq or deSilvaFreq)
myfreq &lt;- data.frame(row.names=c("popA","popB"), Genomes=c(120,100),
                     locG.152=c(0.1,0.4), locG.156=c(0.5, 0.3),
                     locG.160=c(0.4, 0.3), locK.179=c(0.15, 0.25),
                     locK.181=c(0.35, 0.6), locK.183=c(0.5, 0.15))
myfreq

# convert to adegenet format
gpfreq &lt;- freq.to.genpop(myfreq)
gpfreq

## Not run: 
# If you have adegenet installed, you can now make this into a
# genpop object.
require(adegenet)
mygenpop &lt;- genpop(gpfreq, ploidy=as.integer(4), type="codom")

# examine the object that has been created
mygenpop
popNames(mygenpop)
mygenpop@tab
mygenpop@all.names

# Perform a distance calculation with the object
dist.genpop(mygenpop)

## End(Not run)
</code></pre>


</div>