<div class="container">

<table style="width: 100%;"><tr>
<td>genobaypass2pooldata</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Convert BayPass read count and haploid pool size input files into a pooldata object</h2>

<h3>Description</h3>

<p>Convert BayPass read count and haploid pool size input files into a pooldata object
</p>


<h3>Usage</h3>

<pre><code class="language-R">genobaypass2pooldata(
  genobaypass.file = "",
  poolsize.file = "",
  snp.pos = NA,
  poolnames = NA,
  min.cov.per.pool = -1,
  max.cov.per.pool = 1e+06,
  min.maf = -1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genobaypass.file</code></td>
<td>
<p>The name (or a path) of the BayPass read count file (see the BayPass manual <a href="https://forgemia.inra.fr/mathieu.gautier/baypass_public/">https://forgemia.inra.fr/mathieu.gautier/baypass_public/</a>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>poolsize.file</code></td>
<td>
<p>The name (or a path) of the BayPass (haploid) pool size file (see the BayPass manual <a href="https://forgemia.inra.fr/mathieu.gautier/baypass_public/">https://forgemia.inra.fr/mathieu.gautier/baypass_public/</a>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp.pos</code></td>
<td>
<p>An optional two column matrix with nsnps rows containing the chromosome (or contig/scaffold) of origin and the position of each markers</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>poolnames</code></td>
<td>
<p>A character vector with the names of pool</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.cov.per.pool</code></td>
<td>
<p>Minimal allowed read count (per pool). If at least one pool is not covered by at least min.cov.perpool reads, the position is discarded</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.cov.per.pool</code></td>
<td>
<p>Maximal allowed read count (per pool). If at least one pool is covered by more than min.cov.perpool reads, the position is discarded</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.maf</code></td>
<td>
<p>Minimal allowed Minor Allele Frequency (computed from the ratio overall read counts for the reference allele over the read coverage)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>If TRUE extra information is printed on the terminal</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Information on SNP position is only required for some graphical display or to carried out block-jacknife sampling estimation of confidence intervals. If no mapping information is given (default), SNPs will be assumed to be ordered on the same chromosome and separated by 1 bp. As blocks are defined with a number of consecutive SNPs (rather than a length), the latter assumption has actually no effect (except in the reported estimated block sizes in Mb).
</p>


<h3>Value</h3>

<p>A pooldata object containing 7 elements:
</p>

<ol>
<li>
<p> "refallele.readcount": a matrix with nsnp rows and npools columns containing read counts for the reference allele (chosen arbitrarily) in each pool
</p>
</li>
<li>
<p> "readcoverage": a matrix with nsnp rows and npools columns containing read coverage in each pool
</p>
</li>
<li>
<p> "snp.info": a matrix with nsnp rows and four columns containing respectively the contig (or chromosome) name (1st column) and position (2nd column) of the SNP; the allele taken as reference in the refallele.readcount matrix (3rd column); and the alternative allele (4th column)
</p>
</li>
<li>
<p> "poolsizes": a vector of length npools containing the haploid pool sizes
</p>
</li>
<li>
<p> "poolnames": a vector of length npools containing the names of the pools
</p>
</li>
<li>
<p> "nsnp": a scalar corresponding to the number of SNPs
</p>
</li>
<li>
<p> "npools": a scalar corresponding to the number of pools
</p>
</li>
</ol>
<h3>Examples</h3>

<pre><code class="language-R"> make.example.files(writing.dir=tempdir())
 pooldata=popsync2pooldata(sync.file=paste0(tempdir(),"/ex.sync.gz"),poolsizes=rep(50,15))
 pooldata2genobaypass(pooldata=pooldata,writing.dir=tempdir())
 pooldata=genobaypass2pooldata(genobaypass.file=paste0(tempdir(),"/genobaypass"),
                               poolsize.file=paste0(tempdir(),"/poolsize"))
</code></pre>


</div>