<div class="container">

<table style="width: 100%;"><tr>
<td>pairwisefst-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>S4 class to represent a pairwise Fst results obtained with the compute.pairwiseFST</h2>

<h3>Description</h3>

<p>S4 class to represent a pairwise Fst results obtained with the compute.pairwiseFST
</p>


<h3>Slots</h3>


<dl>
<dt><code>values</code></dt>
<dd>
<p>A data frame with npop*(npop-1)/2 rows and 3 (or 7 if blockjackknife is TRUE) columns containing for both the Fst and Q2, estimates over all the SNPs and if blockjackknife=TRUE, the estimated block-jackknife and standard error (s.e.). The seventh (or third if blockjackknife=FALSE) column gives the number of SNPs.</p>
</dd>
<dt><code>PairwiseFSTmatrix</code></dt>
<dd>
<p>A npxnp matrix containing the pairwise FST estimates</p>
</dd>
<dt><code>PairwiseSnpFST</code></dt>
<dd>
<p>A matrix (nsnp rows and npops columns) with read count data for the reference allele</p>
</dd>
<dt><code>PairwiseSnpQ1</code></dt>
<dd>
<p>A matrix (nsnp rows and npops columns) with overall read coverage</p>
</dd>
<dt><code>PairwiseSnpQ2</code></dt>
<dd>
<p>A matrix (nsnp rows and 4 columns) detailing for each SNP, the chromosome (or scaffold), the position, allele 1 and allele 2</p>
</dd>
<dt><code>blockjacknife</code></dt>
<dd>
<p>A logical indicating whether block-jackknife estimates of standard errors are available (TRUE) or not (FALSE)</p>
</dd>
</dl>
<h3>See Also</h3>

<p>To generate pairwise object, see <code>compute.pairwiseFST</code>
</p>


</div>