<div class="container">

<table style="width: 100%;"><tr>
<td>compute_H1</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>compute_H1</h2>

<h3>Description</h3>

<p>Compute (uncorrected) 1-Q1 for each block-jackknife block (if any) and over all the SNPs (i.e., either within or outside blocks)
</p>


<h3>Usage</h3>

<pre><code class="language-R">.compute_H1(refcount, totcount, nblocks, block_id, verbose)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>refcount</code></td>
<td>
<p>Matrix of nsnpxnpop with counts (genotype or reads) for the reference allele</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>totcount</code></td>
<td>
<p>Matrix of nsnpxnpop with total counts or read coverages</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nblocks</code></td>
<td>
<p>Integer giving the number of block-jackknife blocs (may be 0 if no block-jackknife)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>block_id</code></td>
<td>
<p>Integer vector of length nsnps with the (0-indexed) id of the block to which each SNP belongs (-1 for SNPs outside blocks)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Logical (if TRUE progression bar is printed on the terminal)</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Compute all the (uncorrected) H1=1-Q1 for each block-jackknife block (if any) and overall SNPs (within or outside blocks). 
It is indeed more convenient to compute H1 (rather than Q1) to apply corrections afterwards within R function
</p>


<h3>Value</h3>

<p>Return a matrix with npops rows and nblocks+1 column giving the mean H1 of each pop within each block and for all SNPs (last column)
</p>


<h3>Examples</h3>

<pre><code class="language-R">#
</code></pre>


</div>