<div class="container">

<table style="width: 100%;"><tr>
<td>numberReference</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute the number of reference reads at multiple loci</h2>

<h3>Description</h3>

<p>This function computes the number of reference reads over multiple loci and
for a single population.
</p>


<h3>Usage</h3>

<pre><code class="language-R">numberReference(genotypes, indContribution, error)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genotypes</code></td>
<td>
<p>is a list, where each entry corresponds to a different
locus. Each entry should be a matrix containing the genotypes (coded as 0,
1 or 2). Each column of that matrix should be a different site and each row
a different individual.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>indContribution</code></td>
<td>
<p>either a list or a matrix (that the function will
convert to a list). Each list entry should be a matrix of individual
contributions.  Each row of that matrix is a different individual and each
column is a different site. Thus, each entry of the matrix should contain
the number of reads contributed by that individual at that particular site.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>error</code></td>
<td>
<p>a numeric value with error rate associated with the sequencing
and mapping process. This error rate is assumed to be symmetric:
error(reference -&gt; alternative) = error(alternative -&gt; reference). This
number should be between 0 and 1.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Note that this function will also work on a single locus, provided that the
input is in the list format.
</p>


<h3>Value</h3>

<p>a list with one entry per locus. Each of those entries is a matrix
with the number of reference allele reads contributed by each individual.
Each matrix row represents a different individual and each column is a
different site.
</p>


</div>