<div class="container">

<table style="width: 100%;"><tr>
<td>Pfreqs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute allele frequencies from pooled sequencing data</h2>

<h3>Description</h3>

<p>Computes the frequency of the alternative allele in Pool-seq data and removes
any site with too few minor-allele reads from both the pool frequencies and
the frequencies computed directly from genotypes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Pfreqs(reference, alternative, coverage, min.minor, ifreqs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>reference</code></td>
<td>
<p>a matrix with the number of reference allele reads. Each row
should be a different population and each column a different site.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alternative</code></td>
<td>
<p>a matrix with the number of alternative allele reads. Each
row should be a different population and each column a different site.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coverage</code></td>
<td>
<p>a matrix with the total coverage. Each row should be a
different population and each column a different site.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.minor</code></td>
<td>
<p>is an integer representing the minimum allowed number of
minor-allele reads. Sites that, across all populations, have less
minor-allele reads than this threshold will be removed from the data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ifreqs</code></td>
<td>
<p>a vector of allele frequencies computed directly from the
genotypes where each entry corresponds to a different site.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The frequency at a given SNP is calculated according to: <code>pi = c/r</code>, where c
= number of alternative allele reads and r = total number of observed reads.
Additionally, if a site has less minor-allele reads than <code>min.minor</code>
across all populations, that site is removed from the data.
</p>


<h3>Value</h3>

<p>a list with two entries. The <code>ifreqs</code> entry contains the allele
frequencies computed directly from genotypes and <code>pfreqs</code> the allele
frequencies computed from pooled sequencing data.
</p>


<h3>Examples</h3>

<pre><code class="language-R">set.seed(10)
# create a vector of allele frequencies
freqs &lt;- runif(20)
set.seed(10)
# create a matrix with the number of reads with the alternative allele
alternative &lt;- matrix(sample(x = c(0,5,10), size = 20, replace = TRUE), nrow = 1)
# create a matrix with the depth of coverage
coverage &lt;- matrix(sample(100:150, size = 20), nrow = 1)
# the number of reads with the reference allele is obtained by subtracting
# the number of alternative allele reads from the depth of coverage
reference &lt;- coverage - alternative
# compute allele frequencies from pooled sequencing data
Pfreqs(reference = reference, alternative = alternative, coverage = coverage,
min.minor = 2, ifreqs = freqs)

</code></pre>


</div>