<div class="container">

<table style="width: 100%;"><tr>
<td>remove_by_reads</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Apply a coverage-based filter over a list</h2>

<h3>Description</h3>

<p>This function removes sites that have a coverage below a <code>minimum</code> value and
sites with a coverage above a <code>maximum</code> value. This is done over multiple
loci, assuming that each entry of the <code>reads</code> list is a different locus. If a
list of genotypes is also supplied, then those same sites are also removed
from each locus of the genotypes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">remove_by_reads(nLoci, reads, minimum, maximum, genotypes = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nLoci</code></td>
<td>
<p>an integer that represents how many independent loci were
simulated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reads</code></td>
<td>
<p>a list with the total depth of coverage. Each entry of the list
should be a matrix corresponding to a different locus. Each row of that
matrix should be the coverage of a different population and each column a
different site.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minimum</code></td>
<td>
<p>an integer representing the minimum coverage allowed. Sites
where any population has a depth of coverage below this threshold are
removed from the data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maximum</code></td>
<td>
<p>an integer representing the maximum coverage allowed. Sites
where any population has a depth of coverage above this threshold are
removed from the data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotypes</code></td>
<td>
<p>an optional list input with the genotypes. Each entry of the
list should be a matrix corresponding to a different locus. Each column of
the matrix should be a different site and each row a different individual.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a list with the total depth of coverage similar to the <code>reads</code> input
argument but without sites where the coverage was below the <code>minimum</code> or
above the <code>maximum</code>. If the genotypes were included, a second list entry
will also be included in the output, containing the genotypes minus the
sites that were removed.
</p>


<h3>Examples</h3>

<pre><code class="language-R">set.seed(10)

# simulate coverage for 10 locus
reads &lt;- simulateCoverage(mean = c(25, 25), variance = c(200, 200), nSNPs = 10, nLoci = 10)

# remove sites with coverage below 10x or above 100x
reads &lt;- remove_by_reads(nLoci = 10, reads = reads, minimum = 5, maximum = 100)
# notice that some locus no longer have 10 SNPs - those sites were removed
reads

</code></pre>


</div>