<div class="container">

<table style="width: 100%;"><tr>
<td>vcfinfo</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create vcf table with relevant information</h2>

<h3>Description</h3>

<p>Creates a data frame in the VCF format for all SNPs and across all loci in
the data set.
</p>


<h3>Usage</h3>

<pre><code class="language-R">vcfinfo(string, pos = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>string</code></td>
<td>
<p>is a character vector or a list where each entry contains a
character vector for a different locus. Each entry of this character vector
contains the information for a single SNP coded as R,A:DP. The output of
the <code>vcflocus</code> or <code>vcfloci</code> is the intended input
here.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos</code></td>
<td>
<p>is an optional input (default is NULL). If the actual position of
the SNPs are known, they can be used as input here. When working with a
single locus, this should be a numeric vector with each entry corresponding
to the position of each SNP. If the data has multiple loci, this should be
a list where each entry is a numeric vector with the position of the SNPs
for a different locus.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function combines the information coded as R,A:DP with other necessary
information such as the chromosome of each SNP, the position of the SNP and
the quality of the genotype among others. Note that in the character string,
R is the number of reads of the reference allele, A is the number of reads of
the alternative allele and DP is the total depth of coverage. Each row of the
data frame corresponds to a different SNP.
</p>


<h3>Value</h3>

<p>a data frame with 10 different columns
</p>
<table>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>Chromosome. Each locus is treated as different linkage group.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos</code></td>
<td>
<p>Co-ordinate. The coordinate of the SNP.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ID</code></td>
<td>
<p>Identifier.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>REF</code></td>
<td>
<p>Reference allele. We assume that the reference allele is always
an A. Note that this is not necessarily the major allele.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ALT</code></td>
<td>
<p>Alternative allele. We assume that the alternative allele is always
a T.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>QUAL</code></td>
<td>
<p>Quality score out of 100. We assume that this score is always
100.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FILTER</code></td>
<td>
<p>If this SNP passed quality filters.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>INFO</code></td>
<td>
<p>Further information. Provides further information on the
variants.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FORMAT</code></td>
<td>
<p>Information about the following columns. This column tells us
how the number of reads is coded in the next column.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pop1</code></td>
<td>
<p>Number of reference-allele reads, alternative-allele reads and
total depth of coverage observed for this population at this SNP.</p>
</td>
</tr>
</table>
</div>