<div class="container">

<table style="width: 100%;"><tr>
<td>compute_Q2</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>compute_Q2</h2>

<h3>Description</h3>

<p>Compute all Q2 for each block-jackknife block (if any) and overall SNPs (within or outside blocks)
</p>


<h3>Usage</h3>

<pre><code class="language-R">.compute_Q2(refcount, totcount, nblocks, block_id, verbose)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>refcount</code></td>
<td>
<p>Matrix of nsnpxnpop with counts (genotype or reads) for the reference allele</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>totcount</code></td>
<td>
<p>Matrix of nsnpxnpop with total counts or read coverages</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nblocks</code></td>
<td>
<p>Integer giving the number of block-jackknife blocs (may be 0 if no block-jackknife)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>block_id</code></td>
<td>
<p>Integer vector of length nsnps with the (0-indexed) id of the block to which each SNP belongs (-1 for SNPs outside blocks)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Logical (if TRUE progression bar is printed on the terminal)</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Compute all Q2 for each block-jackknife block (if any) and overall SNPs (within or outside blocks).
</p>


<h3>Value</h3>

<p>Return a matrix with npops*(npops-1)/2 and nblocks+1 column giving the mean Q2 of each pairwise pop comp. within each block and for all SNPs (last column)
</p>


<h3>Examples</h3>

<pre><code class="language-R">#
</code></pre>


</div>