<div class="container">

<table style="width: 100%;"><tr>
<td>popsync2pooldata</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Convert Popoolation Sync files into a pooldata object</h2>

<h3>Description</h3>

<p>Convert Popoolation Sync files into a pooldata object
</p>


<h3>Usage</h3>

<pre><code class="language-R">popsync2pooldata(
  sync.file = "",
  poolsizes = NA,
  poolnames = NA,
  min.rc = 1,
  min.cov.per.pool = -1,
  max.cov.per.pool = 1e+06,
  min.maf = 0.01,
  noindel = TRUE,
  nlines.per.readblock = 1e+06,
  nthreads = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sync.file</code></td>
<td>
<p>The name (or a path) of the Popoolation sync file (might be in compressed format)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>poolsizes</code></td>
<td>
<p>A numeric vector with haploid pool sizes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>poolnames</code></td>
<td>
<p>A character vector with the names of pool</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.rc</code></td>
<td>
<p>Minimal allowed read count per base. Bases covered by less than min.rc reads are discarded and considered as sequencing error. For instance, if nucleotides A, C, G and T are covered by respectively 100, 15, 0 and 1 over all the pools, setting min.rc to 0 will lead to discard the position (the polymorphism being considered as tri-allelic), while setting min.rc to 1 (or 2, 3..14) will make the position be considered as a SNP with two alleles A and C (the only read for allele T being disregarded).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.cov.per.pool</code></td>
<td>
<p>Minimal allowed read count (per pool). If at least one pool is not covered by at least min.cov.perpool reads, the position is discarded</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.cov.per.pool</code></td>
<td>
<p>Maximal allowed read count (per pool). If at least one pool is covered by more than min.cov.perpool reads, the position is discarded</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.maf</code></td>
<td>
<p>Minimal allowed Minor Allele Frequency (computed from the ratio overal read counts for the reference allele over the read coverage)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>noindel</code></td>
<td>
<p>If TRUE, positions with at least one indel count are discarded</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nlines.per.readblock</code></td>
<td>
<p>Number of Lines read simultaneously. Should be adapted to the available RAM.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nthreads</code></td>
<td>
<p>Number of available threads for parallelization of some part of the parsing (default=1, i.e., no parallelization)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A pooldata object containing 7 elements:
</p>

<ol>
<li>
<p> "refallele.readcount": a matrix with nsnp rows and npools columns containing read counts for the reference allele (chosen arbitrarily) in each pool
</p>
</li>
<li>
<p> "readcoverage": a matrix with nsnp rows and npools columns containing read coverage in each pool
</p>
</li>
<li>
<p> "snp.info": a matrix with nsnp rows and four columns containing respectively the contig (or chromosome) name (1st column) and position (2nd column) of the SNP; the allele taken as reference in the refallele.readcount matrix (3rd column); and the alternative allele (4th column)
</p>
</li>
<li>
<p> "poolsizes": a vector of length npools containing the haploid pool sizes
</p>
</li>
<li>
<p> "poolnames": a vector of length npools containing the names of the pools
</p>
</li>
<li>
<p> "nsnp": a scalar corresponding to the number of SNPs
</p>
</li>
<li>
<p> "npools": a scalar corresponding to the number of pools
</p>
</li>
</ol>
<h3>Examples</h3>

<pre><code class="language-R"> make.example.files(writing.dir=tempdir())
 pooldata=popsync2pooldata(sync.file=paste0(tempdir(),"/ex.sync.gz"),poolsizes=rep(50,15))
</code></pre>


</div>