<div class="container">

<table style="width: 100%;"><tr>
<td>poolPops</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create Pooled DNA sequencing data for multiple populations</h2>

<h3>Description</h3>

<p>This function combines the information for each individual of each population
into information at the population level.
</p>


<h3>Usage</h3>

<pre><code class="language-R">poolPops(nPops, nLoci, indContribution, readsReference)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nPops</code></td>
<td>
<p>An integer representing the total number of populations in the
dataset.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nLoci</code></td>
<td>
<p>An integer that represents the total number of independent loci
in the dataset.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>indContribution</code></td>
<td>
<p>Either a list or a matrix (when dealing with a single
locus).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>readsReference</code></td>
<td>
<p>A list, where each entry contains the information for a
single locus. Each list entry should then have one separate entry per
population. Each of these entries should be a matrix, with each row
corresponding to a single individual and each column a different site.
Thus, each entry of the matrix contains the number of observed reads with
the reference allele for that individual at a given site. The output of the
<code>numberReference</code> or <code>numberReferencePop</code> functions should be the input
here.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>In other words, the information of all individuals in a given population is
combined into a single population value and this is done for the various
populations. In this situation, each entry of the <code>indContribution</code> and
<code>readsReference</code> lists should contain one entry per population - being, in
essence, a list within a list. Please note that this function is intended to
work for multiple populations and should not be used with a single
population.
</p>


<h3>Value</h3>

<p>a list with three names entries
</p>
<table>
<tr style="vertical-align: top;">
<td><code>reference</code></td>
<td>
<p>a list with one entry per locus. Each entry is a matrix
with the number of reference allele reads for each population. Each column
represents a different site and each row a different population.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alternative</code></td>
<td>
<p>a list with one entry per locus. Each entry is a matrix
with the number of alternative allele reads for each population. Each
column represents a different site and each row a different population.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>total</code></td>
<td>
<p>a list with one entry per locus. Each entry is a matrix with
the coverage of each population. Each column represents a different site
and each row a different population.</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R"># simulate coverage at 5 SNPs for two populations, assuming 20x mean coverage
reads &lt;- simulateCoverage(mean = c(20, 20), variance = c(100, 100), nSNPs = 5, nLoci = 1)

# simulate the number of reads contributed by each individual
# for each population there are two pools, each with 5 individuals
indContribution &lt;- popsReads(list_np = rep(list(rep(5, 2)), 2), coverage = reads, pError = 5)

# set seed and create a random matrix of genotypes for the 20 individuals - 10 per population
set.seed(10)
genotypes &lt;- matrix(rpois(100, 0.5), nrow = 20)

# simulate the number of reference reads for the two populations
readsReference &lt;- numberReferencePop(genotypes = genotypes, indContribution = indContribution,
size = rep(list(rep(5, 2)), 2), error = 0.01)

# create Pooled DNA sequencing data for these two populations and for a single locus
poolPops(nPops = 2, nLoci = 1, indContribution = indContribution, readsReference = readsReference)

</code></pre>


</div>