<div class="container">

<table style="width: 100%;"><tr>
<td>maePool</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Average absolute difference between allele frequencies</h2>

<h3>Description</h3>

<p>Calculates the average absolute difference between the allele frequencies
computed directly from genotypes and from pooled sequencing data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">maePool(
  nDip,
  nloci,
  pools,
  pError,
  sError,
  mCov,
  vCov,
  min.minor,
  minimum = NA,
  maximum = NA,
  theta = 10
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nDip</code></td>
<td>
<p>an integer representing the total number of diploid individuals
to simulate. Note that <code>scrm::scrm()</code> actually simulates haplotypes, so the
number of simulated haplotypes is double of this.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nloci</code></td>
<td>
<p>is an integer that represents how many independent loci should
be simulated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pools</code></td>
<td>
<p>a list with a vector containing the size (in number of diploid
individuals) of each pool. Thus, if a population was sequenced using a
single pool, the vector should contain only one entry. If a population was
sequenced using two pools, each with 10 individuals, this vector should
contain two entries and both will be 10.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pError</code></td>
<td>
<p>an integer representing the value of the error associated with
DNA pooling. This value is related with the unequal contribution of both
individuals and pools towards the total number of reads observed for a
given population - the higher the value the more unequal are the individual
and pool contributions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sError</code></td>
<td>
<p>a numeric value with error rate associated with the sequencing
and mapping process. This error rate is assumed to be symmetric:
error(reference -&gt; alternative) = error(alternative -&gt; reference). This
number should be between 0 and 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mCov</code></td>
<td>
<p>an integer that defines the mean depth of coverage to simulate.
Please note that this represents the mean coverage across all sites.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vCov</code></td>
<td>
<p>an integer that defines the variance of the depth of coverage
across all sites.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.minor</code></td>
<td>
<p>is an integer representing the minimum allowed number of
minor-allele reads. Sites that, across all populations, have less
minor-allele reads than this threshold will be removed from the data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minimum</code></td>
<td>
<p>an optional integer representing the minimum coverage allowed.
Sites where the population has a depth of coverage below this threshold are
removed from the data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maximum</code></td>
<td>
<p>an optional integer representing the maximum coverage allowed.
Sites where the population has a depth of coverage above this threshold are
removed from the data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>theta</code></td>
<td>
<p>a value for the mutation rate assuming theta = 4Nu, where u is
the neutral mutation rate per locus.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Different combinations of parameters can be tested to check the effect of the
various parameters. The average absolute difference is computed with the
mae function, assuming the frequencies computed directly from
the genotypes as the <code>actual</code> input argument and the frequencies from
pooled data as the <code>predicted</code> input argument.
</p>


<h3>Value</h3>

<p>a data.frame with columns detailing the number of diploid
individuals, the pool error, the number of pools, the number of individuals
per pool, the mean coverage, the variance of the coverage and the average
absolute difference between the frequencies computed from genotypes and
from pooled data.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># single population sequenced with a single pool of 100 individuals
maePool(nDip = 100, nloci = 10, pools = list(100), pError = 100, sError = 0.01,
mCov = 100, vCov = 250, min.minor = 2)

# single population sequenced with two pools, each with 50 individuals
maePool(nDip = 100, nloci = 10, pools = list(c(50, 50)), pError = 100, sError = 0.01,
mCov = 100, vCov = 250, min.minor = 2)

# single population sequenced with two pools, each with 50 individuals
# removing sites with coverage below 10x or above 180x
maePool(nDip = 100, nloci = 10, pools = list(c(50, 50)), pError = 100, sError = 0.01,
mCov = 100, vCov = 250, min.minor = 2, minimum = 10, maximum = 180)

</code></pre>


</div>