<div class="container">

<table style="width: 100%;"><tr>
<td>compute.pairwiseFST</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute pairwise population population FST matrix (and possibly all pairwise SNP-specific FST)</h2>

<h3>Description</h3>

<p>Compute pairwise population population FST matrix (and possibly all pairwise SNP-specific FST)
</p>


<h3>Usage</h3>

<pre><code class="language-R">compute.pairwiseFST(
  x,
  method = "Anova",
  min.cov.per.pool = -1,
  max.cov.per.pool = 1e+06,
  min.indgeno.per.pop = -1,
  min.maf = -1,
  output.snp.values = FALSE,
  nsnp.per.bjack.block = 0,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A pooldata object containing Pool-Seq information or a countdata object containing allele count information</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Either "Anova" (default method as described in the manuscript) or "Identity" (relies on an alternative modeling consisting in estimating unbiased Probability of Identity within and across pairs of pools)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.cov.per.pool</code></td>
<td>
<p>For Pool-Seq data (i.e., pooldata objects) only: minimal allowed read count (per pool). If at least one pool is not covered by at least min.cov.perpool reads, the position is discarded in the corresponding pairwise comparisons</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.cov.per.pool</code></td>
<td>
<p>For Pool-Seq data (i.e., pooldata objects) only: maximal allowed read count (per pool). If at least one pool is covered by more than min.cov.perpool reads, the position is discarded in the corresponding pairwise comparisons.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.indgeno.per.pop</code></td>
<td>
<p>For allele count data (i.e., countdata objects) only: minimal number of overall counts required in each population. If at least one pop is not genotyped for at least min.indgeno.per.pop (haploid) individual, the position is discarded</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.maf</code></td>
<td>
<p>Minimal allowed Minor Allele Frequency (computed from the ratio overal read counts for the reference allele over the read coverage)  in the pairwise comparisons.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output.snp.values</code></td>
<td>
<p>If TRUE, provide SNP-specific pairwise FST for each comparisons (may lead to a huge result object if the number of pools and/or SNPs is large)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nsnp.per.bjack.block</code></td>
<td>
<p>Number of consecutive SNPs within a block for block-jackknife (default=0, i.e., no block-jackknife sampling)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>If TRUE extra information is printed on the terminal</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An object of class pairwisefst (see help(pairwisefst) for details)
</p>


<h3>See Also</h3>

<p>To generate pooldata object, see <code>vcf2pooldata</code>, <code>popsync2pooldata</code>,<code>genobaypass2pooldata</code> or <code>genoselestim2pooldata</code>. To generate coundata object, see <code>genobaypass2countdata</code> or <code>genotreemix2countdata</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R"> make.example.files(writing.dir=tempdir())
 pooldata=popsync2pooldata(sync.file=paste0(tempdir(),"/ex.sync.gz"),poolsizes=rep(50,15))
 PairwiseFST=compute.pairwiseFST(pooldata)
</code></pre>


</div>