<div class="container">

<table style="width: 100%;"><tr>
<td>countdata-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>S4 class to represent a Count data set.</h2>

<h3>Description</h3>

<p>S4 class to represent a Count data set.
</p>


<h3>Slots</h3>


<dl>
<dt><code>npops</code></dt>
<dd>
<p>The number of populations</p>
</dd>
<dt><code>nsnp</code></dt>
<dd>
<p>The number of SNPs</p>
</dd>
<dt><code>refallele.count</code></dt>
<dd>
<p>A matrix (nsnp rows and npops columns) with the allele counts for the reference allele</p>
</dd>
<dt><code>total.count</code></dt>
<dd>
<p>A matrix (nsnp rows and npops columns) with the total number of counts (i.e., twice the number of genotyped individual for diploid species and autosomal markers)</p>
</dd>
<dt><code>snp.info</code></dt>
<dd>
<p>A data frame (nsnp rows and 4 columns) detailing for each SNP, the chromosome (or scaffold), the position, Reference allele name and Alternate allele name (if available)</p>
</dd>
<dt><code>popnames</code></dt>
<dd>
<p>A vector of length npops with the corresponding population names</p>
</dd>
</dl>
<h3>See Also</h3>

<p>To generate countdata object, see <code>genobaypass2countdata</code> and <code>genotreemix2countdata</code>
</p>


</div>