<div class="container">

<table style="width: 100%;"><tr>
<td>Ifreqs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute allele frequencies from genotypes</h2>

<h3>Description</h3>

<p>Computes alternative allele frequencies from genotypes by dividing the total
number of alternative alleles by the total number of gene copies.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Ifreqs(nDip, genotypes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nDip</code></td>
<td>
<p>an integer representing the total number of diploid individuals
to simulate. Note that <code>scrm::scrm()</code> actually simulates haplotypes, so the
number of simulated haplotypes is double of this.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotypes</code></td>
<td>
<p>a list of simulated genotypes, where each entry is a matrix
corresponding to a different locus. At each matrix, each column is a
different SNP and each row is a different individual.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a list of allele frequencies. Each entry of the list corresponds to a
different locus.
</p>


<h3>Examples</h3>

<pre><code class="language-R">genotypes &lt;- run_scrm(nDip = 10, nloci = 10)
Ifreqs(nDip = 10, genotypes)

</code></pre>


</div>