<div class="container">

<table style="width: 100%;"><tr>
<td>computeReference</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute the number of reference reads over a matrix</h2>

<h3>Description</h3>

<p>This function works over all the rows and columns of a matrix and computes
the number of reads containing the reference allele at each site and for each
individual.
</p>


<h3>Usage</h3>

<pre><code class="language-R">computeReference(genotypes, indContribution, error)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genotypes</code></td>
<td>
<p>is a matrix of genotypes. Each column of the matrix should
be a different site and each row a different individual. Genotypes should
be encoded as 0: reference homozygote, 1: heterozygote and 2: alternative
homozygote.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>indContribution</code></td>
<td>
<p>is a matrix of individual contributions. Each row of
that matrix is a different individual and each column is a different site.
Thus, each entry of the matrix should contain the number of reads
contributed by that individual at that particular site.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>error</code></td>
<td>
<p>a numeric value with error rate associated with the sequencing
and mapping process. This error rate is assumed to be symmetric:
error(reference -&gt; alternative) = error(alternative -&gt; reference). This
number should be between 0 and 1.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a matrix with the number of reference allele reads contributed by
each individual. Each row of the matrix represents a different individual
and each column is a different site.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># probability of contribution for 10 individuals at 5 sites
probs &lt;- indProbs(np = 10, nSNPs = 5, pError = 5)

# simulate the number of reads contributed, assuming 20 coverage for each site
indContribution &lt;- indReads(np = 10, coverage = rep(20, 5), probs = probs)

# set seed and create a random matrix of genotypes
set.seed(10)
genotypes &lt;- matrix(rpois(50, 0.5), nrow = 10)

# simulate the number of reads with the reference allele
computeReference(genotypes = genotypes, indContribution = indContribution, error = 0.01)

</code></pre>


</div>