<div class="container">

<table style="width: 100%;"><tr>
<td>computeFST</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute FST from Pool-Seq data or Count data</h2>

<h3>Description</h3>

<p>Compute FST from Pool-Seq data or Count data
</p>


<h3>Usage</h3>

<pre><code class="language-R">computeFST(
  x,
  method = "Anova",
  nsnp.per.bjack.block = 0,
  sliding.window.size = 0,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A pooldata object containing Pool-Seq information or countdata object containing allele counts information</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Either "Anova" (default method as described in Hivert et al (2018, eq. 9) for pool-seq data and Weir (1996, eq. 5.2) for count data) or "Identity" (relying on unbiased estimators of Probability of Identity within and across pairs of pools/populations)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nsnp.per.bjack.block</code></td>
<td>
<p>Number of consecutive SNPs within a block for block-jackknife (default=0, i.e., no block-jackknife sampling)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sliding.window.size</code></td>
<td>
<p>Number of consecutive SNPs within a window for multi-locus computation of Fst over sliding window with half-window size step (default=0, i.e., no sliding-window scan)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>If TRUE extra information is printed on the terminal</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list with the four following elements:
</p>

<ol>
<li>
<p> "FST": a scalar corresponding to the estimate of the genome-wide FST over all the populations
</p>
</li>
<li>
<p> "snp.FST": a vector containing estimates of SNP-specific FST
</p>
</li>
<li>
<p> "snp.Q1": a vector containing estimates of the overall within pop. SNP-specific probability of identity
</p>
</li>
<li>
<p> "snp.Q2": a vector containing estimates of the overall between pop. SNP-specific probability of identity
</p>
</li>
<li>
<p> "mean.fst" (if nsnp.per.bjack.block&gt;0): genome-wide Fst estimate as the mean over block-jackknife samples (may slight differ from "FST" estimate since it is only computed on SNPs eligible for Block-Jackknife)
</p>
</li>
<li>
<p> "se.fst" (if nsnp.per.bjack.block&gt;0): standard-error of the genome-wide Fst estimate computed block-jackknife samples
</p>
</li>
<li>
<p> "fst.bjack.samples" (if nsnp.per.bjack.block&gt;0): a vector containing estimates of the overall between pop. SNP-specific probability of identity
</p>
</li>
<li>
<p> "sliding.windows.fst" (if sliding.window.size&gt;0): a 4-columns data frame containing information on multi-locus Fst computed for sliding windows of SNPs over the whole genome with i) column with the chromosome/contig of origin of each window; ii) the mid-position of each window; iii) the cumulated mid-position of each window (to facilitate further plotting); and iv) the estimated multi-locus Fst
</p>
</li>
</ol>
<h3>See Also</h3>

<p>To generate pooldata object, see <code>vcf2pooldata</code>, <code>popsync2pooldata</code>,<code>genobaypass2pooldata</code> or <code>genoselestim2pooldata</code>. To generate coundata object, see <code>genobaypass2countdata</code> or <code>genotreemix2countdata</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R"> make.example.files(writing.dir=tempdir())
 pooldata=popsync2pooldata(sync.file=paste0(tempdir(),"/ex.sync.gz"),poolsizes=rep(50,15))
 res.fst=computeFST(pooldata)
</code></pre>


</div>