<div class="container">

<table style="width: 100%;"><tr>
<td>extract_vscan_counts</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>extract_vscan_counts</h2>

<h3>Description</h3>

<p>Extract VarScan counts
</p>


<h3>Usage</h3>

<pre><code class="language-R">.extract_vscan_counts(vcf_data, ad_idx, rd_idx)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vcf_data</code></td>
<td>
<p>a matrix of String containing count information in VarScan format</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ad_idx</code></td>
<td>
<p>the index of the FORMAT AD field</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rd_idx</code></td>
<td>
<p>the index of the FORMAT RD field</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Extract VarScan counts and return read counts for the reference and alternate allele.
For VarScan generated vcf, SNPs with more than one alternate allele are discarded 
(because only a single count is then reported in the AD fields) making the min.rc unavailable (of vcf2pooldata).
The VarScan â€“min-reads2 option might replace to some extent the min.rc functionality although 
SNP where the two major alleles in the Pool-Seq data are different from the reference allele 
(e.g., expected to be more frequent when using a distantly related reference genome for mapping) 
will be disregarded.
</p>


<h3>Value</h3>

<p>A numeric matrix of read count with nsnp rows and 2*npools columns.
The first npools columns consist of read count for the reference allele (RD),
columns npools+1 to 2*npools consist of read coverage (RD+AD)
</p>


<h3>Examples</h3>

<pre><code class="language-R">.extract_vscan_counts(rbind(c("0/0:0:20","1/1:18:1"),c("0/1:12:15","1/1:27:2")),3,2)

</code></pre>


</div>