<div class="container">

<table style="width: 100%;"><tr>
<td>genotreemix2countdata</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Convert allele count input files from the Treemix program into a coundata object</h2>

<h3>Description</h3>

<p>Convert allele count input files from the Treemix program into a coundata object
</p>


<h3>Usage</h3>

<pre><code class="language-R">genotreemix2countdata(
  genotreemix.file = "",
  snp.pos = NA,
  min.indgeno.per.pop = -1,
  min.maf = -1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genotreemix.file</code></td>
<td>
<p>The name (or a path) of the Treemix allele count file (see the Treemix manual <a href="https://bitbucket.org/nygcresearch/treemix/wiki/Home">https://bitbucket.org/nygcresearch/treemix/wiki/Home</a>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp.pos</code></td>
<td>
<p>An optional two column matrix with nsnps rows containing the chromosome (or contig/scaffold) of origin and the position of each markers</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.indgeno.per.pop</code></td>
<td>
<p>Minimal number of overall counts required in each population. If at least one pop is not genotyped for at least min.indgeno.per.pop (haploid) individual, the position is discarded</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.maf</code></td>
<td>
<p>Minimal allowed Minor Allele Frequency (computed from the ratio overall counts for the reference allele over the overall number of (haploid) individual genotyped)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>If TRUE extra information is printed on the terminal</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Information on SNP position is only required for some graphical display or to carried out block-jacknife sampling estimation of confidence intervals. If no mapping information is given (default), SNPs will be assumed to be ordered on the same chromosome and separated by 1 bp. As blocks are defined with a number of consecutive SNPs (rather than a length), the latter assumption has actually no effect (except in the reported estimated block sizes in Mb).
</p>


<h3>Value</h3>

<p>A countdata object containing 6 elements:
</p>

<ol>
<li>
<p> "refallele.count": a matrix (nsnp rows and npops columns) with the allele counts for the reference allele
</p>
</li>
<li>
<p> "total.count": a matrix (nsnp rows and npops columns) with the total number of counts (i.e., twice the number of genotyped individual for diploid species and autosomal markers)
</p>
</li>
<li>
<p> "snp.info": a matrix with nsnp rows and four columns containing respectively the contig (or chromosome) name (1st column) and position (2nd column) of the SNP; the allele taken as reference in the refallele.count matrix (3rd column); and the alternative allele (4th column)
</p>
</li>
<li>
<p> "popnames": a vector of length npops containing the names of the pops
</p>
</li>
<li>
<p> "nsnp": a scalar corresponding to the number of SNPs
</p>
</li>
<li>
<p> "npops": a scalar corresponding to the number of populations
</p>
</li>
</ol>
<h3>Examples</h3>

<pre><code class="language-R"> make.example.files(writing.dir=tempdir())
 pooldata=popsync2pooldata(sync.file=paste0(tempdir(),"/ex.sync.gz"),poolsizes=rep(50,15))
 ##NOTE: This example is just for the sake of illustration as it amounts 
 ##to interpret read count as allele count which must not be done in practice!
 dum=matrix(paste(pooldata@refallele.readcount,
   pooldata@readcoverage-pooldata@refallele.readcount,sep=","),
   ncol=pooldata@npools)
 colnames(dum)=pooldata@poolnames
 write.table(dum,file=paste0(tempdir(),"/genotreemix"),quote=FALSE,row.names=FALSE)
 countdata=genotreemix2countdata(genotreemix.file=paste0(tempdir(),"/genotreemix"))
</code></pre>


</div>