<div class="container">

<table style="width: 100%;"><tr>
<td>nei.dist</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate Genetic Distance for a genind or genclone object.</h2>

<h3>Description</h3>

<p>These functions are modified from the function dist.genpop to
be applicable for distances between individuals.
</p>


<h3>Usage</h3>

<pre><code class="language-R">nei.dist(x, warning = TRUE)

edwards.dist(x)

rogers.dist(x)

reynolds.dist(x)

provesti.dist(x)

prevosti.dist
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a genind, genclone, or matrix object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>warning</code></td>
<td>
<p>If <code>TRUE</code>, a warning will be printed if any infinite 
values are detected and replaced. If <code>FALSE</code>, these values will be 
replaced without warning. See Details below.</p>
</td>
</tr>
</table>
<h3>Format</h3>

<p>An object of class <code>function</code> of length 1.
</p>


<h3>Details</h3>

<p>It is important to be careful with the interpretation of these 
distances as they were originally intended for calculation of 
between-population distance. As Nei's distance is the negative log of 0:1, 
this means that it is very possible to obtain distances of infinity. When 
this happens, infinite values are corrected to be 10 * max(D) where D is
the distance matrix without infinite values.
</p>


<h3>Value</h3>

<p>an object of class dist with the same number of observations as the 
number of individuals in your data.
</p>


<h3>Note</h3>

<p>Prevosti's distance is identical to <code>diss.dist</code>, except 
that <code>diss.dist</code> is optimized for a larger number of 
individuals (n &gt; 125) at the cost of required memory. Both
<code>prevosti.dist</code> and <code>provesti.dist</code> are the same function,
<code>provesti.dist</code> is a spelling error and exists for backwards
compatibility.
</p>
<p>These distances were adapted from the <span class="pkg">adegenet</span> function
<code>dist.genpop</code> to work with <code>genind</code> objects.
</p>


<h3>Author(s)</h3>

<p>Zhian N. Kamvar (poppr adaptation)
Thibaut Jombart (adegenet adaptation)
Daniel Chessel (ade4)
</p>


<h3>References</h3>

<p>Nei, M. (1972) Genetic distances between populations. American Naturalist,
106, 283-292. 
</p>
<p>Nei M. (1978) Estimation of average heterozygosity and genetic
distance from a small number of individuals. Genetics, 23, 341-369. 
</p>
<p>Avise, J. C. (1994) Molecular markers, natural history and evolution. Chapman &amp; Hall,
London.
</p>
<p>Edwards, A.W.F. (1971) Distance between populations on the basis of gene
frequencies. Biometrics, 27, 873-881. 
</p>
<p>Cavalli-Sforza L.L. and Edwards A.W.F.
(1967) Phylogenetic analysis: models and estimation procedures. Evolution,
32, 550-570. 
</p>
<p>Hartl, D.L. and Clark, A.G. (1989) Principles of population
genetics. Sinauer Associates, Sunderland, Massachussetts (p. 303).
</p>
<p>Reynolds, J. B., B. S. Weir, and C. C. Cockerham. (1983) Estimation of the
coancestry coefficient: basis for a short-term genetic distance. Genetics,
105, 767-779.
</p>
<p>Rogers, J.S. (1972) Measures of genetic similarity and genetic distances.
Studies in Genetics, Univ. Texas Publ., 7213, 145-153. 
</p>
<p>Avise, J. C. (1994)
Molecular markers, natural history and evolution. Chapman &amp; Hall, London.
</p>
<p>Prevosti A. (1974) La distancia genetica entre poblaciones. Miscellanea
Alcobe, 68, 109-118. 
</p>
<p>Prevosti A., Ocana J. and Alonso G. (1975) Distances
between populations of Drosophila subobscura, based on chromosome
arrangements frequencies. Theoretical and Applied Genetics, 45, 231-241. 
</p>
<p>For more information on dissimilarity indexes: 
</p>
<p>Gower J. and Legendre P. (1986)
Metric and Euclidean properties of dissimilarity coefficients. Journal of
Classification, 3, 5-48 
</p>
<p>Legendre P. and Legendre L. (1998) Numerical Ecology,
Elsevier Science B.V. 20, pp274-288.
</p>


<h3>See Also</h3>

<p><code>aboot</code> <code>diss.dist</code> <code>poppr.amova</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(nancycats)
(nan9   &lt;- popsub(nancycats, 9))
(neinan &lt;- nei.dist(nan9))
(ednan  &lt;- edwards.dist(nan9))
(rodnan &lt;- rogers.dist(nan9))
(reynan &lt;- reynolds.dist(nan9))
(pronan &lt;- prevosti.dist(nan9))

</code></pre>


</div>