<div class="container">

<table style="width: 100%;"><tr>
<td>poppr</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Produce a basic summary table for population genetic analyses.</h2>

<h3>Description</h3>

<p>For the <span class="pkg">poppr</span> package description, please see <code>package?poppr</code>
</p>
<p>This function allows the user to quickly view indices of heterozygosity,
evenness, and linkage to aid in the decision of a path to further analyze a
specified dataset. It natively takes adegenet::genind and
genclone objects, but can convert any raw data formats
that adegenet can take (fstat, structure, genetix, and genpop) as well as
genalex files exported into a csv format (see <code>read.genalex()</code> for details).
</p>


<h3>Usage</h3>

<pre><code class="language-R">poppr(
  dat,
  total = TRUE,
  sublist = "ALL",
  exclude = NULL,
  blacklist = NULL,
  sample = 0,
  method = 1,
  missing = "ignore",
  cutoff = 0.05,
  quiet = FALSE,
  clonecorrect = FALSE,
  strata = 1,
  keep = 1,
  plot = TRUE,
  hist = TRUE,
  index = "rbarD",
  minsamp = 10,
  legend = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dat</code></td>
<td>
<p>a adegenet::genind object OR a genclone
object OR any fstat, structure, genetix, genpop, or genalex formatted
file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>total</code></td>
<td>
<p>When <code>TRUE</code> (default), indices will be calculated for the
pooled populations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sublist</code></td>
<td>
<p>a list of character strings or integers to indicate specific
population names (accessed via <code>adegenet::popNames()</code>).
Defaults to "ALL".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exclude</code></td>
<td>
<p>a <code>vector</code> of population names or indexes that the user
wishes to discard. Default to <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>blacklist</code></td>
<td>
<p>DEPRECATED, use exclude.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample</code></td>
<td>
<p>an integer indicating the number of permutations desired to
obtain p-values. Sampling will shuffle genotypes at each locus to simulate
a panmictic population using the observed genotypes. Calculating the
p-value includes the observed statistics, so set your sample number to one
off for a round p-value (eg. <code>sample = 999</code> will give you p = 0.001 and
<code>sample = 1000</code> will give you p = 0.000999001).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>an integer from 1 to 4 indicating the method of sampling
desired. see <code>shufflepop()</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>missing</code></td>
<td>
<p>how should missing data be treated? <code>"zero"</code> and
<code>"mean"</code> will set the missing values to those documented in
<code>tab()</code>. <code>"loci"</code> and <code>"geno"</code> will remove any loci or
genotypes with missing data, respectively (see <code>missingno()</code> for
more information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cutoff</code></td>
<td>
<p><code>numeric</code> a number from 0 to 1 indicating the percent
missing data allowed for analysis. This is to be used in conjunction with
the flag <code>missing</code> (see <code>missingno()</code> for details)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p><code>FALSE</code> (default) will display a progress bar for each
population analyzed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clonecorrect</code></td>
<td>
<p>default <code>FALSE</code>. must be used with the <code>strata</code>
parameter, or the user will potentially get undesired results. see
<code>clonecorrect()</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>strata</code></td>
<td>
<p>a <code>formula</code> indicating the hierarchical levels to be used.
The hierarchies should be present in the <code>strata</code> slot. See
<code>strata()</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep</code></td>
<td>
<p>an <code>integer</code>. This indicates which strata you wish to keep
after clone correcting your data sets. To combine strata, just set keep
from 1 to the number of straifications set in strata. see
<code>clonecorrect()</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>
<p><code>logical</code> if <code>TRUE</code> (default) and <code>sampling &gt; 0</code>,
a histogram will be produced for each population.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hist</code></td>
<td>
<p><code>logical</code> Deprecated. Use plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>index</code></td>
<td>
<p><code>character</code> Either "Ia" or "rbarD". If <code>hist = TRUE</code>,
this will determine the index used for the visualization.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minsamp</code></td>
<td>
<p>an <code>integer</code> indicating the minimum number of individuals
to resample for rarefaction analysis. See <code>vegan::rarefy()</code> for
details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend</code></td>
<td>
<p><code>logical</code>. When this is set to <code>TRUE</code>, a legend describing the
resulting table columns will be printed. Defaults to <code>FALSE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>arguments to be passed on to <code>diversity_stats()</code></p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This table is intended to be a first look into the dynamics of mutlilocus
genotype diversity. Many of the statistics (except for the the index of
association) are simply based on counts of multilocus genotypes and do not
take into account the actual allelic states. <strong>Descriptions of the
statistics can be found in the Algorithms and Equations vignette</strong>:
<code>vignette("algo", package = "poppr")</code>.
</p>


<h4>sampling</h4>

<p>The sampling procedure is explicitly for testing the index of association.
None of the other diversity statistics (H, G, lambda, E.5) are tested with
this sampling due to the differing data types. To obtain confidence
intervals for these statistics, please see <code>diversity_ci()</code>.
</p>



<h4>rarefaction</h4>

<p>Rarefaction analysis is performed on the number of multilocus genotypes
because it is relatively easy to estimate (Grünwald et al., 2003). To
obtain rarefied estimates of diversity, it is possible to use
<code>diversity_ci()</code> with the argument <code>rarefy = TRUE</code>
</p>



<h4>graphic</h4>

<p>This function outputs a <span class="pkg">ggplot2</span> graphic of histograms. These can be
manipulated to be visualized in another manner by retrieving the plot with
the <code>last_plot()</code> command from <span class="pkg">ggplot2</span>. A useful manipulation would
be to arrange the graphs into a single column so that the values of the
statistic line up: <code style="white-space: pre;">⁠p &lt;- last_plot(); p + facet_wrap(~population, ncol = 1, scales = "free_y")⁠</code> The name for the groupings is
"population" and the name for the x axis is "value".
</p>



<h3>Value</h3>

<p>A data frame with populations in rows and the following columns:
</p>

<ul>
<li> <p><strong>Pop</strong>: A vector indicating the population factor
</p>
</li>
<li> <p><strong>N</strong>: An integer vector indicating the number of individuals/isolates in
the specified population.
</p>
</li>
<li> <p><strong>MLG</strong>: An integer vector indicating the number of multilocus genotypes
found in the specified population, (see: <code>mlg()</code>)
</p>
</li>
<li> <p><strong>eMLG</strong>: The expected number of MLG at the lowest common sample size (set
by the parameter <code>minsamp</code>).
</p>
</li>
<li> <p><strong>SE</strong>: The standard error for the rarefaction analysis
</p>
</li>
<li> <p><strong>H</strong>: Shannon-Weiner Diversity index
</p>
</li>
<li> <p><strong>G</strong>: Stoddard and Taylor's Index
</p>
</li>
<li> <p><strong>lambda</strong>: Simpson's index
</p>
</li>
<li> <p><strong>E.5</strong>: Evenness
</p>
</li>
<li> <p><strong>Hexp</strong>: Nei's gene diversity (expected heterozygosity)
</p>
</li>
<li> <p><strong>Ia</strong>: A numeric vector giving the value of the Index of Association for
each population factor, (see <code>ia()</code>).
</p>
</li>
<li> <p><strong>p.Ia</strong>: A numeric vector indicating the p-value for Ia from the number
of reshufflings indicated in <code>sample</code>. Lowest value is 1/n where n is the
number of observed values.
</p>
</li>
<li> <p><strong>rbarD</strong>: A numeric vector giving the value of the Standardized Index of
Association for each population factor, (see <code>ia()</code>).
</p>
</li>
<li> <p><strong>p.rD</strong>: A numeric vector indicating the p-value for rbarD from the
number of reshuffles indicated in <code>sample</code>. Lowest value is 1/n where n is
the number of observed values.
</p>
</li>
<li> <p><strong>File</strong>: A vector indicating the name of the original data file.
</p>
</li>
</ul>
<h3>Note</h3>

<p>The calculation of <code>Hexp</code> has changed from <span class="pkg">poppr</span> 1.x. It was
previously calculated based on the diversity of multilocus genotypes,
resulting in a value of 1 for sexual populations. This was obviously not
Nei's 1978 expected heterozygosity. We have thus changed the statistic to
be the true value of Hexp by calculating <code class="reqn">(\frac{n}{n-1}) 1 - \sum_{i =
  1}^k{p^{2}_{i}}</code> where p is the allele
frequencies at a given locus and n is the number of observed alleles (Nei,
1978) in each locus and then returning the average. Caution should be
exercised in interpreting the results of Hexp with polyploid organisms with
ambiguous ploidy. The lack of allelic dosage information will cause rare
alleles to be over-represented and artificially inflate the index. This is
especially true with small sample sizes.
</p>


<h3>Author(s)</h3>

<p>Zhian N. Kamvar
</p>


<h3>References</h3>

<p>Paul-Michael Agapow and Austin Burt. Indices of multilocus
linkage disequilibrium. <em>Molecular Ecology Notes</em>, 1(1-2):101-102,
2001
</p>
<p>A.H.D. Brown, M.W. Feldman, and E. Nevo. Multilocus structure of natural
populations of <em>Hordeum spontaneum</em>. <em>Genetics</em>, 96(2):523-536,
1980.
</p>
<p>Niklaus J. Gr\"unwald, Stephen B. Goodwin, Michael G. Milgroom, and William
E. Fry. Analysis of genotypic diversity data for populations of
microorganisms. Phytopathology, 93(6):738-46, 2003
</p>
<p>Bernhard Haubold and Richard R. Hudson. Lian 3.0: detecting linkage
disequilibrium in multilocus data. Bioinformatics, 16(9):847-849, 2000.
</p>
<p>Kenneth L.Jr. Heck, Gerald van Belle, and Daniel Simberloff. Explicit
calculation of the rarefaction diversity measurement and the determination
of sufficient sample size. Ecology, 56(6):pp. 1459-1461, 1975
</p>
<p>Masatoshi Nei. Estimation of average heterozygosity and genetic distance
from a small number of individuals. Genetics, 89(3):583-590, 1978.
</p>
<p>S H Hurlbert. The nonconcept of species diversity: a critique and
alternative parameters. Ecology, 52(4):577-586, 1971.
</p>
<p>J.A. Ludwig and J.F. Reynolds. Statistical Ecology. A Primer on Methods and
Computing. New York USA: John Wiley and Sons, 1988.
</p>
<p>Simpson, E. H. Measurement of diversity. Nature 163: 688, 1949
doi:10.1038/163688a0
</p>
<p>Good, I. J. (1953). On the Population Frequency of Species and the
Estimation of Population Parameters. <em>Biometrika</em> 40(3/4): 237-264.
</p>
<p>Lande, R. (1996). Statistics and partitioning of species diversity, and
similarity among multiple communities. <em>Oikos</em> 76: 5-13.
</p>
<p>Jari Oksanen, F. Guillaume Blanchet, Roeland Kindt, Pierre Legendre, Peter
R. Minchin, R. B. O'Hara, Gavin L. Simpson, Peter Solymos, M. Henry H.
Stevens, and Helene Wagner. vegan: Community Ecology Package, 2012. R
package version 2.0-5.
</p>
<p>E.C. Pielou. Ecological Diversity. Wiley, 1975.
</p>
<p>Claude Elwood Shannon. A mathematical theory of communication. Bell Systems
Technical Journal, 27:379-423,623-656, 1948
</p>
<p>J M Smith, N H Smith, M O'Rourke, and B G Spratt. How clonal are bacteria?
Proceedings of the National Academy of Sciences, 90(10):4384-4388, 1993.
</p>
<p>J.A. Stoddart and J.F. Taylor. Genotypic diversity: estimation and
prediction in samples. Genetics, 118(4):705-11, 1988.
</p>


<h3>See Also</h3>

<p><code>clonecorrect()</code>,
<code>poppr.all()</code>,
<code>ia()</code>,
<code>missingno()</code>,
<code>mlg()</code>,
<code>diversity_stats()</code>,
<code>diversity_ci()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(nancycats)
poppr(nancycats)

## Not run: 
# Sampling
poppr(nancycats, sample = 999, total = FALSE, plot = TRUE)

# Customizing the plot
library("ggplot2")
p &lt;- last_plot()
p + facet_wrap(~population, scales = "free_y", ncol = 1)

# Turning off diversity statistics (see get_stats)
poppr(nancycats, total=FALSE, H = FALSE, G = FALSE, lambda = FALSE, E5 = FALSE)

# The previous version of poppr contained a definition of Hexp, which
# was calculated as (N/(N - 1))*lambda. It basically looks like an unbiased 
# Simpson's index. This statistic was originally included in poppr because it
# was originally included in the program multilocus. It was finally figured
# to be an unbiased Simpson's diversity metric (Lande, 1996; Good, 1953).

data(Aeut)

uSimp &lt;- function(x){
  lambda &lt;- vegan::diversity(x, "simpson")
  x &lt;- drop(as.matrix(x))
  if (length(dim(x)) &gt; 1){
    N &lt;- rowSums(x)
  } else {
    N &lt;- sum(x)
  }
  return((N/(N-1))*lambda)
}
poppr(Aeut, uSimp = uSimp)


# Demonstration with viral data
# Note: this is a larger data set that could take a couple of minutes to run
# on slower computers. 
data(H3N2)
strata(H3N2) &lt;- data.frame(other(H3N2)$x)
setPop(H3N2) &lt;- ~country
poppr(H3N2, total = FALSE, sublist=c("Austria", "China", "USA"), 
  clonecorrect = TRUE, strata = ~country/year)

## End(Not run)
</code></pre>


</div>