<div class="container">

<table style="width: 100%;"><tr>
<td>mlg</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create counts, vectors, and matrices of multilocus genotypes.</h2>

<h3>Description</h3>

<p>Create counts, vectors, and matrices of multilocus genotypes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">mlg(gid, quiet = FALSE)

mlg.table(
  gid,
  strata = NULL,
  sublist = "ALL",
  exclude = NULL,
  blacklist = NULL,
  mlgsub = NULL,
  bar = TRUE,
  plot = TRUE,
  total = FALSE,
  color = FALSE,
  background = FALSE,
  quiet = FALSE
)

mlg.vector(gid, reset = FALSE)

mlg.crosspop(
  gid,
  strata = NULL,
  sublist = "ALL",
  exclude = NULL,
  blacklist = NULL,
  mlgsub = NULL,
  indexreturn = FALSE,
  df = FALSE,
  quiet = FALSE
)

mlg.id(gid)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gid</code></td>
<td>
<p>a adegenet::genind, genclone, adegenet::genlight, or snpclone object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p><code>Logical</code>. If FALSE, progress of functions will be printed
to the screen.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>strata</code></td>
<td>
<p>a formula specifying the strata at which computation is to be
performed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sublist</code></td>
<td>
<p>a <code>vector</code> of population names or indices that the user
wishes to keep. Default to "ALL".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exclude</code></td>
<td>
<p>a <code>vector</code> of population names or indexes that the user
wishes to discard. Default to <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>blacklist</code></td>
<td>
<p>DEPRECATED, use exclude.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mlgsub</code></td>
<td>
<p>a <code>vector</code> of multilocus genotype indices with which to
subset <code>mlg.table</code> and <code>mlg.crosspop</code>. NOTE: The resulting table
from <code>mlg.table</code> will only contain countries with those MLGs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bar</code></td>
<td>
<p>deprecated. Same as <code>plot</code>. Retained for compatibility.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>
<p><code>logical</code> If <code>TRUE</code>, a bar graph for each population
will be displayed showing the relative abundance of each MLG within the
population.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>total</code></td>
<td>
<p><code>logical</code> If <code>TRUE</code>, a row containing the sum of all
represented MLGs is appended to the matrix produced by mlg.table.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>color</code></td>
<td>
<p>an option to display a single barchart for mlg.table, colored by
population (note, this becomes facetted if 'background = TRUE').</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>background</code></td>
<td>
<p>an option to display the the total number of MLGs across
populations per facet in the background of the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reset</code></td>
<td>
<p>logical. For genclone objects, the MLGs are defined by the input
data, but they do not change if more or less information is added (i.e.
loci are dropped). Setting 'reset = TRUE' will recalculate MLGs.
Default is 'FALSE', returning the MLGs defined in the @mlg slot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>indexreturn</code></td>
<td>
<p><code>logical</code> If <code>TRUE</code>, a vector will be returned
to index the columns of <code>mlg.table</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>df</code></td>
<td>
<p><code>logical</code> If <code>TRUE</code>, return a data frame containing the
counts of the MLGs and what countries they are in. Useful for making graphs
with ggplot.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Multilocus genotypes are the unique combination of alleles across
all loci. For details of how these are calculated see <code>vignette("mlg", package = "poppr")</code>. In short, for genind and genclone objects, they are
calculated by using a rank function on strings of alleles, which is
sensitive to missing data. For genlight and snpclone objects, they are
calculated with distance methods via bitwise.dist and
mlg.filter, which means that these are insensitive to missing
data. Three different types of MLGs can be defined in <span class="pkg">poppr</span>:
</p>

<ul>
<li> <p><strong>original</strong> the default definition of multilocus genotypes as
detailed above
</p>
</li>
<li> <p><strong>contracted</strong> these are multilocus genotypes collapsed into multilocus
lineages (mll) with genetic distance via mlg.filter
</p>
</li>
<li> <p><strong>custom</strong> user-defined multilocus genotypes. These are useful for
information such as mycelial compatibility groups
</p>
</li>
</ul>
<p><strong>All of the functions documented here will work on any of the MLG types
defined in <span class="pkg">poppr</span></strong>
</p>


<h3>Value</h3>



<h4>mlg</h4>

<p>an integer describing the number of multilocus genotypes observed.
</p>



<h4>mlg.table</h4>

<p>a matrix with columns indicating unique multilocus genotypes and rows
indicating populations. This table can be used with the funciton
diversity_stats to calculate the Shannon-Weaver index (H), Stoddart and
Taylor's index (aka inverse Simpson's index; G), Simpson's index (lambda),
and evenness (E5).
</p>



<h4>mlg.vector</h4>

<p>a numeric vector naming the multilocus genotype of each individual in the
dataset.
</p>



<h4>mlg.crosspop</h4>


<ul>
<li> <p><strong>default</strong> a <code>list</code> where each element contains a named integer
vector representing the number of individuals represented from each
population in that MLG
</p>
</li>
<li> <p><code>indexreturn = TRUE</code> a <code>vector</code> of integers defining the multilocus
genotypes that have individuals crossing populations
</p>
</li>
<li> <p><code>df = TRUE</code> A long form data frame with the columns: MLG, Population,
Count. Useful for graphing with ggplot2
</p>
</li>
</ul>
<h4>mlg.id</h4>

<p>a list of multilocus genotypes with the associated individual names per MLG.
</p>



<h3>Note</h3>

<p>The resulting matrix of 'mlg.table' can be used for analysis with 
the <span class="pkg">vegan</span> package.
</p>
<p>mlg.vector will recalculate the mlg vector for
[adegenet::genind] objects and will return the contents of the mlg
slot in [genclone][genclone-class] objects. This means that MLGs will be
different for subsetted [adegenet::genind] objects.
</p>


<h3>Author(s)</h3>

<p>Zhian N. Kamvar
</p>


<h3>See Also</h3>

<p><code>vegan::diversity()</code>
diversity_stats
popsub
mll
mlg.filter
mll.custom
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# Load the data set
data(Aeut)

# Investigate the number of multilocus genotypes.
amlg &lt;- mlg(Aeut)
amlg # 119

# show the multilocus genotype vector 
avec &lt;- mlg.vector(Aeut)
avec 

# Get a table
atab &lt;- mlg.table(Aeut, color = TRUE)
atab

# See where multilocus genotypes cross populations
acrs &lt;- mlg.crosspop(Aeut) # MLG.59: (2 inds) Athena Mt. Vernon

# See which individuals belong to each MLG
aid &lt;- mlg.id(Aeut)
aid["59"] # individuals 159 and 57

## Not run: 

# For the mlg.table, you can also choose to display the number of MLGs across
# populations in the background

mlg.table(Aeut, background = TRUE)
mlg.table(Aeut, background = TRUE, color = TRUE)

# A simple example. 10 individuals, 5 genotypes.
mat1 &lt;- matrix(ncol=5, 25:1)
mat1 &lt;- rbind(mat1, mat1)
mat &lt;- matrix(nrow=10, ncol=5, paste(mat1,mat1,sep="/"))
mat.gid &lt;- df2genind(mat, sep="/")
mlg(mat.gid)
mlg.vector(mat.gid)
mlg.table(mat.gid)

# Now for a more complicated example.
# Data set of 1903 samples of the H3N2 flu virus genotyped at 125 SNP loci.
data(H3N2)
mlg(H3N2, quiet = FALSE)

H.vec &lt;- mlg.vector(H3N2)

# Changing the population vector to indicate the years of each epidemic.
pop(H3N2) &lt;- other(H3N2)$x$country
H.tab &lt;- mlg.table(H3N2, plot = FALSE, total = TRUE)

# Show which genotypes exist accross populations in the entire dataset.
res &lt;- mlg.crosspop(H3N2, quiet = FALSE)

# Let's say we want to visualize the multilocus genotype distribution for the
# USA and Russia
mlg.table(H3N2, sublist = c("USA", "Russia"), bar=TRUE)

# An exercise in subsetting the output of mlg.table and mlg.vector.
# First, get the indices of each MLG duplicated across populations.
inds &lt;- mlg.crosspop(H3N2, quiet = FALSE, indexreturn = TRUE)

# Since the columns of the table from mlg.table are equal to the number of
# MLGs, we can subset with just the columns.
H.sub &lt;- H.tab[, inds]

# We can also do the same by using the mlgsub flag.
H.sub &lt;- mlg.table(H3N2, mlgsub = inds)

# We can subset the original data set using the output of mlg.vector to
# analyze only the MLGs that are duplicated across populations. 
new.H &lt;- H3N2[H.vec %in% inds, ]


## End(Not run)
</code></pre>


</div>