<div class="container">

<table style="width: 100%;"><tr>
<td>pgen</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Genotype Probability</h2>

<h3>Description</h3>

<p>Calculate the probability of genotypes based on the product of allele
frequencies over all loci.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pgen(gid, pop = NULL, by_pop = TRUE, log = TRUE, freq = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gid</code></td>
<td>
<p>a genind or genclone object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pop</code></td>
<td>
<p>either a formula to set the population factor from the 
<code>strata</code> slot or a vector specifying the population factor for 
each sample. Defaults to <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>by_pop</code></td>
<td>
<p>When this is <code>TRUE</code> (default), the calculation will be
done by population.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log</code></td>
<td>
<p>a <code>logical</code> if <code>log =TRUE</code> (default), the values
returned will be log(Pgen). If <code>log = FALSE</code>, the values returned will
be Pgen.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>freq</code></td>
<td>
<p>a vector or matrix of allele frequencies. This defaults to 
<code>NULL</code>, indicating that the frequencies will be determined via 
round-robin approach in <code>rraf</code>. <strong>If this matrix or 
vector is not provided, zero-value allele frequencies will automatically be
corrected.</strong> For details, please see the documentation on
correcting rare alleles.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>options from correcting rare
alleles. The default is to correct allele frequencies to 1/n</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Pgen is the probability of a given genotype occuring in a population
assuming HWE. Thus, the value for diploids is 
</p>
<p style="text-align: center;"><code class="reqn">P_{gen} = \left(\prod_{i=1}^m p_i\right)2^h</code>
</p>
 
<p>where <code class="reqn">p_i</code> are the allele frequencies and <em>h</em> is the count of the
number of heterozygous sites in the sample (Arnaud-Haond et al. 2007; Parks
and Werth, 1993). The allele frequencies, by default, are calculated using
a round-robin approach where allele frequencies at a particular locus are 
calculated on the clone-censored genotypes without that locus.
</p>
<p>To avoid issues with numerical precision of small numbers, this function 
calculates pgen per locus by adding up log-transformed values of allele 
frequencies. These can easily be transformed to return the true value (see
examples).
</p>


<h3>Value</h3>

<p>A vector containing Pgen values per locus for each genotype in the 
object.
</p>


<h3>Note</h3>

<p>For haploids, Pgen at a particular locus is the allele frequency. This 
function cannot handle polyploids. Additionally, when the argument 
<code>pop</code> is not <code>NULL</code>, <code>by_pop</code> is automatically <code>TRUE</code>.
</p>


<h3>Author(s)</h3>

<p>Zhian N. Kamvar, Jonah Brooks, Stacy A. Krueger-Hadfield, Erik Sotka
</p>


<h3>References</h3>

<p>Arnaud-Haond, S., Duarte, C. M., Alberto, F., &amp; Serr√£o, E. A. 2007.
Standardizing methods to address clonality in population studies.
<em>Molecular Ecology</em>, 16(24), 5115-5139.
</p>
<p>Parks, J. C., &amp; Werth, C. R. 1993. A study of spatial features of clones in a
population of bracken fern, <em>Pteridium aquilinum</em> (Dennstaedtiaceae).
<em>American Journal of Botany</em>, 537-544.
</p>


<h3>See Also</h3>

<p><code>psex</code>, <code>rraf</code>, <code>rrmlg</code>, 
<code>rare_allele_correction</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(Pram)
head(pgen(Pram, log = FALSE))

## Not run: 
# You can also supply the observed allele frequencies
pramfreq &lt;- Pram %&gt;% genind2genpop() %&gt;% tab(freq = TRUE)
head(pgen(Pram, log = FALSE, freq = pramfreq))

# You can get the Pgen values over all loci by summing over the logged results:
pgen(Pram, log = TRUE) %&gt;%  # calculate pgen matrix
  rowSums(na.rm = TRUE) %&gt;% # take the sum of each row
  exp()                     # take the exponent of the results

# You can also take the product of the non-logged results:
apply(pgen(Pram, log = FALSE), 1, prod, na.rm = TRUE)

## Rare Allele Correction ---------------------------------------------------
##
# If you don't supply a table of frequencies, they are calculated with rraf 
# with correction = TRUE. This is normally benign when analyzing large 
# populations, but it can have a great effect on small populations. To help 
# control this, you can supply arguments described in 
# help("rare_allele_correction"). 


# Default is to correct by 1/n per population. Since the calculation is 
# performed on a smaller sample size due to round robin clone correction, it
# would be more appropriate to correct by 1/rrmlg at each locus. This is 
# acheived by setting d = "rrmlg". Since this is a diploid, we would want to
# account for the number of chromosomes, and so we set mul = 1/2
head(pgen(Pram, log = FALSE, d = "rrmlg", mul = 1/2)) # compare with the output above

# If you wanted to treat all alleles as equally rare, then you would set a
# specific value (let's say the rare alleles are 1/100):
head(pgen(Pram, log = FALSE, e = 1/100))

## End(Not run)
</code></pre>


</div>