<div class="container">

<table style="width: 100%;"><tr>
<td>bruvo.msn</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create minimum spanning network of selected populations using Bruvo's 
distance.</h2>

<h3>Description</h3>

<p>Create minimum spanning network of selected populations using Bruvo's 
distance.
</p>


<h3>Usage</h3>

<pre><code class="language-R">bruvo.msn(
  gid,
  replen = 1,
  add = TRUE,
  loss = TRUE,
  mlg.compute = "original",
  palette = topo.colors,
  sublist = "All",
  exclude = NULL,
  blacklist = NULL,
  vertex.label = "MLG",
  gscale = TRUE,
  glim = c(0, 0.8),
  gadj = 3,
  gweight = 1,
  wscale = TRUE,
  showplot = TRUE,
  include.ties = FALSE,
  threshold = NULL,
  clustering.algorithm = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gid</code></td>
<td>
<p>a <code>genind</code> or <code>genclone</code> object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>replen</code></td>
<td>
<p>a <code>vector</code> of <code>integers</code> indicating the length of the
nucleotide repeats for each microsatellite locus.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>add</code></td>
<td>
<p>if <code>TRUE</code>, genotypes with zero values will be treated under 
the genome addition model presented in Bruvo et al. 2004.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loss</code></td>
<td>
<p>if <code>TRUE</code>, genotypes with zero values will be treated under 
the genome loss model presented in Bruvo et al. 2004.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mlg.compute</code></td>
<td>
<p>if the multilocus genotypes are set to "custom" (see 
<code>mll.custom</code> for details) in your genclone object, this will 
specify which mlg level to calculate the nodes from. See details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>palette</code></td>
<td>
<p>a <code>vector</code> or <code>function</code> defining the color palette 
to be used to color the populations on the graph. It defaults to 
<code>topo.colors</code>. See examples for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sublist</code></td>
<td>
<p>a <code>vector</code> of population names or indexes that the user 
wishes to keep. Default to "ALL".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exclude</code></td>
<td>
<p>a <code>vector</code> of population names or indexes that the user
wishes to discard. Default to <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>blacklist</code></td>
<td>
<p>DEPRECATED, use exclude.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vertex.label</code></td>
<td>
<p>a <code>vector</code> of characters to label each vertex. There
are two defaults: <code>"MLG"</code> will label the nodes with the multilocus 
genotype from the original data set and <code>"inds"</code> will label the nodes 
with the representative individual names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gscale</code></td>
<td>
<p>"grey scale". If this is <code>TRUE</code>, this will scale the color
of the edges proportional to the observed distance, with the lines becoming
darker for more related nodes. See <code>greycurve</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>glim</code></td>
<td>
<p>"grey limit". Two numbers between zero and one. They determine 
the upper and lower limits for the <code>gray</code> function. Default is 
0 (black) and 0.8 (20% black). See <code>greycurve</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gadj</code></td>
<td>
<p>"grey adjust". a positive <code>integer</code> greater than zero that 
will serve as the exponent to the edge weight to scale the grey value to 
represent that weight. See <code>greycurve</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gweight</code></td>
<td>
<p>"grey weight". an <code>integer</code>. If it's 1, the grey scale 
will be weighted to emphasize the differences between closely related 
nodes. If it is 2, the grey scale will be weighted to emphasize the 
differences between more distantly related nodes. See 
<code>greycurve</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>wscale</code></td>
<td>
<p>"width scale". If this is <code>TRUE</code>, the edge widths will be 
scaled proportional to the inverse of the observed distance , with the 
lines becoming thicker for more related nodes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>showplot</code></td>
<td>
<p>logical. If <code>TRUE</code>, the graph will be plotted. If 
<code>FALSE</code>, it will simply be returned.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>include.ties</code></td>
<td>
<p>logical. If <code>TRUE</code>, the graph will include all edges
that were arbitrarily passed over in favor of another edge of equal weight.
If <code>FALSE</code>, which is the default, one edge will be arbitrarily 
selected when two or more edges are tied, resulting in a pure minimum
spanning network.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>numeric. By default, this is <code>NULL</code>, which will have no
effect. Any threshold value passed to this argument will be used in
<code>mlg.filter</code> prior to creating the MSN. If you have a data set
that contains contracted MLGs, this argument will override the threshold in
the data set. See Details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clustering.algorithm</code></td>
<td>
<p>string. By default, this is <code>NULL</code>. If
<code>threshold = NULL</code>, this argument will have no effect. When supplied
with either "farthest_neighbor", "average_neighbor", or "nearest_neighbor",
it will be passed to <code>mlg.filter</code> prior to creating the MSN. If
you have a data set that contains contracted MLGs, this argument will
override the algorithm in the data set. See Details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>any other arguments that could go into plot.igraph</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The minimum spanning network generated by this function is generated
via igraph's <code>minimum.spanning.tree</code>. The resultant
graph produced can be plotted using igraph functions, or the entire object
can be plotted using the function <code>plot_poppr_msn</code>, which will
give the user a scale bar and the option to layout your data.
</p>


<h4>node sizes</h4>

<p>The area of the nodes are representative of the number of samples. Because
<span class="pkg">igraph</span> scales nodes by radius, the node sizes in the graph are 
represented as the square root of the number of samples.</p>



<h4>mlg.compute</h4>

<p>Each node on the graph represents a different multilocus genotype. 
The edges on the graph represent genetic distances that connect the
multilocus genotypes. In genclone objects, it is possible to set the
multilocus genotypes to a custom definition. This creates a problem for
clone correction, however, as it is very possible to define custom lineages
that are not monophyletic. When clone correction is performed on these
definitions, information is lost from the graph. To circumvent this, The
clone correction will be done via the computed multilocus genotypes, either
"original" or "contracted". This is specified in the <code>mlg.compute</code>
argument, above.</p>



<h4>contracted multilocus genotypes</h4>

<p>If your incoming data set is of the class <code>genclone</code>,
and it contains contracted multilocus genotypes, this function will retain
that information for creating the minimum spanning network. You can use the
arguments <code>threshold</code> and <code>clustering.algorithm</code> to change the
threshold or clustering algorithm used in the network. For example, if you
have a data set that has a threshold of 0.1 and you wish to have a minimum
spanning network without a threshold, you can simply add 
<code>threshold = 0.0</code>, and no clustering will happen. 
</p>
<p>The <code>threshold</code> and <code>clustering.algorithm</code> arguments can also be
used to filter un-contracted data sets.
</p>



<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>graph</code></td>
<td>
<p>a minimum spanning network with nodes corresponding to 
MLGs within the data set. Colors of the nodes represent population 
membership. Width and color of the edges represent distance.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>populations</code></td>
<td>
<p>a vector of the population names corresponding to the 
vertex colors</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colors</code></td>
<td>
<p>a vector of the hexadecimal representations of
the colors used in the vertex colors</p>
</td>
</tr>
</table>
<h3>Note</h3>

 <ul>
<li> <p><strong>Please see the documentation for
<code>bruvo.dist</code> for details on the algorithm</strong>. </p>
</li>
<li>
<p> The edges of
these graphs may cross each other if the graph becomes too large. </p>
</li>
<li>
<p> The
nodes in the graph represent multilocus genotypes. The colors of the nodes
are representative of population membership. It is not uncommon to see
different populations containing the same multilocus genotype.</p>
</li>
</ul>
<h3>Author(s)</h3>

<p>Zhian N. Kamvar, Javier F. Tabima
</p>


<h3>References</h3>

<p>Ruzica Bruvo, Nicolaas K. Michiels, Thomas G. D'Souza, and 
Hinrich Schulenburg. A simple method for the calculation of microsatellite 
genotype distances irrespective of ploidy level. Molecular Ecology, 
13(7):2101-2106, 2004.
</p>


<h3>See Also</h3>

<p><code>bruvo.dist</code>, <code>nancycats</code>, 
<code>plot_poppr_msn</code>, <code>mst</code> 
<code>bruvo.boot</code>, <code>greycurve</code> <code>poppr.msn</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# Load the data set.
data(nancycats)

# View populations 8 and 9 with default colors. 
bruvo.msn(nancycats, replen = rep(2, 9), sublist=8:9, vertex.label="inds", 
          vertex.label.cex=0.7, vertex.label.dist=0.4)
## Not run: 
# View heat colors.
bruvo.msn(nancycats, replen=rep(2, 9), sublist=8:9, vertex.label="inds", 
palette=heat.colors, vertex.label.cex=0.7, vertex.label.dist=0.4)

# View custom colors. Here, we use black and orange.
bruvo.msn(nancycats, replen=rep(2, 9), sublist=8:9, vertex.label="inds", 
palette = colorRampPalette(c("orange", "black")), vertex.label.cex=0.7, 
vertex.label.dist=0.4)

# View with darker shades of grey (setting the upper limit to 1/2 black 1/2 white).
bruvo.msn(nancycats, replen=rep(2, 9), sublist=8:9, vertex.label="inds", 
palette = colorRampPalette(c("orange", "black")), vertex.label.cex=0.7, 
vertex.label.dist=0.4, glim=c(0, 0.5))

# View with no grey scaling.
bruvo.msn(nancycats, replen=rep(2, 9), sublist=8:9, vertex.label="inds", 
palette = colorRampPalette(c("orange", "black")), vertex.label.cex=0.7, 
vertex.label.dist=0.4, gscale=FALSE)

# View with no line widths.
bruvo.msn(nancycats, replen=rep(2, 9), sublist=8:9, vertex.label="inds", 
palette = colorRampPalette(c("orange", "black")), vertex.label.cex=0.7, 
vertex.label.dist=0.4, wscale=FALSE)

# View with no scaling at all.
bruvo.msn(nancycats, replen=rep(2, 9), sublist=8:9, vertex.label="inds", 
palette = colorRampPalette(c("orange", "black")), vertex.label.cex=0.7, 
vertex.label.dist=0.4, gscale=FALSE)

# View the whole population, but without labels.
bruvo.msn(nancycats, replen=rep(2, 9), vertex.label=NA)

## End(Not run)
</code></pre>


</div>