<div class="container">

<table style="width: 100%;"><tr>
<td>informloci</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Remove all non-phylogentically informative loci</h2>

<h3>Description</h3>

<p>This function will facilitate in removing phylogenetically uninformative loci
from a <code>genclone</code> or <code>genind</code> object. 
The user has the ability to define what uninformative means by setting a 
cutoff value for either percentage of differentiating genotypes or minor 
allele frequency.
</p>


<h3>Usage</h3>

<pre><code class="language-R">informloci(pop, cutoff = 2/nInd(pop), MAF = 0.01, quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pop</code></td>
<td>
<p>a <code>genclone</code> or <code>genind</code> 
object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cutoff</code></td>
<td>
<p><code>numeric</code>. A number from 0 to 1 defining the minimum 
number of differentiating samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAF</code></td>
<td>
<p><code>numeric</code>. A number from 0 to 1 defining the minimum minor
allele frequency. This is passed as the <code>thresh</code> parameter of
<code>isPoly</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p><code>logical</code>. When <code>quiet = TRUE</code> (default), messages 
indicating the loci removed will be printed to screen. When <code>quiet = 
FALSE</code>, nothing will be printed to screen.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function will remove uninformative loci using a traditional MAF
cutoff (using <code>isPoly</code> from <span class="pkg">adegenet</span>) as well
as analyzing the number of observed genotypes in a locus. This is important
for clonal organisms that can have fixed heterozygous sites not detected by
MAF methods.
</p>


<h3>Value</h3>

<p>A <code>genind</code> object with user-defined informative loci.
</p>


<h3>Note</h3>

<p>This will have a few side effects that affect certain analyses. First, 
the number of multilocus genotypes might be reduced due to the reduced 
number of markers (if you are only using a genind object). Second, if you 
plan on using this data for analysis of the index of association, be sure 
to use the standardized version (rbarD) that corrects for the number of 
observed loci.
</p>


<h3>Author(s)</h3>

<p>Zhian N. Kamvar
</p>


<h3>Examples</h3>

<pre><code class="language-R"># We will use a dummy data set to demonstrate how this detects uninformative
# loci using both MAF and a cutoff.

genos &lt;- c("A/A", "A/B", "A/C", "B/B", "B/C", "C/C")

v &lt;- sample(genos, 100, replace = TRUE)
w &lt;- c(rep(genos[2], 99), genos[3])           # found by cutoff
x &lt;- c(rep(genos[1], 98), genos[3], genos[2]) # found by MAF
y &lt;- c(rep(genos[1], 99), genos[2])           # found by both
z &lt;- sample(genos, 100, replace = TRUE)
dat &lt;- df2genind(data.frame(v = v, w = w, x = x, y = y, z = z), sep = "/")

informloci(dat)

## Not run: 
# Ignore MAF
informloci(dat, MAF = 0)

# Ignore cutoff
informloci(dat, cutoff = 0)

# Real data
data(H3N2)
informloci(H3N2)


## End(Not run)
</code></pre>


</div>