<div class="container">

<table style="width: 100%;"><tr>
<td>bruvo.boot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create a tree using Bruvo's Distance with non-parametric bootstrapping.</h2>

<h3>Description</h3>

<p>Create a tree using Bruvo's Distance with non-parametric bootstrapping.
</p>


<h3>Usage</h3>

<pre><code class="language-R">bruvo.boot(
  pop,
  replen = 1,
  add = TRUE,
  loss = TRUE,
  sample = 100,
  tree = "upgma",
  showtree = TRUE,
  cutoff = NULL,
  quiet = FALSE,
  root = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pop</code></td>
<td>
<p>a <code>genind</code> or <code>genclone</code> object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>replen</code></td>
<td>
<p>a <code>vector</code> of <code>integers</code> indicating the length of the
nucleotide repeats for each microsatellite locus.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>add</code></td>
<td>
<p>if <code>TRUE</code>, genotypes with zero values will be treated under 
the genome addition model presented in Bruvo et al. 2004.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loss</code></td>
<td>
<p>if <code>TRUE</code>, genotypes with zero values will be treated under 
the genome loss model presented in Bruvo et al. 2004.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample</code></td>
<td>
<p>an <code>integer</code> indicated the number of bootstrap replicates 
desired.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tree</code></td>
<td>
<p>any function that can generate a tree from a distance matrix.
Default is <code>upgma</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>showtree</code></td>
<td>
<p><code>logical</code> if <code>TRUE</code>, a tree will be plotted with 
nodelabels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cutoff</code></td>
<td>
<p><code>integer</code> the cutoff value for bootstrap node label values
(between 0 and 100).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p><code>logical</code> defaults to <code>FALSE</code>. If <code>TRUE</code>, a 
progress bar and messages will be suppressed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>root</code></td>
<td>
<p><code>logical</code> This is a parameter passed on to
<code>boot.phylo</code>. If the <code>tree</code> argument produces a
rooted tree (e.g. "upgma"), then this value should be <code>TRUE</code>. If it
produces an unrooted tree (e.g. "nj"), then the value should be
<code>FALSE</code>. By default, it is set to <code>NULL</code>, which will assume an
unrooted phylogeny unless the function name contains "upgma".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>any argument to be passed on to <code>boot.phylo</code>. eg. 
<code>quiet = TRUE</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function will calculate a tree based off of Bruvo's distance
and then utilize <code>boot.phylo</code> to randomly sample loci with
replacement, recalculate the tree, and tally up the bootstrap support
(measured in percent success). While this function can take any tree
function, it has native support for two algorithms: <code>nj</code>
and <code>upgma</code>. If you want to use any other functions,
you must load the package before you use them (see examples).
</p>


<h3>Value</h3>

<p>a tree of class phylo with nodelables
</p>


<h3>Note</h3>

<p><strong>Please refer to the documentation for bruvo.dist for details on
the algorithm.</strong> If the user does not provide a vector of appropriate length
for <code>replen</code> , it will be estimated by taking the minimum difference
among represented alleles at each locus. IT IS NOT RECOMMENDED TO RELY ON
THIS ESTIMATION.
</p>


<h3>Author(s)</h3>

<p>Zhian N. Kamvar, Javier F. Tabima
</p>


<h3>References</h3>

<p>Ruzica Bruvo, Nicolaas K. Michiels, Thomas G. D'Souza, and
Hinrich Schulenburg. A simple method for the calculation of microsatellite
genotype distances irrespective of ploidy level. Molecular Ecology,
13(7):2101-2106, 2004.
</p>


<h3>See Also</h3>

<p><code>bruvo.dist</code>, <code>nancycats</code>, 
<code>upgma</code>, <code>nj</code>, <code>boot.phylo</code>, 
<code>nodelabels</code>, <code>tab</code>, 
<code>missingno</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Please note that the data presented is assuming that the nancycat dataset 
# contains all dinucleotide repeats, it most likely is not an accurate
# representation of the data.

# Load the nancycats dataset and construct the repeat vector.
data(nancycats)
ssr &lt;- rep(2, 9)

# Analyze the 1st population in nancycats

bruvo.boot(popsub(nancycats, 1), replen = ssr)

## Not run: 

# Always load the library before you specify the function.
library("ape")

# Estimate the tree based off of the BIONJ algorithm.

bruvo.boot(popsub(nancycats, 9), replen = ssr, tree = bionj)

# Utilizing  balanced FastME
bruvo.boot(popsub(nancycats, 9), replen = ssr, tree = fastme.bal)

# To change parameters for the tree, wrap it in a function.
# For example, let's build the tree without utilizing subtree-prune-regraft

myFastME &lt;- function(x) fastme.bal(x, nni = TRUE, spr = FALSE, tbr = TRUE)
bruvo.boot(popsub(nancycats, 9), replen = ssr, tree = myFastME)


## End(Not run)

</code></pre>


</div>