<div class="container">

<table style="width: 100%;"><tr>
<td>x.val</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Estimate genome-wide prediction accuracy using cross-validation</h2>

<h3>Description</h3>

<p><code>x.val</code> performs cross-validation (CV) to estimate the accuracy of genome-wide prediction (otherwise known as genomic selection) for a specific training population (TP), i.e. a set of individuals for which phenotypic and genotypic data is available. Cross-validation can be conducted via one of two methods within <code>x.val</code>, see <code>Details</code> for more information.
</p>
<div class="sourceCode"><pre>         NOTE - \code{x.val}, specifically \code{\link[BGLR]{BGLR}} writes and reads files to disk so it is highly recommended to set your working directory
</pre></div>


<h3>Usage</h3>

<pre><code class="language-R">x.val(
  G.in = NULL,
  y.in = NULL,
  min.maf = 0.01,
  mkr.cutoff = 0.5,
  entry.cutoff = 0.5,
  remove.dups = TRUE,
  impute = "EM",
  frac.train = 0.6,
  nCV.iter = 100,
  nFold = NULL,
  nFold.reps = 1,
  return.estimates = FALSE,
  CV.burnIn = 750,
  CV.nIter = 1500,
  models = c("rrBLUP", "BayesA", "BayesB", "BayesC", "BL", "BRR"),
  saveAt = tempdir()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>G.in</code></td>
<td>
<p><code>Matrix</code> of genotypic data. First row contains marker names and the first column contains entry (taxa) names. Genotypes should be coded as follows: </p>

<ul>
<li> <p><code>1</code>: homozygous for minor allele
</p>
</li>
<li> <p><code>0</code>: heterozygous
</p>
</li>
<li> <p><code>-1</code>: homozygous for major allele
</p>
</li>
<li> <p><code>NA</code>: missing data
</p>
</li>
<li>
<p> Imputed genotypes can be passed, see <code>impute</code> below for details
</p>
</li>
</ul>
<p>TIP - Set header=<code>FALSE</code> within <code>read.table</code> or <code>read.csv</code> when importing a tab-delimited file containing data for <code>G.in</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y.in</code></td>
<td>
<p><code>Matrix</code> of phenotypic data. First column contains entry (taxa) names found in <code>G.in</code>, regardless of whether the entry has a phenotype for any or all traits. Additional columns contain phenotypic data; column names should reflect the trait name(s). TIP - Set header=<code>TRUE</code> within <code>read.table</code> or <code>read.csv</code> when importing a tab-delimited file containing dat</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.maf</code></td>
<td>
<p>Optional <code>numeric</code> indicating a minimum minor allele frequency (MAF) when filtering <code>G.in</code>. Markers with an MAF &lt; <code>min.maf</code> will be removed. Default is <code>0.01</code> to remove monomorphic markers. Set to <code>0</code> for no filtering.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mkr.cutoff</code></td>
<td>
<p>Optional <code>numeric</code> indicating the maximum missing data per marker when filtering <code>G.in</code>. Markers missing &gt; <code>mkr.cutoff</code> data will be removed. Default is <code>0.50</code>. Set to <code>1</code> for no filtering.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>entry.cutoff</code></td>
<td>
<p>Optional <code>numeric</code> indicating the maximum missing genotypic data per entry allowed when filtering <code>G.in</code>. Entries missing &gt; <code>entry.cutoff</code> marker data will be removed. Default is <code>0.50</code>. Set to <code>1</code> for no filtering.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remove.dups</code></td>
<td>
<p>Optional <code>logical</code>. If <code>TRUE</code> duplicate entries in the genotype matrix, if present, will be removed. This step may be necessary for missing marker imputation (see <code>impute</code>). Default is <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>impute</code></td>
<td>
<p>Options include <code>c("EM", "mean", "pass")</code>. By default (i.e. <code>"EM"</code>), after filtering missing genotypic data will be imputed via the EM algorithm implemented in <code>rrBLUP-package</code> (<cite>Endelman, 2011</cite>; <cite>Poland et al., 2012</cite>). If <code>"mean"</code> missing genotypic data will be imputed via the 'marker mean' method, also implemented in <code>rrBLUP-package</code>. Enter <code>"pass"</code> if a pre-filtered and imputed genotype matrix is provided to <code>G.in</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>frac.train</code></td>
<td>
<p>Optional <code>numeric</code> indicating the fraction of the TP that is used to estimate marker effects (i.e. the prediction set) under cross-validation (CV) method 1 (see <code>Details</code>). The remaining <code class="reqn">(1-frac.trait)</code> of the TP will then comprise the prediction set.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nCV.iter</code></td>
<td>
<p>Optional <code>integer</code> indicating the number of times to iterate <em>CV method 1</em> described in <code>Details</code>. Default is <code>100</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nFold</code></td>
<td>
<p>Optional <code>integer</code>. If a number is provided, denoting the number of "folds", then CV will be conducted using <em>CV method 2</em> (see <code>Details</code>). Default is <code>NULL</code>, resulting in the default use of the <em>CV method 1</em>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nFold.reps</code></td>
<td>
<p>Optional <code>integer</code> indicating the number of times <em>CV method 2</em> is repeated. The CV accuracy returned is the average <em>r</em> of each rep. Default is <code>1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return.estimates</code></td>
<td>
<p>Optional <code>logical</code>. If <code>TRUE</code> additional items including the marker effect and beta estimates from the selected prediction model (i.e. highest CV accuracy) will be returned.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CV.burnIn</code></td>
<td>
<p>Optional <code>integer</code> argument used by <code>BGLR</code> when fitting Bayesian models. Default is <code>750</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CV.nIter</code></td>
<td>
<p>Optional <code>integer</code> argument used by <code>BGLR</code> (<cite>de los Compos and Rodriguez, 2014</cite>) when fitting Bayesian models. Default is <code>1500</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>models</code></td>
<td>
<p>Optional <code>character vector</code> of the regression models to be used in CV and to estimate marker effects. Options include <code>rrBLUP, BayesA, BayesB, BayesC, BL, BRR</code>, one or more may be included at a time. By default all models are tested.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>saveAt</code></td>
<td>
<p>When using models other than "rrBLUP" (i.e. Bayesian models), this is a path and prefix for saving temporary files
the are produced by the <code>BGLR</code> function.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Two CV methods are available within <code>PopVar</code>: </p>

<ul>
<li> <p><code>CV method 1</code>: During each iteration a training (i.e. model training) set will be <strong>randomly sampled</strong> from the TP of size <code class="reqn">N*(frac.train)</code>, where <em>N</em> is the size of the TP, and the remainder of the TP is assigned to the validation set. The accuracies of individual models are expressed as average Pearson's correlation coefficient (<em>r</em>) between the genome estimated breeding value (GEBV) and observed phenotypic values in the validation set across all <code>nCV.iter</code> iterations. Due to its amendibility to various TP sizes, <em>CV method 1</em> is the default CV method in <code>pop.predict</code>.
</p>
</li>
<li> <p><code>CV method 2</code>: <code>nFold</code> <strong>independent</strong> validation sets are sampled from the TP and predicted by the remainder. For example, if <code class="reqn">nFold = 10</code> the TP will be split into 10 equal sets, each containing <code class="reqn">1/10</code>-th of the TP, which will be predicted by the remaining <code class="reqn">9/10</code>-ths of the TP. The accuracies of individual models are expressed as the average (<em>r</em>) between the GEBV and observed phenotypic values in the validation set across all <code>nFold</code> folds. The process can be repeated <code>nFold.reps</code> times with <code>nFold</code> new independent sets being sampled each replication, in which case the reported prediction accuracies are averages across all folds and replications.
</p>
</li>
</ul>
<h3>Value</h3>

<p>A list containing: </p>

<ul>
<li> <p><code>CVs</code> A <code>dataframe</code> of CV results for each trait/model combination specified
</p>
</li>
<li>
<p> If <code>return.estimates</code> is <code>TRUE</code> the additional items will be returned: </p>

<ul>
<li> <p><code>models.used</code> A <code>list</code> of the models chosen to estimate marker effects for each trait
</p>
</li>
<li> <p><code>mkr.effects</code> A <code>vector</code> of marker effect estimates for each trait generated by the respective prediction model used
</p>
</li>
<li> <p><code>betas</code> A <code>list</code> of beta values for each trait generated by the respective prediction model used
</p>
</li>
</ul>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">
## CV using method 1 with 25 iterations
CV.mthd1 &lt;- x.val(G.in = G.in_ex, y.in = y.in_ex, nCV.iter = 25)
CV.mthd1$CVs

## CV using method 2 with 5 folds and 3 replications
x.val(G.in = G.in_ex, y.in = y.in_ex, nFold = 5, nFold.reps = 3)

</code></pre>


</div>