<div class="container">

<table style="width: 100%;"><tr>
<td>think_barley.rda</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>An example barley dataset</h2>

<h3>Description</h3>

<p>A sample dataset, previously described in <cite>Sallam et al. (2014)</cite> is provided as an example of the proper formatting of input files and also for users to become familiar with <code>PopVar</code>; the <code>think_barley</code> dataset is useful in demonstrating both <code>pop.predict</code> and <code>x.val</code>. Note that a number of entries are missing data for one or both traits,
which is representative of a real breeding scenario where phenotypic data may not be available for all parent candidates.
</p>


<h3>Format</h3>

<p>The names of the example files are:
</p>

<dl>
<dt>G.in_ex</dt>
<dd>
<p>A set of 245 barley lines genotyped with 742 SNP markers</p>
</dd>
<dt>G.in_ex_mat</dt>
<dd>
<p>A n x p matrix of n = 245 barley lines genotyped with p = 742 SNP markers</p>
</dd>
<dt>G.in_ex_imputed</dt>
<dd>
<p>A n x p matrix of n = 245 barley lines and p = 742 <em>imputed</em> SNP marker genotypes</p>
</dd>
<dt>y.in_ex</dt>
<dd>
<p>Phenotypes of four traits for a portion of the 245 barley lines, Fusarium head blight (FHB), deoxynivalenol (DON) in ppm, grain yield in bushels/acre, and plant height in cm.</p>
</dd>
<dt>map.in_ex</dt>
<dd>
<p>Genetic map (i.e. chromosome assignment and genetic distance (cM) between markers) of the 742 SNP markers based on <cite>Munoz-Amatriain et al., 2011</cite></p>
</dd>
<dt>cross.tab_ex</dt>
<dd>
<p>A table of user-defined crosses</p>
</dd>
</dl>
<h3>References</h3>

<p>Sallam, A.H., J.B. Endelman, J-L. Jannink, and K.P. Smith. 2015. Assessing Genomic Selection Prediction Accuracy in a Dynamic Barley Breeding Population. Plant Gen. 8(1)
</p>


</div>