<div class="container">

<table style="width: 100%;"><tr>
<td>cALD</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculation of the <code class="reqn">D'</code>, <code class="reqn">Wn</code>, and conditional Asymmetric LD Measures</h2>

<h3>Description</h3>

<p>Calculates <code class="reqn">D'</code>, <code class="reqn">Wn</code> (Cramer's V) and Thomson and Single's conditional asymmetric LD (<code class="reqn">ALD</code>) measures for pairs of loci.
</p>


<h3>Usage</h3>

<pre><code class="language-R">cALD(
  dataSet,
  inPhase = FALSE,
  verbose = TRUE,
  saveVector = FALSE,
  vectorName = "",
  vectorPrefix = "",
  vecDir = tempdir()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dataSet</code></td>
<td>
<p>A data frame or tab delimited file consisting of four columns of genotype data named, e.g. locus1_1 locus1_2 locus2_1 locus2_2, with 1 row per sample. The columns must be organized in this exact order, but the column names should not have _1 or _2 appended; use the same locus name for each column of a given locus. For phased data, locus1_1 is in phase with locus2_1, and locus1_2 is in phase with locus_2_2. Because this funciton operates on locus pairs, any rows with missing data should be excluded from the input genotype data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>inPhase</code></td>
<td>
<p>A boolean identifying the genotyping data as phased or unphased (TRUE = phased; FALSE = unphased); default is unphased.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>A boolean identifying if results should be printed to the console (verbose = TRUE), or returned in a vector of (<code class="reqn">D'</code>, <code class="reqn">Wn</code>, <code class="reqn">WLocus2/Locus1</code>, <code class="reqn">WLocus1/Locus2</code>, number of haplotypes) (verbose = FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>saveVector</code></td>
<td>
<p>A boolean identifying if the vector of all haplotypes should be exported as a text file (saveVector = TRUE), or not (saveVector = FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vectorName</code></td>
<td>
<p>A name for the exported haplotype vector file; this name is not used if saveVector = FALSE. If a name is unspecified, then a filename including the locus-pair and a timestamp is generated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vectorPrefix</code></td>
<td>
<p>An optional prefix for the haplotpe vector to be used if saveVector = TRUE. This prefix will be appended, along with the phase status, before the locus name and timestamp. LDWrap() uses this parameter to identify the dataset and haplotype information passed to cALD().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vecDir</code></td>
<td>
<p>The directory into which the haplotype vector should be written if saveVector = TRUE. The default is the directory specified by tempdir().</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>LD results can be directed to the console or to a data file or data frame object. This function can generate a haplotype vector file for each locus pair analyzed, and will return the LD results eiher in the console, or as a data frame object.
The implementation of ALD applied here is calculated using individual <code class="reqn">Dij</code> LD values and allele frequencies.
</p>


<h3>Value</h3>

<p>A vector of D', Wn, WLocus2/Locus1, WLocus1/Locus2 values, and the number of haplotypes evaluated
</p>


<h3>References</h3>

<p>Thomson G. &amp; Single R.M. GENETICS 2014;198(1):321-31. https://doi.org/10.1534/genetics.114.165266
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Analyze the first 10 rows of the included drb1.dqb1.demo genotype dataset
# and report LD results to the console.
cALD(drb1.dqb1.demo[1:10,])
# Alternatively, return a vector of LD results.
LDvec &lt;- cALD(drb1.dqb1.demo[1:10,],verbose=FALSE)
</code></pre>


</div>