<div class="container">

<table style="width: 100%;"><tr>
<td>hla.hap.demo</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Example Six-Locus HLA Haplotype Data in GL String Format</h2>

<h3>Description</h3>

<p>A data frame of experimentally phased genotype data for the HLA-A, -C, -B, -DRB1, -DQB1 and DQB1 loci. These haplotypes are
for 419 unrelated European American control subjects included in a study of Multiple Sclerosis. These haplotypes were experimentally phased using the EM algorithm; low-frequency haplotypes (with counts &lt; 3) are assigned stochasically by the EM method, so the phase "quality" of these haplotypes should be considered lower than that for haplotypes determined via family segregation analysis.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(hla.hap.demo)
</code></pre>


<h3>Format</h3>

<p>A data frame with 419 rows and two columns
</p>

<ul>
<li>
<p>Relation: the genotyped individual's status in a family
</p>
</li>
<li>
<p>Gl String: GL String formatted multilocus phased HLA genotype
</p>
</li>
</ul>
<h3>Note</h3>

<p>This data is formatted as example input for the LDWrap() function, but is not actual family data. Actual family data would include 'mother', 'father' and 'child' in the "Relation" field. Nevertheless, including 'Subject' in this field is sufficient for analysis using LDWrap().
</p>


<h3>Source</h3>

<p>immport.org study SDY1045 doi:10.21430/M3QW34U2SG
</p>


<h3>References</h3>


<ul>
<li>
<p>Published study: Mack et al. Genes Immun. 2019;20(4):308-326. doi: 10.1038/s41435-017-0006-8.
</p>
</li>
<li>
<p>GL String format: Milius et al. Tissue Antigens. 2013;82(2):106-12. doi: 10.1111/tan.12150.
</p>
</li>
</ul>
</div>