<div class="container">

<table style="width: 100%;"><tr>
<td>powerEQTL.scRNAseq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Power Calculation for Association Between Genotype and Gene Expression Based on Single Cell RNAseq Data
</h2>

<h3>Description</h3>

<p>Power calculation for association between genotype and gene expression based on single cell RNAseq data. This function can be used to calculate one of the 4 parameters (power, sample size, minimum detectable slope, and minimum allowable MAF) by setting the corresponding parameter
as NULL and providing values for the other 3 parameters.
</p>


<h3>Usage</h3>

<pre><code class="language-R">powerEQTL.scRNAseq(
  slope, 
  n, 
  m, 
  power = NULL,
  sigma.y, 
  MAF = 0.2, 
  rho = 0.8, 
  FWER = 0.05,
  nTests = 1,
  n.lower = 2.01,
  n.upper = 1e+30)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>slope</code></td>
<td>

<p>numeric. Slope under alternative hypothesis.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>

<p>integer. Total number of subjects.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m</code></td>
<td>

<p>integer. Number of cells per subject.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>power</code></td>
<td>

<p>numeric. Power for testing if the slope is equal to zero.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigma.y</code></td>
<td>

<p>numeric. Standard deviation of the gene expression.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAF</code></td>
<td>

<p>numeric. Minor allele frequency (between 0 and 0.5).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rho</code></td>
<td>

<p>numeric. Intra-class correlation (i.e., correlation between 
<code class="reqn">y_{ij}</code> and <code class="reqn">y_{ik}</code> for the <code class="reqn">j</code>-th  and <code class="reqn">k</code>-th cells of the <code class="reqn">i</code>-th subject).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FWER</code></td>
<td>

<p>numeric. Family-wise Type I error rate for one pair (SNP, gene).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nTests</code></td>
<td>

<p>integer. Number of tests (i.e., number of all (SNP, gene) pairs) in eQTL analysis.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.lower</code></td>
<td>

<p>numeric. Lower bound of the total number of subjects. Only used when calculating total number of subjects.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.upper</code></td>
<td>

<p>numeric. Upper bound of the total number of subjects. Only used when calculating total number of subjects.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>We assume the following simple linear mixed effects model for each
(SNP, gene) pair to characterize the association between
genotype and gene expression:
</p>
<p style="text-align: center;"><code class="reqn">y_{ij} = \beta_{0i} + \beta_1 * x_i + \epsilon_{ij},</code>
</p>

<p>where 
</p>
<p style="text-align: center;"><code class="reqn">\beta_{0i} \sim N\left(\beta_0, \sigma^2_{\beta}\right),</code>
</p>

<p>and
</p>
<p style="text-align: center;"><code class="reqn">\epsilon_{ij} \sim N\left(0, \sigma^2\right),</code>
</p>

<p><code class="reqn">i=1,\ldots, n</code>, <code class="reqn">j=1,\ldots, m</code>, <code class="reqn">n</code> is the number of
subjects, <code class="reqn">m</code> is the number of cells per subject,
<code class="reqn">y_{ij}</code> is the gene expression level for the <code class="reqn">j</code>-th cell
of the <code class="reqn">i</code>-th subject, <code class="reqn">x_i</code> is the genotype for the
<code class="reqn">i</code>-th subject using additive coding. That is, <code class="reqn">x_i=0</code> indicates the <code class="reqn">i</code>-th subject is a wildtype homozygote,
<code class="reqn">x_i=1</code> indicates the <code class="reqn">i</code>-th subject is a heterozygote,
and <code class="reqn">x_i=2</code> indicates the <code class="reqn">i</code>-th subject is a mutation
homozygote.
</p>
<p>We would like to test the following hypotheses:
</p>
<p style="text-align: center;"><code class="reqn">H_0: \beta_1=0,</code>
</p>

<p>and
</p>
<p style="text-align: center;"><code class="reqn">H_1: \beta_1 = \delta,</code>
</p>

<p>where <code class="reqn">\delta\neq 0</code>.
</p>
<p>For a given SNP, we assume Hardy-Weinberg Equilibrium and denote the minor allele
frequency of the SNP as <code class="reqn">\theta</code>.
</p>
<p>We can derive the power calculation formula is
</p>
<p style="text-align: center;"><code class="reqn">power=1- \Phi\left(z_{\alpha^{*}/2}-a\times b\right)
+\Phi\left(-z_{\alpha^{*}/2} - a\times b\right),</code>
</p>

<p>where
</p>
<p style="text-align: center;"><code class="reqn">a=
\frac{\sqrt{2\theta\left(1-\theta\right)}}{\sigma_y}</code>
</p>

<p>and
</p>
<p style="text-align: center;"><code class="reqn">
b=\frac{\delta\sqrt{m(n-1)}}{\sqrt{1+(m-1)\rho}}
</code>
</p>

<p>and <code class="reqn">z_{\alpha^{*}/2}</code> is the upper <code class="reqn">100\alpha^{*}/2</code>
percentile of the standard normal distribution,
<code class="reqn">\alpha^{*}=FWER/nTests</code>, nTests is the number of
(SNP, gene) pairs,
<code class="reqn">\sigma_y=\sqrt{\sigma^2_{\beta}+\sigma^2}</code>,
and <code class="reqn">\rho=\sigma^2_{\beta}/\left(\sigma^2_{\beta}+\sigma^2\right)</code> is the intra-class correlation.
</p>


<h3>Value</h3>

<p>power if the input parameter <code>power = NULL</code>.
</p>
<p>sample size (total number of subjects) if the input parameter <code>n = NULL</code>;
</p>
<p>minimum detectable slope if the input parameter <code>slope = NULL</code>;
</p>
<p>minimum allowable MAF if the input parameter <code>MAF = NULL</code>.
</p>


<h3>Author(s)</h3>

<p>Xianjun Dong &lt;XDONG@rics.bwh.harvard.edu&gt;,
Xiaoqi Li&lt;xli85@bwh.harvard.edu&gt;,
Tzuu-Wang Chang &lt;Chang.Tzuu-Wang@mgh.harvard.edu&gt;,
Scott T. Weiss &lt;restw@channing.harvard.edu&gt;,
Weiliang Qiu &lt;weiliang.qiu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Dong X, Li X, Chang T-W, Scherzer CR, Weiss ST, and Qiu W. 
powerEQTL: An R package and shiny application for sample size and power calculation of bulk tissue and single-cell eQTL analysis.
Bioinformatics, 2021;, btab385
</p>


<h3>Examples</h3>

<pre><code class="language-R">  n = 102
  m = 227868
  
  # calculate power
  power = powerEQTL.scRNAseq(
    slope = 0.6, 
    n = n, 
    m = m,
    power = NULL,
    sigma.y = 0.29, 
    MAF = 0.05, 
    rho = 0.8, 
    nTests = 1e+6)

  print(power)
  
  # calculate sample size (total number of subjects)
  n = powerEQTL.scRNAseq(
    slope = 0.6, 
    n = NULL, 
    m = m,
    power = 0.9567288,
    sigma.y = 0.29, 
    MAF = 0.05, 
    rho = 0.8, 
    nTests = 1e+6)

  print(n)

  # calculate slope
  slope = powerEQTL.scRNAseq(
    slope = NULL, 
    n = n, 
    m = m,
    power = 0.9567288,
    sigma.y = 0.29, 
    MAF = 0.05, 
    rho = 0.8, 
    nTests = 1e+6)

  print(slope)
  
  # calculate MAF
  MAF = powerEQTL.scRNAseq(
    slope = 0.6, 
    n = n, 
    m = m,
    power = 0.9567288,
    sigma.y = 0.29, 
    MAF = NULL, 
    rho = 0.8, 
    nTests = 1e+6)
  print(MAF)

</code></pre>


</div>