<div class="container">

<table style="width: 100%;"><tr>
<td>powerGG</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Power for GxG interactions in genetic association studies</h2>

<h3>Description</h3>

<p>This routine carries out (analytical, approximate) power calculations for identifying Gene-Gene interactions in Genome Wide Association Studies</p>


<h3>Usage</h3>

<pre><code class="language-R">powerGG(n, power, model, caco, alpha, alpha1)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>Sample size: combined number of cases and controls. Note: exactly one of <code>n</code> and <code>power</code> should be specified.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>power</code></td>
<td>
<p>Power: targeted power. Note: exactly one of <code>n</code> and <code>power</code> should be specified.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>List specifying the genetic model. This list contains the following objects:
</p>

<ul>
<li> <p><code>prev</code> Prevalence of the outcome in the <em>population</em>. Note that for case-only and empirical Bayes estimators to be valid,
the prevalence needs to be low.
</p>
</li>
<li> <p><code>pGene1</code> Probability that the first <em>binary</em> SNP is 1 (i.e. not the minor allele frequency for a three level SNP).
</p>
</li>
<li> <p><code>pGene2</code> Probability that the first <em>binary</em> SNP is 1 (i.e. not the minor allele frequency for a three level SNP).
</p>
</li>
<li> <p><code>beta.LOR</code> Vector of length three with the odds ratios of the first genetic, second genetic, and GxG interaction effect, respectively.
</p>
</li>
<li> <p><code>nSNP</code> Number of SNPs (genes) being tested.</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>caco</code></td>
<td>
<p>Fraction of the sample that are cases (default = 0.5).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>Overall (family-wise) Type 1 error (default = 0.05).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha1</code></td>
<td>
<p>Significance level at which testing during the first stage (screening) takes place. If alpha1 = 1, there is no screening.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The routine computes power calculations for a two-stage procedure with marginal screening followed by either case-control or case-only testing.</p>


<h3>Value</h3>

<p>A data frame consisting of two numbers: the power for the case-control and case-only approaches if <code>n</code> is specified or the required combined sample size 
for the case-control and case-only approaches if <code>power</code> is specified.</p>


<h3>Author(s)</h3>

<p>Charles Kooperberg, <a href="mailto:clk@fredhutch.org">clk@fredhutch.org</a></p>


<h3>References</h3>

<p>Kooperberg C, LeBlanc M (2008). Increasing the power of identifying gene x gene interactions in genome-wide association studies.
<em>Genetic Epidemiology</em>, <b>32</b>, 255-263.</p>


<h3>See Also</h3>

<p>powerGG</p>


<h3>Examples</h3>

<pre><code class="language-R">mod1 &lt;- list(prev=0.05, pGene1=0.3, pGene2=0.3, beta.LOR=c(0,0,.6),nSNP=500000)
powerGG(n=10000,mod=mod1,caco=0.5,alpha=.05,alpha1=.001)
powerGG(power=0.8,mod=mod1,caco=0.5,alpha=.05,alpha1=.001)
</code></pre>


</div>