<div class="container">

<table style="width: 100%;"><tr>
<td>powerEQTL.ANOVA</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Power Calculation for EQTL Analysis Based on Un-Balanced One-Way ANOVA
</h2>

<h3>Description</h3>

<p>Power calculation for eQTL analysis that tests if a SNP is associated to a gene probe by using un-balanced one-way ANOVA. This function can be used to calculate one of the 3 parameters (power, sample size, and minimum allowable MAF) by setting the corresponding parameter
as NULL and providing values for the other 2 parameters.
</p>


<h3>Usage</h3>

<pre><code class="language-R">powerEQTL.ANOVA(MAF,
                deltaVec=c(-0.13, 0.13),
                n=200,
                power = NULL,
                sigma = 0.13,
                FWER = 0.05,
                nTests = 200000,
                n.lower = 4,
                n.upper = 1e+30)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>MAF</code></td>
<td>

<p>numeric. Minor allele frequency.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>deltaVec</code></td>
<td>

<p>numeric. A vector having 2 elements. The first element is equal to
<code class="reqn">\mu_2-\mu_1</code> and the second element is equal to
<code class="reqn">\mu_3-\mu_2</code>, where <code class="reqn">\mu_1</code> is
the mean gene expression level for the mutation homozygotes,
<code class="reqn">\mu_2</code> is the mean gene expression level for the heterozygotes,
and <code class="reqn">\mu_3</code> is the mean gene expression level for the 
wild-type gene expression level.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>

<p>integer. Total number of subjects.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>power</code></td>
<td>

<p>numeric. Power for testing if 3 genotypes have the same mean 
gene expression levels.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigma</code></td>
<td>

<p>numeric. Standard deviation of the random error.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FWER</code></td>
<td>

<p>numeric. Family-wise Type I error rate.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nTests</code></td>
<td>

<p>integer. Number of tests (i.e., number of all (SNP, gene) pairs) in eQTL analysis.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.lower</code></td>
<td>

<p>numeric. Lower bound of the total number of subjects. Only used when calculating total number of subjects.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.upper</code></td>
<td>

<p>numeric. Upper bound of the total number of subjects. Only used when calculating total number of subjects.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If we would like to test potential non-linear relationship between genotype of a SNP and expression of a gene, we can use un-balanced one-way ANOVA. Actually, an article published by the GTEx Consortium in 2013 used this approach. 
</p>
<p>Suppose there are <code class="reqn">k=3</code> groups of subjects: (1) mutation homozygotes; (2) heterozygotes; and (3) wildtype homozygotes. We would like to test if the mean expression <code class="reqn">\mu_i</code>, <code class="reqn">i=1, \ldots, k</code>, of the gene is the same among the <code class="reqn">k</code> groups of subjects. We can use the following one-way ANOVA model to characterize the relationship between observed gene expression level <code class="reqn">y_{ij}</code> and the population mean expression level <code class="reqn">\mu_i</code>:
</p>
<p style="text-align: center;"><code class="reqn">y_{ij} = \mu_i + \epsilon_{ij}, \quad \epsilon_{ij} \sim N\left(0, \sigma^2\right),</code>
</p>

<p>where <code class="reqn">i=1,\ldots, k</code>, <code class="reqn">j= 1, \ldots, n_i</code>,
<code class="reqn">y_{ij}</code> is the observed gene expression level for the <code class="reqn">j</code>-th subject in the <code class="reqn">i</code>-th group, <code class="reqn">\mu_i</code> is the mean gene expression level of the <code class="reqn">i</code>-th group, <code class="reqn">\epsilon_{ij}</code> is the random error, <code class="reqn">k</code> is the number of groups, <code class="reqn">n_i</code> is the number of subjects in the <code class="reqn">i</code>-th group. 
Denote the total number of subjects as <code class="reqn">N = \sum_{i=1}^{k} n_i</code>.
That is, we have <code class="reqn">n_1</code> mutation homozygotes, <code class="reqn">n_2</code> heterozygotes, and <code class="reqn">n_3</code> wildtype homozygotes.
</p>
<p>We would like to test the null hypothesis <code class="reqn">H_0</code> and alternative hypothesis <code class="reqn">H_1</code>:
</p>
<p style="text-align: center;"><code class="reqn">H_0:  \mu_1 = \mu_2 = \mu_3,</code>
</p>

<p style="text-align: center;"><code class="reqn">H_1:  \mbox{not all means are the same}.</code>
</p>

<p>According to O'Brien and Muller (1993), the power calculation formula for unbalanced one-way ANOVA is
</p>
<p style="text-align: center;"><code class="reqn">power=Pr\left(\left.F\geq F_{1-\alpha}\left(k-1, N-k\right)\right| F\sim
F_{k-1, N-k, \lambda}\right),</code>
</p>

<p>where <code class="reqn">k=3</code> is the number of groups of subjects, <code class="reqn">N</code> is the total number
of subjects, <code class="reqn">F_{1-\alpha}\left(k-1, N-k\right)</code> is the 
<code class="reqn">100(1-\alpha)</code>-th percentile of central F distribution with degrees of freedoms <code class="reqn">k-1</code> and <code class="reqn">N-k</code>, and <code class="reqn">F_{k-1, N-k, \lambda}</code> is the non-central F distribution
with degrees of freedoms <code class="reqn">k-1</code> and <code class="reqn">N-k</code> and non-central parameter (ncp)
<code class="reqn">\lambda</code>. The ncp <code class="reqn">\lambda</code> is equal to
</p>
<p style="text-align: center;"><code class="reqn">\lambda=\frac{N}{\sigma^2}\sum_{i=1}^{k} w_i \left(\mu_i-\mu\right)^2,</code>
</p>
<p> where <code class="reqn">\mu_i</code> is the mean gene expression level
for the <code class="reqn">i</code>-th group of subjects, <code class="reqn">w_i</code> is the weight for the <code class="reqn">i</code>-th group of subjects, <code class="reqn">\sigma^2</code> is the variance of the random errors in ANOVA (assuming each group has equal variance), and
<code class="reqn">\mu</code> is the weighted mean gene expression level
</p>
<p style="text-align: center;"><code class="reqn">\mu=\sum_{i=1}^{k}w_i \mu_i.</code>
</p>

<p>The weights <code class="reqn">w_i=n_i/N</code> are the sample proportions for the 3 groups of subjects, where <code class="reqn">N=n_1+n_2+n_3</code> is the total number of subjects. Hence, 
<code class="reqn">\sum_{i=1}^{3}w_i = 1</code>.
Based on Hardy-Weinberg Equilibrium, we have
<code class="reqn">w_1 = \theta^2</code>, <code class="reqn">w_2 = 2\theta(1-\theta)</code>, and <code class="reqn">w_3 = (1-\theta)^2</code>, 
where <code class="reqn">\theta</code> is MAF.
</p>
<p>Without loss of generality, we set <code class="reqn">\mu_1 = -\delta_1</code>, <code class="reqn">\mu_2=0</code>,
and <code class="reqn">\mu_3 = \delta_2</code>.
</p>
<p>We adopted the parameters from the GTEx cohort (see the Power analysis" section of Nature Genetics, 2013; https://www.nature.com/articles/ng.2653), where they modeled the expression data as having a log-normal distribution with a log standard deviation of 0.13 within each genotype class (AA, AB, BB). This level of noise is based on estimates from initial GTEx data. In their power analysis, they assumed the across-genotype difference delta = 0.13 (i.e., equivalent to detecting a log expression change similar to the standard deviation within a single genotype class). 
</p>


<h3>Value</h3>

<p>power if the input parameter <code>power = NULL</code>.
</p>
<p>sample size (total number of subjects) if the input parameter <code>n = NULL</code>;
</p>
<p>minimum detectable slope if the input parameter <code>slope = NULL</code>.
</p>


<h3>Author(s)</h3>

<p>Xianjun Dong &lt;XDONG@rics.bwh.harvard.edu&gt;,
Xiaoqi Li&lt;xli85@bwh.harvard.edu&gt;,
Tzuu-Wang Chang &lt;Chang.Tzuu-Wang@mgh.harvard.edu&gt;,
Scott T. Weiss &lt;restw@channing.harvard.edu&gt;,
Weiliang Qiu &lt;weiliang.qiu@gmail.com&gt;
</p>


<h3>References</h3>

<p>The GTEx Consortium. 
The Genotype-Tissue Expression (GTEx) project. 
Nature Genetics, 45:580-585, 2013.
</p>
<p>O'Brien, RG and Muller, KE. 
Unified power analysis for t-tests through multivariate hypotheses. 
In LK Edwards, editor, Applied Analysis of Variance in Behavioral Science,
pages 297-344. New York: Dekker, 1993.
</p>
<p>Dong X, Li X, Chang T-W, Scherzer CR, Weiss ST, and Qiu W. 
powerEQTL: An R package and shiny application for sample size and power calculation of bulk tissue and single-cell eQTL analysis.
Bioinformatics, 2021;, btab385
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# calculate power
powerEQTL.ANOVA(MAF = 0.1,
                deltaVec = c(-0.13, 0.13),
                n = 282,
                power = NULL,
                sigma = 0.13,
                FWER = 0.05,
                nTests = 200000)
                
# calculate sample size (total number of subjects)
powerEQTL.ANOVA(MAF = 0.1,
                deltaVec = c(-0.13, 0.13),
                n = NULL,
                power = 0.8,
                sigma = 0.13,
                FWER = 0.05,
                nTests = 200000) 
                
# calculate minimum allowable MAF
powerEQTL.ANOVA(MAF = NULL,
                deltaVec = c(-0.13, 0.13),
                n = 282,
                power = 0.8,
                sigma = 0.13,
                FWER = 0.05,
                nTests = 200000)                 
                       
                       
</code></pre>


</div>