<div class="container">

<table style="width: 100%;"><tr>
<td>powerEQTL.scRNAseq.sim</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Power Calculation for Association Between Genotype and Gene Expression Based on Single Cell RNAseq Data via Simulation from ZINB Mixed Effects Regression Model
</h2>

<h3>Description</h3>

<p>Power calculation for association between genotype and gene expression based on single cell RNAseq data via simulation from ZINB mixed effects regression model. This function can be used to calculate one of the 4 parameters (power, sample size, minimum detectable slope, and minimum allowable MAF) by setting the corresponding parameter
as NULL and providing values for the other 3 parameters.
</p>


<h3>Usage</h3>

<pre><code class="language-R">powerEQTL.scRNAseq.sim(slope, 
		       n, 
		       m, 
                       power = NULL,
		       m.int = -1, 
		       sigma.int = 1, 
		       zero.p = 0.1, 
		       theta = 1, 
		       MAF = 0.2, 
		       FWER = 0.05, 
		       nTests = 1, 
		       nSim = 1000, 
		       estMethod = "GLMMadaptive", 
		       nCores = 1,
                       n.lower = 2.01,
                       n.upper = 1e+4,
                       slope.lower =  1e-6,
                       slope.upper = log(1.0e+6),
                       MAF.lower = 0.05,
                       MAF.upper = 0.49
)

</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>slope</code></td>
<td>

<p>numeric. Slope (see details).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>

<p>integer. Total number of subjects.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m</code></td>
<td>

<p>integer. Number of cells per subject.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>power</code></td>
<td>

<p>numeric. Power for testing if the slope is equal to zero.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m.int</code></td>
<td>

<p>numeric. Mean of random intercept (see details).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigma.int</code></td>
<td>

<p>numeric. Standard deviation of random intercept (see details).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>zero.p</code></td>
<td>

<p>numeric. Probability that an excess zero occurs.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>theta</code></td>
<td>

<p>numeric. Dispersion parameter of negative binomial distribution.
The smaller <code>theta</code> is, the larger variance of NB random variable is.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAF</code></td>
<td>

<p>numeric. Minor allele frequency of the SNP.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FWER</code></td>
<td>

<p>numeric. Family-wise Type I error rate.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nTests</code></td>
<td>

<p>integer. Number of tests (i.e., number of all (SNP, gene) pairs) in eQTL analysis.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nSim</code></td>
<td>

<p>integer. Number of simulated datasets to be generated.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>estMethod</code></td>
<td>

<p>character. Indicates which method would be used to fit zero inflated 
negative binomial mixed effects model.
Currently, the possible choice is “GLMMadaptive”.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nCores</code></td>
<td>

<p>integer. Number of computer cores used by <code>mclapply</code> for parallel computing. For Windows, <code>nCores=1</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.lower</code></td>
<td>

<p>numeric. Lower bound of the total number of subjects. Only used when calculating total number of subjects.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.upper</code></td>
<td>

<p>numeric. Upper bound of the total number of subjects. Only used when calculating total number of subjects.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>slope.lower</code></td>
<td>

<p>numeric. Lower bound of the slope. Only used when calculating minimum slope.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>slope.upper</code></td>
<td>

<p>numeric. Upper bound of the slope Only used when calculating minimum slope.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAF.lower</code></td>
<td>

<p>numeric. Lower bound of the MAF. Only used when calculating minimum MAF.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAF.upper</code></td>
<td>

<p>numeric. Upper bound of the MAF Only used when calculating minimum MAF.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function calculates the power for testing if genotypes of a SNP is associated with the expression of a gene via <code>nSim</code> datasets generated from zero-inflated negative binomial (ZINB) regression model.
</p>
<p>Each dataset is generated from zero-inflated negative binomial mixed effects regression model with only one covariate: genotype. That is, the read counts of a gene follows a mixture of 2-component distributions. One component takes only one value: zero. The other component is negative binomial distribution, which takes non-negative values 0, 1, 2, .... The log mean of the negative binomial distribution is linear function of the genotype.
</p>
<p>For each dataset, the p-value for testing if the slope for genotype is equal to zero will be calculated.
</p>
<p>The proportion of p-values <code class="reqn">&lt; \alpha</code> is the estimated power, where <code class="reqn">\alpha = FWER/nTests</code>.
</p>
<p>Each simulated dataset contains gene expression levels of one gene and genotypes of one SNP for subjects with multiple cells.
The gene expression levels (read counts) follow zero-inflated negative binomial distribution. Denote <code class="reqn">Y_{ij}</code> as the read counts for the <code class="reqn">j</code>-th cell of
the <code class="reqn">i</code>-th subject, <code class="reqn">i=1,\ldots, n</code>, <code class="reqn">j=1,\ldots, m</code>,
<code class="reqn">n</code> is the number of subjects, and <code class="reqn">m</code> is the number of cells per subject.
Denote <code class="reqn">p</code> as the probability that <code class="reqn">Y_{ij}=0</code> is an excess zero.
With probability <code class="reqn">1-p</code>, <code class="reqn">Y_{ij}</code> follows a negative binomial distribution <code class="reqn">NB(\mu, \theta)</code>, where <code class="reqn">\mu</code> is the mean (i.e., <code class="reqn">\mu=E(Y_{ij})</code>) and <code class="reqn">\theta</code> is the dispersion parameter.
The variance of the NB distribution is <code class="reqn">\mu + \mu^2/\theta</code>.
The relationship between gene expression and genotype for the <code class="reqn">i</code>-th subject is characterized by the equation 
</p>
<p style="text-align: center;"><code class="reqn">\mu_i = \exp(\beta_{0i} + \beta_{1} x_i),</code>
</p>

<p>where <code class="reqn">\beta_{0i}</code> is the random intercept following a normal
distribution <code class="reqn">N(\beta_0, \sigma^2)</code> to account for within-subject correlation of gene expression, <code class="reqn">\beta_0</code> is the mean of the random intercept, <code class="reqn">\sigma</code> is the standard deviation of the random intercept, <code class="reqn">\beta_1</code> is the slope, and <code class="reqn">x_i</code> is the additive-coded genotype for the SNP with minor allele frequency <code class="reqn">MAF</code>.
</p>
<p>We assume that the SNP satisfies the Hardy-Weinberg Equilibrium. That is, the
probabilities of the 3 genotypes <code class="reqn">(0, 1, 2)</code> are <code class="reqn">(1-MAF)^2, 2 MAF (1-MAF), MAF^2</code>, respectively.
</p>
<p>For simplicity, we assume that excess zeros are caused by technical issues, hence are not related to genotypes.
</p>


<h3>Value</h3>

<p>power if the input parameter <code>power = NULL</code>.
</p>
<p>sample size (total number of subjects) if the input parameter <code>n = NULL</code>;
</p>
<p>minimum detectable slope if the input parameter <code>slope = NULL</code>;
</p>
<p>minimum allowable MAF if the input parameter <code>MAF = NULL</code>.
</p>


<h3>Note</h3>

<p>The speed of simulation approach is slow. It is recommended to run the function <code>powerEQTL.scRNAseq.sim</code> in parallel computing environment (e.g., Unix/Linux) and set <code>nCores &gt; 1</code>.
Also, it is important to set appropriate ranges to search sample size, minimum detectable slope, or minimum allowable MAF. If the function (e.g., GLMMadaptive) to fit data returns an error message 
for a simulated dataset, then the function <code>powerEQTL.scRNAseq.sim</code> will try a few more runs with different simulated datasets. If all the runs failed, zero (for power estimation) or <code>NA</code> or <code>NaN</code> (for estimating slope, sample size, or MAF) would be output.  It is recommended to set <code>nSim &gt;= 1000</code> to get stable results.
</p>


<h3>Author(s)</h3>

<p>Xianjun Dong &lt;XDONG@rics.bwh.harvard.edu&gt;,
Xiaoqi Li&lt;xli85@bwh.harvard.edu&gt;,
Tzuu-Wang Chang &lt;Chang.Tzuu-Wang@mgh.harvard.edu&gt;,
Scott T. Weiss &lt;restw@channing.harvard.edu&gt;,
Weiliang Qiu &lt;weiliang.qiu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Dong X, Li X, Chang T-W, Scherzer CR, Weiss ST, and Qiu W. 
powerEQTL: An R package and shiny application for sample size and power calculation of bulk tissue and single-cell eQTL analysis.
Bioinformatics, 2021;, btab385
</p>


<h3>Examples</h3>

<pre><code class="language-R">
nSubj = 10
nCellPerSubj = 10

# calculate power
power = powerEQTL.scRNAseq.sim(
  slope = 1.62, # slope
  n = nSubj, # total number of subjects
  m = nCellPerSubj, # number of cells per subject 
  power = NULL, # power to be estimated
  m.int = -1, # mean of random intercept
  sigma.int = 1, # SD of the random intercept
  zero.p = 0.01, # probability that an excess zero occurs
  theta = 1, # dispersion parameter of NB distribution NB(mu, theta)
  MAF = 0.45, 
  FWER = 0.05,
  nTests = 1,
  nSim  =  5, # number of simulations
  estMethod = "GLMMadaptive", # parameter estimation method for ZINB
  nCores = 1 # number of computer cores used by 'mclapply'
)

print(power)


</code></pre>


</div>