<div class="container">

<table style="width: 100%;"><tr>
<td>powerLMEnoCov</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Power Calculation for Simple Linear Mixed Effects Model Without Covariate
</h2>

<h3>Description</h3>

<p>Power calculation for simple linear mixed effects model without covariate.
This function can be used to calculate one of the 3 parameters (power, sample size, and minimum detectable slope) by setting the corresponding parameter
as NULL and providing values for the other 2 parameters.
</p>


<h3>Usage</h3>

<pre><code class="language-R">powerLMEnoCov(
  slope, 
  n, 
  m, 
  sigma.y,
  power = NULL,
  rho = 0.8, 
  FWER = 0.05,
  nTests = 1,
  n.lower = 2.01,
  n.upper = 1e+30)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>slope</code></td>
<td>

<p>numeric. Slope under alternative hypothesis.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>

<p>integer. Total number of subjects.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m</code></td>
<td>

<p>integer. Number of pairs of replicates per subject.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigma.y</code></td>
<td>

<p>numeric. Standard deviation of the outcome y.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>power</code></td>
<td>

<p>numeric. Desired power.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rho</code></td>
<td>

<p>numeric. Intra-class correlation (i.e., correlation between 
<code class="reqn">y_{ij}</code> and <code class="reqn">y_{ik}</code> for the <code class="reqn">j</code>-th and <code class="reqn">k</code>-th observations of the <code class="reqn">i</code>-th subject).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FWER</code></td>
<td>

<p>numeric. Family-wise Type I error rate. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nTests</code></td>
<td>

<p>integer. Number of tests (e.g., number of genes in differential expression analysis based on scRNAseq to compare gene expression before and after treatment).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.lower</code></td>
<td>

<p>numeric. Lower bound of the total number of subjects. Only used when calculating total number of subjects.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.upper</code></td>
<td>

<p>numeric. Upper bound of the total number of subjects. Only used when calculating total number of subjects.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>In an experiment, there are <code class="reqn">n</code> samples. For each sample, we get 
<code class="reqn">m</code> pairs of replicates. For each pair, one replicate will receive no treatment. The other
replicate will receive treatment. The outcome is the expression of a gene. The overall goal of the experiment is to check if the treatment affects gene expression level or not. Or equivalently, the overall goal of the experiment is to test if the mean within-pair difference of gene expression is equal to zero or not. In the design stage, we would like to calculate the power/sample size of the experiment for testing if the mean within-pair difference of gene expression is equal to zero or not. 
</p>
<p>We assume the following linear mixed effects model to characterize the 
relationship between the within-pair difference of gene expression <code class="reqn">y_{ij}</code> and
the mean of the within-pair difference <code class="reqn">\beta_{0i}</code>:
</p>
<p style="text-align: center;"><code class="reqn">y_{ij} = \beta_{0i} + \epsilon_{ij},</code>
</p>

<p>where
</p>
<p style="text-align: center;"><code class="reqn">\beta_{0i} \sim N\left(\beta_0, \sigma^2_{\beta}\right),</code>
</p>

<p>and
</p>
<p style="text-align: center;"><code class="reqn">\epsilon_{ij} \sim N\left(0, \sigma^2\right),</code>
</p>

<p><code class="reqn">i=1,\ldots, n</code>, <code class="reqn">j=1,\ldots, m</code>, <code class="reqn">n</code> is the number of
subjects, <code class="reqn">m</code> is the number of pairs of replicates per subject,
<code class="reqn">y_{ij}</code> is the within-pair difference of outcome value for the <code class="reqn">j</code>-th pair of the <code class="reqn">i</code>-th subject. 
</p>
<p>We would like to test the following hypotheses:
</p>
<p style="text-align: center;"><code class="reqn">H_0: \beta_0=0,</code>
</p>

<p>and
</p>
<p style="text-align: center;"><code class="reqn">H_1: \beta_0 = \delta,</code>
</p>

<p>where <code class="reqn">\delta\neq 0</code>. If we reject the null hypothesis <code class="reqn">H_0</code>
based on a sample, we then get evidence that the treatment affects
the gene expression level.
</p>
<p>We can derive the power calculation formula:
</p>
<p style="text-align: center;"><code class="reqn">power=1- \Phi\left(z_{\alpha^{*}/2}-a\times b\right)
+\Phi\left(-z_{\alpha^{*}/2} - a\times b\right),</code>
</p>

<p>where
</p>
<p style="text-align: center;"><code class="reqn">a= 
\frac{1
}{\sigma_y}</code>
</p>

<p>and
</p>
<p style="text-align: center;"><code class="reqn">
b=\frac{\delta\sqrt{mn}}{\sqrt{1+(m-1)\rho}}
</code>
</p>

<p>and <code class="reqn">z_{\alpha^{*}/2}</code> is the upper <code class="reqn">100\alpha^{*}/2</code>
percentile of the standard normal distribution,
<code class="reqn">\alpha^{*}=\alpha/nTests</code>, nTests is the number of 
tests,
<code class="reqn">\sigma_y=\sqrt{\sigma^2_{\beta}+\sigma^2}</code>
and <code class="reqn">\rho=\sigma^2_{\beta}/\left(\sigma^2_{\beta}+\sigma^2\right)</code> is the intra-class correlation.
</p>


<h3>Value</h3>

<p>power if the input parameter <code>power = NULL</code>.
</p>
<p>sample size (total number of subjects) if the input parameter <code>n = NULL</code>;
</p>
<p>minimum detectable slope if the input parameter <code>slope = NULL</code>.
</p>


<h3>Author(s)</h3>

<p>Xianjun Dong &lt;XDONG@rics.bwh.harvard.edu&gt;,
Xiaoqi Li&lt;xli85@bwh.harvard.edu&gt;,
Tzuu-Wang Chang &lt;Chang.Tzuu-Wang@mgh.harvard.edu&gt;,
Scott T. Weiss &lt;restw@channing.harvard.edu&gt;,
Weiliang Qiu &lt;weiliang.qiu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Dong X, Li X, Chang T-W, Scherzer CR, Weiss ST, and Qiu W. 
powerEQTL: An R package and shiny application for sample size and power calculation of bulk tissue and single-cell eQTL analysis.
Bioinformatics, 2021;, btab385
</p>


<h3>Examples</h3>

<pre><code class="language-R">  n = 17
  m = 5
  sigma.y = 0.68
  slope = 1.3*sigma.y
  print(slope)
  
  # estimate power
  power = powerLMEnoCov(
    slope = slope, 
    n = n, 
    m = m,
    sigma.y = sigma.y,
    power = NULL,
    rho = 0.8, 
    FWER = 0.05,
    nTests = 20345)

  print(power)

  # estimate sample size (total number of subjects)
  n = powerLMEnoCov(
    slope = slope, 
    n = NULL, 
    m = m,
    sigma.y = sigma.y,
    power = 0.8721607,
    rho = 0.8, 
    FWER = 0.05,
    nTests = 20345)

  print(n)
  
  # estimate slope
  slope = powerLMEnoCov(
    slope = NULL, 
    n = n, 
    m = m,
    sigma.y = sigma.y,
    power = 0.8721607,
    rho = 0.8, 
    FWER = 0.05,
    nTests = 20345)

  print(slope)
</code></pre>


</div>