<div class="container">

<table style="width: 100%;"><tr>
<td>powerEQTL.SLR</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Power Calculation for EQTL Analysis Based on Simple Linear Regression
</h2>

<h3>Description</h3>

<p>Power calculation for eQTL analysis that tests if a SNP is associated to a gene probe by using simple linear regression. This function can be used to calculate one of the 4 parameters (power, sample size, minimum detectable slope, and minimum allowable MAF) by setting the corresponding parameter
as NULL and providing values for the other 3 parameters.
</p>


<h3>Usage</h3>

<pre><code class="language-R">powerEQTL.SLR(
  MAF,
  slope = 0.13,
  n = 200,
  power = NULL,
  sigma.y = 0.13,
  FWER = 0.05,
  nTests = 2e+05,
  n.lower = 2.01,
  n.upper = 1e+30)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>MAF</code></td>
<td>

<p>numeric. Minor allele frequency.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>slope</code></td>
<td>

<p>numeric. Slope of the simple linear regression.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>

<p>integer. Total number of subjects.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>power</code></td>
<td>

<p>numeric. Power for testing if the slope is equal to zero.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigma.y</code></td>
<td>

<p>numeric. Standard deviation of the outcome <code class="reqn">y_i</code> in simple linear regression.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FWER</code></td>
<td>

<p>numeric. Family-wise Type I error rate.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nTests</code></td>
<td>

<p>integer. Number of tests (i.e., number of all (SNP, gene) pairs) in eQTL analysis.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.lower</code></td>
<td>

<p>numeric. Lower bound of the total number of subjects. Only used when calculating total number of subjects.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.upper</code></td>
<td>

<p>numeric. Upper bound of the total number of subjects. Only used when calculating total number of subjects.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>To test if a SNP is associated with a gene probe, we use the simple linear regression
</p>
<p style="text-align: center;"><code class="reqn">y_i = \beta_0+\beta_1 x_i + \epsilon_i,</code>
</p>

<p>where <code class="reqn">y_i</code> is the gene expression level of the <code class="reqn">i</code>-th subject,
<code class="reqn">x_i</code> is the genotype of the <code class="reqn">i</code>-th subject, and
<code class="reqn">\epsilon_i</code> is the random error term with mean zero and standard deviation <code class="reqn">\sigma</code>. Additive coding for genotype is used. That is,
<code class="reqn">x_i=0</code> indicates wildtype homozygotes;
<code class="reqn">x_i=1</code> indicates heterozygotes; and <code class="reqn">x_i=2</code> indicates mutation heterozygotes.
</p>
<p>To test if the SNP is associated with the gene probe, we test the null hypothesis <code class="reqn">H_0: \beta_1=0</code> versus the alternative hypothesis <code class="reqn">H_1: \beta_1 = \delta</code>, where <code class="reqn">\delta\neq 0</code>.
</p>
<p>Denote <code class="reqn">\theta</code> as the minor allele frequency (MAF) of the SNP. Under Hardy-Weinberg equilibrium, we can calculate the variance of genotype of the SNP: 
<code class="reqn">\sigma^2_x=2 \theta (1-\theta)</code>, 
where <code class="reqn">\sigma^2_x</code> is the variance of the
predictor (i.e. the SNP) <code class="reqn">x_i</code>.
</p>
<p>The exact power calculation formula can be derived as
</p>
<p style="text-align: center;"><code class="reqn">1-T_{n-2, \lambda}(t_{n-2}(\alpha/2)) + T_{n-2, \lambda}(-t_{n-2}(\alpha/2)),</code>
</p>

<p>where <code class="reqn">T_{n-2, \lambda}(a)</code> is the value at <code class="reqn">a</code> of cumulative distribution function of non-central t distribution with <code class="reqn">n-2</code> degrees of freedom
and non-centrality parameter <code class="reqn">\lambda=\delta/\sqrt{\sigma^2/[(n-1)\tilde{\sigma}^2_{x}]}</code>. And <code class="reqn">\tilde{\sigma}^2_{x}=\sum_{i=1}^n(x_i - \bar{x})^2/(n-1)</code>. 
</p>
<p>Dupont and Plummer (1998) mentioned the following relationship:
</p>
<p style="text-align: center;"><code class="reqn">\sigma^2 = \sigma^2_y - \beta_1^2 \sigma^2_x.</code>
</p>

<p>So we can plug in the above equation to the power calculation formula.
</p>
<p>Under Hardy-Weinberg equilibrium, we have <code class="reqn">\sigma_x^2=2\theta(1-\theta)</code>,
where <code class="reqn">\theta</code> is the minor allele frequency (MAF).
</p>
<p>Hence, the non-centrality parameter can be rewritten as
</p>
<p style="text-align: center;"><code class="reqn">\lambda=\frac{\delta}{\sqrt{
\left(\sigma_y^2 - \delta^2 2\left(1-\hat{\theta}\right)\hat{\theta}\right)/
\left[(n-1)2\left(1-\hat{\theta}\right)\hat{\theta}\right]
}}</code>
</p>

<p>We adopted the parameters from the GTEx cohort (see the Power analysis" section of Nature Genetics, 2013; https://www.nature.com/articles/ng.2653), where they modeled the expression data as having a log-normal distribution with a log standard deviation of 0.13 within each genotype class (AA, AB, BB). This level of noise is based on estimates from initial GTEx data. In their power analysis, they assumed the across-genotype difference delta = 0.13 (i.e., equivalent to detecting a log expression change similar to the standard deviation within a single genotype class). 
</p>


<h3>Value</h3>

<p>power if the input parameter <code>power = NULL</code>.
</p>
<p>sample size (total number of subjects) if the input parameter <code>n = NULL</code>;
</p>
<p>minimum detectable slope if the input parameter <code>slope = NULL</code>;
</p>
<p>minimum allowable MAF if the input parameter <code>MAF = NULL</code>.
</p>


<h3>Author(s)</h3>

<p>Xianjun Dong &lt;XDONG@rics.bwh.harvard.edu&gt;,
Xiaoqi Li&lt;xli85@bwh.harvard.edu&gt;,
Tzuu-Wang Chang &lt;Chang.Tzuu-Wang@mgh.harvard.edu&gt;,
Scott T. Weiss &lt;restw@channing.harvard.edu&gt;,
Weiliang Qiu &lt;weiliang.qiu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Dupont, W.D. and Plummer, W.D.. 
Power and Sample Size Calculations for Studies Involving Linear Regression.
Controlled Clinical Trials. 1998;19:589-601.
</p>
<p>Dong X, Li X, Chang T-W, Scherzer CR, Weiss ST, and Qiu W. 
powerEQTL: An R package and shiny application for sample size and power calculation of bulk tissue and single-cell eQTL analysis.
Bioinformatics, 2021;, btab385
</p>


<h3>Examples</h3>

<pre><code class="language-R"># calculate power
powerEQTL.SLR(
  MAF = 0.1,
  slope = 0.13,
  n = 179,
  power = NULL,
  sigma.y = 0.13,
  FWER = 0.05,
  nTests = 2e+05)
  
# calculate sample size (total number of subjects)
powerEQTL.SLR(
  MAF = 0.1,
  slope = 0.13,
  n = NULL,
  power = 0.8,
  sigma.y = 0.13,
  FWER = 0.05,
  nTests = 2e+05)
  
# calculate minimum detectable slope
powerEQTL.SLR(
  MAF = 0.1,
  slope = NULL,
  n = 179,
  power = 0.8,
  sigma.y = 0.13,
  FWER = 0.05,
  nTests = 2e+05)  
  
# calculate minimum allowable MAF
powerEQTL.SLR(
  MAF = NULL,
  slope = 0.13,
  n = 179,
  power = 0.8,
  sigma.y = 0.13,
  FWER = 0.05,
  nTests = 2e+05) 

</code></pre>


</div>