<div class="container">

<table style="width: 100%;"><tr>
<td>simDat.eQTL.scRNAseq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Generate Gene Expression Levels Of One Gene And Genotypes Of One SNP For Subjects With Multiple Cells Based On ZINB Mixed Effects Regression Model
</h2>

<h3>Description</h3>

<p>Generate gene expression levels of one gene and genotypes of one SNP for subjects with multiple cells based on ZINB mixed effects regression model.
</p>


<h3>Usage</h3>

<pre><code class="language-R">simDat.eQTL.scRNAseq(nSubj = 50, 
		     nCellPerSubj = 100, 
		     zero.p = 0.01, 
		     m.int = 0, 
		     sigma.int = 1, 
		     slope = 1, 
		     theta = 1, 
		     MAF = 0.45)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nSubj</code></td>
<td>

<p>integer. Total number of subjects.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nCellPerSubj</code></td>
<td>

<p>integer. Number of cells per subject.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>zero.p</code></td>
<td>

<p>numeric. Probability that an excess zero occurs.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m.int</code></td>
<td>

<p>numeric. Mean of random intercept (see details).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigma.int</code></td>
<td>

<p>numeric. Standard deviation of random intercept (see details).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>slope</code></td>
<td>

<p>numeric. Slope (see details).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>theta</code></td>
<td>

<p>numeric. dispersion parameter of negative binomial distribution.
The smaller <code>theta</code> is, the larger variance of NB random variable is.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAF</code></td>
<td>

<p>numeric. Minor allele frequency of the SNP.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function simulates gene expression levels of one gene and genotypes of one SNP for subjects with multiple cells based on zero-inflated negative binomial (ZINB) regression model with only one 
covariate: genotype.  
That is, the read counts of a gene follows a mixture of 2-component distributions. One component takes only one value: zero. The other component is negative binomial distribution, which takes non-negative values 0, 1, 2, .... The log mean of the negative binomial distribution is linear function of the genotype.
</p>
<p>Denote <code class="reqn">Y_{ij}</code> as the read counts for the <code class="reqn">j</code>-th cell of
the <code class="reqn">i</code>-th subject, <code class="reqn">i=1,\ldots, n</code>, <code class="reqn">j=1,\ldots, m</code>,
<code class="reqn">n</code> is the number of subjects, and <code class="reqn">m</code> is the number of cells per subject.
</p>
<p>Denote <code class="reqn">p</code> as the probability that <code class="reqn">Y_{ij}=0</code> is an excess zero.
With probability <code class="reqn">1-p</code>, <code class="reqn">Y_{ij}</code> follows a negative binomial distribution <code class="reqn">NB(\mu, \theta)</code>, where <code class="reqn">\mu</code> is the mean (i.e., <code class="reqn">\mu=E(Y_{ij})</code>) and <code class="reqn">\theta</code> is the dispersion parameter.
The variance of the NB distribution is <code class="reqn">\mu + \mu^2/\theta</code>.
The relationship between gene expression and genotype for the <code class="reqn">i</code>-th subject is characterized by the equation 
</p>
<p style="text-align: center;"><code class="reqn">\mu_i = \exp(\beta_{0i} + \beta_{1} x_i),</code>
</p>

<p>where <code class="reqn">\beta_{0i}</code> is the random intercept following a normal
distribution <code class="reqn">N(\beta_0, \sigma^2)</code> to account for within-subject correlation of gene expression, <code class="reqn">\beta_0</code> is the mean of the random intercept, <code class="reqn">\sigma</code> is the standard deviation of the random intercept, <code class="reqn">\beta_1</code> is the slope, and <code class="reqn">x_i</code> is the additive-coded genotype for the SNP with minor allele frequency <code class="reqn">MAF</code>.
</p>
<p>We assume that the SNP satisfies the Hardy-Weinberg Equilibrium. That is, the
probabilities of the 3 genotypes <code class="reqn">(0, 1, 2)</code> are <code class="reqn">(1-MAF)^2,
		2 MAF (1-MAF), MAF^2</code>, respectively.
</p>
<p>For simplicity, we assume that excess zeros are caused by technical issues, hence are not related to genotypes.
</p>


<h3>Value</h3>

<p>A data frame with 3 columns:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>id </code></td>
<td>
<p>subject id</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno </code></td>
<td>
<p>additive-coded genotype of the SNP</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>counts </code></td>
<td>
<p>gene expression of the gene</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Xianjun Dong &lt;XDONG@rics.bwh.harvard.edu&gt;,
Xiaoqi Li&lt;xli85@bwh.harvard.edu&gt;,
Tzuu-Wang Chang &lt;Chang.Tzuu-Wang@mgh.harvard.edu&gt;,
Scott T. Weiss &lt;restw@channing.harvard.edu&gt;,
Weiliang Qiu &lt;weiliang.qiu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Dong X, Li X, Chang T-W, Scherzer CR, Weiss ST, and Qiu W. 
powerEQTL: An R package and shiny application for sample size and power calculation of bulk tissue and single-cell eQTL analysis.
Bioinformatics, 2021;, btab385
</p>


<h3>Examples</h3>

<pre><code class="language-R">frame = simDat.eQTL.scRNAseq(nSubj = 5, 
		     nCellPerSubj = 3, 
		     zero.p = 0.01, 
		     m.int = 0, 
		     sigma.int = 1, 
		     slope = 1, 
		     theta = 1, 
		     MAF = 0.45)
print(dim(frame))
print(frame[1:10,])

</code></pre>


</div>