<div class="container">

<table style="width: 100%;"><tr>
<td>PMR_summary</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>PMR model with summary data</h2>

<h3>Description</h3>

<p>Fit the probabilistic MR model with summary data while accounting for the 
correlated instruments and horizontal pleiotropy in TWAS framework.
</p>


<h3>Usage</h3>

<pre><code class="language-R">PMR_summary(betaxin, betayin, Sigma1sin, Sigma2sin, samplen1, samplen2,
  gammain, alphain, max_iterin, epsin)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>betaxin</code></td>
<td>
<p>the cis-SNP effect size vector for one specific gene in eQTL 
data, which must be calculated based on both the standardized gene expression 
value and the standardized cis-genotype matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>betayin</code></td>
<td>
<p>the cis-SNP effect size vector for one specific gene in GWAS 
data, which be calculated based on both the standardized complex trait value 
and the standardized cis-genotype matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Sigma1sin</code></td>
<td>
<p>the LD matrix in eQTL data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Sigma2sin</code></td>
<td>
<p>the LD matrix in GWAS data. Both <code>Sigma2sin</code> and 
<code>sigma1sin</code> are often the same from the reference panel.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>samplen1</code></td>
<td>
<p>the sample size of eQTL data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>samplen2</code></td>
<td>
<p>the sample size of GWAS data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gammain</code></td>
<td>
<p>indicator variable for constrained model, with 1 for the null 
model that there is no horizontal pleiotropy.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alphain</code></td>
<td>
<p>indicator variable for constrained model, with 1 for the null 
model that there is no causal effect.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_iterin</code></td>
<td>
<p>The maximum iteration.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>epsin</code></td>
<td>
<p>The convergence tolerance of the absolute value of the difference 
between the nth and (n+1)th log likelihood.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list of estimates of model parameters, including the causal effect 
<code>alpha</code>, the horizontal pleiotropy effect <code>gamma</code>, and the two corresponding p values.
</p>


<h3>Author(s)</h3>

<p>Zhongshang Yuan, Xiang Zhou.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(Examplesummary)
attach(Examplesummary)
fmH1=PMR_summary(betaxin=betax,betayin=betay,Sigma1sin=Sigma1,Sigma2sin=Sigma2,
      samplen1=n1,samplen2=n2,gammain=0,alphain=0,max_iterin =1000, epsin=1e-5)
fmH0alpha=PMR_summary(betaxin=betax,betayin=betay,Sigma1sin=Sigma1,Sigma2sin=Sigma2,
           samplen1=n1,samplen2=n2,gammain=0,alphain=1,max_iterin =1000, epsin=1e-5)
fmH0gamma=PMR_summary(betaxin=betax,betayin=betay,Sigma1sin=Sigma1,Sigma2sin=Sigma2,
           samplen1=n1,samplen2=n2,gammain=1,alphain=0,max_iterin =1000, epsin=1e-5)
loglikH1=max(fmH1$loglik,na.rm=TRUE)
loglikH0alpha=max(fmH0alpha$loglik,na.rm=TRUE)
loglikH0gamma=max(fmH0gamma$loglik,na.rm=TRUE)
stat_alpha = 2 * (loglikH1 - loglikH0alpha)
pvalue_alpha = pchisq(stat_alpha,1,lower.tail=FALSE)
stat_gamma = 2 * (loglikH1 - loglikH0gamma)
pvalue_gamma = pchisq(stat_gamma,1,lower.tail=FALSE)

</code></pre>


</div>