<div class="container">

<table style="width: 100%;"><tr>
<td>PMR_individual</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>PMR model with individual level data</h2>

<h3>Description</h3>

<p>Fit the probabilistic MR model with individual level data while accounting for 
the correlated instruments and horizontal pleiotropy in TWAS framework.
</p>


<h3>Usage</h3>

<pre><code class="language-R">PMR_individual(yin, zin, x1in, x2in, gammain, alphain, max_iterin, epsin)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>yin</code></td>
<td>
<p>standardized exposure vector (e.g. gene expression in TWAS).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>zin</code></td>
<td>
<p>standardized complex trait vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x1in</code></td>
<td>
<p>standardized cis-genotype matrix in eQTL data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x2in</code></td>
<td>
<p>standardized cis-genotype matrix in GWAS data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gammain</code></td>
<td>
<p>indicator variable for constrained PMR model, with 1 for the null 
model that there is no horizontal pleiotropy.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alphain</code></td>
<td>
<p>indicator variable for constrained PMR model, with 1 for the null 
model that there is no causal effect.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_iterin</code></td>
<td>
<p>The maximum iteration.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>epsin</code></td>
<td>
<p>The convergence tolerance of the absolute value of the difference 
between the nth and (n+1)th log likelihood.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a list of estimates of model parameters, including the causal effect 
<code>alpha</code>, the horizontal pleiotropy effect <code>gamma</code>, and the two 
corresponding p values
</p>


<h3>Author(s)</h3>

<p>Zhongshang Yuan, Xiang Zhou.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(Exampleindividual)
attach(Exampleindividual)
fmH1 = PMR_individual(yin=x, zin=y, x1in=zx, x2in=zy,
                      gammain=0,alphain = 0,max_iterin =1000,epsin=1e-5)
alpha&lt;-fmH1$alpha
gamma&lt;-fmH1$gamma
fmH0gamma = PMR_individual(yin=x,zin= y, x1in=zx, x2in=zy,gammain=1, 
                           alphain = 0,max_iterin =1000,epsin=1e-5)
fmH0alpha = PMR_individual(yin=x, zin=y, x1in=zx, x2in=zy,gammain=0,
                           alphain = 1,max_iterin =1000, epsin=1e-5)
loglikH1=max(fmH1$loglik,na.rm=TRUE)
loglikH0gamma=max(fmH0gamma$loglik,na.rm=TRUE)
loglikH0alpha=max(fmH0alpha$loglik,na.rm=TRUE)
stat_alpha = 2 * (loglikH1 - loglikH0alpha)
pvalue_alpha = pchisq(stat_alpha,1,lower.tail=FALSE)
stat_gamma = 2 * (loglikH1 - loglikH0gamma)
pvalue_gamma = pchisq(stat_gamma,1,lower.tail=FALSE)

</code></pre>


</div>