<div class="container">

<table style="width: 100%;"><tr>
<td>adjpval</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>adjpval</h2>

<h3>Description</h3>

<p>A function to retrieve a table with adjusted p-values after running PRANA.
The table includes all variables that were included in the pseudo-value regression model.
</p>


<h3>Usage</h3>

<pre><code class="language-R">adjpval(PRANAres)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>PRANAres</code></td>
<td>
<p>An object called after running PRANA.</p>
</td>
</tr></table>
<h3>Value</h3>

<p>A table that includes adjusted p-values for all variables included in the fitted model.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(combinedCOPDdat_RGO) # A complete data containing expression and clinical data.

# A gene expression data part of the downloaded data.
rnaseqdat = combinedCOPDdat_RGO[ , 8:ncol(combinedCOPDdat_RGO)]
rnaseqdat = as.data.frame(apply(rnaseqdat, 2, as.numeric))

# A clinical data with additional covariates sorted by current smoking groups:
# The first column is ID, so do not include.
phenodat = combinedCOPDdat_RGO[order(combinedCOPDdat_RGO$currentsmoking), 2:7]

# Indices of non-current smoker (namely Group A)
index_grpA = which(combinedCOPDdat_RGO$currentsmoking == 0)
# Indices of current smoker (namely Group B)
index_grpB = which(combinedCOPDdat_RGO$currentsmoking == 1)

# Use PRANA() function to perform the pseudo-value regression analysis.
PRANAres &lt;- PRANA(RNASeqdat = rnaseqdat, clindat = phenodat,
 groupA = index_grpA, groupB = index_grpB)

# Now, we want to keep the table with adjusted p-values only.
adjpval(PRANAres)
</code></pre>


</div>