<div class="container">

<table style="width: 100%;"><tr>
<td>selectIntFeatures</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Select common genes for multiple data batches</h2>

<h3>Description</h3>

<p>selectIntFeatures prioritizes genes based on the number of times they were selected as HVGs/SVGs in all data baches, and chose the top genes as the input for the analysis.  We broke ties by examining the ranks of the tied genes in each original dataset and taking those with the highest median rank.</p>


<h3>Usage</h3>

<pre><code class="language-R">  selectIntFeatures(seulist, spaFeatureList, IntFeatures=2000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seulist</code></td>
<td>
<p>a list consisting of Seurat objects, where each object is a SRT data batch.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>spaFeatureList</code></td>
<td>
<p>an list consisting of SVGs vectors, where each vector is the top HVGs/SVGs for each SRT data batch.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>IntFeatures</code></td>
<td>
<p>the number of common HVGs/SVGs genes to be chosen.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Nothing
</p>


<h3>Value</h3>

<p>Return a string vector, the selected gene list for integration in PRECAST.
</p>


<h3>Note</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Wei Liu
</p>


<h3>References</h3>

<p><a href="https://www.nature.com/articles/s41467-023-35947-w">Wei Liu, Liao, X., Luo, Z. et al, Jin Liu* (2023). Probabilistic embedding, clustering, and alignment for integrating spatial transcriptomics data with PRECAST. Nature Communications, 14, 296</a>
</p>


<h3>See Also</h3>

<p>None
</p>


</div>