<div class="container">

<table style="width: 100%;"><tr>
<td>PredCRG_Enc</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Encoding of protein sequence data in to numeric feature vector based on PredCRG features.
</h2>

<h3>Description</h3>

<p>Before using the protein sequences for prediction using the proposed model, the sequences must be transformed into numeric feature vectors. The function <code>PredCRG_Enc</code> will transform each protein sequnces to a numeric vector of 62 observations, based on the compositional, physico-chemical and transitional features used in the <code>PredCRG</code> model.
</p>


<h3>Usage</h3>

<pre><code class="language-R">PredCRG_Enc(prot_seq)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>prot_seq</code></td>
<td>

<p>Sequence dataset to be supplied as input, must be an object of class <code>AAStringSet</code>
</p>
</td>
</tr></table>
<h3>Details</h3>

<p>The dataset must contains the protein sequences having standard amino acid residues only. The clas <code>AAStringSet</code> can be obtained by reading the FASTA file using <code>readAAStringSet</code> available in bioconductor package <b><code>Biostrings</code></b>.
</p>


<h3>Value</h3>

<p>A matrix of dimension n*62, for n number of sequences.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher, ICAR-Indian Agricultural Statistics Research Institute, New Delhi-110012, INDIA
</p>


<h3>See Also</h3>

<p>PredCRG, PredCRG_training, model1, model2,model3,model4
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(test)
enc &lt;- PredCRG_Enc(test)#encoding of test sequence data
enc[1:5,1:5]
  </code></pre>


</div>