<div class="container">

<table style="width: 100%;"><tr>
<td>PredCRG</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Prediction of circadian proteins using the proposed PredCRG model.
</h2>

<h3>Description</h3>

<p>The user can predict the protein sequences as CRG (circadian protein) or non-CRG (non-circadian protein) with certain probability by supplying the test sequences.
</p>


<h3>Usage</h3>

<pre><code class="language-R">PredCRG(seq_data)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>seq_data</code></td>
<td>

<p>Sequence dataset in FASTA format consisting of protein sequences with standard amino acid residues only. It must be an object of class <code>AAStringSet</code> which can be obtained by reading sequences with <code>readAAStringSet</code> available in <b><code>Biostrings</code></b> package.
</p>
</td>
</tr></table>
<h3>Details</h3>

<p>The user has to supply only the <code>seq_data</code> for which the prediction is to be made. 
</p>


<h3>Value</h3>

<p>A dataframe with three columns consisting of sequence name, predicted labels of sequences (CRG or non-CRG) and probabilities of prediction.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher, ICAR-Indian Agricultural Statsitics Research Institute, New Delhi-110012, INDIA
</p>


<h3>See Also</h3>

<p>PredCRG_Enc, PredCRG_training,model1, model2,model3,model4
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(test)
tst &lt;- test[1:10]
PredCRG(seq_data=tst)
</code></pre>


</div>