<div class="container">

<table style="width: 100%;"><tr>
<td>convertgen</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Convert Genotype</h2>

<h3>Description</h3>

<p>Convert genotypes in HapMap format or in numeric format for hypred package.
</p>


<h3>Usage</h3>

<pre><code class="language-R">convertgen(
  input_geno,
  type = c("hmp1", "hmp2", "num"),
  missingrate = 0.2,
  maf = 0.05,
  impute = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>input_geno</code></td>
<td>
<p>genotype in HapMap format or in numeric format. The names of individuals should be provided. Missing (NA) values are allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>the type of genotype. There are three options: "hmp1" for genotypes in HapMap format with single bit, "hmp2" for genotypes in HapMap format with double bit, and "num" for genotypes in numeric format.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>missingrate</code></td>
<td>
<p>max missing percentage for each SNP, default is 0.2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maf</code></td>
<td>
<p>minor allele frequency for each SNP, default is 0.05.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>impute</code></td>
<td>
<p>logical. If TRUE, imputation. Default is TRUE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A matrix of genotypes in numeric format, coded as 1, 0, -1 for AA, Aa, aa. Each row represents an individual and each column represents a marker. The rownames of the matrix are the names of individuals.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## load genotype in HapMap format with double bit
data(input_geno)

## convert genotype for hypred package
inbred_gen &lt;- convertgen(input_geno, type = "hmp2")


## load genotype in numeric format
data(input_geno1)
head(input_geno1)

## convert genotype for hypred package
inbred_gen1 &lt;- convertgen(input_geno1, type = "num")


</code></pre>


</div>