<div class="container">

<table style="width: 100%;"><tr>
<td>ChannelType-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Class "ChannelType"</h2>

<h3>Description</h3>

<p>This class represents the "type" of a microarray channel. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">ChannelType(mk, md, nc, nr, gl, design="")
setDesign(object, design)
getDesign(object)
## S4 method for signature 'ChannelType'
print(x, ...)
## S4 method for signature 'ChannelType'
show(object)
## S4 method for signature 'ChannelType'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>mk</code></td>
<td>
<p>character string specifying the name of the manufacturer of the
microarray (e.g., 'Affymetrix') </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>md</code></td>
<td>
<p>character string specifying the model of the microarray
(e.g., 'Hu95A')</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nc</code></td>
<td>
<p>scalar integer specifying the number of columns in the array</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nr</code></td>
<td>
<p>scalar integer specifying the number of rows in the array</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gl</code></td>
<td>
<p>character string specifying the material used to label samples</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>design</code></td>
<td>
<p>character string containing the name of an object describing
details about the design of the microarray</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>object of class <code>ChannelType</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>object of class <code>ChannelType</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>extra arguments for generic or plotting routines</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Microarrays come in numerous flavors. At present, the two most common
types are the synthesized oligonucleotide arrays produced by
Affymetrix and the printed cDNA arrays on glass, which started in Pat
Brown's lab at Stanford. In earlier days, it was also common to find
nylon microarrays, with the samples labeled using a radioactive
isotope. The glass arrays are distinguished from other kinds of
arrays in that they typically cohybridize two different samples
simultaneously, using two different fluorescent dyes. The fluorescence
from each dye is scanned separately, producing two images and thus two
related sets of data from the same microarray. We refer to these
parallel data sets within an array as “channels”.
</p>
<p>An object of the <code>ChannelType</code> class represents a combination of
the kind of microarray along with the kind of labeling procedure.
These objects are intended to be passed around as part of more complex
objects representing the actual gene expression data collected from
particular experiments, in order to be able to eventually tie back into
the description of what spots were laid down when the array was produced.
</p>
<p>The <code>ChannelType</code> object only contains a high level description
of the microarray, however. Detailed information about what
biological material was laid down at each spot on the microarray is
stored elsewhere, in a “design” object. Within a <code>ChannelType</code>
object, the design is represented simply by a character string. This
string should be the name of a separate object containing the detailed
design information. This implementation allows us to defer the design
details until later. It also saves space by putting the details in a
single object instead of copying them into every microarray. Finally,
it allows that single object to be updated when better biological
annotations are available, with the benefits spreading immediately to
all the microarray projects that use that design.
</p>


<h3>Value</h3>

<p>The <code>ChannelType</code> constructor returns a valid object of the
class.
</p>
<p>The <code>setDesign</code> function invisibly returns the <code>ChannelType</code>
object on which it was invoked.
</p>
<p>The <code>getDesign</code> function returns the design object referred to by
the <code>design</code> slot in the <code>ChannelType</code> object. If this string
does not evaluate to the name of an object, then <code>getDesign</code> returns
a <code>NULL</code> value.
</p>


<h3>Slots</h3>


<dl>
<dt>
<code>maker</code>:</dt>
<dd>
<p>character string specifying the name of the
manufacturer of the microarray</p>
</dd>
<dt>
<code>model</code>:</dt>
<dd>
<p>character string specifying the model of the
microarray</p>
</dd> 
<dt>
<code>nCol</code>:</dt>
<dd>
<p>scalar integer specifying number of columns in the array</p>
</dd>
<dt>
<code>nRow</code>:</dt>
<dd>
<p>scalar integer specifying number of rows in the array</p>
</dd>
<dt>
<code>glow</code>:</dt>
<dd>
<p>character string specifying the material used to label
samples</p>
</dd>
<dt>
<code>design</code>:</dt>
<dd>
<p>character string containing the name of an object
describing details about the design of the microarray</p>
</dd>
</dl>
<h3>Methods</h3>


<dl>
<dt>print(x, ...)</dt>
<dd>
<p>Prints all the information in the object</p>
</dd>
<dt>show(object)</dt>
<dd>
<p>Prints all the information in the object</p>
</dd>
<dt>summary(object, ...)</dt>
<dd>
<p>Writes out a summary of the object</p>
</dd>
</dl>
<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
P. Roebuck <a href="mailto:proebuck@mdanderson.org">proebuck@mdanderson.org</a>
</p>


<h3>See Also</h3>

<p><code>Channel</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">showClass("ChannelType")

x &lt;- ChannelType('Affymetrix', 'oligo', 100, 100, 'fluor')
x
print(x)
summary(x)

y &lt;- setDesign(x, 'fake.design')
print(y)
summary(y)
d &lt;- getDesign(y)
d

rm(d, x, y) # cleanup
</code></pre>


</div>