<div class="container">

<table style="width: 100%;"><tr>
<td>preseqR-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Predicting <code class="reqn">r</code>-species accumulation curves
</h2>

<h3>Description</h3>

<p>The functionality of this package is to predict <code class="reqn">r</code>-species
accumulaiton curves. The method is based on a nonparametric empirical Bayes approach
with rational function approximation. The estimator is
excellent in accuracy for both large values of <code class="reqn">r</code> and long-range
extrapolations, which are essential to large-scale applications. Some
examples are predicting the molecular complexity of sequencing
libraries, estimating the minimum sufficient sequencing depths for
whole-exome sequencing experiments and optimizing depths for single-cell
whole-genome sequencing experiments. 
</p>


<h3>Details</h3>

<p>main functions: 
</p>
<p>preseqR.rSAC
</p>
<p>preseqR.rSAC.bootstrap
</p>
<p>preseqR.optimal.sequencing
</p>
<p>preseqR.rSAC.sequencing.rmdup
</p>
<p>preseqR.sample.cov
</p>
<p>preseqR.sample.cov.bootstrap
</p>


<h3>Author(s)</h3>

<p>Chao Deng, Timothy Daley, and Andrew D. Smith
</p>
<p>Maintainer: Chao Deng &lt;chaodeng@usc.edu&gt;
</p>


<h3>References</h3>

<p>Baker, G. A., &amp; Graves-Morris, P. (1996). Pade approximants
(Encyclopedia of Mathematics and its Applications vol 59).
</p>
<p>Boneh, S., Boneh, A., &amp; Caron, R. J. (1998). Estimating the prediction function
and the number of unseen species in sampling with replacement.
Journal of the American Statistical Association, 93(441), 372-379.
</p>
<p>Chao, A., &amp; Shen, T. J. (2004). Nonparametric prediction in species sampling.
Journal of agricultural, biological, and environmental statistics, 9(3), 253-269.
</p>
<p>Cohen Jr, A. C. (1960). Estimating the parameters of a modified Poisson distribution.
Journal of the American Statistical Association, 55(289), 139-143.
</p>
<p>Daley, T., &amp; Smith, A. D. (2013). Predicting the molecular complexity of
sequencing libraries. Nature methods, 10(4), 325-327.
</p>
<p>Deng C, Daley T &amp; Smith AD (2015). Applications of species accumulation curves
in large-scale biological data analysis. Quantitative Biology, 3(3), 135-144. 
URL http://dx.doi.org/10.1007/s40484-015-0049-7.
</p>
<p>Deng, C., Daley, T., Calabrese, P., Ren, J., &amp; Smith, A.D. (2016). Estimating
the number of species to attain sufficient representation in a random sample.
arXiv preprint arXiv:1607.02804v3.
</p>
<p>Efron, B., &amp; Thisted, R. (1976). Estimating the number of unseen species:
How many words did Shakespeare know?. Biometrika, 63(3), 435-447.
</p>
<p>Efron, B. (1979). Bootstrap methods: another look at the jackknife.
The annals of Statistics, 1-26.
</p>
<p>Efron, B., &amp; Tibshirani, R. J. (1994). An introduction to the bootstrap. CRC press.
</p>
<p>Fisher, R. A., Corbet, A. S., and Williams, C. B. ,1943, The Relation Between 
the Number of Species and the Number of Individuals in a Random Sample of an  
Animal Population, Journal of Animal Ecology, 12, 42-58.
</p>
<p>Good, I. J., &amp; Toulmin, G. H. (1956). The number of new species, and the
increase in population coverage, when a sample is increased.
Biometrika, 43(1-2), 45-63.
</p>
<p>Heck Jr, K. L., van Belle, G., &amp; Simberloff, D. (1975). 
Explicit calculation of the rarefaction diversity measurement and the 
determination of sufficient sample size. Ecology, 1459-1461.
</p>
<p>Kalinin V (1965). Functionals related to the poisson distribution and               
statistical structure of a text. Articles on Mathematical Statistics and            
the Theory of Probability pp. 202-220.
</p>


</div>