<div class="container">

<table style="width: 100%;"><tr>
<td>kmer.frac.curve</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Fraction of <code class="reqn">k</code>-mers observed at least <code class="reqn">r</code> times 
</h2>

<h3>Description</h3>

<p><code>kmer.frac.curve</code> predicts the expected fraction of <code class="reqn">k</code>-mers observed at
least <code class="reqn">r</code> times in a high-throughput sequencing experiment given the 
amount of sequencing
</p>


<h3>Usage</h3>

<pre><code class="language-R">  kmer.frac.curve(n, k, read.len, seq, r=2, mt=20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of <code class="reqn">k</code>-mers observed exactly <code class="reqn">j</code>
times in the initial experiment. The first column must be sorted in an
ascending order.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>

<p>The number of nucleotides in a <code class="reqn">k</code>-mer.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>read.len</code></td>
<td>

<p>The average length of a read.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq</code></td>
<td>

<p>The amount of nucleotides sequenced..
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r</code></td>
<td>

<p>A positive integer. Default is 1.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mt</code></td>
<td>

<p>An positive integer constraining possible rational function
approximations. Default is 20.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>kmer.frac.curve</code> is mainly designed for metagenomics to evaluate how 
saturated a metagenomic data is.
</p>
<p><code>kmer.frac.curve</code> is the fast version of <code>kmer.frac.curve.bootstrap</code>.
The function does not provide the confidence interval. To obtain the
confidence interval along with the estimates, one should use the function
<code>kmer.frac.curve.bootstrap</code>.
</p>


<h3>Value</h3>

<p>A two-column matrix. The first column is the amount of sequencing in an 
experiment. The second column is the estimate of the fraction of <code class="reqn">k</code>-mers observed at least
<code class="reqn">r</code> times in the experiment.
</p>


<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>References</h3>

<p>Deng, C and Smith, AD (2016). Estimating the number of species to attain 
sufficient representation in a random sample. arXiv preprint arXiv:1607.02804
</p>


<h3>Examples</h3>

<pre><code class="language-R">## load library
library(preseqR)

## import data
data(SRR061157_k31)

## the fraction of 31-mers represented at least 10 times in an experiment when
## sequencing 1M, 10M, 100M, 1G, 10G, 100G, 1T nucleotides
kmer.frac.curve(n=SRR061157_k31, k=31, read.len=100, seq=10^(6:12), r=10, mt=20) 
</code></pre>


</div>