<div class="container">

<table style="width: 100%;"><tr>
<td>preseqR.optimal.sequencing</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Optimal amount of sequencing for scWGS
</h2>

<h3>Description</h3>

<p><code>preseqR.optimal.sequencing</code> predicts the optimal amount of sequencing in
a single-cell whole-genome sequencing (scWGS) experiment based on a shallow sequencing experiment.
</p>


<h3>Usage</h3>

<pre><code class="language-R">preseqR.optimal.sequencing(n, efficiency=0.05, bin=1e8, r=1, mt=20, 
                           times=30, conf=0.95)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">n_j</code>, the number of species with each species represented exactly <code class="reqn">j</code>
times in the initial sample. The first column must be sorted in an
ascending order.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>efficiency</code></td>
<td>

<p>The minimum benefit-cost ratio
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bin</code></td>
<td>

<p>One unit of sequencing effort. Default is 1e8.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r</code></td>
<td>

<p>A positive integer. Default is 1.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mt</code></td>
<td>

<p>An positive integer constraining possible rational function
approximations. Default is 20.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>times</code></td>
<td>

<p>The number of bootstrap samples.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>conf</code></td>
<td>

<p>The confidence level. Default is 0.95
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>preseqR.optimal.sequencing</code> predicts the optimal amount of sequencing
in a scWGS experiment. The term optimal is interpreted as the maximum 
amount of sequencing with its benefit-cost ratio greater than a given threshold.
The benefit-cost ratio is defined as the probability of a new nucleotide in the
genome represented at least <code class="reqn">r</code> times when one more base is sequenced. 
In order to improve the numeric stability, we use the mean of new nucleotdies
with coverage at least <code class="reqn">r</code> in one unit of sequencing effort to approximate the
ratio. The amount of sequences in one unit of sequencing effort is defined by
the variable <code>bin</code>.
</p>
<p>Note that the benefit-cost ratio is not monotonic. The ratio first increases
and then decrease as the amount of sequencing increase. To predicte the
optimal amount of sequencing, we consider only the areas after the peak, where
the ratio starts to decrease.
</p>


<h3>Value</h3>

<p>A vector of three dimensions. The first coordinate is the optimal amount of
sequencing. The second and the third coordinates are the lower and upper bound
of the confidence interval.
</p>


<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>References</h3>

<p>Deng, C., Daley, T., Calabrese, P., Ren, J., &amp; Smith, A.D. (2016). Estimating
the number of species to attain sufficient representation in a random sample.
arXiv preprint arXiv:1607.02804v3.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## load library
#library(preseqR)

## import data
# data(SRR611492_5M)
## the optimal amount of sequencing with the benefit-cost ratio greater than 
## 0.05 for r = 4
# preseqR.optimal.sequencing(n=SRR611492_5M, efficiency=0.05, bin=1e8, r=4)
## the optimal amount of sequencing with the benefit-cost ratio greater than 
## 0.05 for r = 10
# preseqR.optimal.sequencing(n=SRR611492_5M, efficiency=0.05, bin=1e8, r=10)
</code></pre>


</div>