<div class="container">

<table style="width: 100%;"><tr>
<td>preseqR.rSAC.sequencing.rmdup</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Predicting <code class="reqn">r</code>-SAC in WES/WGS
</h2>

<h3>Description</h3>

<p><code>preseqR.rSAC.sequencing.rmdup</code> predicts the expected number of 
nucleotides in the genome sequenced at least r times in a sequencing
experiment, based on a shallow sequencing experiment.
</p>


<h3>Usage</h3>

<pre><code class="language-R">preseqR.rSAC.sequencing.rmdup(n_base, n_read, r=1, mt=20, times=30, conf=0.95)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n_base</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N_j</code>, the number of nucleotides in the genome sequenced exactly 
<code class="reqn">j</code> times in the initial experiment. The first column must be sorted in
an ascending order.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_read</code></td>
<td>

<p>A two-column matrix.  
The first column is the frequency <code class="reqn">j = 1,2,\dots</code>; and the second column
is <code class="reqn">N'_j</code>, the number of distinct reads with exactly <code class="reqn">j</code> duplicates
in the initial experiment. The first column must be sorted in an
ascending order.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r</code></td>
<td>

<p>A positive integer. Default is 1.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mt</code></td>
<td>

<p>An positive integer constraining possible rational function
approximations. Default is 20.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>times</code></td>
<td>

<p>The number of bootstrap samples. Default is 30.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>conf</code></td>
<td>

<p>The confidence level. Default is 0.95
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>preseqR.rSAC.sequencing.rmdup</code> is designed for sequencing experiments, 
where duplicate reads are removed. The procedure is commonly used in 
whole-exome sequencing experiments and sometimes appeared in WGS as well.
To use the function, one must have two histograms. The first histogram
is the coverage histogram, which is based on distinct reads.
The second histogram is the counts of reads with exactly <code class="reqn">j</code> duplicates.
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>f</code></td>
<td>

<p>The estimator for the expected number of nucleotides in the genome
sequenced at least <code class="reqn">r</code> times given the amount of sequencing.
The input of the estimator is a vector of sequencing efforts t, i.e. 
the relative amount of sequencing comparing with the amount in the initial
experiment. For example, t = 2 means sequencing twice the amount of the initial
experiment.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>se</code></td>
<td>

<p>The standard error for the estimator. The input is a vector of sequencing
efforts t.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lb</code></td>
<td>

<p>The lower bound of the confidence interval.The input is a vector of
sequencing efforts t.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ub</code></td>
<td>

<p>The upper bound of the confidence interval.The input is a vector of
sequencing efforts t.
</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Chao Deng
</p>


<h3>References</h3>

<p>Deng, C., Daley, T., Calabrese, P., Ren, J., &amp; Smith, A.D. (2016). Estimating
the number of species to attain sufficient representation in a random sample.
arXiv preprint arXiv:1607.02804v3.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## load library
#library(preseqR)

## import data
# data(SRR1301329_1M_base)
# data(SRR1301329_1M_read)
# construct the estimator
# estimator1 &lt;- preseqR.rSAC.sequencing.rmdup(
#                 n_base=SRR1301329_1M_base, n_read=SRR5365359_5M_read, 
#                 r=4, mt=20, times=100, conf=0.95)
## The number of nucleotides in the genome covered at least 4 times, when the
## amount of sequencing is 10 or 20 times of the intial experiment
## 10 or 20 times of the initial sample
# estimator1$f(c(10, 20))
## The standard error of the estiamtes
# estimator1$se(c(10, 20))
## The confidence interval of the estimates
# lb &lt;- estimator1$lb(c(10, 20))
# ub &lt;- estimator1$ub(c(10, 20))
# matrix(c(lb, ub), byrow=FALSE, ncol=2)

# construct the estimator
# estimator2 &lt;- preseqR.rSAC.sequencing.rmdup(
#                 n_base=SRR1301329_1M_base, n_read=SRR5365359_5M_read, 
#                 r=10, mt=20, times=100, conf=0.95)
## The number of nucleotides in the genome covered at least 10 times, when the
## amount of sequencing is 10 or 20 times of the intial experiment
## 10 or 20 times of the initial sample
# estimator2$f(c(10, 20))
## The standard error of the estiamtes
# estimator2$se(c(10, 20))
## The confidence interval of the estimates
# lb &lt;- estimator2$lb(c(10, 20))
# ub &lt;- estimator2$ub(c(10, 20))
# matrix(c(lb, ub), byrow=FALSE, ncol=2)

</code></pre>


</div>