<div class="container">

<table style="width: 100%;"><tr>
<td>find_dep</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Identify differentially expressed proteins between groups</h2>

<h3>Description</h3>

<p>This function performs differential expression analysis
on protein intensity data with limma.
</p>


<h3>Usage</h3>

<pre><code class="language-R">find_dep(
  df,
  save_output = FALSE,
  save_tophits = FALSE,
  file_path = NULL,
  adj_method = "BH",
  cutoff = 0.05,
  lfc = 1,
  n_top = 20
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>df</code></td>
<td>
<p>A <code>norm_df</code> object or an <code>imp_df</code> object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save_output</code></td>
<td>
<p>Logical. If <code>TRUE</code> saves results from the
differential expression analysis in a text file labeled "limma_output.txt"
in the directory specified by <code>file_path</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save_tophits</code></td>
<td>
<p>Logical. If <code>TRUE</code> saves <code>n_top</code>
number of top hits from the differential expression analysis in a text file
labeled "TopHits.txt" in the directory specified by <code>file_path</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file_path</code></td>
<td>
<p>A string containing the directory path to save the file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>adj_method</code></td>
<td>
<p>Method used for adjusting the p-values for multiple
testing. Default is <code>"BH"</code> for "Benjamini-Hochberg" method.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cutoff</code></td>
<td>
<p>Cutoff value for p-values and adjusted p-values. Default is
0.05.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lfc</code></td>
<td>
<p>Minimum absolute log2-fold change to use as threshold for
differential expression.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_top</code></td>
<td>
<p>The number of top differentially expressed proteins to save in
the "TopHits.txt" file. Default is <code>20</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<ul>
<li>
<p>It is important that the data is first log-transformed, ideally,
imputed, and normalized before performing differential expression analysis.
</p>
</li>
<li> <p><code>save_output</code> saves the complete results table from the
differential expression analysis.
</p>
</li>
<li> <p><code>save_tophits</code> first subsets the results to those with absolute
log fold change of more than 1, performs multiple correction with
the method specified in <code>adj_method</code> and outputs the top <code>n_top</code>
results based on lowest p-value and adjusted p-value.
</p>
</li>
<li>
<p> If the number of hits with absolute log fold change of more than 1 is
less than <code>n_top</code>, <code>find_dep</code> prints only those with
log-fold change &gt; 1 to "TopHits.txt".
</p>
</li>
<li>
<p> If the <code>file_path</code> is not specified, text files will be saved in
a temporary directory.</p>
</li>
</ul>
<h3>Value</h3>

<p>A <code>fit_df</code> object, which is similar to a <code>limma</code>
<code>fit</code> object.
</p>


<h3>Author(s)</h3>

<p>Chathurani Ranathunge
</p>


<h3>References</h3>

<p>Ritchie, Matthew E., et al. "limma powers differential expression
analyses for RNA-sequencing and microarray studies." Nucleic acids research
43.7 (2015): e47-e47.
</p>


<h3>See Also</h3>

<ul>
<li>
<p><code>normalize_data</code>
</p>
</li>
<li>
<p><code>lmFit</code>,
<code>eBayes</code>,
<code>topTable</code>, and
<code>write.fit</code> functions from the
<code>limma</code> package.</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">
## Perform differential expression analysis using default settings
fit_df1 &lt;- find_dep(ecoli_norm_df)

## Change p-value and adjusted p-value cutoff
fit_df2 &lt;- find_dep(ecoli_norm_df, cutoff = 0.1)

</code></pre>


</div>