<div class="container">

<table style="width: 100%;"><tr>
<td>Sel.Features</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Gene (Feature) Selection.
</h2>

<h3>Description</h3>

<p><code>Sel.Feature</code> selects the most discriminative genes (features) among the given ones.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Sel.Features(ES, Y, K = "Min", Verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ES</code></td>
<td>

<p>gene (feature) matrix: P, number of genes, by N, number of samples (observations).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Y</code></td>
<td>

<p>a vector of length N for samples' class label.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K</code></td>
<td>

<p>the number of genes to be selected. The default is to give the minimum subset of genes that correctly classify the maximum number of the given tissue samples (observations). Alternatively, <code>K</code> should be a positive integer.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Verbose</code></td>
<td>

<p>logical. If <code>TRUE</code>, more information about the selected genes are returned. 
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>Sel.Feature</code> selects the most relevant genes (features) in the high-dimensional binary classification problems. The discriminative genes are identified using analyzing the overlap between the expression values across both classes. The “<b>POS</b>” technique has been applied to produce the selected set of genes. A proportional overlapping score measures the overlapping degree avoiding the outliers effect for each gene. Each gene is described by a robust mask that represents its discriminative power. The constructed masks along with the gene scores are exploited to produce the selected subset of genes.
</p>


<h3>Value</h3>

<p>If <code>K</code> is specified as ‘Min’ (the default), a list containing the following components is returned:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>Features</code></td>
<td>

<p>A matrix of the indices of selected genes with their POS measures. See <code>POS</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Covered.Obs</code></td>
<td>

<p>A vector showing the indices of the observations that have been covered by the returned minimum subset of genes.
</p>
</td>
</tr>
</table>
<p>If <code>K</code> is specified as a positive integer, a list containing the following components is returned:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>features</code></td>
<td>

<p>A vector of the indices of the selected genes.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nMin.Features</code></td>
<td>

<p>The number of genes included in the minimum subset.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Measures</code></td>
<td>

<p>Available only when <code>Verbose</code> is <code>TRUE</code>. It is an object with class “<code>data.frame</code>” which contains 3 columns: the indices of the selected genes; the POS measures of the selected genes (see <code>POS</code>); the status that reports on which basis a gene is selected (“Min.Set”: the gene is a member of the selected minimum subset, 1: the gene has a low POS score and its relative dominant class is the first class or 2: the gene has a low POS score and its relative dominant class is the second class), see <code>RDC</code>.
</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p><code>Verbose</code> is only needed when <code>K</code> is specified. If <code>K</code> is set to “Min” (default), all information are automatically returned.
</p>


<h3>Author(s)</h3>

<p>Osama Mahmoud <a href="mailto:ofamah@essex.ac.uk">ofamah@essex.ac.uk</a>
</p>


<h3>References</h3>

<p>Mahmoud O., Harrison A., Perperoglou A., Gul A., Khan Z., Metodiev M. and Lausen B. (2014) <em>A feature selection method for classification within functional genomics experiments based on the proportional overlapping score</em>. BMC Bioinformatics, 2014, 15:274.
</p>


<h3>See Also</h3>

<p><code>POS</code> for calculating the proportional overlapping scores and <code>RDC</code> for assigning the relative dominant class.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(leukaemia)
GenesExpression &lt;- leukaemia[1:7129,] #define the features matrix
Class           &lt;- leukaemia[7130,]   #define the observations' class labels
## select the minimum subset of features
Selection       &lt;- Sel.Features(GenesExpression, Class)
attributes(Selection)
(Candidates      &lt;- Selection$Features)   #return the selected features
(Covered.observations &lt;- Selection$Covered.Obs) #return the covered observations by the selection
## select a specific number of features
Selection.k      &lt;- Sel.Features(GenesExpression, Class, K=10, Verbose=TRUE)
Selection.k$Features
Selection.k$nMin.Features   #return the size of the minimum subset of genes
Selection.k$Measures        #return the selected features' information
</code></pre>


</div>