<div class="container">

<table style="width: 100%;"><tr>
<td>sel_resist</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculating resistance values</h2>

<h3>Description</h3>

<p>Calculates the phenotypic resistance value (Renton et al. 2011) for each genotype in <code>dfgenotype</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">sel_resist(Rmx, epis, dom)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Rmx</code></td>
<td>
<p>maximum resistance value, if all gene loci under consideration are homozygous resistant. <code class="reqn">\code{numeric}, \ge 1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>epis</code></td>
<td>
<p>epistasis value, describing the interaction between resistance alleles. <code>numeric</code>. <code>epis</code> = 0: no interaction i.e. additive effects of resistance alleles, <code>epis</code> &lt; 0: effect of resistance alleles is smaller than additive, <code>epis</code> &gt; 0: effect of resistance alleles is higher than additive.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dom</code></td>
<td>
<p>dominance of resistance alleles. <code>numeric vector</code> with length of <code>n_loci</code> (number of genes) and elements in [0,1].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function is used in <code>struc_preparation2</code> to calculate the resistance value using the following term:
</p>
<p style="text-align: center;"><code class="reqn">  1 + (Rmx - 1) * (sum(dom)/n\_loci)^{2^{epis}}). </code>
</p>

<p>Resistance values range from 1 to Rmx. The higher the resistance value is, the less the plant is susceptible to the herbicide.
<code>sel_resist</code> is only used by <code>struc_preparation2</code>.
</p>


<h3>Value</h3>

<p><code>numeric vector</code> with resistance values
</p>


<h3>References</h3>

<p>Renton, M.; Diggle, A.; Manalil, S. &amp; Powles, S. (2011): Does cutting herbicide rates threaten the sustainability of weed management in cropping systems? Journal of Theoretical Biology, 283, 14-27.
</p>


<h3>See Also</h3>

<p><code>sel_herb</code> <code>intern_herbicide</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#dfgenotype is usually generated by the function 'struc_preparation2'.
#Here, a simple example is done by hand.
var1 &lt;- c("00","01","02","10","11","12","20","21","21")
var2 &lt;- c(0,0,0,1,1,1,2,2,2)
var3 &lt;- c(0,1,2,0,1,2,0,1,2)
dfgenotype &lt;- data.frame(genotype=var1,l1=var2,l2=var3, stringsAsFactors = TRUE)
sel_resist(Rmx=10, epis=0, dom=1)
</code></pre>


</div>