<div class="container">

<table style="width: 100%;"><tr>
<td>gen_diploid</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Diploid genetics: recombination of alleles</h2>

<h3>Description</h3>

<p><code>gen_diploid</code> recombines the alleles of weeds with diploid genome.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gen_diploid(
  start,
  start_comb = NA,
  result,
  newseeds = get0("newseeds", envir = parent.frame(n = 1)),
  max_vec_length = 1e+07
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>start</code></td>
<td>
<p>column names of parental cohorts. <code>character vector</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start_comb</code></td>
<td>
<p>all named columns in <code>start</code> are joined and returned as a column with the name of <code>start_comb</code> (<code>character</code>). Only used when <code>start</code> longer than 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>result</code></td>
<td>
<p>name of the results column. <code>character</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>newseeds</code></td>
<td>
<p>number of new seeds calculated by <code>pop_reprod</code>. <code>positive</code> <code>integer</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_vec_length</code></td>
<td>
<p>used internally, a technical term, defining the maximum length of vectors to be used.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>gen_diploid</code> assumes independent allele recombination.
</p>


<h3>Value</h3>

<p>A column with the name of <code>result</code> and, if necessary, a column with the name of <code>start_comb</code> is added to the <code>data.frame</code> <code>dfgenotype</code>.
</p>


<h3>warning</h3>

<p>newseeds are always coerced to a whole number.
</p>


<h3>See Also</h3>

<p><code>pop_reprod</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# generate a 'dfgenotype' data.frame:
struc_preparation2(Rmx=10, af=c(0.01, 0.2), epis=0, dom=1)
#Distribute 10000 individuals of the starting population across the genotypes. 
#The two gene loci have initial frequencies of 0.01 and 0.8.
gen_freq(af=c(0.01,0.8), n_seeds=10000)
# The column "initialSB" represents the parent generation, which recombines and 
#therefore defines the genetics of the new seeds
newseeds &lt;- 10000
gen_diploid(start="initialSB", result="followingSB")
# If a second cohort is reprodusing in the same time
gen_diploid(start=c("initialSB", "followingSB"), 
                  start_comb="two_cohorts_combind", result="followingSB2")
rm(dfgenotype, mf, newseeds, xprobab)

## End(Not run)
</code></pre>


</div>