<div class="container">

<table style="width: 100%;"><tr>
<td>struc_preparation2</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generating data input and output structures</h2>

<h3>Description</h3>

<p><code>struc_preparation</code> creates the data input and output structures (data.frames and table) for the simulation run, 'dfgenotype', 'xprobab' and 'mf'.
</p>


<h3>Usage</h3>

<pre><code class="language-R">struc_preparation2(Rmx = NA, af = NA, epis = NA, dom = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Rmx</code></td>
<td>
<p>maximum resistance value, if all gene loci under consideration are homozygous resistant. <code class="reqn">\code{numeric}, \ge 1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>af</code></td>
<td>
<p>initial frequency of resistance alleles in the population. <code>numeric vector</code> with length of <code>ng</code> (number of genes) and elements in [0,1].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>epis</code></td>
<td>
<p>epistasis value, describing the interaction between resistance alleles. <code>numeric</code>. <code>epis</code> = 0: no interaction i.e. additive effects of resistance alleles, <code>epis</code> &lt; 0: effect of resistance alleles is smaller than additive, <code>epis</code> &gt; 0: effect of resistance alleles is higher than additive.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dom</code></td>
<td>
<p>dominance of resistance alleles. <code>numeric vector</code> with length of <code>n_loci</code> (number of genes) and elements in [0,1].</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Prior to the simulation, a data.frame is generated to save results (<code>dfgenotype</code>). Additionally, a table with recombination probabilities (<code>xprobab</code>) is calculated. During the simulation run, probability values are not computed again but looked up in the table. PROSPER assumes diploid plants and maximum four resistance genes. To calculate the phenotypic resistance value for each genotype <code>sel_resist</code> is called. If <code>n_loci &lt; 0</code> the structure is set up for no genetics at all in the simulation.
</p>


<h3>Value</h3>

<p>Returns a <code>list</code> of two <code>data.frame</code> and a <code>table</code>:
</p>

<ol>
<li>
<p> mf: all possible combinations of parental genotypes (see 'dfgenotype$genotype') are saved in one column 'mf' (male, female). The column 'mf' is a character vector. Each string of the vector has twice the length of the number of resistance loci under consideration.
</p>
</li>
<li>
<p> dfgenotype: the structure to save the results of one simulation cycle (year). After each cycle the data is reset to the new start values. The first column 'genotype' is a character vector. Each string of the vector has the length of the number of resistance loci under consideration. Each locus can have 0, 1 or 2 resistance alleles. The second column 'resist' saves resistance values that are calculated according to the equation in the section 'details'.
</p>
</li>
<li>
<p> xprobab: the probabilities of occurrence for all possible genotypes in the offspring (F-generation) with all possible parent genotypes (P-generation). Free recombination is assumed. Column names are the combinations of parental genes, row names are genotypes of the offspring.
</p>
</li>
</ol>
<h3>Warning</h3>

<p>The run of <code>struc_preparation</code> is time consuming. Duration strongly increases with the number of genes under consideration, <code>n_loci</code>.
</p>


<h3>See Also</h3>

<p><code>sel_resist</code> <code>gen_freq</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#generate the genotype and probability tables for a simulation with two resistance 
#loci with one dominant and one partial dominant resistant allele, no epistasis, and a 
#maximumx resistance value of 10. 
ls()
struc_preparation2(Rmx=10, af=c(0.01,0.5), epis=0, dom=c(1,0.3))
ls()
rm(dfgenotype, mf, xprobab)
</code></pre>


</div>