<div class="container">

<table style="width: 100%;"><tr>
<td>extractDescScales</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Scales-Based Descriptors with 20+ classes of Molecular Descriptors</h2>

<h3>Description</h3>

<p>This function calculates the scales-based descriptors with
molecular descriptors sets calculated by Dragon, Discovery Studio and MOE.
Users can specify which molecular descriptors to select from one of these
deseriptor sets by specify the numerical or character index of the
molecular descriptors in the descriptor set.
</p>


<h3>Usage</h3>

<pre><code class="language-R">extractDescScales(
  x,
  propmat,
  index = NULL,
  pc,
  lag,
  scale = TRUE,
  silent = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>propmat</code></td>
<td>
<p>The matrix containing the descriptor set for the amino acids,
which can be chosen from
<code>AAMOE2D</code>, <code>AAMOE3D</code>, <code>AACPSA</code>, <code>AADescAll</code>,
<code>AA2DACOR</code>, <code>AA3DMoRSE</code>, <code>AAACF</code>, <code>AABurden</code>,
<code>AAConn</code>, <code>AAConst</code>, <code>AAEdgeAdj</code>, <code>AAEigIdx</code>,
<code>AAFGC</code>, <code>AAGeom</code>, <code>AAGETAWAY</code>, <code>AAInfo</code>,
<code>AAMolProp</code>, <code>AARandic</code>, <code>AARDF</code>, <code>AATopo</code>,
<code>AATopoChg</code>, <code>AAWalk</code>, and <code>AAWHIM</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>index</code></td>
<td>
<p>Integer vector or character vector. Specify which
molecular descriptors to select from one of these deseriptor
sets by specify the numerical or character index of the molecular
descriptors in the descriptor set.
Default is <code>NULL</code>, which means selecting all the molecular
descriptors in this descriptor set.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pc</code></td>
<td>
<p>Integer. The maximum dimension of the space which the data
are to be represented in. Must be no greater than the number of
amino acid properties provided.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lag</code></td>
<td>
<p>The lag parameter. Must be less than the amino acids.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale</code></td>
<td>
<p>Logical. Should we auto-scale the property matrix
(<code>propmat</code>) before doing MDS? Default is <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>silent</code></td>
<td>
<p>Logical. Whether we print the standard deviation,
proportion of variance and the cumulative proportion of
the selected principal components or not. Default is <code>TRUE</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A length <code>lag * p^2</code> named vector,
<code>p</code> is the number of scales selected.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>Examples</h3>

<pre><code class="language-R">x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
descscales &lt;- extractDescScales(
  x,
  propmat = "AATopo", index = c(37:41, 43:47),
  pc = 5, lag = 7, silent = FALSE
)
</code></pre>


</div>