<div class="container">

<table style="width: 100%;"><tr>
<td>extractMoran</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Moran Autocorrelation Descriptor</h2>

<h3>Description</h3>

<p>This function calculates the Moran
autocorrelation descriptor (dim: <code>length(props) * nlag</code>).
</p>


<h3>Usage</h3>

<pre><code class="language-R">extractMoran(
  x,
  props = c("CIDH920105", "BHAR880101", "CHAM820101", "CHAM820102", "CHOC760101",
    "BIGC670101", "CHAM810101", "DAYM780201"),
  nlag = 30L,
  customprops = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>props</code></td>
<td>
<p>A character vector, specifying the
Accession Number of the target properties.
8 properties are used by default, as listed below:
</p>

<dl>
<dt>AccNo. CIDH920105</dt>
<dd>
<p>Normalized average hydrophobicity scales (Cid et al., 1992)</p>
</dd>
<dt>AccNo. BHAR880101</dt>
<dd>
<p>Average flexibility indices (Bhaskaran-Ponnuswamy, 1988)</p>
</dd>
<dt>AccNo. CHAM820101</dt>
<dd>
<p>Polarizability parameter (Charton-Charton, 1982)</p>
</dd>
<dt>AccNo. CHAM820102</dt>
<dd>
<p>Free energy of solution in water, kcal/mole (Charton-Charton, 1982)</p>
</dd>
<dt>AccNo. CHOC760101</dt>
<dd>
<p>Residue accessible surface area in tripeptide (Chothia, 1976)</p>
</dd>
<dt>AccNo. BIGC670101</dt>
<dd>
<p>Residue volume (Bigelow, 1967)</p>
</dd>
<dt>AccNo. CHAM810101</dt>
<dd>
<p>Steric parameter (Charton, 1981)</p>
</dd>
<dt>AccNo. DAYM780201</dt>
<dd>
<p>Relative mutability (Dayhoff et al., 1978b)</p>
</dd>
</dl>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nlag</code></td>
<td>
<p>Maximum value of the lag parameter. Default is <code>30</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>customprops</code></td>
<td>
<p>A <code>n x 21</code> named data frame contains <code>n</code>
customized property. Each row contains one property.
The column order for different amino acid types is
<code>'AccNo'</code>, <code>'A'</code>, <code>'R'</code>, <code>'N'</code>,
<code>'D'</code>, <code>'C'</code>, <code>'E'</code>, <code>'Q'</code>,
<code>'G'</code>, <code>'H'</code>, <code>'I'</code>, <code>'L'</code>,
<code>'K'</code>, <code>'M'</code>, <code>'F'</code>, <code>'P'</code>,
<code>'S'</code>, <code>'T'</code>, <code>'W'</code>, <code>'Y'</code>,
<code>'V'</code>, and the columns should also be <em>exactly</em> named like this.
The <code>AccNo</code> column contains the properties' names.
Then users should explicitly specify these properties
with these names in the argument <code>props</code>.
See the examples below for a demonstration.
The default value for <code>customprops</code> is <code>NULL</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A length <code>length(props) * nlag</code> named vector.
</p>


<h3>Note</h3>

<p>For this descriptor type, users need to intelligently evaluate
the underlying details of the descriptors provided, instead of using
this function with their data blindly. It would be wise to use some
negative and positive control comparisons where relevant to help guide
interpretation of the results.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>AAindex: Amino acid index database.
<a href="https://www.genome.jp/dbget/aaindex.html">https://www.genome.jp/dbget/aaindex.html</a>
</p>
<p>Feng, Z.P. and Zhang, C.T. (2000)
Prediction of membrane protein types based on the hydrophobic
index of amino acids.
<em>Journal of Protein Chemistry</em>, 19, 269-275.
</p>
<p>Horne, D.S. (1988)
Prediction of protein helix content from
an autocorrelation analysis of sequence hydrophobicities.
<em>Biopolymers</em>, 27, 451-477.
</p>
<p>Sokal, R.R. and Thomson, B.A. (2006)
Population structure inferred by local spatial autocorrelation:
an usage from an Amerindian tribal population.
<em>American Journal of Physical Anthropology</em>, 129, 121-131.
</p>


<h3>See Also</h3>

<p>See <code>extractMoreauBroto</code> and <code>extractGeary</code>
for Moreau-Broto autocorrelation descriptors and
Geary autocorrelation descriptors.
</p>


<h3>Examples</h3>

<pre><code class="language-R">x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
extractMoran(x)

myprops &lt;- data.frame(
  AccNo = c("MyProp1", "MyProp2", "MyProp3"),
  A = c(0.62, -0.5, 15), R = c(-2.53, 3, 101),
  N = c(-0.78, 0.2, 58), D = c(-0.9, 3, 59),
  C = c(0.29, -1, 47), E = c(-0.74, 3, 73),
  Q = c(-0.85, 0.2, 72), G = c(0.48, 0, 1),
  H = c(-0.4, -0.5, 82), I = c(1.38, -1.8, 57),
  L = c(1.06, -1.8, 57), K = c(-1.5, 3, 73),
  M = c(0.64, -1.3, 75), F = c(1.19, -2.5, 91),
  P = c(0.12, 0, 42), S = c(-0.18, 0.3, 31),
  T = c(-0.05, -0.4, 45), W = c(0.81, -3.4, 130),
  Y = c(0.26, -2.3, 107), V = c(1.08, -1.5, 43)
)

# Use 4 properties in the AAindex database, and 3 cutomized properties
extractMoran(
  x,
  customprops = myprops,
  props = c(
    "CIDH920105", "BHAR880101",
    "CHAM820101", "CHAM820102",
    "MyProp1", "MyProp2", "MyProp3"
  )
)
</code></pre>


</div>