<div class="container">

<table style="width: 100%;"><tr>
<td>removeGaps</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Remove or replace gaps from protein sequences.</h2>

<h3>Description</h3>

<p>Remove/replace gaps or any irregular characters from protein sequences,
to make them suitable for feature extraction or sequence alignment
based similarity computation.
</p>


<h3>Usage</h3>

<pre><code class="language-R">removeGaps(x, pattern = "-", replacement = "", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>character vector, containing the input protein sequence(s).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pattern</code></td>
<td>
<p>character string contains the gap (or other irregular)
character to be removed or replaced. Default is <code>"-"</code>.
For advanced usage, see <code>gsub</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>replacement</code></td>
<td>
<p>a replacement for matched characters.
Default is <code>""</code> (remove the matched character).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>addtional parameters for <code>gsub</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a vector of protein sequence(s) with gaps or
irregular characters removed/replaced.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>Examples</h3>

<pre><code class="language-R"># amino acid sequences that contain gaps ("-")
aaseq &lt;- list(
  "MHGDTPTLHEYMLDLQPETTDLYCYEQLSDSSE-EEDEIDGPAGQAEPDRAHYNIVTFCCKCDSTLRLCVQS",
  "MHGDTPTLHEYMLDLQPETTDLYCYEQLNDSSE-EEDEIDGPAGQAEPDRAHYNIVTFCCKCDSTLRLCVQS"
)
## Not run: 
#' # gaps create issues for alignment
parSeqSim(aaseq)

# remove the gaps
nogapseq &lt;- removeGaps(aaseq)
parSeqSim(nogapseq)

## End(Not run)
</code></pre>


</div>