<div class="container">

<table style="width: 100%;"><tr>
<td>extractPSSM</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute PSSM (Position-Specific Scoring Matrix) for given protein sequence</h2>

<h3>Description</h3>

<p>This function calculates the PSSM (Position-Specific Scoring Matrix) derived
by PSI-Blast for given protein sequence or peptides.
</p>


<h3>Usage</h3>

<pre><code class="language-R">extractPSSM(
  seq,
  start.pos = 1L,
  end.pos = nchar(seq),
  psiblast.path = NULL,
  makeblastdb.path = NULL,
  database.path = NULL,
  iter = 5,
  silent = TRUE,
  evalue = 10L,
  word.size = NULL,
  gapopen = NULL,
  gapextend = NULL,
  matrix = "BLOSUM62",
  threshold = NULL,
  seg = "no",
  soft.masking = FALSE,
  culling.limit = NULL,
  best.hit.overhang = NULL,
  best.hit.score.edge = NULL,
  xdrop.ungap = NULL,
  xdrop.gap = NULL,
  xdrop.gap.final = NULL,
  window.size = NULL,
  gap.trigger = 22L,
  num.threads = 1L,
  pseudocount = 0L,
  inclusion.ethresh = 0.002
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seq</code></td>
<td>
<p>Character vector, as the input protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start.pos</code></td>
<td>
<p>Optional integer denoting the start position of the
fragment window. Default is <code>1</code>,
i.e. the first amino acid of the given sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>end.pos</code></td>
<td>
<p>Optional integer denoting the end position of the
fragment window. Default is <code>nchar(seq)</code>,
i.e. the last amino acid of the given sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>psiblast.path</code></td>
<td>
<p>Character string indicating the path of the
<code>psiblast</code> program.
If NCBI Blast+ was previously installed in the operation system,
the path will be automatically detected.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>makeblastdb.path</code></td>
<td>
<p>Character string indicating the path of the
<code>makeblastdb</code> program.
If NCBI Blast+ was previously installed in the system,
the path will be automatically detected.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>database.path</code></td>
<td>
<p>Character string indicating the path of
a reference database (a FASTA file).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>iter</code></td>
<td>
<p>Number of iterations to perform for PSI-Blast.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>silent</code></td>
<td>
<p>Logical. Whether the PSI-Blast running output
should be shown or not (May not work on some Windows versions and
PSI-Blast versions), default is <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>evalue</code></td>
<td>
<p>Expectation value (E) threshold for saving hits.
Default is <code>10</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>word.size</code></td>
<td>
<p>Word size for wordfinder algorithm. An integer &gt;= 2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gapopen</code></td>
<td>
<p>Integer. Cost to open a gap.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gapextend</code></td>
<td>
<p>Integer. Cost to extend a gap.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>matrix</code></td>
<td>
<p>Character string. The scoring matrix name
(default is <code>"BLOSUM62"</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>Minimum word score such that the word is added to
the BLAST lookup table. A real value &gt;= 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seg</code></td>
<td>
<p>Character string. Filter query sequence with SEG (<code>"yes"</code>,
<code>"window locut hicut"</code>, or <code>"no"</code> to disable).
Default is <code>"no"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>soft.masking</code></td>
<td>
<p>Logical. Apply filtering locations as soft masks?
Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>culling.limit</code></td>
<td>
<p>An integer &gt;= 0. If the query range of a hit is
enveloped by that of at least this many higher-scoring hits,
delete the hit. Incompatible with <code>best.hit.overhang</code> and
<code>best_hit_score_edge</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>best.hit.overhang</code></td>
<td>
<p>Best Hit algorithm overhang value
(A real value &gt;= 0 and =&lt; 0.5, recommended value: 0.1).
Incompatible with <code>culling_limit</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>best.hit.score.edge</code></td>
<td>
<p>Best Hit algorithm score edge value
(A real value &gt;=0 and =&lt; 0.5, recommended value: 0.1).
Incompatible with <code>culling_limit</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xdrop.ungap</code></td>
<td>
<p>X-dropoff value (in bits) for ungapped extensions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xdrop.gap</code></td>
<td>
<p>X-dropoff value (in bits) for preliminary gapped extensions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xdrop.gap.final</code></td>
<td>
<p>X-dropoff value (in bits) for final gapped alignment.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>window.size</code></td>
<td>
<p>An integer &gt;= 0. Multiple hits window size,
To specify 1-hit algorithm, use <code>0</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gap.trigger</code></td>
<td>
<p>Number of bits to trigger gapping. Default is <code>22</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.threads</code></td>
<td>
<p>Integer. Number of threads (CPUs) to use in the
BLAST search. Default is <code>1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pseudocount</code></td>
<td>
<p>Integer. Pseudo-count value used when constructing PSSM.
Default is <code>0</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>inclusion.ethresh</code></td>
<td>
<p>E-value inclusion threshold for pairwise alignments.
Default is <code>0.002</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For given protein sequences or peptides, PSSM represents the
log-likelihood of the substitution of the 20 types of amino acids at that
position in the sequence. Note that the output value is not normalized.
</p>


<h3>Value</h3>

<p>The original PSSM, a numeric matrix which has
<code>end.pos - start.pos + 1</code> columns and <code>20</code> named rows.
</p>


<h3>Note</h3>

<p>The function requires the <code>makeblastdb</code> and <code>psiblast</code> programs
to be properly installed in the operation system or
their paths provided.
</p>
<p>The two command-line programs are included in the NCBI-BLAST+
software package. To install NCBI Blast+ for your operating system, see
<a href="https://blast.ncbi.nlm.nih.gov/doc/blast-help/downloadblastdata.html">https://blast.ncbi.nlm.nih.gov/doc/blast-help/downloadblastdata.html</a>
for detailed instructions.
</p>
<p>Ubuntu/Debian users can directly use the command
<code>sudo apt-get install ncbi-blast+</code> to install NCBI Blast+.
For OS X users, download <code>ncbi-blast- ... .dmg</code> then install.
For Windows users, download <code>ncbi-blast- ... .exe</code> then install.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Altschul, Stephen F., et al.
"Gapped BLAST and PSI-BLAST: a new generation of protein database search programs."
<em>Nucleic acids research</em> 25.17 (1997): 3389–3402.
</p>
<p>Ye, Xugang, Guoli Wang, and Stephen F. Altschul.
"An assessment of substitution scores for protein profile-profile comparison."
<em>Bioinformatics</em> 27.24 (2011): 3356–3363.
</p>
<p>Rangwala, Huzefa, and George Karypis.
"Profile-based direct kernels for remote homology detection and fold recognition."
<em>Bioinformatics</em> 21.23 (2005): 4239–4247.
</p>


<h3>See Also</h3>

<p>extractPSSMFeature extractPSSMAcc
</p>


<h3>Examples</h3>

<pre><code class="language-R">if (Sys.which("makeblastdb") == "" | Sys.which("psiblast") == "") {
  cat("Cannot find makeblastdb or psiblast. Please install NCBI Blast+ first")
} else {
  x &lt;- readFASTA(system.file(
    "protseq/P00750.fasta",
    package = "protr"
  ))[[1]]
  dbpath &lt;- tempfile("tempdb", fileext = ".fasta")
  invisible(file.copy(from = system.file(
    "protseq/Plasminogen.fasta",
    package = "protr"
  ), to = dbpath))

  pssmmat &lt;- extractPSSM(seq = x, database.path = dbpath)

  dim(pssmmat) # 20 x 562 (P00750: length 562, 20 Amino Acids)
}
</code></pre>


</div>