<div class="container">

<table style="width: 100%;"><tr>
<td>extractCTDDClass</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>CTD Descriptors - Distribution (with customized amino acid
classification support)</h2>

<h3>Description</h3>

<p>This function calculates the Distribution descriptor of the
CTD descriptors, with customized amino acid classification support.
</p>


<h3>Usage</h3>

<pre><code class="language-R">extractCTDDClass(x, aagroup1, aagroup2, aagroup3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aagroup1</code></td>
<td>
<p>A named list which contains the first group of customized
amino acid classification. See example below.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aagroup2</code></td>
<td>
<p>A named list which contains the second group of customized
amino acid classification. See example below.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aagroup3</code></td>
<td>
<p>A named list which contains the third group of customized
amino acid classification. See example below.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A length <code>k * 15</code> named vector, <code>k</code> is the number of
amino acid properties used.
</p>


<h3>Note</h3>

<p>For this descriptor type, users need to intelligently evaluate
the underlying details of the descriptors provided, instead of using
this function with their data blindly. It would be wise to use some
negative and positive control comparisons where relevant to help guide
interpretation of the results.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Inna Dubchak, Ilya Muchink, Stephen R. Holbrook and Sung-Hou Kim.
Prediction of protein folding class using global description of
amino acid sequence. <em>Proceedings of the National Academy of Sciences</em>.
USA, 1995, 92, 8700-8704.
</p>
<p>Inna Dubchak, Ilya Muchink, Christopher Mayor, Igor Dralyuk and Sung-Hou Kim.
Recognition of a Protein Fold in the Context of the SCOP classification.
<em>Proteins: Structure, Function and Genetics</em>, 1999, 35, 401-407.
</p>


<h3>See Also</h3>

<p>See <code>extractCTDCClass</code> and
<code>extractCTDTClass</code> for Composition and Transition of
the CTD descriptors with customized amino acid classification support.
</p>


<h3>Examples</h3>

<pre><code class="language-R">x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]

# using five customized amino acid property classification
group1 &lt;- list(
  "hydrophobicity" = c("R", "K", "E", "D", "Q", "N"),
  "normwaalsvolume" = c("G", "A", "S", "T", "P", "D", "C"),
  "polarizability" = c("G", "A", "S", "D", "T"),
  "secondarystruct" = c("E", "A", "L", "M", "Q", "K", "R", "H"),
  "solventaccess" = c("A", "L", "F", "C", "G", "I", "V", "W")
)

group2 &lt;- list(
  "hydrophobicity" = c("G", "A", "S", "T", "P", "H", "Y"),
  "normwaalsvolume" = c("N", "V", "E", "Q", "I", "L"),
  "polarizability" = c("C", "P", "N", "V", "E", "Q", "I", "L"),
  "secondarystruct" = c("V", "I", "Y", "C", "W", "F", "T"),
  "solventaccess" = c("R", "K", "Q", "E", "N", "D")
)

group3 &lt;- list(
  "hydrophobicity" = c("C", "L", "V", "I", "M", "F", "W"),
  "normwaalsvolume" = c("M", "H", "K", "F", "R", "Y", "W"),
  "polarizability" = c("K", "M", "H", "F", "R", "Y", "W"),
  "secondarystruct" = c("G", "N", "P", "S", "D"),
  "solventaccess" = c("M", "S", "P", "T", "H", "Y")
)

extractCTDDClass(x, aagroup1 = group1, aagroup2 = group2, aagroup3 = group3)
</code></pre>


</div>