<div class="container">

<table style="width: 100%;"><tr>
<td>extractBLOSUM</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>BLOSUM and PAM Matrix-Derived Descriptors</h2>

<h3>Description</h3>

<p>This function calculates BLOSUM matrix-derived descriptors.
For users' convenience, <code>protr</code> provides the BLOSUM45, BLOSUM50,
BLOSUM62, BLOSUM80, BLOSUM100, PAM30, PAM40, PAM70, PAM120, and PAM250
matrices for the 20 amino acids to select from.
</p>


<h3>Usage</h3>

<pre><code class="language-R">extractBLOSUM(x, submat = "AABLOSUM62", k, lag, scale = TRUE, silent = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>submat</code></td>
<td>
<p>Substitution matrix for the 20 amino acids. Should be one of
<code>AABLOSUM45</code>, <code>AABLOSUM50</code>, <code>AABLOSUM62</code>,
<code>AABLOSUM80</code>, <code>AABLOSUM100</code>, <code>AAPAM30</code>,
<code>AAPAM40</code>, <code>AAPAM70</code>, <code>AAPAM120</code>, or <code>AAPAM250</code>.
Default is <code>"AABLOSUM62"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>Integer. The number of selected scales (i.e. the first
<code>k</code> scales) derived by the substitution matrix.
This can be selected according to the printed relative importance values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lag</code></td>
<td>
<p>The lag parameter. Must be less than the amino acids.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale</code></td>
<td>
<p>Logical. Should we auto-scale the substitution matrix
(<code>submat</code>) before doing eigen decomposition? Default is <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>silent</code></td>
<td>
<p>Logical. Whether we print the relative importance of
each scales (diagnal value of the eigen decomposition result matrix B)
or not. Default is <code>TRUE</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A length <code>lag * p^2</code> named vector, <code>p</code> is the number
of scales selected.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Georgiev, A. G. (2009).
Interpretable numerical descriptors of amino acid space.
Journal of Computational Biology, 16(5), 703â€“723.
</p>


<h3>Examples</h3>

<pre><code class="language-R">x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
blosum &lt;- extractBLOSUM(x, submat = "AABLOSUM62", k = 5, lag = 7, scale = TRUE, silent = FALSE)
</code></pre>


</div>