<div class="container">

<table style="width: 100%;"><tr>
<td>extractScales</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Scales-Based Descriptors derived by Principal Components Analysis</h2>

<h3>Description</h3>

<p>This function calculates scales-based descriptors
derived by Principal Components Analysis (PCA).
Users can provide customized amino acid property matrices.
This function implements the core computation procedure needed for
the scales-based descriptors derived by AA-Properties (AAindex)
and scales-based descriptors derived by 20+ classes of 2D and 3D
molecular descriptors (Topological, WHIM, VHSE, etc.) in the protr package.
</p>


<h3>Usage</h3>

<pre><code class="language-R">extractScales(x, propmat, pc, lag, scale = TRUE, silent = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>propmat</code></td>
<td>
<p>A matrix containing the properties for the amino acids.
Each row represent one amino acid type, each column represents
one property. Note that the one-letter row names must be provided
for we need them to seek the properties for each AA type.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pc</code></td>
<td>
<p>Integer. Use the first pc principal components as the scales.
Must be no greater than the number of AA properties provided.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lag</code></td>
<td>
<p>The lag parameter. Must be less than the amino acids.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale</code></td>
<td>
<p>Logical. Should we auto-scale the property matrix
(<code>propmat</code>) before PCA? Default is <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>silent</code></td>
<td>
<p>Logical. Whether we print the standard deviation,
proportion of variance and the cumulative proportion of
the selected principal components or not. Default is <code>TRUE</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A length <code>lag * p^2</code> named vector,
<code>p</code> is the number of scales (principal components) selected.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>See Also</h3>

<p>See <code>extractDescScales</code> scales descriptors based on
20+ classes of molecular descriptors, and <code>extractProtFP</code>
for amino acid property based scales descriptors (protein fingerprint).
</p>


<h3>Examples</h3>

<pre><code class="language-R">x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
data(AAindex)
AAidxmat &lt;- t(na.omit(as.matrix(AAindex[, 7:26])))
scales &lt;- extractScales(x, propmat = AAidxmat, pc = 5, lag = 7, silent = FALSE)
</code></pre>


</div>