<div class="container">

<table style="width: 100%;"><tr>
<td>extractCTriadClass</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Conjoint Triad Descriptor (with customized amino acid
classification support)</h2>

<h3>Description</h3>

<p>This function calculates the Conjoint Triad descriptor, with customized
amino acid classification support.
</p>


<h3>Usage</h3>

<pre><code class="language-R">extractCTriadClass(x, aaclass)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aaclass</code></td>
<td>
<p>A list containing the customized amino acid
classification. See example below.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A length <code>k^3</code> named vector, where <code>k</code> is the number
of customized classes of the amino acids.
</p>


<h3>Note</h3>

<p>For this descriptor type, users need to intelligently evaluate
the underlying details of the descriptors provided, instead of using
this function with their data blindly. It would be wise to use some
negative and positive control comparisons where relevant to help guide
interpretation of the results.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>J.W. Shen, J. Zhang, X.M. Luo, W.L. Zhu,
K.Q. Yu, K.X. Chen, Y.X. Li, H.L. Jiang.
Predicting Protein-protein Interactions Based Only on Sequences Information.
<em>Proceedings of the National Academy of Sciences</em>. 007, 104, 4337â€“4341.
</p>


<h3>Examples</h3>

<pre><code class="language-R">x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]

# use customized amino acid classification (normalized van der Waals volume)
newclass &lt;- list(
  c("G", "A", "S", "T", "P", "D", "C"),
  c("N", "V", "E", "Q", "I", "L"),
  c("M", "H", "K", "F", "R", "Y", "W")
)

extractCTriadClass(x, aaclass = newclass)
</code></pre>


</div>