<div class="container">

<table style="width: 100%;"><tr>
<td>hmm_ac</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>hmm_ac</h2>

<h3>Description</h3>

<p>This feature calculates the covariance between two residues separated by a lag value within
the same amino acid emission frequency column along the protein sequence.
</p>


<h3>Usage</h3>

<pre><code class="language-R">hmm_ac(hmm, lg = 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>hmm</code></td>
<td>
<p>The name of a profile hidden markov model file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lg</code></td>
<td>
<p>The lag value, which indicates the distance between residues.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A vector of length 20 <code class="reqn">\times</code> the lag value; by default this is a vector of length 80.
</p>


<h3>Note</h3>

<p>The lag value must be less than the length of the protein sequence
</p>


<h3>References</h3>

<p>Dong, Q., Zhou, S., &amp; Guan, J. (2009).
A new taxonomy-based protein fold recognition approach based on autocross-covariance transformation.
Bioinformatics, 25(20), 2655â€“2662.
</p>


<h3>Examples</h3>

<pre><code class="language-R">h&lt;- hmm_ac(system.file("extdata", "1DLHA2-7", package="protHMM"))
</code></pre>


</div>