<div class="container">

<table style="width: 100%;"><tr>
<td>twoGOSim</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Protein Similarity Calculation based on Gene Ontology (GO) Similarity</h2>

<h3>Description</h3>

<p>This function calculates the Gene Ontology (GO) similarity
between two groups of GO terms or two Entrez gene IDs.
</p>


<h3>Usage</h3>

<pre><code class="language-R">twoGOSim(
  id1,
  id2,
  type = c("go", "gene"),
  ont = c("MF", "BP", "CC"),
  organism = "human",
  measure = "Resnik",
  combine = "BMA"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>id1</code></td>
<td>
<p>Character vector. When length &gt; 1: each element is a GO term;
when length = 1: the Entrez Gene ID.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id2</code></td>
<td>
<p>Character vector. When length &gt; 1: each element is a GO term;
when length = 1: the Entrez Gene ID.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>Input type of id1 and id2, <code>'go'</code> for GO Terms,
<code>"gene"</code> for gene ID.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ont</code></td>
<td>
<p>Default is <code>"MF"</code>, can be one of <code>"MF"</code>,
<code>"BP"</code>, or <code>"CC"</code> subontologies.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>organism</code></td>
<td>
<p>Organism name. Default is <code>"human"</code>, can be one of
<code>"anopheles"</code>, <code>"arabidopsis"</code>, <code>"bovine"</code>, <code>"canine"</code>,
<code>"chicken"</code>, <code>"chimp"</code>, <code>"coelicolor"</code>, <code>"ecolik12"</code>,
<code>"ecsakai"</code>, <code>"fly"</code>, <code>"human"</code>, <code>"malaria"</code>,
<code>"mouse"</code>, <code>"pig"</code>, <code>"rat"</code>, <code>"rhesus"</code>,
<code>"worm"</code>, <code>"xenopus"</code>, <code>"yeast"</code> or <code>"zebrafish"</code>.
Before specifying the organism, please install the corresponding genome wide
<a href="https://bioconductor.org/packages/release/BiocViews.html#___OrgDb">annotation data package</a>
for the selected organism.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>measure</code></td>
<td>
<p>Default is <code>"Resnik"</code>, can be one of
<code>"Resnik"</code>, <code>"Lin"</code>, <code>"Rel"</code>, <code>"Jiang"</code>
or <code>"Wang"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>combine</code></td>
<td>
<p>Default is <code>"BMA"</code>, can be one of <code>"max"</code>,
<code>"average"</code>, <code>"rcmax"</code> or <code>"BMA"</code>
for combining semantic similarity scores of multiple GO terms
associated with proteins.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Similarity value.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>See Also</h3>

<p>See <code>parGOSim</code> for protein similarity calculation
based on Gene Ontology (GO) semantic similarity.
See <code>parSeqSim</code> for paralleled protein similarity
calculation based on Smith-Waterman local alignment.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 

# Be careful when testing this since it involves GO similarity computation
# and might produce unpredictable results in some environments

library("GOSemSim")
library("org.Hs.eg.db")

# By GO terms
go1 &lt;- c("GO:0004022", "GO:0004024", "GO:0004023")
go2 &lt;- c("GO:0009055", "GO:0020037")
twoGOSim(go1, go2, type = "go", ont = "MF", measure = "Wang")

# By Entrez gene id
gene1 &lt;- "1956" # EGFR
gene2 &lt;- "2261" # FGFR3
twoGOSim(gene1, gene2, type = "gene", ont = "BP", measure = "Lin")

## End(Not run)
</code></pre>


</div>