<div class="container">

<table style="width: 100%;"><tr>
<td>extractPSSMFeature</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Profile-based protein representation derived by PSSM
(Position-Specific Scoring Matrix)</h2>

<h3>Description</h3>

<p>This function calculates the profile-based protein representation
derived by PSSM. The feature vector is based on the PSSM computed by
<code>extractPSSM</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">extractPSSMFeature(pssmmat)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>pssmmat</code></td>
<td>
<p>The PSSM computed by <code>extractPSSM</code>.</p>
</td>
</tr></table>
<h3>Details</h3>

<p>For a given sequence, the PSSM feature represents the log-likelihood
of the substitution of the 20 types of amino acids at that position
in the sequence.
</p>
<p>Each PSSM feature value in the vector represents the degree of
conservation of a given amino acid type. The value is normalized
to interval (0, 1) by the transformation 1/(1+e^(-x)).
</p>


<h3>Value</h3>

<p>A numeric vector which has <code>20 x N</code> named elements,
where <code>N</code> is the size of the window (number of rows of the PSSM).
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Ye, Xugang, Guoli Wang, and Stephen F. Altschul.
"An assessment of substitution scores for protein profile-profile comparison."
<em>Bioinformatics</em> 27.24 (2011): 3356–3363.
</p>
<p>Rangwala, Huzefa, and George Karypis.
"Profile-based direct kernels for remote homology detection and fold recognition."
<em>Bioinformatics</em> 21.23 (2005): 4239–4247.
</p>


<h3>See Also</h3>

<p>extractPSSM extractPSSMAcc
</p>


<h3>Examples</h3>

<pre><code class="language-R">if (Sys.which("makeblastdb") == "" | Sys.which("psiblast") == "") {
  cat("Cannot find makeblastdb or psiblast. Please install NCBI Blast+")
} else {
  x &lt;- readFASTA(system.file(
    "protseq/P00750.fasta",
    package = "protr"
  ))[[1]]
  dbpath &lt;- tempfile("tempdb", fileext = ".fasta")
  invisible(file.copy(from = system.file(
    "protseq/Plasminogen.fasta",
    package = "protr"
  ), to = dbpath))

  pssmmat &lt;- extractPSSM(seq = x, database.path = dbpath)
  pssmfeature &lt;- extractPSSMFeature(pssmmat)
  head(pssmfeature)
}
</code></pre>


</div>