<div class="container">

<table style="width: 100%;"><tr>
<td>twoSeqSim</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Protein Sequence Alignment for Two Protein Sequences</h2>

<h3>Description</h3>

<p>Sequence alignment between two protein sequences.
</p>


<h3>Usage</h3>

<pre><code class="language-R">twoSeqSim(
  seq1,
  seq2,
  type = "local",
  submat = "BLOSUM62",
  gap.opening = 10,
  gap.extension = 4
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seq1</code></td>
<td>
<p>Character string, containing one protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq2</code></td>
<td>
<p>Character string, containing another protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>Type of alignment, default is <code>"local"</code>,
could be <code>"global"</code> or <code>"local"</code>,
where <code>"global"</code> represents Needleman-Wunsch global alignment;
<code>"local"</code> represents Smith-Waterman local alignment.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>submat</code></td>
<td>
<p>Substitution matrix, default is <code>"BLOSUM62"</code>,
can be one of <code>"BLOSUM45"</code>, <code>"BLOSUM50"</code>, <code>"BLOSUM62"</code>,
<code>"BLOSUM80"</code>, <code>"BLOSUM100"</code>, <code>"PAM30"</code>, <code>"PAM40"</code>,
<code>"PAM70"</code>, <code>"PAM120"</code>, or <code>"PAM250"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gap.opening</code></td>
<td>
<p>The cost required to open a gap of any length
in the alignment. Defaults to 10.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gap.extension</code></td>
<td>
<p>The cost to extend the length of an existing
gap by 1. Defaults to 4.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A <code>Biostrings</code> object containing the alignment scores
and other alignment information.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>See Also</h3>

<p>See <code>parSeqSim</code> for paralleled pairwise
protein similarity calculation based on sequence alignment.
See <code>twoGOSim</code> for calculating the GO semantic
similarity between two groups of GO terms or two Entrez gene IDs.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 

# Be careful when testing this since it involves sequence alignment
# and might produce unpredictable results in some environments

library("Biostrings")

s1 &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
s2 &lt;- readFASTA(system.file("protseq/P10323.fasta", package = "protr"))[[1]]
seqalign &lt;- twoSeqSim(s1, s2)
summary(seqalign)
score(seqalign)

## End(Not run)
</code></pre>


</div>