<div class="container">

<table style="width: 100%;"><tr>
<td>extractFAScales</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Scales-Based Descriptors derived by Factor Analysis</h2>

<h3>Description</h3>

<p>This function calculates scales-based descriptors
derived by Factor Analysis (FA).
Users can provide customized amino acid property matrices.
</p>


<h3>Usage</h3>

<pre><code class="language-R">extractFAScales(
  x,
  propmat,
  factors,
  scores = "regression",
  lag,
  scale = TRUE,
  silent = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A character vector, as the input protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>propmat</code></td>
<td>
<p>A matrix containing the properties for the amino acids.
Each row represent one amino acid type, each column represents one property.
Note that the one-letter row names must be provided for we need them to seek
the properties for each AA type.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>factors</code></td>
<td>
<p>Integer. The number of factors to be fitted.
Must be no greater than the number of AA properties provided.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scores</code></td>
<td>
<p>Type of scores to produce. The default is <code>"regression"</code>,
which gives Thompson's scores, <code>"Bartlett"</code> given Bartlett's weighted
least-squares scores.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lag</code></td>
<td>
<p>The lag parameter. Must be less than the amino acids number
in the protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale</code></td>
<td>
<p>Logical. Should we auto-scale the property matrix
(<code>propmat</code>) before doing Factor Analysis? Default is <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>silent</code></td>
<td>
<p>Logical. Whether we print the SS loadings,
proportion of variance and the cumulative proportion of
the selected factors or not. Default is <code>TRUE</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A length <code>lag * p^2</code> named vector,
<code>p</code> is the number of scales (factors) selected.
</p>


<h3>Author(s)</h3>

<p>Nan Xiao &lt;<a href="https://nanx.me">https://nanx.me</a>&gt;
</p>


<h3>References</h3>

<p>Atchley, W. R., Zhao, J., Fernandes, A. D., &amp; Druke, T. (2005).
Solving the protein sequence metric problem.
Proceedings of the National Academy of Sciences of the United States of America,
102(18), 6395-6400.
</p>


<h3>Examples</h3>

<pre><code class="language-R">x &lt;- readFASTA(system.file("protseq/P00750.fasta", package = "protr"))[[1]]
data(AATopo)
tprops &lt;- AATopo[, c(37:41, 43:47)] # select a set of topological descriptors
fa &lt;- extractFAScales(x, propmat = tprops, factors = 5, lag = 7, silent = FALSE)
</code></pre>


</div>