<div class="container">

<table style="width: 100%;"><tr>
<td>PRS_PGx_Bayes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Construct PGx PRS using Bayesian regression</h2>

<h3>Description</h3>

<p>Flexibly shrink prognostic and predictive effect sizes simutaneously with glocal-local shrinkage parameters
</p>


<h3>Usage</h3>

<pre><code class="language-R">PRS_PGx_Bayes(
  PGx_GWAS,
  G_reference,
  n.itr = 1000,
  n.burnin = 500,
  n.gap = 10,
  paras,
  standardize = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>PGx_GWAS</code></td>
<td>
<p>a numeric list containing PGx GWAS summary statistics (with SNP ID, position, <code class="reqn">\beta</code>, <code class="reqn">\alpha</code>, 2-df p-value, MAF and N), SD(Y), and mean(T)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>G_reference</code></td>
<td>
<p>a numeric matrix containing the individual-level genotype information from the reference panel (e.g., 1KG)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.itr</code></td>
<td>
<p>a numeric value indicating the total number of MCMC iteration</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.burnin</code></td>
<td>
<p>a numeric value indicating the number of burn in</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.gap</code></td>
<td>
<p>a numeric value indicating the MCMC gap</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>paras</code></td>
<td>
<p>a numeric vector containg hyper-parameters (<code class="reqn">v</code>, <code class="reqn">\phi</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>standardize</code></td>
<td>
<p>a logical flag indicating should phenotype and genotype be standardized</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>PRS-PGx-Bayes only needs PGx summary statistics and external reference genotype
</p>


<h3>Value</h3>

<p>A numeric list, the first sublist contains estimated prognostic effect sizes, the second sublist contains estimated predictive effect sizes
</p>


<h3>Author(s)</h3>

<p>Song Zhai
</p>


<h3>References</h3>

<p>Ge, T., Chen, CY., Ni, Y. et al. Polygenic prediction via Bayesian regression and continuous shrinkage priors. Nat. Commun. 10, 1776 (2019).
</p>
<p>Zhai, S., Zhang, H., Mehrotra, D.V. &amp; Shen, J. Paradigm Shift from Disease PRS to PGx PRS for Drug Response Prediction using PRS-PGx Methods (submitted).
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(PRSPGx.example); attach(PRSPGx.example)
paras = c(3, 5)
coef_est &lt;- PRS_PGx_Bayes(PGx_GWAS, G_reference, paras = paras, n.itr = 10, n.burnin = 5, n.gap = 1)
summary(coef_est$coef.G)
summary(coef_est$coef.TG)


</code></pre>


</div>