<div class="container">

<table style="width: 100%;"><tr>
<td>PRS_Dis_CT</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Construct disease PRS unadjusted or using clumping and thresholding</h2>

<h3>Description</h3>

<p>Shrink prognostic effect sizes by p-value cutoff (PRS-Dis-CT turns out to be PRS-Dis-Unadj when setting p-value cutoff = 1)
</p>


<h3>Usage</h3>

<pre><code class="language-R">PRS_Dis_CT(
  DIS_GWAS,
  G_reference,
  pcutoff = 1e-05,
  clumping = TRUE,
  p1 = 1e-04,
  d1 = 250000,
  r1 = 0.8
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>DIS_GWAS</code></td>
<td>
<p>a numeric matrix containing disease GWAS summary statistics, including SNP ID, position, <code class="reqn">\beta</code>, SE(<code class="reqn">\beta</code>), p-value, N, and MAF</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>G_reference</code></td>
<td>
<p>a numeric matrix containing the individual-level genotype information from the reference panel (e.g., 1KG)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pcutoff</code></td>
<td>
<p>a numeric value indicating the p-value cutoff</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clumping</code></td>
<td>
<p>a logical flag indicating should clumping be performed</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p1</code></td>
<td>
<p>a numeric value indicating p-value threshold to decide flag SNPs in clumping</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d1</code></td>
<td>
<p>a numeric value indicating window size in clumping</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r1</code></td>
<td>
<p>a numeric value indicating correlation in clumping</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>PRS-Dis-CT automatically sets predictive effect sizes equivalent to the prognostic effect sizes; and only need disease GWAS summary statistics
</p>


<h3>Value</h3>

<p>A numeric list, the first sublist contains estimated prognostic effect sizes, the second sublist contains estimated predictive effect sizes
</p>


<h3>Author(s)</h3>

<p>Song Zhai
</p>


<h3>References</h3>

<p>Euesden, J., Lewis, C.M. &amp; O'Reilly, P.F. PRSice: Polygenic Risk Score software. Bioinformatics 564, 1466-1468 (2015).
</p>
<p>Zhai, S., Zhang, H., Mehrotra, D.V. &amp; Shen, J. Paradigm Shift from Disease PRS to PGx PRS for Drug Response Prediction using PRS-PGx Methods (submitted).
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(PRSPGx.example); attach(PRSPGx.example)
coef_est &lt;- PRS_Dis_CT(DIS_GWAS, G_reference, pcutoff = 0.01, clumping = TRUE)
summary(coef_est$coef.G)
summary(coef_est$coef.TG)

</code></pre>


</div>