<div class="container">

<table style="width: 100%;"><tr>
<td>PRSPGx.example</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulated example data</h2>

<h3>Description</h3>

<p>Simulated example data required by PRS-DIS and PRS-PGx functions.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(PRSPGx.example)
</code></pre>


<h3>Format</h3>

<p>A list with 8 sublists:
</p>

<dl>
<dt>PGx_GWAS</dt>
<dd>
<p>PGx GWAS including SNP ID, MAF, position, <code class="reqn">\beta</code>, <code class="reqn">\alpha</code>, 2-df p-value, and N; SD(Y), and mean(T)</p>
</dd>
<dt>DIS_GWAS</dt>
<dd>
<p>disease GWAS including SNP ID, MAF, position, <code class="reqn">\beta</code>, SE(<code class="reqn">\beta</code>), p-value, and N</p>
</dd>
<dt>G_reference</dt>
<dd>
<p>simulated individual-level genotype from the reference panel matched with the simulated sample PGx genotype</p>
</dd>
<dt>Y</dt>
<dd>
<p>simulated phenotype (continuous)</p>
</dd>
<dt>T</dt>
<dd>
<p>simulated treatment assignment, 1 = treatment, 0 = placebo</p>
</dd>
<dt>G</dt>
<dd>
<p>simulated sample PGx genotype with 100 SNPs and 4000 subjects</p>
</dd>
<dt>beta</dt>
<dd>
<p>simulated prognostic effect sizes (i.e., the underlying true prognostic effect sizes)</p>
</dd>
<dt>alpha</dt>
<dd>
<p>simulated predictive effect sizes (i.e., the underlying true predictive effect sizes)</p>
</dd>
</dl>
</div>