<div class="container">

<table style="width: 100%;"><tr>
<td>findLargeGaps</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Identifies gaps of a genome where there exist no observations</h2>

<h3>Description</h3>

<p>Identifies gaps of a genome where there exist no observations.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## Default S3 method:
findLargeGaps(chromosome=NULL, x, minLength, resolution=1L, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>chromosome</code></td>
<td>
<p>(Optional) An <code>integer</code> <code>vector</code> of length J of
chromosome indices.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A <code>numeric</code> <code>vector</code> of J of genomic locations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minLength</code></td>
<td>
<p>A positive <code>numeric</code> scalar specifying the minimum
length of a gap.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>resolution</code></td>
<td>
<p>A non-negative <code>numeric</code> specifying the minimum
length unit, which by default equals one nucleotide/base pair.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Not used.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns <code>data.frame</code> zero or more rows and with columns
<code>chromosome</code> (if given), <code>start</code>, <code>stop</code>,
and <code>length</code>.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p>Use <code>gapsToSegments</code>() to turn the set of identified gaps into
the complementary set of segments such that they can be passed
to <code>segmentByCBS</code>(), <code>segmentByPairedPSCBS</code>() and
<code>segmentByNonPairedPSCBS</code>() via argument <code>knownSegments</code>.
</p>


</div>