<div class="container">

<table style="width: 100%;"><tr>
<td>gapsToSegments.data.frame</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Gets the genomic segments that are complementary to the gaps</h2>

<h3>Description</h3>

<p>Gets the genomic segments that are complementary to the gaps, with default chromosome boundaries being <code>-Inf</code>
and <code>+Inf</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'data.frame'
gapsToSegments(gaps, resolution=1L, minLength=0L, dropGaps=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gaps</code></td>
<td>
<p>A <code>data.frame</code> with columns <code>chromosome</code>, <code>start</code>,
and <code>stop</code>. Any overlapping gaps will throw an error.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>resolution</code></td>
<td>
<p>A non-negative <code>numeric</code> specifying the minimum
length unit, which by default equals one nucleotide/base pair.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minLength</code></td>
<td>
<p>Minimum length of segments to be kept.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dropGaps</code></td>
<td>
<p>If <code>TRUE</code>, the gaps themselves are not part of the output.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Not used.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns <code>data.frame</code> of least one row with columns <code>chromosome</code>
if that argument is given), <code>start</code>, <code>stop</code> and <code>length</code>.
The segments are ordered along the genome.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>See Also</h3>

<p><code>findLargeGaps</code>().
</p>


</div>