<div class="container">

<table style="width: 100%;"><tr>
<td>PSCBS-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Package PSCBS</h2>

<h3>Description</h3>

<p>Segmentation of allele-specific DNA copy number data and detection of regions with abnormal copy number within each parental chromosome.  Both tumor-normal paired and tumor-only analyses are supported..
</p>
<p>This package should be considered to be in an alpha or beta phase.
You should expect the API to be changing over time.
</p>


<h3>Installation and updates</h3>

<p>To install this package, use <code>install.packages("PSCBS")</code>.
</p>


<h3>To get started</h3>

<p>To get started, see:
</p>

<ol>
<li>
<p> Vignettes '<a href="../doc/index.html">Parent-specific copy-number segmentation using Paired PSCBS</a>' and '<a href="../doc/index.html">Total copy-number segmentation using CBS</a>'.
</p>
</li>
<li> <p><code>segmentByCBS</code>() - segments total copy-numbers, or any
other unimodal genomic signals, using the CBS method [3,4].
</p>
</li>
<li> <p><code>segmentByPairedPSCBS</code>() - segments allele-specific
tumor signal from a tumor <em>with</em> a matched normal
using the Paired PSCBS method [1,2].
</p>
</li>
<li> <p><code>segmentByNonPairedPSCBS</code>() - segments allele-specific
tumor signal from a tumor <em>without</em> a matched normal
using the Non-Paired PSCBS method adopted from [1,2].
</p>
</li>
</ol>
<h3>How to cite</h3>

<p>Please use [1] and [2] to cite when using Paired PSCBS,
and [3] and [4] when using CBS.
When using Non-Paired PSCBS, please cite [1] and [2] as well.
</p>


<h3>License</h3>

<p>GPL (&gt;= 2).
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


<h3>References</h3>

<p>[1] A.B. Olshen, H. Bengtsson, P. Neuvial, P.T. Spellman, R.A. Olshen, V.E. Seshan, <em>Parent-specific copy number in paired tumor-normal studies using circular binary segmentation</em>, Bioinformatics, 2011
<br>
[2] H. Bengtsson, P. Neuvial and T.P. Speed, <em>TumorBoost: Normalization of allele-specific tumor copy numbers from a single pair of tumor-normal genotyping microarrays</em>, BMC Bioinformatics, 2010
<br>
[3] A.B. Olshen, E.S. Venkatraman (aka Venkatraman E. Seshan), R. Lucito and M. Wigler, <em>Circular binary segmentation for the analysis of array-based DNA copy number data</em>, Biostatistics, 2004
<br>
[4] E.S. Venkatraman and A.B. Olshen, <em>A faster circular binary segmentation algorithm for the analysis of array CGH data</em>, Bioinformatics, 2007
<br></p>


</div>