<div class="container">

<table style="width: 100%;"><tr>
<td>plotTracks1.PairedPSCBS</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plots parental specific copy numbers along the genome</h2>

<h3>Description</h3>

<p>Plots parental specific copy numbers along the genome for one or more chromosomes.
It is possible to specify what type of tracks to plot.
Each type of track is plotted in its own panel.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'PairedPSCBS'
plotTracks1(x, tracks=c("tcn", "dh", "tcn,c1,c2", "tcn,c1", "tcn,c2", "c1,c2", "betaN",
  "betaT", "betaTN")[1:3], scatter="*", calls=".*", pch=".", col=NULL, cex=1,
  changepoints=FALSE, grid=FALSE, quantiles=c(0.05, 0.95), xlim=NULL,
  Clim=c(0, 3 * ploidy(x)), Blim=c(0, 1), xScale=1e-06, ..., add=FALSE,
  subplots=!add &amp;&amp; (length(tracks) &gt; 1), verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A result object returned by <code>segmentByPairedPSCBS</code>().</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tracks</code></td>
<td>
<p>A <code>character</code> <code>vector</code> specifying what types of tracks to plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scatter</code></td>
<td>
<p>A <code>character</code> <code>vector</code> specifying which of the tracks should
have scatter plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>calls</code></td>
<td>
<p>A <code>character</code> <code>vector</code> of regular expression identifying
call labels to be highlighted in the panels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pch</code></td>
<td>
<p>The type of the scatter points, if any.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>The color of the scatter points, if any.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex</code></td>
<td>
<p>The size of the scatter points, if any.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>changepoints</code></td>
<td>
<p>If <code>TRUE</code>, changepoints are drawn as vertical lines.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grid</code></td>
<td>
<p>If <code>TRUE</code>, horizontal lines are displayed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quantiles</code></td>
<td>
<p>A <code>numeric</code> <code>vector</code> in [0,1] specifying the quantiles
of the confidence bands to be drawn, if any.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlim</code></td>
<td>
<p>(Optional) The genomic range to plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Clim</code></td>
<td>
<p>The range of copy numbers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Blim</code></td>
<td>
<p>The range of allele B fractions (BAFs) and
decrease of heterozygosity (DHs).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xScale</code></td>
<td>
<p>The scale factor used for genomic positions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Not used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>add</code></td>
<td>
<p>If <code>TRUE</code>, the panels plotted are added to the existing plot,
otherwise a new plot is created.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subplots</code></td>
<td>
<p>If <code>TRUE</code>, then subplots are automatically setup.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>See <code>Verbose</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns nothing.
</p>


<h3>Author(s)</h3>

<p>Henrik Bengtsson</p>


</div>