<div class="container">

<table style="width: 100%;"><tr>
<td>RPM_PSSM</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>RPM-PSSM feature vector</h2>

<h3>Description</h3>

<p>In this feature The idea is similar to the probe concept used in microarray technologies, where probes are used to identify
genes. For the convenience, we call it residue probing method. In our application, each probe is an amino acid, which
corresponds to a particular column in the PSSM profiles. For each probe, we average the PSSM scores of all the amino acids
in the associated column with a PSSM value greater than zero in the sequence, which leads to a 1  20 feature vector. Once
again, for the 20 probes, the final feature for each protein sequence is a 1  400 vector.
</p>


<h3>Usage</h3>

<pre><code class="language-R">RPM_PSSM(pssm_name)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>pssm_name</code></td>
<td>
<p>name of PSSM Matrix file</p>
</td>
</tr></table>
<h3>Value</h3>

<p>RPM-PSSM feature vector of length 400
</p>


<h3>References</h3>

<p>Jeong, J.C., Lin, X. and Chen, X.W. (2011) On position-specific scoring matrix for protein function prediction
, IEEE/ACM transactions on computational biology and bioinformatics / IEEE, ACM, 8, 308-315.
</p>


<h3>Examples</h3>

<pre><code class="language-R">X&lt;- RPM_PSSM(system.file("extdata","C7GRQ3.txt.pssm",package="PSSMCOOL"))
</code></pre>


</div>