<div class="container">

<table style="width: 100%;"><tr>
<td>ptycho-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Bayesian Variable Selection with Hierarchical Priors</h2>

<h3>Description</h3>

<p>Bayesian variable selection for linear regression models using hierarchical
priors. There is a prior that combines information across responses and one
that combines information across covariates, as well as a standard spike and
slab prior for comparison. An MCMC samples from the marginal posterior
distribution for the 0-1 variables indicating if each covariate belongs to the
model for each response.
</p>


<h3>Details</h3>

<p>This package provides functions to carry out Bayesian model selection combining
different layers of information: across multiple traits or across multiple
variants in the same gene.  The priors are described by Stell and Sabatti
(2015).  To sample the posterior distribution for specified genotype and
phenotype matrices, use <code>ptycho</code>.
</p>
<p>This package also provides functions to generate simulated data as in Stell and
Sabatti (2015); see <code>createData</code> and
<a href="web.stanford.edu/~lstell/ptycho/">web.stanford.edu/~lstell/ptycho/</a>.  Those datasets are not included in this
package because they have images about 20 MB or larger.  Instead small data
objects are included for examples; see Data.
</p>
<p>Functions for post-processing <code>ptycho</code> objects are described at
<code>checkConvergence</code> and PosteriorStatistics.
</p>


<h3>Author(s)</h3>

<p>Laurel Stell and Chiara Sabatti<br>
Maintainer: Laurel Stell &lt;lstell@stanford.edu&gt;
</p>


<h3>References</h3>

<p>Stell, L. and Sabatti, C. (2015) Genetic variant selection: learning across
traits and sites, arXiv:1504.00946.
</p>


</div>