<div class="container">

<table style="width: 100%;"><tr>
<td>estimateParams</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Estimated pulse model parameters!</h2>

<h3>Description</h3>

<p>It is used to estimate pulse model parameters.
Transcription rates, processing rates, and degradation correspond to different pulse model parameters, respectively.
The transcription rate has 6 parameters, the processing rate has 5 parameters, and the degradation rate has 4 parameters.
</p>


<h3>Usage</h3>

<pre><code class="language-R">estimateParams(labexon, totexon, totintr, TimeGrid, tL,
  clusterNumber = NULL, loopnumber = 50, message = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>labexon</code></td>
<td>
<p>A matrix containing expression levels of 4su exons.For the calculation of the expression value, see <code>estimateExpression</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>totexon</code></td>
<td>
<p>A matrix containing expression levels of total exons.For the calculation of the expression value, see <code>estimateExpression</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>totintr</code></td>
<td>
<p>A matrix containing expression levels of total introns.For the calculation of the expression value, see <code>estimateExpression</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>TimeGrid</code></td>
<td>
<p>A vector of points in time at which experimental data is collected, not allowed to be repeated</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tL</code></td>
<td>
<p>4sU labeled time during the experiment</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clusterNumber</code></td>
<td>
<p>Given the number of cluster cores, the default is the maximum number of available cores</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loopnumber</code></td>
<td>
<p>The number of iterations of the gradient descent when solving the parameter. The default is 50.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>message</code></td>
<td>
<p>Whether to print the log, the default is TRUE</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A 'pulseTDmodel' containing the expression values of the filtered genes, a list of solved parameters, and some basic parameter information
</p>


<h3>Examples</h3>

<pre><code class="language-R">data('rpkmSim', package='pulseTD')
rpkm_TL &lt;- rpkmSim$labexon[1:2,]
rpkm_PT &lt;- rpkmSim$totintr[1:2,]
rpkm_TT &lt;- rpkmSim$totexon[1:2,]
TimeGrid &lt;- c(0, 15, 30, 45, 60, 75, 90, 105, 120, 135, 150, 165, 180)
tL &lt;- 10

pulseRates&lt;- estimateParams(rpkm_TL,rpkm_TT,rpkm_PT,TimeGrid,tL,clusterNumber=1,loopnumber=10)


</code></pre>


</div>