<div class="container">

<table style="width: 100%;"><tr>
<td>estimateExpression</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Estimated Expression</h2>

<h3>Description</h3>

<p>This function is used to calculate the expression value of the gene.
It includs total exon expression values, total intron expression values, labeled exon expression values, and labeled intron expression values.
</p>


<h3>Usage</h3>

<pre><code class="language-R">estimateExpression(txdb, filelist, by = "gene", mode = "Union",
  singleEnd = TRUE, ignore.strand = TRUE, fragments = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>txdb</code></td>
<td>
<p>Genomic annotation file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filelist</code></td>
<td>
<p>A vector of sam file paths containing 4sU labeled sam files and unlabeled files</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>by</code></td>
<td>
<p>In terms of genes or transcripts, the default is the gene</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mode</code></td>
<td>
<p>mode can be one of the pre-defined count methods such as "Union", "IntersectionStrict", or "IntersectionNotEmpty" or it a user supplied count function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>singleEnd</code></td>
<td>
<p>(Default TRUE) A logical indicating if reads are single or paired-end.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ignore.strand</code></td>
<td>
<p>A logical indicating if strand should be considered when matching</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fragments</code></td>
<td>
<p>(Default FALSE) A logical; applied to paired-end data only. fragments control which function is used to read the data which subsequently affects which records are included in counting.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>The parameters of the <code>summarizeOverlaps</code> method</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list of expression
</p>


<h3>See Also</h3>

<p>Other parameters <code>summarizeOverlaps</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(TxDb.Hsapiens.UCSC.hg19.knownGene)
txdb &lt;- TxDb.Hsapiens.UCSC.hg19.knownGene
test_path &lt;- file.path(system.file(package="pulseTD"),'extdata/test1.sorted.bam')
test_path2 &lt;- file.path(system.file(package="pulseTD"),'extdata/test2.sorted.bam')
rpkmres &lt;- estimateExpression(txdb,c(test_path,test_path2), by='gene')
data('rpkmres', package='pulseTD')
head(rpkmres$total_exp)
head(rpkmres$pre_exp)
</code></pre>


</div>