<div class="container">

<table style="width: 100%;"><tr>
<td>predict_topologies_from_sequences</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Run PureseqTM directy on a protein sequence</h2>

<h3>Description</h3>

<p>Run PureseqTM directy on a protein sequence
</p>


<h3>Usage</h3>

<pre><code class="language-R">predict_topologies_from_sequences(
  protein_sequences,
  folder_name = get_default_pureseqtm_folder(),
  temp_fasta_filename = tempfile(fileext = ".fasta")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>protein_sequences</code></td>
<td>
<p>one ore more protein sequence,
each sequence with the amino acids as capitals, for
example <code>MEILCEDNTSLSSIPNSL</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>folder_name</code></td>
<td>
<p>superfolder of PureseqTM.
The superfolder's name is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by
get_default_pureseqtm_folder</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>temp_fasta_filename</code></td>
<td>
<p>temporary FASTA filename,
which will deleted after usage</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a topology as a string of zeroes and ones, where a one denotes
that the corresponding amino acid is located within the membrane.
</p>


<h3>Author(s)</h3>

<p>Rich√®l J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use mock_predict_topologies_from_sequences
to mock the prediction of protein sequences, as can be useful
in testing
</p>


<h3>Examples</h3>

<pre><code class="language-R">if (is_pureseqtm_installed()) {
  protein_sequence &lt;- paste0(
    "QEKNWSALLTAVVIILTIAGNILVIMAVSLEKKLQNATNYFLM",
    "SLAIADMLLGFLVMPVSMLTILYGYRWP"
  )
  predict_topology_from_sequence(protein_sequence)
}
</code></pre>


</div>