<div class="container">

<table style="width: 100%;"><tr>
<td>lung</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>DNA Microarray Data of Lung Tumors</h2>

<h3>Description</h3>

<p>DNA Microarray data of 73 lung tissues including 67 lung tumors.
There are 916 observations of genes for each lung tissue.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(lung)</code></pre>


<h3>Format</h3>

<p>data frame of size <code class="reqn">916 \times 73</code>.</p>


<h3>Details</h3>

<p>This dataset has been modified from original data. Each one
observation of duplicate genes has been removed. See <code>source</code>
section in this help for original data source.
</p>


<h3>Source</h3>

<p><a href="http://genome-www.stanford.edu/lung_cancer/adeno/">http://genome-www.stanford.edu/lung_cancer/adeno/</a></p>


<h3>References</h3>

<p>Garber, M. E. et al. (2001)
"Diversity of gene expression in adenocarcinoma of the lung",
<em>Proceedings of the National Academy of Sciences</em>,
98, 13784-13789.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## Reading the data
data(lung)

## Multiscale Bootstrap Resampling
lung.pv &lt;- pvclust(lung, nboot=100)

## CAUTION: nboot=100 may be too small for actual use.
##          We suggest nboot=1000 or larger.
##          plot/print functions will be useful for diagnostics.

## Plot the result
plot(lung.pv, cex=0.8, cex.pv=0.7)

ask.bak &lt;- par()$ask
par(ask=TRUE)

pvrect(lung.pv, alpha=0.9)
msplot(lung.pv, edges=c(51,62,68,71))

par(ask=ask.bak)

## Print a cluster with high p-value
lung.pp &lt;- pvpick(lung.pv, alpha=0.9)
lung.pp$clusters[[2]]

## Print its edge number
lung.pp$edges[2]


## We recommend parallel computing for large dataset as this one
## Not run: 
library(snow)
cl &lt;- makeCluster(10, type="MPI")
lung.pv &lt;- parPvclust(cl, lung, nboot=1000)

## End(Not run)
</code></pre>


</div>